<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A complete analysis of single cell subpopulations, from identifying
    subpopulations to analysing their relationship
    (scGPS = single cell Global Predictions of Subpopulation) • scGPS</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="A complete analysis of single cell subpopulations, from identifying
    subpopulations to analysing their relationship
    (scGPS = single cell Global Predictions of Subpopulation)">
<meta property="og:description" content="The package implements two main algorithms to answer two key
    questions: a SCORE (Stable Clustering at Optimal REsolution) to find
    subpopulations, followed by scGPS to investigate the relationships between
    subpopulations.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">scGPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>

<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Benchmarking
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="benchmarking/benchMarking_baron.html" target="_blank">Baron Cluster</a>
      <a href="benchmarking/benchMarking_camp.html" target="_blank">Camp Cluster</a>
      <a href="benchmarking/benchMarking_klein.html" target="_blank">Klein Cluster</a>
      <a href="benchmarking/benchMarking_koh.html" target="_blank">Koh Cluster</a>
      <a href="benchmarking/benchMarking_kumar.html" target="_blank">Kumar Cluster</a>
      <a href="benchmarking/benchMarking_yan.html" target="_blank">Yan Cluster</a>
      <a href="benchmarking/buettner.html" target="_blank">Buettner Trajectory</a>
      <a href="benchmarking/engel.html" target="_blank">Engel Trajectory</a>
      <a href="benchmarking/guo.html" target="_blank">Guo Trajectory</a>
      <a href="benchmarking/hayashi.html" target="_blank">Hayashi Trajectory</a>
      <a href="benchmarking/kowalczyk.html" target="_blank">Kowalczyk Trajectory</a>
      <a href="benchmarking/loh.html" target="_blank">Loh Trajectory</a>
      <a href="benchmarking/petropoulos.html" target="_blank">Petropoulos Trajectory</a>
      <a href="benchmarking/saliba.html" target="_blank">Saliba Trajectory</a>
      <a href="benchmarking/schlitzer.html" target="_blank">Schlitzer Trajectory</a>
      <a href="benchmarking/trapnell.html" target="_blank">Trapnell Trajectory</a>
      <a href="benchmarking/zhang.html" target="_blank">Zhang Trajectory</a>
    </li>
  </ul>
</li>

      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/IMB-Computational-Genomics-Lab/scGPS/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="scgps---single-cell-global-fate-potential-of-subpopulations" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#scgps---single-cell-global-fate-potential-of-subpopulations" class="anchor"></a><em>scGPS</em> - Single Cell Global fate Potential of Subpopulations</h1></div>
<p align="center">
<img src="reference/figures/scGPSlogo.png" width="200px"></p>
<p>The <em>scGPS</em> package website is available at: <a href="https://imb-computational-genomics-lab.github.io/scGPS/index.html" class="uri">https://imb-computational-genomics-lab.github.io/scGPS/index.html</a></p>
<p>The usage instruction can be found at: <a href="https://imb-computational-genomics-lab.github.io/scGPS/articles/vignette.html" class="uri">https://imb-computational-genomics-lab.github.io/scGPS/articles/vignette.html</a></p>
<div id="scgps-general-description" class="section level2">
<h2 class="hasAnchor">
<a href="#scgps-general-description" class="anchor"></a><em>scGPS</em> general description</h2>
<p><em>scGPS</em> is a complete single cell RNA analysis framework from decomposing a mixed population into clusters (<em>SCORE</em>) to analysing the relationship between clusters (<em>scGPS</em>). <em>scGPS</em> also performs unsupervised selection of predictive genes defining a subpopulation and/or driving transition between subpopulations.</p>
<p>The package implements two new algorithms <em>SCORE</em> and <em>scGPS</em>.</p>
<p>Key features of the <em>SCORE</em> clustering algorithm</p>
<ul>
<li>Unsupervised (no prior number of clusters), stable (with automated selection of stability and resolution parameters through scanning a range of search windows for each run, together with a boostrapping aggregation approach to determine stable clusters), fast (with Rcpp implementation)</li>
<li>
<em>SCORE</em> first builds a reference cluster (the highest resolution) and then runs iterative clustering through 40 windows (or more) in the dendrogram</li>
<li>Resolution is quantified as the divergence from reference by applying adjusted Rand index</li>
<li>Stability is the proportional to the number of executive runs without Rand index change while changing the cluster search space</li>
<li>Optimal resolution is the combination of: stable and high resolution</li>
<li>Bagging algorithm (bootstrap aggregation) can detect a rare subpopulation, which appears multiple times during different decision tree runs</li>
</ul>
<p>Key features of the <em>scGPS</em> algorithm</p>
<ul>
<li>Estimates transition scores between any two subpopulations</li>
<li>
<em>scGPS</em> prediction model is based on Elastic Net procedure, which enables to select predictive genes and train interpretable models to predict each subpopulation</li>
<li>Genes identified by <em>scGPS</em> perform better than known gene markers in predicting cell subpopulations</li>
<li>Transition scores are percents of target cells classified as the same class to the original subpopulation</li>
<li>For cell subtype comparision, transition scores are similarity between two subpopulations</li>
<li>The scores are average values from 100 bootstrap runs</li>
<li>For comparison, a non-shrinkage procedure with linear discriminant analysis (LDA) is used</li>
</ul>
</div>
<div id="scgps-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#scgps-workflow" class="anchor"></a><em>scGPS</em> workflow</h2>
<p><em>scGPS</em> takes scRNA expression dataset(s) from one or more unknown sample(s) to find subpopulations and relationship between these subpopulations. The input dataset(s) contains mixed, heterogeous cells. <em>scGPS</em> first uses <em>SCORE</em> (or <em>CORE</em> V2.0) to identify homogenous subpopulations. <em>scGPS</em> contains a number of functions to verify the subpopulations identified by <em>SCORE</em> (e.g. functions to compare with results from PCA, tSNE and the imputation method CIDR). <em>scGPS</em> also has options to find gene markers that distinguish a subpopulation from the remaining cells and performs pathway enrichment analysis to annotate subpopulation. In the second stage, <em>scGPS</em> applies a machine learning procedure to select optimal gene predictors and to build prediction models that can estimate between-subpopulation transition scores, which are the probability of cells from one subpopulation that can likely transition to the other subpopulation.</p>
<p align="center">
<img src="reference/figures/packagePlan.png" width="450px"><br> Figure 1. scGPS workflow. Yellow boxes show inputs, and green boxes show main scGPS analysis.
</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/scGPS">https://​www.bioconductor.org/​packages/​scGPS</a>
</li>
<li>Browse source code at <br><a href="https://github.com/IMB-Computational-Genomics-Lab/scGPS/">https://​github.com/​IMB-Computational-Genomics-Lab/​scGPS/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/IMB-Computational-Genomics-Lab/scGPS/issues">https://​github.com/​IMB-Computational-Genomics-Lab/​scGPS/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Quan Nguyen <br><small class="roles"> Author, maintainer </small>  </li>
<li>Michael Thompson <br><small class="roles"> Author </small>  </li>
<li>Anne Senabouth <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Senabouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
