<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGPS introduction â€¢ scGPS</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scGPS introduction">
<meta property="og:description" content="scGPS is a single cell RNA analysis framework to decompose a mixed population into clusters (SCORE), followed by analysing the relationship between clusters (scGPS)">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scGPS introduction</h1>
                        <h4 class="author">Quan Nguyen and Michael Thompson</h4>
            
            <h4 class="date">2019-05-15</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># To install scGPS from github (Depending on the configuration of the local</span>
<span class="co"># computer or HPC, possible custom C++ compilation may be required - see</span>
<span class="co"># installation trouble-shootings below)</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"IMB-Computational-Genomics-Lab/scGPS"</span>)

<span class="co"># for C++ compilation trouble-shooting, manual download and installation can be</span>
<span class="co"># done from github</span>

git clone https<span class="op">:</span><span class="er">//</span>github.com<span class="op">/</span>IMB<span class="op">-</span>Computational<span class="op">-</span>Genomics<span class="op">-</span>Lab<span class="op">/</span>scGPS

<span class="co"># then check in scGPS/src if any of the precompiled (e.g.  those with *.so and</span>
<span class="co"># *.o) files exist and delete them before recompiling</span>

<span class="co"># then with the scGPS as the R working directory, manually install and load</span>
<span class="co"># using devtools functionality</span>
<span class="co"># Install the package</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/install">install</a></span>()
<span class="co">#load the package to the workspace </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scGPS)</code></pre></div>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: </em> <em>given a mixed population with known subpopulations, </em> <em>estimate transition scores between these subpopulation.</em></p>
<div id="create-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#create-scgps-objects" class="anchor"></a>2.1 Create scGPS objects</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load mixed population 1 (loaded from sample1 dataset, named it as day2)</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scGPS)

day2 &lt;-<span class="st"> </span>sample1
mixedpop1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/new_scGPS_object.html">new_scGPS_object</a></span>(<span class="dt">ExpressionMatrix =</span> day2<span class="op">$</span>dat2_counts,
    <span class="dt">GeneMetadata =</span> day2<span class="op">$</span>dat2geneInfo, <span class="dt">CellMetadata =</span> day2<span class="op">$</span>dat2_clusters)

<span class="co"># load mixed population 2 (loaded from sample2 dataset, named it as day5)</span>
day5 &lt;-<span class="st"> </span>sample2
mixedpop2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/new_scGPS_object.html">new_scGPS_object</a></span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts,
    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> day5<span class="op">$</span>dat5_clusters)</code></pre></div>
</div>
<div id="run-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#run-prediction" class="anchor"></a>2.2 Run prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># select a subpopulation</span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co"># load gene list (this can be any lists of user selected genes)</span>
genes &lt;-<span class="st"> </span>GeneList
genes &lt;-<span class="st"> </span>genes<span class="op">$</span>Merged_unique
<span class="co"># load cluster information </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#run training (running nboots = 3 here, but recommend to use nboots = 50-100)</span>
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap.html">bootstrap</a></span>(<span class="dt">nboots =</span> <span class="dv">3</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID,
    <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2, <span class="dt">trainset_ratio =</span> <span class="fl">0.7</span>)</code></pre></div>
<pre><code>## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev   Lambda
##  [1,]  0 -1.442e-15 0.144900
##  [2,]  1  1.029e-02 0.132100
##  [3,]  1  1.885e-02 0.120300
##  [4,]  1  2.598e-02 0.109600
##  [5,]  1  3.193e-02 0.099890
##  [6,]  1  3.689e-02 0.091020
##  [7,]  1  4.104e-02 0.082930
##  [8,]  1  4.450e-02 0.075560
##  [9,]  1  4.739e-02 0.068850
## [10,]  1  4.980e-02 0.062730
## [11,]  1  5.182e-02 0.057160
## [12,]  2  5.414e-02 0.052080
## [13,]  3  5.832e-02 0.047460
## [14,]  3  6.194e-02 0.043240
## [15,]  3  6.506e-02 0.039400
## [16,]  3  6.775e-02 0.035900
## [17,]  3  7.007e-02 0.032710
## [18,]  4  7.223e-02 0.029800
## [19,]  4  7.415e-02 0.027160
## [20,]  4  7.582e-02 0.024740
## [21,]  4  7.725e-02 0.022550
## [22,]  4  7.848e-02 0.020540
## [23,]  4  7.954e-02 0.018720
## [24,]  4  8.044e-02 0.017060
## [25,]  4  8.122e-02 0.015540
## [26,]  4  8.189e-02 0.014160
## [27,]  4  8.246e-02 0.012900
## [28,]  5  8.294e-02 0.011760
## [29,]  5  8.335e-02 0.010710
## [30,]  5  8.371e-02 0.009760
## [31,]  5  8.400e-02 0.008893
## [32,]  5  8.426e-02 0.008103
## [33,]  5  8.447e-02 0.007383
## [34,]  5  8.465e-02 0.006727
## [35,]  5  8.480e-02 0.006129
## [36,]  5  8.493e-02 0.005585
## [37,]  5  8.504e-02 0.005089
## [38,]  5  8.513e-02 0.004637
## [39,]  5  8.521e-02 0.004225
## [40,]  5  8.527e-02 0.003849
## [41,]  5  8.533e-02 0.003507
## [42,]  5  8.537e-02 0.003196
## [43,]  5  8.541e-02 0.002912
## [44,]  5  8.544e-02 0.002653
## [45,]  5  8.547e-02 0.002418
## [46,]  5  8.549e-02 0.002203
## [47,]  5  8.551e-02 0.002007
## [48,]  5  8.552e-02 0.001829
## [49,]  5  8.554e-02 0.001666
## [50,]  5  8.555e-02 0.001518
## [51,]  5  8.556e-02 0.001383
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev   Lambda
##  [1,]  0 1.281e-15 0.180200
##  [2,]  1 1.591e-02 0.164200
##  [3,]  1 2.916e-02 0.149600
##  [4,]  1 4.021e-02 0.136300
##  [5,]  1 4.946e-02 0.124200
##  [6,]  1 5.719e-02 0.113100
##  [7,]  1 6.368e-02 0.103100
##  [8,]  1 6.911e-02 0.093930
##  [9,]  1 7.367e-02 0.085590
## [10,]  1 7.749e-02 0.077990
## [11,]  1 8.069e-02 0.071060
## [12,]  1 8.338e-02 0.064750
## [13,]  1 8.563e-02 0.058990
## [14,]  3 8.993e-02 0.053750
## [15,]  3 9.409e-02 0.048980
## [16,]  3 9.766e-02 0.044630
## [17,]  3 1.007e-01 0.040660
## [18,]  3 1.033e-01 0.037050
## [19,]  3 1.056e-01 0.033760
## [20,]  4 1.077e-01 0.030760
## [21,]  4 1.095e-01 0.028030
## [22,]  4 1.110e-01 0.025540
## [23,]  4 1.123e-01 0.023270
## [24,]  4 1.135e-01 0.021200
## [25,]  4 1.145e-01 0.019320
## [26,]  4 1.153e-01 0.017600
## [27,]  4 1.160e-01 0.016040
## [28,]  4 1.166e-01 0.014610
## [29,]  4 1.172e-01 0.013310
## [30,]  4 1.176e-01 0.012130
## [31,]  4 1.180e-01 0.011050
## [32,]  4 1.183e-01 0.010070
## [33,]  4 1.186e-01 0.009177
## [34,]  4 1.189e-01 0.008362
## [35,]  4 1.191e-01 0.007619
## [36,]  4 1.192e-01 0.006942
## [37,]  4 1.194e-01 0.006326
## [38,]  4 1.195e-01 0.005764
## [39,]  4 1.196e-01 0.005252
## [40,]  4 1.197e-01 0.004785
## [41,]  4 1.198e-01 0.004360
## [42,]  4 1.198e-01 0.003973
## [43,]  4 1.199e-01 0.003620
## [44,]  4 1.199e-01 0.003298
## [45,]  4 1.199e-01 0.003005
## [46,]  4 1.200e-01 0.002738
## [47,]  4 1.200e-01 0.002495
## [48,]  4 1.200e-01 0.002273
## [49,]  4 1.200e-01 0.002071
## [50,]  4 1.201e-01 0.001887
## [51,]  4 1.201e-01 0.001720
## [52,]  4 1.201e-01 0.001567
## [53,]  4 1.201e-01 0.001428
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev   Lambda
##  [1,]  0 -3.204e-15 0.126800
##  [2,]  1  7.881e-03 0.115500
##  [3,]  1  1.443e-02 0.105300
##  [4,]  1  1.989e-02 0.095930
##  [5,]  1  2.443e-02 0.087400
##  [6,]  1  2.822e-02 0.079640
##  [7,]  1  3.138e-02 0.072570
##  [8,]  1  3.401e-02 0.066120
##  [9,]  1  3.621e-02 0.060240
## [10,]  2  3.923e-02 0.054890
## [11,]  3  4.281e-02 0.050020
## [12,]  3  4.652e-02 0.045570
## [13,]  3  4.969e-02 0.041520
## [14,]  3  5.239e-02 0.037840
## [15,]  3  5.471e-02 0.034470
## [16,]  3  5.670e-02 0.031410
## [17,]  3  5.841e-02 0.028620
## [18,]  3  5.988e-02 0.026080
## [19,]  3  6.114e-02 0.023760
## [20,]  3  6.223e-02 0.021650
## [21,]  3  6.316e-02 0.019730
## [22,]  3  6.397e-02 0.017970
## [23,]  3  6.466e-02 0.016380
## [24,]  3  6.525e-02 0.014920
## [25,]  3  6.575e-02 0.013600
## [26,]  3  6.618e-02 0.012390
## [27,]  3  6.655e-02 0.011290
## [28,]  3  6.687e-02 0.010290
## [29,]  3  6.713e-02 0.009372
## [30,]  3  6.736e-02 0.008540
## [31,]  3  6.755e-02 0.007781
## [32,]  3  6.772e-02 0.007090
## [33,]  3  6.786e-02 0.006460
## [34,]  3  6.797e-02 0.005886
## [35,]  3  6.807e-02 0.005363
## [36,]  4  6.817e-02 0.004887
## [37,]  4  6.826e-02 0.004453
## [38,]  4  6.833e-02 0.004057
## [39,]  4  6.839e-02 0.003697
## [40,]  4  6.844e-02 0.003368
## [41,]  4  6.848e-02 0.003069
## [42,]  4  6.851e-02 0.002796
## [43,]  4  6.854e-02 0.002548
## [44,]  4  6.857e-02 0.002322
## [45,]  4  6.859e-02 0.002115
## [46,]  4  6.861e-02 0.001927
## [47,]  4  6.862e-02 0.001756
## [48,]  4  6.863e-02 0.001600
## [49,]  4  6.864e-02 0.001458</code></pre>
</div>
<div id="summarise-results" class="section level2">
<h2 class="hasAnchor">
<a href="#summarise-results" class="anchor"></a>2.3 Summarise results</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># display the list of result information in the LASOLDA_dat object </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(LSOLDA_dat)</code></pre></div>
<pre><code>## [1] "Accuracy"          "ElasticNetGenes"   "Deviance"         
## [4] "ElasticNetFit"     "LDAFit"            "predictor_S1"     
## [7] "ElasticNetPredict" "LDAPredict"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>ElasticNetPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## NULL
## 
## [[1]][[2]]
## NULL
## 
## [[1]][[3]]
## NULL
## 
## [[1]][[4]]
## NULL
## 
## [[1]][[5]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[1]][[6]]
## [1] "NA"
## 
## [[1]][[7]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 65
## 
## [[1]][[9]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[1]][[10]]
## [1] 90.22556
## 
## [[1]][[11]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[1]][[12]]
## [1] 77.5
## 
## 
## [[2]]
## [[2]][[1]]
## NULL
## 
## [[2]][[2]]
## NULL
## 
## [[2]][[3]]
## NULL
## 
## [[2]][[4]]
## NULL
## 
## [[2]][[5]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 14.4385
## 
## [[2]][[7]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 55.71429
## 
## [[2]][[9]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[2]][[10]]
## [1] 91.72932
## 
## [[2]][[11]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[2]][[12]]
## [1] 75
## 
## 
## [[3]]
## [[3]][[1]]
## NULL
## 
## [[3]][[2]]
## NULL
## 
## [[3]][[3]]
## NULL
## 
## [[3]][[4]]
## NULL
## 
## [[3]][[5]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 29.41176
## 
## [[3]][[7]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 47.14286
## 
## [[3]][[9]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[3]][[10]]
## [1] 94.73684
## 
## [[3]][[11]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[3]][[12]]
## [1] 75</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>LDAPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## NULL
## 
## [[1]][[2]]
## NULL
## 
## [[1]][[3]]
## NULL
## 
## [[1]][[4]]
## NULL
## 
## [[1]][[5]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[1]][[6]]
## [1] 0
## 
## [[1]][[7]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 43.57143
## 
## [[1]][[9]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[1]][[10]]
## [1] 1.503759
## 
## [[1]][[11]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[1]][[12]]
## [1] 12.5
## 
## 
## [[2]]
## [[2]][[1]]
## NULL
## 
## [[2]][[2]]
## NULL
## 
## [[2]][[3]]
## NULL
## 
## [[2]][[4]]
## NULL
## 
## [[2]][[5]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 16.57754
## 
## [[2]][[7]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 43.57143
## 
## [[2]][[9]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[2]][[10]]
## [1] 27.06767
## 
## [[2]][[11]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[2]][[12]]
## [1] 27.5
## 
## 
## [[3]]
## [[3]][[1]]
## NULL
## 
## [[3]][[2]]
## NULL
## 
## [[3]][[3]]
## NULL
## 
## [[3]][[4]]
## NULL
## 
## [[3]][[5]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 33.68984
## 
## [[3]][[7]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 25
## 
## [[3]][[9]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[3]][[10]]
## [1] 45.11278
## 
## [[3]][[11]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[3]][[12]]
## [1] 32.5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary results LDA</span>
sum_pred_lda &lt;-<span class="st"> </span><span class="kw"><a href="../reference/summary_prediction_lda.html">summary_prediction_lda</a></span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># summary results Lasso to show the percent of cells</span>
<span class="co"># classified as cells belonging </span>
sum_pred_lasso &lt;-<span class="st"> </span><span class="kw"><a href="../reference/summary_prediction_lasso.html">summary_prediction_lasso</a></span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat,
    <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># plot summary results </span>
plot_sum &lt;-<span class="cf">function</span>(sum_dat){
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(sum_dat)
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/na.fail">na.omit</a></span>(sum_dat_tf)
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(sum_dat[, <span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(sum_dat)],<span class="dv">1</span>,
        <span class="cf">function</span>(x){<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(x))})
    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"for subpop "</span>,<span class="st">"sp"</span>,  sum_dat<span class="op">$</span>names )
    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"in target mixedpop"</span>,<span class="st">"in p"</span>,  sum_dat<span class="op">$</span>names) 
    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sum_dat_tf) &lt;-<span class="st"> </span>sum_dat<span class="op">$</span>names
    <span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(sum_dat_tf, <span class="dt">las=</span><span class="dv">2</span>)
}
<span class="kw">plot_sum</span>(sum_pred_lasso)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_sum</span>(sum_pred_lda)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw"><a href="../reference/summary_accuracy.html">summary_accuracy</a></span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## [1] 66.51163 63.25581 65.58140</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary maximum deviance explained by the model </span>
<span class="kw"><a href="../reference/summary_deviance.html">summary_deviance</a></span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## $allDeviance
## [1] "0.08552" "0.1007"  "0.06575"
## 
## $DeviMax
##          dat_DE$Dfd          Deviance           DEgenes
## 1                 0            0.1007    genes_cluster1
## 2                 1            0.1007    genes_cluster1
## 3                 3            0.1007    genes_cluster1
## 4 remaining DEgenes remaining DEgenes remaining DEgenes
## 
## $LassoGenesMax
## NULL</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">###############################################
<span class="co"># for building time in bioconductor, the following </span>
<span class="co"># is not run, but users can run it in their machine</span>
###############################################
<span class="co"># #3. A complete workflow of the scGPS:</span>
<span class="co"># * The purpose of this workflow is to solve the following task: given an unknown*</span>
<span class="co">#   * mixed population, find clusters and estimate relationship between clusters*</span>
<span class="co">#   </span>
<span class="co">#   ##3.1 Identify clusters in a dataset using CORE</span>
<span class="co">#   *(skip this step if clusters are known)*</span>
<span class="co">#   ```{r CORE, warning = FALSE, message = FALSE, eval = FALSE}</span>
<span class="co"># </span>
<span class="co"># # find clustering information in an expresion data using CORE</span>
<span class="co"># day5 &lt;- sample2</span>
<span class="co"># cellnames &lt;- colnames(day5$dat5_counts)</span>
<span class="co"># cluster &lt;-day5$dat5_clusters</span>
<span class="co"># cellnames &lt;-data.frame("Cluster"=cluster, "cellBarcodes" = cellnames)</span>
<span class="co"># mixedpop2 &lt;-new_scGPS_object(ExpressionMatrix = day5$dat5_counts,</span>
<span class="co">#                      GeneMetadata = day5$dat5geneInfo, CellMetadata = cellnames)</span>
<span class="co"># </span>
<span class="co"># CORE_cluster &lt;- CORE(mixedpop2, remove_outlier = c(0), PCA=FALSE)</span>
<span class="co"># </span>
<span class="co"># # to update the clustering information, users can ...</span>
<span class="co"># key_height &lt;- CORE_cluster$optimalClust$KeyStats$Height</span>
<span class="co"># optimal_res &lt;- CORE_cluster$optimalClust$OptimalRes</span>
<span class="co"># optimal_index = which(key_height == optimal_res)</span>
<span class="co"># </span>
<span class="co"># clustering_after_outlier_removal &lt;- unname(unlist(</span>
<span class="co">#   CORE_cluster$Cluster[[optimal_index]]))</span>
<span class="co"># corresponding_cells_after_outlier_removal &lt;- CORE_cluster$cellsForClustering</span>
<span class="co"># original_cells_before_removal &lt;- colData(mixedpop2)[,2]</span>
<span class="co"># corresponding_index &lt;- match(corresponding_cells_after_outlier_removal,</span>
<span class="co">#                              original_cells_before_removal )</span>
<span class="co"># # check the matching</span>
<span class="co"># identical(as.character(original_cells_before_removal[corresponding_index]),</span>
<span class="co">#           corresponding_cells_after_outlier_removal)</span>
<span class="co"># # create new object with the new clustering after removing outliers</span>
<span class="co"># mixedpop2_post_clustering &lt;- mixedpop2[,corresponding_index]</span>
<span class="co"># colData(mixedpop2_post_clustering)[,1] &lt;- clustering_after_outlier_removal</span>
<span class="co"># </span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##3.2 Identify clusters in a dataset using SCORE</span>
<span class="co"># ##(Stable Clustering at Optimal REsolution)</span>
<span class="co"># *(skip this step if clusters are known)*</span>
<span class="co">#   *(SCORE aims to get stable subpopulation results, *</span>
<span class="co">#       * by introducing bagging aggregation and bootstrapping to the CORE algorithm)*</span>
<span class="co">#   ```{r SCORE with bagging, warning = FALSE, message = FALSE, eval = FALSE}</span>
<span class="co"># </span>
<span class="co"># # find clustering information in an expresion data using SCORE</span>
<span class="co"># day5 &lt;- sample2</span>
<span class="co"># cellnames &lt;- colnames(day5$dat5_counts)</span>
<span class="co"># cluster &lt;-day5$dat5_clusters</span>
<span class="co"># cellnames &lt;-data.frame("Cluster"=cluster, "cellBarcodes" = cellnames)</span>
<span class="co"># mixedpop2 &lt;-new_scGPS_object(ExpressionMatrix = day5$dat5_counts,</span>
<span class="co">#                      GeneMetadata = day5$dat5geneInfo, CellMetadata = cellnames )</span>
<span class="co"># </span>
<span class="co"># SCORE_test &lt;- CORE_bagging(mixedpop2, remove_outlier = c(0), PCA=FALSE,</span>
<span class="co">#                                  bagging_run = 20, subsample_proportion = .8)</span>
<span class="co"># </span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##3.3 Visualise all cluster results in all iterations</span>
<span class="co"># ```{r visualisation, dev='CairoPDF', eval = FALSE}</span>
<span class="co"># dev.off()</span>
<span class="co"># ##3.2.1 plot CORE clustering</span>
<span class="co"># p1 &lt;- plot_CORE(CORE_cluster$tree, CORE_cluster$Cluster,</span>
<span class="co">#                 color_branch = c("#208eb7", "#6ce9d3", "#1c5e39", "#8fca40", "#154975",</span>
<span class="co">#                                  "#b1c8eb"))</span>
<span class="co"># p1</span>
<span class="co"># #extract optimal index identified by CORE</span>
<span class="co"># key_height &lt;- CORE_cluster$optimalClust$KeyStats$Height</span>
<span class="co"># optimal_res &lt;- CORE_cluster$optimalClust$OptimalRes</span>
<span class="co"># optimal_index = which(key_height == optimal_res)</span>
<span class="co"># #plot one optimal clustering bar</span>
<span class="co"># plot_optimal_CORE(original_tree= CORE_cluster$tree,</span>
<span class="co">#                   optimal_cluster = unlist(CORE_cluster$Cluster[optimal_index]),</span>
<span class="co">#                   shift = -2000)</span>
<span class="co"># </span>
<span class="co"># ##3.2.2 plot SCORE clustering</span>
<span class="co"># #plot all clustering bars</span>
<span class="co"># plot_CORE(SCORE_test$tree, list_clusters = SCORE_test$Cluster)</span>
<span class="co"># #plot one stable optimal clustering bar</span>
<span class="co"># plot_optimal_CORE(original_tree= SCORE_test$tree,</span>
<span class="co">#                   optimal_cluster = unlist(SCORE_test$Cluster[SCORE_test$optimal_index]),</span>
<span class="co">#                   shift = -100)</span>
<span class="co"># </span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##3.4 Compare clustering results with other dimensional reduction methods</span>
<span class="co"># ##(e.g., tSNE or CIDR)</span>
<span class="co"># ```{r compare_clustering, eval = FALSE}</span>
<span class="co"># #library(cidr)</span>
<span class="co"># t &lt;- tSNE(expression.mat=assay(mixedpop2))</span>
<span class="co"># p2 &lt;-plotReduced(t, color_fac = factor(colData(mixedpop2)[,1]),</span>
<span class="co">#                        palletes =1:length(unique(colData(mixedpop2)[,1])))</span>
<span class="co"># p2</span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##3.5 Find gene markers and annotate clusters</span>
<span class="co"># </span>
<span class="co"># ```{r find_markers, warning = FALSE, message = FALSE, eval = FALSE}</span>
<span class="co"># #load gene list (this can be any lists of user-selected genes)</span>
<span class="co"># genes &lt;-GeneList</span>
<span class="co"># genes &lt;-genes$Merged_unique</span>
<span class="co"># </span>
<span class="co"># #the gene list can also be objectively identified by differential expression</span>
<span class="co"># #analysis cluster information is requied for find_markers. Here, we use</span>
<span class="co"># #CORE results.</span>
<span class="co"># </span>
<span class="co"># #colData(mixedpop2)[,1] &lt;- unlist(SCORE_test$Cluster[SCORE_test$optimal_index])</span>
<span class="co"># </span>
<span class="co"># suppressMessages(library(locfit))</span>
<span class="co"># suppressMessages(library(DESeq))</span>
<span class="co"># </span>
<span class="co"># DEgenes &lt;- find_markers(expression_matrix=assay(mixedpop2),</span>
<span class="co">#                              cluster = colData(mixedpop2)[,1],</span>
<span class="co">#                              selected_cluster=unique(colData(mixedpop2)[,1]))</span>
<span class="co"># </span>
<span class="co"># #the output contains dataframes for each cluster.</span>
<span class="co"># #the data frame contains all genes, sorted by p-values</span>
<span class="co"># names(DEgenes)</span>
<span class="co"># </span>
<span class="co"># #you can annotate the identified clusters</span>
<span class="co"># DEgeneList_1vsOthers &lt;- DEgenes$DE_Subpop1vsRemaining$id</span>
<span class="co"># </span>
<span class="co"># #users need to check the format of the gene input to make sure they are</span>
<span class="co"># #consistent to the gene names in the expression matrix</span>
<span class="co"># </span>
<span class="co"># #the following command saves the file "PathwayEnrichment.xlsx" to the</span>
<span class="co"># #working dir</span>
<span class="co"># #use 500 top DE genes</span>
<span class="co"># suppressMessages(library(DOSE))</span>
<span class="co"># suppressMessages(library(ReactomePA))</span>
<span class="co"># suppressMessages(library(clusterProfiler))</span>
<span class="co"># enrichment_test &lt;- annotate(DEgeneList_1vsOthers[1:500],</span>
<span class="co">#                                   pvalueCutoff=0.05, gene_symbol=TRUE)</span>
<span class="co"># </span>
<span class="co"># #the enrichment outputs can be displayed by running</span>
<span class="co"># dotplot(enrichment_test, showCategory=10, font.size = 6)</span>
<span class="co"># </span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># #4. Relationship between clusters within one sample or between two samples</span>
<span class="co"># *The purpose of this workflow is to solve the following task: *</span>
<span class="co">#   *given one or two unknown mixed population(s) and clusters in each mixed*</span>
<span class="co">#   *population, estimate and visualise relationship between clusters*</span>
<span class="co">#   </span>
<span class="co">#   ##4.1 Start the scGPS prediction to find relationship between clusters</span>
<span class="co">#   </span>
<span class="co">#   ```{r scGPS_prediction, warning = FALSE, message = FALSE, eval = FALSE}</span>
<span class="co"># </span>
<span class="co"># #select a subpopulation, and input gene list</span>
<span class="co"># c_selectID &lt;- 1</span>
<span class="co"># #note make sure the format for genes input here is the same to the format</span>
<span class="co"># #for genes in the mixedpop1 and mixedpop2</span>
<span class="co"># genes = DEgenes$DE_Subpop1vsRemaining$id[1:500]</span>
<span class="co"># </span>
<span class="co"># #run the test bootstrap with nboots = 2 runs</span>
<span class="co"># </span>
<span class="co"># cluster_mixedpop1 &lt;- colData(mixedpop1)[,1]</span>
<span class="co"># cluster_mixedpop2 &lt;- colData(mixedpop2)[,1]</span>
<span class="co"># </span>
<span class="co"># sink("temp")</span>
<span class="co"># LSOLDA_dat &lt;- bootstrap(nboots = 2, mixedpop1 = mixedpop1,</span>
<span class="co">#                               mixedpop2 = mixedpop2, genes = genes, c_selectID  = c_selectID,</span>
<span class="co">#                               listData = list(),</span>
<span class="co">#                               cluster_mixedpop1 = cluster_mixedpop1,</span>
<span class="co">#                               cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># sink()</span>
<span class="co"># </span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##4.2 Display summary results for the prediction</span>
<span class="co"># </span>
<span class="co"># ```{r summarise_prediction, eval = FALSE}</span>
<span class="co"># #get the number of rows for the summary matrix</span>
<span class="co"># row_cluster &lt;-length(unique(colData(mixedpop2)[,1]))</span>
<span class="co"># </span>
<span class="co"># #summary results LDA to to show the percent of cells classified as cells</span>
<span class="co"># #belonging by LDA classifier</span>
<span class="co"># summary_prediction_lda(LSOLDA_dat=LSOLDA_dat, nPredSubpop = row_cluster )</span>
<span class="co"># </span>
<span class="co"># #summary results Lasso to show the percent of cells classified as cells</span>
<span class="co"># #belonging by Lasso classifier</span>
<span class="co"># summary_prediction_lasso(LSOLDA_dat=LSOLDA_dat, nPredSubpop = row_cluster)</span>
<span class="co"># </span>
<span class="co"># # summary maximum deviance explained by the model during the model training</span>
<span class="co"># summary_deviance(object = LSOLDA_dat)</span>
<span class="co"># </span>
<span class="co"># # summary accuracy to check the model accuracy in the leave-out test set</span>
<span class="co"># summary_accuracy(object = LSOLDA_dat)</span>
<span class="co"># </span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##4.3 Plot the relationship between clusters in one sample</span>
<span class="co"># *Here we look at one example use case to find relationship between clusters</span>
<span class="co"># #within one sample or between two sample*</span>
<span class="co"># </span>
<span class="co"># ```{r prediction_one_sample, warning = FALSE, message = FALSE, eval = FALSE}</span>
<span class="co"># #run prediction for 3 clusters</span>
<span class="co"># cluster_mixedpop1 &lt;- colData(mixedpop1)[,1]</span>
<span class="co"># cluster_mixedpop2 &lt;- colData(mixedpop2)[,1]</span>
<span class="co"># #cluster_mixedpop2 &lt;- as.numeric(as.vector(colData(mixedpop2)[,1]))</span>
<span class="co"># </span>
<span class="co"># c_selectID &lt;- 1</span>
<span class="co"># #top 200 gene markers distinguishing cluster 1</span>
<span class="co"># genes = DEgenes$DE_Subpop1vsRemaining$id[1:200]</span>
<span class="co"># </span>
<span class="co"># LSOLDA_dat1 &lt;- bootstrap(nboots = 2, mixedpop1 = mixedpop2,</span>
<span class="co">#                                mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list(),</span>
<span class="co">#                                cluster_mixedpop1 = cluster_mixedpop2,</span>
<span class="co">#                                cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># c_selectID &lt;- 2</span>
<span class="co"># genes = DEgenes$DE_Subpop2vsRemaining$id[1:200]</span>
<span class="co"># </span>
<span class="co"># LSOLDA_dat2 &lt;- bootstrap(nboots = 2,mixedpop1 = mixedpop2,</span>
<span class="co">#                                mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list(),</span>
<span class="co">#                                cluster_mixedpop1 = cluster_mixedpop2,</span>
<span class="co">#                                cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># c_selectID &lt;- 3</span>
<span class="co"># genes = DEgenes$DE_Subpop3vsRemaining$id[1:200]</span>
<span class="co"># LSOLDA_dat3 &lt;- bootstrap(nboots = 2,mixedpop1 = mixedpop2,</span>
<span class="co">#                                mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list(),</span>
<span class="co">#                                cluster_mixedpop1 = cluster_mixedpop2,</span>
<span class="co">#                                cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># c_selectID &lt;- 4</span>
<span class="co"># genes = DEgenes$DE_Subpop4vsRemaining$id[1:200]</span>
<span class="co"># LSOLDA_dat4 &lt;- bootstrap(nboots = 2,mixedpop1 = mixedpop2,</span>
<span class="co">#                                mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list(),</span>
<span class="co">#                                cluster_mixedpop1 = cluster_mixedpop2,</span>
<span class="co">#                                cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># #prepare table input for sankey plot</span>
<span class="co"># </span>
<span class="co"># LASSO_C1S2  &lt;- reformat_LASSO(c_selectID=1, mp_selectID = 2,</span>
<span class="co">#                               LSOLDA_dat=LSOLDA_dat1,</span>
<span class="co">#                               nPredSubpop = length(unique(colData(mixedpop2)[,1])),</span>
<span class="co">#                               Nodes_group ="#7570b3")</span>
<span class="co"># </span>
<span class="co"># LASSO_C2S2  &lt;- reformat_LASSO(c_selectID=2, mp_selectID =2,</span>
<span class="co">#                               LSOLDA_dat=LSOLDA_dat2,</span>
<span class="co">#                               nPredSubpop = length(unique(colData(mixedpop2)[,1])),</span>
<span class="co">#                               Nodes_group ="#1b9e77")</span>
<span class="co"># </span>
<span class="co"># LASSO_C3S2  &lt;- reformat_LASSO(c_selectID=3, mp_selectID =2,</span>
<span class="co">#                               LSOLDA_dat=LSOLDA_dat3,</span>
<span class="co">#                               nPredSubpop = length(unique(colData(mixedpop2)[,1])),</span>
<span class="co">#                               Nodes_group ="#e7298a")</span>
<span class="co"># </span>
<span class="co"># LASSO_C4S2  &lt;- reformat_LASSO(c_selectID=4, mp_selectID =2,</span>
<span class="co">#                               LSOLDA_dat=LSOLDA_dat4,</span>
<span class="co">#                               nPredSubpop = length(unique(colData(mixedpop2)[,1])),</span>
<span class="co">#                               Nodes_group ="#00FFFF")</span>
<span class="co"># </span>
<span class="co"># combined &lt;- rbind(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2, LASSO_C4S2 )</span>
<span class="co"># combined &lt;- combined[is.na(combined$Value) != TRUE,]</span>
<span class="co"># </span>
<span class="co"># nboots = 2</span>
<span class="co"># #links: source, target, value</span>
<span class="co"># #source: node, nodegroup</span>
<span class="co"># combined_D3obj &lt;-list(Nodes=combined[,(nboots+3):(nboots+4)],</span>
<span class="co">#                       Links=combined[,c((nboots+2):(nboots+1),ncol(combined))])</span>
<span class="co"># </span>
<span class="co"># library(networkD3)</span>
<span class="co"># </span>
<span class="co"># Node_source &lt;- as.vector(sort(unique(combined_D3obj$Links$Source)))</span>
<span class="co"># Node_target &lt;- as.vector(sort(unique(combined_D3obj$Links$Target)))</span>
<span class="co"># Node_all &lt;-unique(c(Node_source, Node_target))</span>
<span class="co"># </span>
<span class="co"># #assign IDs for Source (start from 0)</span>
<span class="co"># Source &lt;-combined_D3obj$Links$Source</span>
<span class="co"># Target &lt;- combined_D3obj$Links$Target</span>
<span class="co"># </span>
<span class="co"># for(i in 1:length(Node_all)){</span>
<span class="co">#   Source[Source==Node_all[i]] &lt;-i-1</span>
<span class="co">#   Target[Target==Node_all[i]] &lt;-i-1</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># combined_D3obj$Links$Source &lt;- as.numeric(Source)</span>
<span class="co"># combined_D3obj$Links$Target &lt;- as.numeric(Target)</span>
<span class="co"># combined_D3obj$Links$LinkColor &lt;- combined$NodeGroup</span>
<span class="co"># </span>
<span class="co"># #prepare node info</span>
<span class="co"># node_df &lt;-data.frame(Node=Node_all)</span>
<span class="co"># node_df$id &lt;-as.numeric(c(0, 1:(length(Node_all)-1)))</span>
<span class="co"># </span>
<span class="co"># suppressMessages(library(dplyr))</span>
<span class="co"># Color &lt;- combined %&gt;% count(Node, color=NodeGroup) %&gt;% select(2)</span>
<span class="co"># node_df$color &lt;- Color$color</span>
<span class="co"># </span>
<span class="co"># suppressMessages(library(networkD3))</span>
<span class="co"># p1&lt;-sankeyNetwork(Links =combined_D3obj$Links, Nodes = node_df,</span>
<span class="co">#                   Value = "Value", NodeGroup ="color", LinkGroup = "LinkColor", NodeID="Node",</span>
<span class="co">#                   Source="Source", Target="Target", fontSize = 22 )</span>
<span class="co"># p1</span>
<span class="co"># </span>
<span class="co"># #saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span>
<span class="co"># </span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##4.3 Plot the relationship between clusters in two samples</span>
<span class="co"># *Here we look at one example use case to find relationship between clusters*</span>
<span class="co">#   *within one sample or between two sample*</span>
<span class="co">#   </span>
<span class="co">#   ```{r prediction_two_samples,warning = FALSE, message = FALSE, eval = FALSE}</span>
<span class="co"># #run prediction for 3 clusters</span>
<span class="co"># cluster_mixedpop1 &lt;- colData(mixedpop1)[,1]</span>
<span class="co"># cluster_mixedpop2 &lt;- colData(mixedpop2)[,1]</span>
<span class="co"># row_cluster &lt;-length(unique(colData(mixedpop2)[,1]))</span>
<span class="co"># </span>
<span class="co"># c_selectID &lt;- 1</span>
<span class="co"># #top 200 gene markers distinguishing cluster 1</span>
<span class="co"># genes = DEgenes$DE_Subpop1vsRemaining$id[1:200]</span>
<span class="co"># LSOLDA_dat1 &lt;- bootstrap(nboots = 2, mixedpop1 = mixedpop1,</span>
<span class="co">#                                mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list(),</span>
<span class="co">#                                cluster_mixedpop1 = cluster_mixedpop1,</span>
<span class="co">#                                cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># c_selectID &lt;- 2</span>
<span class="co"># genes = DEgenes$DE_Subpop2vsRemaining$id[1:200]</span>
<span class="co"># LSOLDA_dat2 &lt;- bootstrap(nboots = 2,mixedpop1 = mixedpop1,</span>
<span class="co">#                                mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list(),</span>
<span class="co">#                                cluster_mixedpop1 = cluster_mixedpop1,</span>
<span class="co">#                                cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># c_selectID &lt;- 3</span>
<span class="co"># genes = DEgenes$DE_Subpop3vsRemaining$id[1:200]</span>
<span class="co"># LSOLDA_dat3 &lt;- bootstrap(nboots = 2,mixedpop1 = mixedpop1,</span>
<span class="co">#                                mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list(),</span>
<span class="co">#                                cluster_mixedpop1 = cluster_mixedpop1,</span>
<span class="co">#                                cluster_mixedpop2 = cluster_mixedpop2)</span>
<span class="co"># </span>
<span class="co"># #prepare table input for sankey plot</span>
<span class="co"># </span>
<span class="co"># LASSO_C1S1  &lt;- reformat_LASSO(c_selectID=1, mp_selectID = 1,</span>
<span class="co">#                               LSOLDA_dat=LSOLDA_dat1, nPredSubpop = row_cluster, Nodes_group = "#7570b3")</span>
<span class="co"># </span>
<span class="co"># LASSO_C2S1  &lt;- reformat_LASSO(c_selectID=2, mp_selectID = 1,</span>
<span class="co">#                               LSOLDA_dat=LSOLDA_dat2, nPredSubpop = row_cluster, Nodes_group = "#1b9e77")</span>
<span class="co"># </span>
<span class="co"># LASSO_C3S1  &lt;- reformat_LASSO(c_selectID=3, mp_selectID = 1,</span>
<span class="co">#                               LSOLDA_dat=LSOLDA_dat3, nPredSubpop = row_cluster, Nodes_group = "#e7298a")</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># combined &lt;- rbind(LASSO_C1S1,LASSO_C2S1,LASSO_C3S1)</span>
<span class="co"># </span>
<span class="co"># nboots = 2</span>
<span class="co"># #links: source, target, value</span>
<span class="co"># #source: node, nodegroup</span>
<span class="co"># combined_D3obj &lt;-list(Nodes=combined[,(nboots+3):(nboots+4)],</span>
<span class="co">#                       Links=combined[,c((nboots+2):(nboots+1),ncol(combined))])</span>
<span class="co"># combined &lt;- combined[is.na(combined$Value) != TRUE,]</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># library(networkD3)</span>
<span class="co"># </span>
<span class="co"># Node_source &lt;- as.vector(sort(unique(combined_D3obj$Links$Source)))</span>
<span class="co"># Node_target &lt;- as.vector(sort(unique(combined_D3obj$Links$Target)))</span>
<span class="co"># Node_all &lt;-unique(c(Node_source, Node_target))</span>
<span class="co"># </span>
<span class="co"># #assign IDs for Source (start from 0)</span>
<span class="co"># Source &lt;-combined_D3obj$Links$Source</span>
<span class="co"># Target &lt;- combined_D3obj$Links$Target</span>
<span class="co"># </span>
<span class="co"># for(i in 1:length(Node_all)){</span>
<span class="co">#   Source[Source==Node_all[i]] &lt;-i-1</span>
<span class="co">#   Target[Target==Node_all[i]] &lt;-i-1</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># combined_D3obj$Links$Source &lt;- as.numeric(Source)</span>
<span class="co"># combined_D3obj$Links$Target &lt;- as.numeric(Target)</span>
<span class="co"># combined_D3obj$Links$LinkColor &lt;- combined$NodeGroup</span>
<span class="co"># </span>
<span class="co"># #prepare node info</span>
<span class="co"># node_df &lt;-data.frame(Node=Node_all)</span>
<span class="co"># node_df$id &lt;-as.numeric(c(0, 1:(length(Node_all)-1)))</span>
<span class="co"># </span>
<span class="co"># suppressMessages(library(dplyr))</span>
<span class="co"># n &lt;- length(unique(node_df$Node))</span>
<span class="co"># getPalette = colorRampPalette(RColorBrewer::brewer.pal(9, "Set1"))</span>
<span class="co"># Color = getPalette(n)</span>
<span class="co"># node_df$color &lt;- Color</span>
<span class="co"># suppressMessages(library(networkD3))</span>
<span class="co"># p1&lt;-sankeyNetwork(Links =combined_D3obj$Links, Nodes = node_df,</span>
<span class="co">#                   Value = "Value", NodeGroup ="color", LinkGroup = "LinkColor",</span>
<span class="co">#                   NodeID="Node", Source="Source", Target="Target", fontSize = 22)</span>
<span class="co"># p1</span>
<span class="co"># #saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span>
<span class="co"># ```</span>
<span class="co"># </span>
<span class="co"># ##4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with</span>
<span class="co"># ##a reference dataset to see which cluster is most similar to the reference</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">session_info</a></span>()</code></pre></div>
<pre><code>## - Session info ----------------------------------------------------------
##  setting  value                       
##  version  R version 3.5.1 (2018-07-02)
##  os       Windows 8.1 x64             
##  system   x86_64, mingw32             
##  ui       RTerm                       
##  language (EN)                        
##  collate  English_Australia.1252      
##  ctype    English_Australia.1252      
##  tz       Australia/Brisbane          
##  date     2019-05-15                  
## 
## - Packages --------------------------------------------------------------
##  ! package              * version     date       lib
##    annotate               1.60.0      2018-10-30 [1]
##    AnnotationDbi          1.44.0      2018-10-30 [1]
##    assertthat             0.2.0       2017-04-11 [1]
##    backports              1.1.2       2017-12-13 [1]
##    base64enc              0.1-3       2015-07-28 [1]
##    bindr                  0.1.1       2018-03-13 [1]
##    bindrcpp             * 0.2.2       2018-03-29 [1]
##    Biobase              * 2.42.0      2018-10-30 [1]
##    BiocGenerics         * 0.28.0      2018-10-30 [1]
##    BiocParallel         * 1.16.0      2018-10-30 [1]
##    bit                    1.1-14      2018-05-29 [1]
##    bit64                  0.9-7       2017-05-08 [1]
##    bitops                 1.0-6       2013-08-17 [1]
##    blob                   1.1.1       2018-03-25 [1]
##    callr                  3.0.0       2018-08-24 [1]
##    caret                * 6.0-81      2018-11-20 [1]
##    class                  7.3-14      2015-08-30 [2]
##    cli                    1.0.1       2018-09-25 [1]
##    codetools              0.2-15      2016-10-05 [2]
##    colorspace             1.3-2       2016-12-14 [1]
##    commonmark             1.6         2018-09-30 [1]
##    crayon                 1.3.4       2017-09-16 [1]
##    data.table             1.11.8      2018-09-30 [1]
##    DBI                    1.0.0       2018-05-02 [1]
##    DelayedArray         * 0.8.0       2018-10-30 [1]
##    desc                   1.2.0       2018-05-01 [1]
##    DESeq                  1.34.0      2018-10-30 [1]
##    devtools               2.0.1.9000  2018-11-25 [1]
##    digest                 0.6.18      2018-10-10 [1]
##    dplyr                  0.7.8       2018-11-10 [1]
##    dynamicTreeCut       * 1.63-1      2016-03-11 [1]
##    e1071                  1.7-0       2018-07-28 [1]
##    evaluate               0.12        2018-10-09 [1]
##    foreach                1.4.4       2017-12-12 [1]
##    fs                     1.2.6       2018-08-23 [1]
##    genefilter             1.64.0      2018-10-30 [1]
##    geneplotter            1.60.0      2018-10-30 [1]
##    generics               0.0.1       2018-10-26 [1]
##    GenomeInfoDb         * 1.18.1      2018-11-12 [1]
##    GenomeInfoDbData       1.2.0       2018-11-25 [1]
##    GenomicRanges        * 1.34.0      2018-10-30 [1]
##    ggplot2              * 3.1.0       2018-10-25 [1]
##    glmnet                 2.0-16      2018-04-02 [1]
##    glue                   1.3.0       2018-07-17 [1]
##    gower                  0.1.2       2017-02-23 [1]
##    gtable                 0.2.0       2016-02-26 [1]
##    htmltools              0.3.6       2017-04-28 [1]
##    ipred                  0.9-8       2018-11-05 [1]
##    IRanges              * 2.16.0      2018-10-30 [1]
##    iterators              1.0.10      2018-07-13 [1]
##    knitr                  1.21        2018-12-10 [1]
##    lattice              * 0.20-35     2017-03-25 [2]
##    lava                   1.6.4       2018-11-25 [1]
##    lazyeval               0.2.1       2017-10-29 [1]
##    locfit                 1.5-9.1     2013-04-20 [1]
##    lubridate              1.7.4       2018-04-11 [1]
##    magrittr               1.5         2014-11-22 [1]
##    MASS                   7.3-50      2018-04-30 [2]
##    Matrix                 1.2-14      2018-04-13 [2]
##    matrixStats          * 0.54.0      2018-07-23 [1]
##    memoise                1.1.0       2017-04-21 [1]
##    ModelMetrics           1.2.2       2018-11-03 [1]
##    munsell                0.5.0       2018-06-12 [1]
##    nlme                   3.1-137     2018-04-07 [2]
##    nnet                   7.3-12      2016-02-02 [2]
##    pillar                 1.3.0       2018-07-14 [1]
##    pkgbuild               1.0.2       2018-10-16 [1]
##    pkgconfig              2.0.2       2018-08-16 [1]
##    pkgdown                1.3.0       2018-12-07 [1]
##    pkgload                1.0.2       2018-10-29 [1]
##    plyr                   1.8.4       2016-06-08 [1]
##    prettyunits            1.0.2       2015-07-13 [1]
##    processx               3.2.0       2018-08-16 [1]
##    prodlim                2018.04.18  2018-04-18 [1]
##    ps                     1.2.1       2018-11-06 [1]
##    purrr                  0.2.5       2018-05-29 [1]
##    R6                     2.3.0       2018-10-04 [1]
##    RColorBrewer           1.1-2       2014-12-07 [1]
##    Rcpp                   1.0.0       2018-11-07 [1]
##    RcppArmadillo          0.9.200.4.0 2018-11-10 [1]
##  D RcppParallel           4.4.1       2018-07-19 [1]
##    RCurl                  1.95-4.11   2018-07-15 [1]
##    recipes                0.1.4       2018-11-19 [1]
##    remotes                2.0.2       2018-10-30 [1]
##    reshape2               1.4.3       2017-12-11 [1]
##    rlang                  0.3.0.1     2018-10-25 [1]
##    rmarkdown              1.10        2018-06-11 [1]
##    roxygen2               6.1.1       2018-11-07 [1]
##    rpart                  4.1-13      2018-02-23 [2]
##    rprojroot              1.3-2       2018-01-03 [1]
##    RSQLite                2.1.1       2018-05-06 [1]
##    rstudioapi             0.8         2018-10-02 [1]
##    S4Vectors            * 0.20.1      2018-11-09 [1]
##    scales                 1.0.0       2018-08-09 [1]
##    scGPS                * 0.99.14     2019-05-14 [1]
##    sessioninfo            1.1.1       2018-11-05 [1]
##    SingleCellExperiment * 1.4.0       2018-10-30 [1]
##    stringi                1.2.4       2018-07-20 [1]
##    stringr                1.3.1       2018-05-10 [1]
##    SummarizedExperiment * 1.12.0      2018-10-30 [1]
##    survival               2.42-3      2018-04-16 [2]
##    testthat               2.0.1       2018-10-13 [1]
##    tibble                 1.4.2       2018-01-22 [1]
##    tidyselect             0.2.5       2018-10-11 [1]
##    timeDate               3043.102    2018-02-21 [1]
##    usethis                1.4.0       2018-08-14 [1]
##    withr                  2.1.2       2018-03-15 [1]
##    xfun                   0.4         2018-10-23 [1]
##    XML                    3.98-1.16   2018-08-19 [1]
##    xml2                   1.2.0       2018-01-24 [1]
##    xtable                 1.8-3       2018-08-29 [1]
##    XVector                0.22.0      2018-10-30 [1]
##    yaml                   2.2.0       2018-07-25 [1]
##    zlibbioc               1.28.0      2018-10-30 [1]
##  source                         
##  Bioconductor                   
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.0)                 
##  CRAN (R 3.5.0)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  Bioconductor                   
##  Bioconductor                   
##  CRAN (R 3.5.0)                 
##  CRAN (R 3.5.0)                 
##  CRAN (R 3.5.0)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  Github (r-lib/devtools@8a3d2c2)
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.0)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  Bioconductor                   
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.2)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.2)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.0)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
##  CRAN (R 3.5.1)                 
##  Bioconductor                   
## 
## [1] C:/Users/s5125601/Documents/R/win-library/3.5
## [2] C:/Program Files/R/R-3.5.1/library
## 
##  D -- DLL MD5 mismatch, broken installation.</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation-instruction">1. Installation instruction</a></li>
      <li>
<a href="#a-simple-workflow-of-the-scgps">2. A simple workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#create-scgps-objects">2.1 Create scGPS objects</a></li>
      <li><a href="#run-prediction">2.2 Run prediction</a></li>
      <li><a href="#summarise-results">2.3 Summarise results</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Asenbouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
