<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Main prediction function applying the optimal ElasticNet and LDA models — predicting • scGPS</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Main prediction function applying the optimal ElasticNet and LDA models — predicting" />
<meta property="og:description" content="Predict a new mixed population after training the model for a
subpopulation in the first mixed population.
All subpopulations in the new target mixed population will be predicted, 
where each targeted subpopulation will have a transition score from the 
orginal subpopulation to the new subpopulation." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/IMB-Computational-Genomics-Lab/scGPS/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Main prediction function applying the optimal ElasticNet and LDA models</h1>
    <small class="dont-index">Source: <a href='https://github.com/IMB-Computational-Genomics-Lab/scGPS/blob/master/R/MainLassoLDATraining.R'><code>R/MainLassoLDATraining.R</code></a></small>
    <div class="hidden name"><code>predicting.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Predict a new mixed population after training the model for a
subpopulation in the first mixed population.
All subpopulations in the new target mixed population will be predicted, 
where each targeted subpopulation will have a transition score from the 
orginal subpopulation to the new subpopulation.</p>
    </div>

    <pre class="usage"><span class='fu'>predicting</span><span class='op'>(</span>
  listData <span class='op'>=</span> <span class='cn'>NULL</span>,
  cluster_mixedpop2 <span class='op'>=</span> <span class='cn'>NULL</span>,
  mixedpop2 <span class='op'>=</span> <span class='cn'>NULL</span>,
  out_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  standardize <span class='op'>=</span> <span class='cn'>TRUE</span>,
  LDA_run <span class='op'>=</span> <span class='cn'>FALSE</span>,
  c_selectID <span class='op'>=</span> <span class='cn'>NULL</span>,
  log_transform <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>listData</th>
      <td><p>a <code>list</code> object containing trained results for the
selected subpopulation in the first mixed population</p></td>
    </tr>
    <tr>
      <th>cluster_mixedpop2</th>
      <td><p>a vector of cluster assignment for mixedpop2</p></td>
    </tr>
    <tr>
      <th>mixedpop2</th>
      <td><p>a SingleCellExperiment object from the target
mixed population of importance, e.g. differentially expressed genes that are
most significant</p></td>
    </tr>
    <tr>
      <th>out_idx</th>
      <td><p>a number to specify index to write results into the list 
output. This is needed for running bootstrap.</p></td>
    </tr>
    <tr>
      <th>standardize</th>
      <td><p>a logical of whether to standardize the data</p></td>
    </tr>
    <tr>
      <th>LDA_run</th>
      <td><p>logical, if the LDA prediction is added to compare to 
ElasticNet, the LDA model needs to be trained from the training before
inputting to this prediction step</p></td>
    </tr>
    <tr>
      <th>c_selectID</th>
      <td><p>a number to specify the trained cluster used for prediction</p></td>
    </tr>
    <tr>
      <th>log_transform</th>
      <td><p>boolean whether log transform should be computed</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a <code>list</code> with prediction results written in to the index
<code>out_idx</code></p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Quan Nguyen, 2017-11-25</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>c_selectID</span><span class='op'>&lt;-</span><span class='fl'>1</span>
<span class='va'>out_idx</span><span class='op'>&lt;-</span><span class='fl'>1</span>
<span class='va'>day2</span> <span class='op'>&lt;-</span> <span class='va'>day_2_cardio_cell_sample</span>
<span class='va'>mixedpop1</span> <span class='op'>&lt;-</span><span class='fu'><a href='new_scGPS_object.html'>new_scGPS_object</a></span><span class='op'>(</span>ExpressionMatrix <span class='op'>=</span> <span class='va'>day2</span><span class='op'>$</span><span class='va'>dat2_counts</span>, 
    GeneMetadata <span class='op'>=</span> <span class='va'>day2</span><span class='op'>$</span><span class='va'>dat2geneInfo</span>, CellMetadata <span class='op'>=</span> <span class='va'>day2</span><span class='op'>$</span><span class='va'>dat2_clusters</span><span class='op'>)</span>
<span class='va'>day5</span> <span class='op'>&lt;-</span> <span class='va'>day_5_cardio_cell_sample</span>
<span class='va'>mixedpop2</span> <span class='op'>&lt;-</span><span class='fu'><a href='new_scGPS_object.html'>new_scGPS_object</a></span><span class='op'>(</span>ExpressionMatrix <span class='op'>=</span> <span class='va'>day5</span><span class='op'>$</span><span class='va'>dat5_counts</span>, 
    GeneMetadata <span class='op'>=</span> <span class='va'>day5</span><span class='op'>$</span><span class='va'>dat5geneInfo</span>, CellMetadata <span class='op'>=</span> <span class='va'>day5</span><span class='op'>$</span><span class='va'>dat5_clusters</span><span class='op'>)</span>
<span class='va'>genes</span> <span class='op'>&lt;-</span><span class='va'>training_gene_sample</span>
<span class='va'>genes</span> <span class='op'>&lt;-</span><span class='va'>genes</span><span class='op'>$</span><span class='va'>Merged_unique</span>
<span class='va'>listData</span>  <span class='op'>&lt;-</span> <span class='fu'><a href='training.html'>training</a></span><span class='op'>(</span><span class='va'>genes</span>, 
    cluster_mixedpop1 <span class='op'>=</span> <span class='fu'>colData</span><span class='op'>(</span><span class='va'>mixedpop1</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>, mixedpop1 <span class='op'>=</span> <span class='va'>mixedpop1</span>, 
    mixedpop2 <span class='op'>=</span> <span class='va'>mixedpop2</span>, <span class='va'>c_selectID</span>, listData <span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>, out_idx<span class='op'>=</span><span class='va'>out_idx</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Total 224 cells as source subpop</span></div><div class='output co'>#&gt; <span class='message'>Total 366 cells in remaining subpops</span></div><div class='output co'>#&gt; <span class='message'>subsampling 112 cells for training source subpop</span></div><div class='output co'>#&gt; <span class='message'>subsampling 112 cells in remaining subpops for training</span></div><div class='output co'>#&gt; <span class='message'>use 6 genes for training model</span></div><div class='output co'>#&gt; <span class='message'>use 6 genes 224 cells for testing model</span></div><div class='output co'>#&gt; <span class='message'>rename remaining subpops to 2_3</span></div><div class='output co'>#&gt; <span class='message'>there are 112 cells in class 2_3 and 112 cells in class 1</span></div><div class='output co'>#&gt; <span class='message'>removing 1 genes with no variance</span></div><div class='output co'>#&gt; <span class='message'>standardizing prediction/target dataset</span></div><div class='output co'>#&gt; <span class='message'>performning elasticnet model training...</span></div><div class='output co'>#&gt; <span class='message'>extracting deviance and best gene features...</span></div><div class='output co'>#&gt; <span class='message'>lambda min is at location 16</span></div><div class='output co'>#&gt; <span class='message'>the leave-out cells in the source subpop is 112</span></div><div class='output co'>#&gt; <span class='message'>use 112 target subpops cells for leave-out test set</span></div><div class='output co'>#&gt; <span class='message'>standardizing the leave-out target and source subpops...</span></div><div class='output co'>#&gt; <span class='message'>start ElasticNet prediction for estimating accuracy...</span></div><div class='output co'>#&gt; <span class='message'>evaluation accuracy ElasticNet 0.611111111111111</span></div><div class='input'><span class='va'>listData</span>  <span class='op'>&lt;-</span> <span class='fu'>predicting</span><span class='op'>(</span>listData <span class='op'>=</span><span class='va'>listData</span>,  mixedpop2 <span class='op'>=</span> <span class='va'>mixedpop2</span>, 
    out_idx<span class='op'>=</span><span class='va'>out_idx</span>, cluster_mixedpop2 <span class='op'>=</span> <span class='fu'>colData</span><span class='op'>(</span><span class='va'>mixedpop2</span><span class='op'>)</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>, 
    c_selectID <span class='op'>=</span> <span class='va'>c_selectID</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>standardizing target subpops before prediction...</span></div><div class='output co'>#&gt; <span class='message'>predicting from source to target subpop 1...</span></div><div class='output co'>#&gt; <span class='message'>number of cells in the target subpop 1 is 187</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the target data, but not in model genes is 4995</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the model present in the target data is 5</span></div><div class='output co'>#&gt; <span class='message'>There are 0 genes that are in the model, but not in target subpopulations</span></div><div class='output co'>#&gt; <span class='message'>the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in model </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in target </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>running elasticNet classification...</span></div><div class='output co'>#&gt; <span class='message'>class probability prediction ElasticNet for target subpop 1 is 3.74331550802139</span></div><div class='output co'>#&gt; <span class='message'>predicting from source to target subpop 2...</span></div><div class='output co'>#&gt; <span class='message'>number of cells in the target subpop 2 is 140</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the target data, but not in model genes is 4995</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the model present in the target data is 5</span></div><div class='output co'>#&gt; <span class='message'>There are 0 genes that are in the model, but not in target subpopulations</span></div><div class='output co'>#&gt; <span class='message'>the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in model </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in target </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>running elasticNet classification...</span></div><div class='output co'>#&gt; <span class='message'>class probability prediction ElasticNet for target subpop 2 is 20.7142857142857</span></div><div class='output co'>#&gt; <span class='message'>predicting from source to target subpop 3...</span></div><div class='output co'>#&gt; <span class='message'>number of cells in the target subpop 3 is 133</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the target data, but not in model genes is 4995</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the model present in the target data is 5</span></div><div class='output co'>#&gt; <span class='message'>There are 0 genes that are in the model, but not in target subpopulations</span></div><div class='output co'>#&gt; <span class='message'>the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in model </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in target </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>running elasticNet classification...</span></div><div class='output co'>#&gt; <span class='message'>class probability prediction ElasticNet for target subpop 3 is 7.5187969924812</span></div><div class='output co'>#&gt; <span class='message'>predicting from source to target subpop 4...</span></div><div class='output co'>#&gt; <span class='message'>number of cells in the target subpop 4 is 40</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the target data, but not in model genes is 4995</span></div><div class='output co'>#&gt; <span class='message'>Number of genes in the model present in the target data is 5</span></div><div class='output co'>#&gt; <span class='message'>There are 0 genes that are in the model, but not in target subpopulations</span></div><div class='output co'>#&gt; <span class='message'>the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in model </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>first 10 genes in target </span></div><div class='output co'>#&gt; <span class='message'>GATA4VIMSNAI2GJA1TMEM88</span></div><div class='output co'>#&gt; <span class='message'>running elasticNet classification...</span></div><div class='output co'>#&gt; <span class='message'>class probability prediction ElasticNet for target subpop 4 is 17.5</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Senabouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


