---
title: "BenchMarking Camp Dataset"
author: "Quan and Michael"
date: "7/9/2018"
output: html_document
---


#using camp dataset
```{r}
#Load everyting for scGPS

#Github Installation
devtools::install_github("IMB-Computational-Genomics-Lab/scGPS")

#Manual Installation with scGPS in working directory
#devtools::load_all()

library(scater)
library(scran)

#Retrieve the dataset
camp_dat <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/camp2.rds"))

#Remove the "unknown" cell type
to_keep <- which(!(colData(camp_dat)$cell_type1 %in% c("Unknown")))
camp_dat <- camp_dat[, to_keep]

#Creates FPKM assay
logs <- (as.matrix(assays(camp_dat)[["logcounts"]]))
fpkm(camp_dat) <- apply(logs, c(1,2), function(x) 2^x-1)

#Find the genes with all zero entries and remove
keep_features <- rowSums(fpkm(camp_dat) > 0) > 0
camp_dat <- camp_dat[keep_features, ]

#Use Scran normalisation
computeSumFactors(camp_dat, assay.type = "fpkm")
normalize(camp_dat, exprs_values = "fpkm")

#Remove the spikes
is.spike <-grepl("^ERCC", rownames(camp_dat))
table(is.spike)
camp_dat <- camp_dat[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Extract the needed variables for scGPS
camp_dat_exprs <- assays(camp_dat)[["logcounts"]]
camp_dat_cellnames <- colnames(camp_dat)
camp_dat_cellnames <- data.frame("cellBarcodes" = camp_dat_cellnames)
camp_dat_GeneMetaData <- rownames(camp_dat)
camp_dat_GeneMetaData <- data.frame("GeneSymbol" = camp_dat_GeneMetaData)

#Store Data in scGPS format
mixedpop <-NewscGPS_SME(ExpressionMatrix = camp_dat_exprs, GeneMetadata = camp_dat_GeneMetaData, CellMetadata = camp_dat_cellnames)

#Cluster and plot data using SCORE
CORE_cluster_bagging <- CORE_scGPS_bagging(mixedpop, remove_outlier = c(0), PCA=FALSE, bagging_run = 20, subsample_proportion = .8)
plot_CORE(CORE_cluster_bagging$tree, list_clusters = CORE_cluster_bagging$Cluster)
plot_optimal_CORE(original_tree= CORE_cluster_bagging$tree, optimal_cluster = unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index]), shift = -100)

#Stop the time here
end_time <- Sys.time()
time_difference_SCORE <- end_time - start_time

#Make a dataframe with the results we want to examine
cell_types1 <- colData(camp_dat)$cell_type1
label_list <- unlist(camp_dat_cellnames$cellBarcodes)
cluster_list <- unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index])
compare_frame <- data.frame("Gene_label" = label_list, "type1" = cell_types1,"cluster" = cluster_list)

#Find the Adjusted Rand Index
AdjustedRandIndex_SCORE <- mclust::adjustedRandIndex(compare_frame$type1, compare_frame$cluster)
HighResRand <- mclust::adjustedRandIndex(compare_frame$type1, unlist(CORE_cluster_bagging$Cluster[1]))

#Store the estimated k from the bagging runs
estimated_k_SCORE <- CORE_cluster_bagging$optimalMax

#Remove unwanted data
rm(list = setdiff(ls(), c("AdjustedRandIndex_SCORE", "time_difference_SCORE", "estimated_k_SCORE", "HighResRand")))

```



#Using SC3 to test the camp dataset
```{r}
#Load everything for SC3
library(SC3)
library(scater)

#Retrieve the dataset
camp_dat <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/camp2.rds"))

#Remove the "unknown" cell type
to_keep <- which(!(colData(sce)$cell_type1 %in% c("Unknown")))
sce <- sce[, to_keep]

#Creates FPKM assay
logs <- (as.matrix(assays(sce)[["logcounts"]]))
fpkm(sce) <- apply(logs, c(1,2), function(x) 2^x-1)
counts(sce) <- fpkm(sce)

#Find the genes with all zero entries and remove
keep_features <- rowSums(fpkm(sce) > 0) > 0
sce <- sce[keep_features, ]

#Remove the spikes
is.spike <-grepl("^ERCC", rownames(sce))
sce <- sce[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Run sc3 with an estimation for k
sce <- sc3_prepare(sce, n_cores = 1, gene_filter = TRUE)
sce <- sc3_estimate_k(sce)
SC3_k_estimate <- as.integer(unlist(metadata(sce)$sc3$k_estimation))
sce <- sc3(sce, ks = SC3_k_estimate, biology = FALSE, gene_filter = TRUE, n_cores = 1)

#Stop the time here
end_time <- Sys.time()
time_difference_SC3 <- end_time - start_time

#Make a dataframe with the results we want to examine
cell_types1 <- colData(sce)$cell_type1
label_list <- rownames(colData(sce))
cluster_list <- as.numeric(colData(sce)[, paste0("sc3_", SC3_k_estimate, "_clusters")])
compare_frame <- data.frame("Gene_label" = label_list, "type1" = cell_types1, "cluster" = cluster_list)

#Save the results to a spreadsheet (requires perl)
library("WriteXLS")
WriteXLS(compare_frame, ExcelFileName = "C:/Users/s5125601/Desktop/SC3.xlsx")

#Find the Adjusted Rand Index
AdjustedRandIndex_SC3 <- mclust::adjustedRandIndex(compare_frame$type1, compare_frame$cluster)

#Remove unwanted data
rm(list = setdiff(ls(), c("AdjustedRandIndex_SC3", "time_difference_SC3", "SC3_k_estimate")))

```




