---
title: "benchMark"
author: "Quan and Michael"
date: "7/9/2018"
output: html_document
---

#Benchmarking SCORE with the koh dataset
```{r}
#Load everyting for scGPS

#Github Installation
devtools::install_github("IMB-Computational-Genomics-Lab/scGPS")

#Manual Installation with scGPS in working directory
#devtools::load_all()

library("MultiAssayExperiment")
library("scater")
library("scran")

#Retrieve the dataset
koh_dat <- readRDS(url("http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/SRP073808.rds"))

#Exract the gene-level length-scaled TPMs
koh_expr <- assays(experiments(koh_dat)[["gene"]])[["count_lstpm"]]

#Extract the phenotype data.
phn <- colData(koh_dat)
phn$phenoid <- as.character(phn$LibraryName)
table(phn$phenoid)

#Create single cell experiment
stopifnot(all(colnames(koh_expr) == rownames(phn)))
SCE <- SingleCellExperiment(
  assays = list(counts = koh_expr),
  colData = phn
)

#Remove features with no gene expression
keep_features <- rowSums(counts(SCE) > 0) > 0
SCE <- SCE[keep_features, ]

#Use scran normalisation
SCE <- computeSumFactors(SCE)
SCE <- normalise(SCE, exprs_values = "counts", return_log = TRUE) 

#Create a count per million assay
cpm(SCE) <- calculateCPM(SCE)

#Remove spikes
is.spike <-grepl("^ERCC", rownames(SCE))
SCE <- SCE[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Extract the needed variables
koh_dat_exprs <- assays(SCE)[["logcounts"]]
koh_dat_cellnames <- colnames(SCE)
koh_dat_cellnames <- data.frame("cellBarcodes" = koh_dat_cellnames)
koh_dat_GeneMetaData <- rownames(SCE)
koh_dat_GeneMetaData <- data.frame("GeneSymbol" = koh_dat_GeneMetaData)

#Store Data in scGPS format
mixedpop <-NewscGPS_SME(ExpressionMatrix = koh_dat_exprs, GeneMetadata = koh_dat_GeneMetaData, CellMetadata = koh_dat_cellnames)

#Cluster and plot data using SCORE
CORE_cluster_bagging <- CORE_scGPS_bagging(mixedpop, remove_outlier = c(0), PCA=FALSE, bagging_run = 20, subsample_proportion = .8)
plot_CORE(CORE_cluster_bagging$tree, list_clusters = CORE_cluster_bagging$Cluster)
plot_optimal_CORE(original_tree= CORE_cluster_bagging$tree, optimal_cluster = unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index]), shift = -100)

#Stop the time here
end_time <- Sys.time()
time_difference_SCORE <- end_time - start_time

#Find data needed for comparisons and store in data frame
phenoid_list <- unlist(colData(SCE)$phenoid)
label_list <- unlist(koh_dat_cellnames$cellBarcodes)
cluster_list <- unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index])
compare_frame <- data.frame("Gene_label" = label_list, "phenoid_list" = phenoid_list, "cluster" = cluster_list)

#Find the adjusted rand index
AdjustedRandIndex_SCORE <- mclust::adjustedRandIndex(compare_frame$phenoid_list, compare_frame$cluster)
HighResRand <- mclust::adjustedRandIndex(compare_frame$phenoid_list, unlist(CORE_cluster_bagging$Cluster[1]))
estimated_k_SCORE <- CORE_cluster_bagging$optimalMax

#Remove clutter from the environment
rm(list = setdiff(ls(), c("AdjustedRandIndex_SCORE", "time_difference_SCORE", "estimated_k_SCORE", "HighResRand")))

```



#Benchmarking SC3 with the Koh dataset
```{r}
#Load everything for SC3
library("SC3")
library("MultiAssayExperiment")
library("scater")

#Get the path to the dataset
path_to_data = "/Users/s5125601/Documents/BenchMarking/Datasets/KohDataset/"
#Read the data

#Retrieve the dataset
koh_dat <- readRDS(url("http://imlspenticton.uzh.ch/robinson_lab/conquer/data-mae/SRP073808.rds"))

#Exract the gene-level length-scaled TPMs
koh_expr <- assays(experiments(koh_dat)[["gene"]])[["count_lstpm"]]

#Extract th phenotype data.
phn <- colData(koh_dat)
phn$phenoid <- as.character(phn$LibraryName)

#Create single cell experiment
stopifnot(all(colnames(koh_expr) == rownames(phn)))
SCE <- SingleCellExperiment(
  assays = list(counts = koh_expr),
  colData = phn
)

#Find the genes with all zero entries and remove
keep_features <- rowSums(counts(SCE) > 0) > 0
SCE <- SCE[keep_features, ]

#Create "logcounts" assays
SCE <- normalise(SCE, exprs_values = "counts", return_log = TRUE)

#Remove the spikes
is.spike <-grepl("^ERCC", rownames(SCE))
SCE <- SCE[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Run sc3 with an estimation for k
rowData(SCE)$feature_symbol <- rownames(counts(SCE))
SCE <- sc3_prepare(SCE, n_cores = 1)
SCE <- sc3_estimate_k(SCE)
SC3_k_estimate <- as.integer(unlist(metadata(SCE)$sc3$k_estimation))
SCE <- sc3(SCE, ks = SC3_k_estimate, biology = FALSE, gene_filter = TRUE, n_cores = 1)

#Stop the time here
end_time <- Sys.time()
time_difference_SC3 <- end_time - start_time

#Make a dataframe with the results we want to examine
phenoid_list <- colData(SCE)$phenoid
label_list <- rownames(colData(SCE))
cluster_list <- as.numeric(colData(SCE)[, paste0("sc3_", SC3_k_estimate, "_clusters")])
compare_frame <- data.frame("Gene_label" = label_list, "phenoid_list" = phenoid_list, "cluster" = cluster_list)

#Find the Adjusted Rand Index
AdjustedRandIndex_SC3 <- mclust::adjustedRandIndex(compare_frame$phenoid_list, compare_frame$cluster)

#Remove unwanted data
rm(list = setdiff(ls(), c("AdjustedRandIndex_SC3", "time_difference_SC3", "SC3_k_estimate")))

```




