---
title: "benchMark"
author: "Quan and Michael"
date: "1/3/2018"
output: html_document
---

#Benchmarking SCORE with the baron dataset
```{r}
#Load everyting for scGPS

#Github Installation
devtools::install_github("IMB-Computational-Genomics-Lab/scGPS")
library(scGPS)

#Manual Installation with scGPS in working directory
#devtools::load_all()

library(SingleCellExperiment)
library(scran)
library(scater)

#Retrieve dataset
baron_dat <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/baron-human.rds"))

#Find the genes with all zero entries and remove
keep_features <- rowSums(counts(baron_dat) > 0) > 0
baron_dat <- baron_dat[keep_features, ]

#Use Scran normalisation
baron_dat <- computeSumFactors(baron_dat)
baron_dat <- normalize(baron_dat)

#Used to add the counts per million
cpm(baron_dat) <- calculateCPM(baron_dat)

#Remove the spikes
is.spike <-grepl("^ERCC", rownames(baron_dat))
baron_dat <- baron_dat[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Extract the needed variables
baron_dat_exprs <- assays(baron_dat)[["logcounts"]]
baron_dat_cellnames <- colnames(baron_dat)
baron_dat_cellnames <- data.frame("cellBarcodes" = baron_dat_cellnames)
baron_dat_GeneMetaData <- rownames(baron_dat)
baron_dat_GeneMetaData <- data.frame("GeneSymbol" = baron_dat_GeneMetaData)

#Store Data in scGPS format
mixedpop <- new_summarized_scGPS_object(ExpressionMatrix = baron_dat_exprs, GeneMetadata = baron_dat_GeneMetaData, CellMetadata = baron_dat_cellnames)

#Cluster and plot data using SCORE
CORE_cluster_bagging <- CORE_bagging(mixedpop, remove_outlier = c(0), PCA=FALSE, bagging_run = 20, subsample_proportion = .8)
plot_CORE(CORE_cluster_bagging$tree, list_clusters = CORE_cluster_bagging$Cluster)
plot_optimal_CORE(original_tree= CORE_cluster_bagging$tree, optimal_cluster = unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index]), shift = -100)

#Stop the time here and record
end_time <- Sys.time()
time_difference_SCORE <- end_time - start_time

#Make a dataframe with the results we want to examine
cell_types1 <- colData(baron_dat)$cell_type1
label_list <- unlist(baron_dat_cellnames$cellBarcodes)
cluster_list <- unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index])
compare_frame <- data.frame("Gene_label" = label_list, "type1" = cell_types1,"cluster" = cluster_list)

#Find the Adjusted Rand Index
AdjustedRandIndex_SCORE <- mclust::adjustedRandIndex(compare_frame$type1, compare_frame$cluster)
HighResRandIndex_SCORE <- mclust::adjustedRandIndex(compare_frame$type1, unlist(CORE_cluster_bagging$Cluster[1]))


#Store the estimated k from the bagging runs
estimated_k_SCORE <- CORE_cluster_bagging$optimalMax

#Remove Unwanted data and save
rm(list = setdiff(ls(), c("AdjustedRandIndex_SCORE", "time_difference_SCORE", "estimated_k_SCORE", "HighResRandIndex_SCORE")))
```


#Benchmarking SC3 with the baron dataset
```{r}
#Load everything for SC3
library(SC3)
library(scater)

#Retrieve the dataset
sce <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/baron-human.rds"))

#Find the genes with all zero entries and remove
keep_features <- rowSums(counts(sce) > 0) > 0
sce <- sce[keep_features, ]

#Remove the spikes
is.spike <-grepl("^ERCC", rownames(sce))
table(is.spike)
sce <- sce[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Estimate the number of clusters to use
sce <- sc3_prepare(sce, n_cores = 1, gene_filter = TRUE)
sce <- sc3_estimate_k(sce)
SC3_k_estimate <- as.integer(unlist(metadata(sce)$sc3$k_estimation))

#Run sc3 with start = 50 as cells > 2000
SC3result <- sc3(sce, ks = SC3_k_estimate, biology = FALSE, gene_filter = TRUE, n_cores = 1, kmeans_nstart = 50)

#Stop the time here
end_time <- Sys.time()
time_difference_SC3 <- end_time - start_time

#Make a dataframe with the results we want to examine
cell_types1 <- colData(SC3result)$cell_type1
label_list <- rownames(colData(SC3result))
cluster_list <- as.numeric(colData(SC3result)[, paste0("sc3_", SC3_k_estimate, "_clusters")])
compare_frame <- data.frame("Gene_label" = label_list, "type1" = cell_types1, "cluster" = cluster_list)

#Find the Adjusted Rand Index
AdjustedRandIndex_SC3 <- mclust::adjustedRandIndex(compare_frame$type1, compare_frame$cluster)

#Remove unwanted data and save
rm(list = setdiff(ls(), c("AdjustedRandIndex_SC3", "time_difference_SC3", "SC3_k_estimate")))
```


