<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Homepage • scGPS</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>



  <link href="extra.css" rel="stylesheet">
  <script src="extra.js"></script>
<meta property="og:title" content="Homepage" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">scGPS</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.9.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Homepage</h1>
    </div>


<hr>
<p>title: “scGPS introduction” author: “Quan Nguyen” date: “<code>r Sys.Date()</code>” output: pdf_document: toc: true highlight: tango html_document: standalone: true highlight: tango self-contained: true keep_md: true toc: true vignette: &gt; %\VignetteIndexEntry{Multi-format vignettes} \usepackage[utf8]{inputenc} %\VignetteEngine{knitr::multiformat} —</p>
<p>```{r setup, out.width = ‘100%’, include = FALSE} knitr::opts_chunk$set( collapse = TRUE, comment = “#&gt;”, fig.path = “man/figures/” )</p>
<div id="homepage" class="section level1">

<p>knitr::include_graphics(“./docs/reference/figures/packagePlan.png”)</p>
</div>
<div id="knitropts_chunkset" class="section level1">
<h1 class="hasAnchor">
<a href="#knitropts_chunkset" class="anchor"></a>knitr::opts_chunk$set(</h1>
</div>
<div id="fig-path--docsreferencefigures" class="section level1">
<h1 class="hasAnchor">
<a href="#fig-path--docsreferencefigures" class="anchor"></a>fig.path = “./docs/reference/figures/”</h1>
</div>
<div id="section" class="section level1">
<h1 class="hasAnchor">
<a href="#section" class="anchor"></a>)</h1>
</div>
<div id="owd-setwdusersquan-nguyendocumentspowell_group_macquanallcodesscgpsvignettes" class="section level1">
<h1 class="hasAnchor">
<a href="#owd-setwdusersquan-nguyendocumentspowell_group_macquanallcodesscgpsvignettes" class="anchor"></a>owd = setwd(‘/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/vignettes/’)</h1>
</div>
<div id="setwdowd" class="section level1">
<h1 class="hasAnchor">
<a href="#setwdowd" class="anchor"></a>setwd(owd)</h1>
<p>```</p>
</div>
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<p>```{r, eval = FALSE}</p>
</div>
<div id="prior-to-installing-scgps-you-need-to-install-the-summarizedexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#prior-to-installing-scgps-you-need-to-install-the-summarizedexperiment" class="anchor"></a>Prior to installing scGPS you need to install the SummarizedExperiment</h1>
</div>
<div id="bioconductor-package-as-the-following" class="section level1">
<h1 class="hasAnchor">
<a href="#bioconductor-package-as-the-following" class="anchor"></a>bioconductor package as the following</h1>
</div>
<div id="sourcehttpsbioconductor-orgbioclite-r-bioclitesummarizedexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#sourcehttpsbioconductor-orgbioclite-r-bioclitesummarizedexperiment" class="anchor"></a>source(‘<a href="https://bioconductor.org/biocLite.R" class="uri">https://bioconductor.org/biocLite.R</a>’) biocLite(‘SummarizedExperiment’)</h1>
</div>
<div id="to-install-scgps-from-github-depending-on-the-configuration-of-the-local" class="section level1">
<h1 class="hasAnchor">
<a href="#to-install-scgps-from-github-depending-on-the-configuration-of-the-local" class="anchor"></a>To install scGPS from github (Depending on the configuration of the local</h1>
</div>
<div id="computer-or-hpc-possible-custom-c-compilation-may-be-required---see" class="section level1">
<h1 class="hasAnchor">
<a href="#computer-or-hpc-possible-custom-c-compilation-may-be-required---see" class="anchor"></a>computer or HPC, possible custom C++ compilation may be required - see</h1>
</div>
<div id="installation-trouble-shootings-below" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-trouble-shootings-below" class="anchor"></a>installation trouble-shootings below)</h1>
<p>devtools::install_github(“IMB-Computational-Genomics-Lab/scGPS”)</p>
</div>
<div id="for-c-compilation-trouble-shooting-manual-download-and-installation-can-be" class="section level1">
<h1 class="hasAnchor">
<a href="#for-c-compilation-trouble-shooting-manual-download-and-installation-can-be" class="anchor"></a>for C++ compilation trouble-shooting, manual download and installation can be</h1>
</div>
<div id="done-from-github" class="section level1">
<h1 class="hasAnchor">
<a href="#done-from-github" class="anchor"></a>done from github</h1>
<p>git clone <a href="https://github.com/IMB-Computational-Genomics-Lab/scGPS" class="uri">https://github.com/IMB-Computational-Genomics-Lab/scGPS</a></p>
</div>
<div id="then-check-in-scgpssrc-if-any-of-the-precompiled-e-g--those-with--so-and" class="section level1">
<h1 class="hasAnchor">
<a href="#then-check-in-scgpssrc-if-any-of-the-precompiled-e-g--those-with--so-and" class="anchor"></a>then check in scGPS/src if any of the precompiled (e.g. those with *.so and</h1>
</div>
<div id="o-files-exist-and-delete-them-before-recompiling" class="section level1">
<h1 class="hasAnchor">
<a href="#o-files-exist-and-delete-them-before-recompiling" class="anchor"></a>*.o) files exist and delete them before recompiling</h1>
</div>
<div id="create-a-makevars-file-in-the-scgpssrc-with-one-line-pkg_libs" class="section level1">
<h1 class="hasAnchor">
<a href="#create-a-makevars-file-in-the-scgpssrc-with-one-line-pkg_libs" class="anchor"></a>create a Makevars file in the scGPS/src with one line: PKG_LIBS =</h1>
</div>
<div id="lapack_libs-blas_libs-flibs" class="section level1">
<h1 class="hasAnchor">
<a href="#lapack_libs-blas_libs-flibs" class="anchor"></a>$(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)</h1>
</div>
<div id="then-with-the-scgps-as-the-r-working-directory-manually-recompile-scgps-in-r" class="section level1">
<h1 class="hasAnchor">
<a href="#then-with-the-scgps-as-the-r-working-directory-manually-recompile-scgps-in-r" class="anchor"></a>then with the scGPS as the R working directory, manually recompile scGPS in R</h1>
</div>
<div id="using-devtools-to-load-and-install-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#using-devtools-to-load-and-install-functions" class="anchor"></a>using devtools to load and install functions</h1>
<p>devtools::document()</p>
</div>
<div id="update-the-namespace-using-the-update_namespace-sh" class="section level1">
<h1 class="hasAnchor">
<a href="#update-the-namespace-using-the-update_namespace-sh" class="anchor"></a>update the NAMESPACE using the update_NAMESPACE.sh</h1>
<p>sh update_NAMESPACE.sh</p>
</div>
<div id="for-window-system-to-update-the-namespace-copy-and-paste-the-content-of-the-file-namespace_toadd_cpp_links-to-end-of-the-file-namespace" class="section level1">
<h1 class="hasAnchor">
<a href="#for-window-system-to-update-the-namespace-copy-and-paste-the-content-of-the-file-namespace_toadd_cpp_links-to-end-of-the-file-namespace" class="anchor"></a>for window system, to update the NAMESPACE: copy and paste the content of the file NAMESPACE_toAdd_cpp_links to end of the file NAMESPACE</h1>
</div>
<div id="load-the-package-to-the-workspace" class="section level1">
<h1 class="hasAnchor">
<a href="#load-the-package-to-the-workspace" class="anchor"></a>load the package to the workspace</h1>
<p>devtools::load_all() ```</p>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given a mixed population with known subpopulations, estimate transition scores between these subpopulation</em></p>
<div id="setup-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#setup-scgps-objects" class="anchor"></a>2.1 Setup scGPS objects</h2>
<p>```{r, warning = FALSE, message = FALSE}</p>
</div>
</div>
<div id="load-mixed-population-1-loaded-from-sample1-dataset-named-it-as-day2" class="section level1">
<h1 class="hasAnchor">
<a href="#load-mixed-population-1-loaded-from-sample1-dataset-named-it-as-day2" class="anchor"></a>load mixed population 1 (loaded from sample1 dataset, named it as day2)</h1>
<p>devtools::load_all()</p>
<p>day2 &lt;- sample1</p>
<p>mixedpop1 &lt;- NewscGPS_SME(ExpressionMatrix = day2$dat2_counts, GeneMetadata = day2$dat2geneInfo, CellMetadata = day2$dat2_clusters)</p>
</div>
<div id="load-mixed-population-2-loaded-from-sample2-dataset-named-it-as-day5" class="section level1">
<h1 class="hasAnchor">
<a href="#load-mixed-population-2-loaded-from-sample2-dataset-named-it-as-day5" class="anchor"></a>load mixed population 2 (loaded from sample2 dataset, named it as day5)</h1>
<p>day5 &lt;- sample2 mixedpop2 &lt;- NewscGPS_SME(ExpressionMatrix = day5$dat5_counts, GeneMetadata = day5$dat5geneInfo, CellMetadata = day5$dat5_clusters)</p>
</div>
<div id="load-gene-list-this-can-be-any-lists-of-user-selected-genes" class="section level1">
<h1 class="hasAnchor">
<a href="#load-gene-list-this-can-be-any-lists-of-user-selected-genes" class="anchor"></a>load gene list (this can be any lists of user selected genes)</h1>
<p>genes &lt;- GeneList genes &lt;- genes$Merged_unique</p>
</div>
<div id="select-a-subpopulation" class="section level1">
<h1 class="hasAnchor">
<a href="#select-a-subpopulation" class="anchor"></a>select a subpopulation</h1>
<p>c_selectID &lt;- 1</p>
<p><code>##2.2 Run predictions</code>{r}</p>
</div>
<div id="run-the-test-bootstrap" class="section level1">
<h1 class="hasAnchor">
<a href="#run-the-test-bootstrap" class="anchor"></a>run the test bootstrap</h1>
<p>LSOLDA_dat &lt;- bootstrap_scGPS(nboots = 2, mixedpop1 = mixedpop1, mixedpop2 = mixedpop2, genes = genes, c_selectID, listData = list())</p>
<p><code>##2.3 Summarise results</code>{r}</p>
<p>LSOLDA_dat &lt;- bootstrap_scGPS(nboots = 2, mixedpop1 = mixedpop1, mixedpop2 = mixedpop2, genes = genes, c_selectID, listData = list())</p>
</div>
<div id="display-the-list-of-result-information-in-the-lasolda_dat-object" class="section level1">
<h1 class="hasAnchor">
<a href="#display-the-list-of-result-information-in-the-lasolda_dat-object" class="anchor"></a>display the list of result information in the LASOLDA_dat object</h1>
<p>names(LSOLDA_dat) LSOLDA_dat$LassoPredict LSOLDA_dat$LDAPredict</p>
</div>
<div id="summary-results-lda" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-results-lda" class="anchor"></a>summary results LDA</h1>
<p>summary_prediction_lda(LSOLDA_dat = LSOLDA_dat, nPredSubpop = 4)</p>
</div>
<div id="summary-results-lasso" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-results-lasso" class="anchor"></a>summary results Lasso</h1>
<p>summary_prediction_lasso(LSOLDA_dat = LSOLDA_dat, nPredSubpop = 4)</p>
</div>
<div id="summary-deviance" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-deviance" class="anchor"></a>summary deviance</h1>
<p>summary_deviance(object = LSOLDA_dat) ```</p>
</div>
<div id="a-complete-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-complete-workflow-of-the-scgps" class="anchor"></a>3. A complete workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given an unknown mixed population, find clusters and estimate relationship between clusters</em></p>
<div id="identify-clusters-in-a-using-core" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-using-core" class="anchor"></a>3.1 Identify clusters in a using CORE</h2>
<p><em>(skip this step if clusters are known)</em> ```{r, warning = FALSE, message = FALSE}</p>
</div>
</div>
<div id="lets-find-clustering-information-in-an-expresion-data" class="section level1">
<h1 class="hasAnchor">
<a href="#lets-find-clustering-information-in-an-expresion-data" class="anchor"></a>Let’s find clustering information in an expresion data</h1>
<p>day5 &lt;- sample2 cellnames &lt;- colnames(day5$dat5_counts) cluster &lt;-day5$dat5_clusters cellnames &lt;-data.frame(“Cluster”=cluster, “cellBarcodes” = cellnames) mixedpop2 &lt;-NewscGPS_SME(ExpressionMatrix = day5$dat5_counts, GeneMetadata = day5$dat5geneInfo, CellMetadata = cellnames )</p>
<p>CORE_cluster &lt;- CORE_scGPS(mixedpop2, remove_outlier = c(0), PCA=FALSE)</p>
<p><code>##3.2 Visualise all cluster results in all iterations</code>{r}</p>
</div>
<div id="plot-with-default-colors" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-with-default-colors" class="anchor"></a>plot with default colors</h1>
<p>plot_CORE(CORE_cluster$tree, CORE_cluster$Cluster)</p>
</div>
<div id="lets-find-the-core-clusters" class="section level1">
<h1 class="hasAnchor">
<a href="#lets-find-the-core-clusters" class="anchor"></a>let’s find the CORE clusters</h1>
<p>CORE_cluster &lt;- CORE_scGPS(mixedpop2, remove_outlier = c(0), PCA=FALSE)</p>
</div>
<div id="lets-plot-all-clusters" class="section level1">
<h1 class="hasAnchor">
<a href="#lets-plot-all-clusters" class="anchor"></a>let’s plot all clusters</h1>
<p>plot_CORE(CORE_cluster$tree, CORE_cluster$Cluster)</p>
</div>
<div id="you-can-customise-the-cluster-color-bars-provide-color_branch-values" class="section level1">
<h1 class="hasAnchor">
<a href="#you-can-customise-the-cluster-color-bars-provide-color_branch-values" class="anchor"></a>you can customise the cluster color bars (provide color_branch values)</h1>
<p>plot_CORE(CORE_cluster$tree, CORE_cluster$Cluster, color_branch = c(“#208eb7”, “#6ce9d3”, “#1c5e39”, “#8fca40”, “#154975”, “#b1c8eb”))</p>
</div>
<div id="you-can-customise-the-cluster-color-bars-provide-color_branch-values-1" class="section level1">
<h1 class="hasAnchor">
<a href="#you-can-customise-the-cluster-color-bars-provide-color_branch-values-1" class="anchor"></a>you can customise the cluster color bars (provide color_branch values)</h1>
<p>plot_CORE(CORE_cluster$tree, CORE_cluster$Cluster, color_branch = c(“#208eb7”, “#6ce9d3”, “#1c5e39”, “#8fca40”, “#154975”, “#b1c8eb”)) ```</p>
<div id="plot-the-optimal-clustering-result" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-optimal-clustering-result" class="anchor"></a>3.3 Plot the optimal clustering result</h2>
<pre class="{r}"><code>#extract optimal index identified by CORE_scGPS
optimal_index = which(CORE_cluster$optimalClust$KeyStats$Height == CORE_cluster$optimalClust$OptimalRes)

#plot the optimal result
plot_optimal_CORE(original_tree= CORE_cluster$tree, optimal_cluster = unlist(CORE_cluster$Cluster[optimal_index]), shift = -100)</code></pre>
</div>
<div id="compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="anchor"></a>3.4 Compare clustering results with other dimensional reduction methods (e.g., CIDR)</h2>
<pre class="{r}"><code>library(cidr)
t &lt;- CIDR_scGPS(expression.matrix=assay(mixedpop2))
p2 &lt;-plotReduced_scGPS(t, color_fac = factor(colData(mixedpop2)[,1]),palletes =1:length(unique(colData(mixedpop2)[,1])))
p2</code></pre>
</div>
<div id="find-gene-markers-and-annotate-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#find-gene-markers-and-annotate-clusters" class="anchor"></a>3.5 Find gene markers and annotate clusters</h2>
<p>```{r, warning = FALSE, message = FALSE}</p>
</div>
</div>
<div id="load-gene-list-this-can-be-any-lists-of-user-selected-genes-1" class="section level1">
<h1 class="hasAnchor">
<a href="#load-gene-list-this-can-be-any-lists-of-user-selected-genes-1" class="anchor"></a>load gene list (this can be any lists of user-selected genes)</h1>
<p>genes &lt;-GeneList genes &lt;-genes$Merged_unique</p>
</div>
<div id="the-gene-list-can-also-be-objectively-identified-by-differential-expression-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#the-gene-list-can-also-be-objectively-identified-by-differential-expression-analysis" class="anchor"></a>the gene list can also be objectively identified by differential expression analysis</h1>
</div>
<div id="cluster-information-is-requied-for-findmarkers_scgps--here-we-use-core-results-" class="section level1">
<h1 class="hasAnchor">
<a href="#cluster-information-is-requied-for-findmarkers_scgps--here-we-use-core-results-" class="anchor"></a>cluster information is requied for findMarkers_scGPS. Here, we use CORE results.</h1>
<p>Optimal_index &lt;- which( CORE_cluster$optimalClust$KeyStats$Height == CORE_cluster$optimalClust$OptimalRes) colData(mixedpop2)[,1] &lt;- unlist(CORE_cluster$Cluster[[Optimal_index]])</p>
<p>suppressMessages(library(locfit)) suppressMessages(library(DESeq))</p>
<p>DEgenes &lt;- findMarkers_scGPS(expression_matrix=assay(mixedpop2), cluster = colData(mixedpop2)[,1], selected_cluster=unique(colData(mixedpop2)[,1]))</p>
</div>
<div id="the-output-contains-dataframes-for-each-cluster-" class="section level1">
<h1 class="hasAnchor">
<a href="#the-output-contains-dataframes-for-each-cluster-" class="anchor"></a>the output contains dataframes for each cluster.</h1>
</div>
<div id="the-data-frame-contains-all-genes-sorted-by-p-values" class="section level1">
<h1 class="hasAnchor">
<a href="#the-data-frame-contains-all-genes-sorted-by-p-values" class="anchor"></a>the data frame contains all genes, sorted by p-values</h1>
<p>names(DEgenes)</p>
</div>
<div id="you-can-annotate-the-identified-clusters" class="section level1">
<h1 class="hasAnchor">
<a href="#you-can-annotate-the-identified-clusters" class="anchor"></a>you can annotate the identified clusters</h1>
<p>DEgeneList_3vsOthers &lt;- DEgenes$DE_Subpop3vsRemaining$id</p>
</div>
<div id="users-need-to-check-the-format-of-the-gene-input-to-make-sure-they-are-consistent-to" class="section level1">
<h1 class="hasAnchor">
<a href="#users-need-to-check-the-format-of-the-gene-input-to-make-sure-they-are-consistent-to" class="anchor"></a>users need to check the format of the gene input to make sure they are consistent to</h1>
</div>
<div id="the-gene-names-in-the-expression-matrix" class="section level1">
<h1 class="hasAnchor">
<a href="#the-gene-names-in-the-expression-matrix" class="anchor"></a>the gene names in the expression matrix</h1>
<p>DEgeneList_3vsOthers &lt;-gsub(“_.*“,”“, DEgeneList_3vsOthers )</p>
</div>
<div id="the-following-command-saves-the-file-pathwayenrichment-xlsx-to-the-working-dir" class="section level1">
<h1 class="hasAnchor">
<a href="#the-following-command-saves-the-file-pathwayenrichment-xlsx-to-the-working-dir" class="anchor"></a>the following command saves the file “PathwayEnrichment.xlsx” to the working dir</h1>
</div>
<div id="use-500-top-de-genes" class="section level1">
<h1 class="hasAnchor">
<a href="#use-500-top-de-genes" class="anchor"></a>use 500 top DE genes</h1>
<p>suppressMessages(library(DOSE)) suppressMessages(library(ReactomePA)) suppressMessages(library(clusterProfiler))</p>
<p>enrichment_test &lt;- annotate_scGPS(DEgeneList_3vsOthers[1:500], pvalueCutoff=0.05, gene_symbol=TRUE,output_filename = “PathwayEnrichment.xlsx”, output_path = NULL )</p>
</div>
<div id="the-enrichment-outputs-can-be-displayed-by-running" class="section level1">
<h1 class="hasAnchor">
<a href="#the-enrichment-outputs-can-be-displayed-by-running" class="anchor"></a>the enrichment outputs can be displayed by running</h1>
<p>dotplot(enrichment_test, showCategory=15)</p>
<p>```</p>
</div>
<div id="relationship-between-clusters-within-one-sample-or-between-two-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples" class="anchor"></a>4. Relationship between clusters within one sample or between two samples</h1>
<p><em>The purpose of this workflow is to solve the following task: given one or two unknown mixed population(s) and clusters in each mixed population, estimate and visualise relationship between clusters</em></p>
<div id="start-the-scgps-prediction-to-find-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#start-the-scgps-prediction-to-find-relationship-between-clusters" class="anchor"></a>4.1 Start the scGPS prediction to find relationship between clusters</h2>
<p>```{r, warning = FALSE, message = FALSE}</p>
</div>
</div>
<div id="select-a-subpopulation-and-input-gene-list" class="section level1">
<h1 class="hasAnchor">
<a href="#select-a-subpopulation-and-input-gene-list" class="anchor"></a>select a subpopulation, and input gene list</h1>
<p>c_selectID &lt;- 1 genes = DEgenes$DE_Subpop1vsRemaining$id[1:500]</p>
</div>
<div id="format-gene-names" class="section level1">
<h1 class="hasAnchor">
<a href="#format-gene-names" class="anchor"></a>format gene names</h1>
<p>genes &lt;- gsub(“_.*“,”“, genes)</p>
</div>
<div id="run-the-test-bootstrap-with-nboots-2-runs" class="section level1">
<h1 class="hasAnchor">
<a href="#run-the-test-bootstrap-with-nboots-2-runs" class="anchor"></a>run the test bootstrap with nboots = 2 runs</h1>
<p>sink(“temp”) LSOLDA_dat &lt;- bootstrap_scGPS(nboots = 2,mixedpop1 = mixedpop2, mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list()) sink()</p>
<p>```</p>
<div id="display-summary-results-for-the-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#display-summary-results-for-the-prediction" class="anchor"></a>4.2 Display summary results for the prediction</h2>
<pre class="{r}"><code>#get the number of rows for the summary matrix 
row_cluster &lt;-length(unique(colData(mixedpop2)[,1]))

#summary results LDA
summary_prediction_lda(LSOLDA_dat=LSOLDA_dat, nPredSubpop = row_cluster )

#summary results Lasso
summary_prediction_lasso(LSOLDA_dat=LSOLDA_dat, nPredSubpop = row_cluster)

#summary deviance 
summary_deviance(LSOLDA_dat)
</code></pre>
</div>
<div id="plot-the-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters" class="anchor"></a>4.3 Plot the relationship between clusters</h2>
<p><em>Here we look at one example use case to find relationship between clusters within one sample or between two sample</em></p>
<p>```{r,warning = FALSE, message = FALSE}</p>
</div>
</div>
<div id="run-prediction-for-3-clusters" class="section level1">
<h1 class="hasAnchor">
<a href="#run-prediction-for-3-clusters" class="anchor"></a>run prediction for 3 clusters</h1>
<p>c_selectID &lt;- 1 genes = DEgenes$DE_Subpop1vsRemaining$id[1:200] #top 200 gene markers distinguishing cluster 1 genes &lt;- gsub(“_.*“,”“, genes)</p>
<p>LSOLDA_dat1 &lt;- bootstrap_scGPS(nboots = 1,mixedpop1 = mixedpop2, mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list())</p>
<p>c_selectID &lt;- 2 genes = DEgenes$DE_Subpop2vsRemaining$id[1:200] genes &lt;- gsub(“_.*“,”“, genes) LSOLDA_dat2 &lt;- bootstrap_scGPS(nboots = 1,mixedpop1 = mixedpop2, mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list())</p>
<p>c_selectID &lt;- 3 genes = DEgenes$DE_Subpop3vsRemaining$id[1:200] genes &lt;- gsub(“_.*“,”“, genes) LSOLDA_dat3 &lt;- bootstrap_scGPS(nboots = 1,mixedpop1 = mixedpop2, mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list())</p>
</div>
<div id="prepare-table-input-for-sankey-plot" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-table-input-for-sankey-plot" class="anchor"></a>prepare table input for sankey plot</h1>
<p>reformat_LASSO &lt;-function(c_selectID = NULL, s_selectID = NULL, LSOLDA_dat = NULL, nPredSubpop = row_cluster, Nodes_group = “#7570b3”){ LASSO_out &lt;- summary_prediction_lasso(LSOLDA_dat=LSOLDA_dat, nPredSubpop = nPredSubpop) LASSO_out &lt;-as.data.frame(LASSO_out) temp_name &lt;- gsub(“LASSO for subpop”, “C”, LASSO_out$names) temp_name &lt;- gsub(" in target mixedpop“,”S“, temp_name) LASSO_out$names &lt;-temp_name source &lt;-rep(paste0(”C“,c_selectID,”S“,s_selectID), length(temp_name)) LASSO_out$Source &lt;- source LASSO_out$Node &lt;- source LASSO_out$Nodes_group &lt;- rep(Nodes_group, length(temp_name)) colnames(LASSO_out) &lt;-c(”Value“,”Target“,”Source“,”Node“,”NodeGroup“) LASSO_out$Value &lt;- as.numeric(as.vector(LASSO_out$Value)) return(LASSO_out) }</p>
<p>LASSO_C1S2 &lt;- reformat_LASSO(c_selectID=1, s_selectID =2, LSOLDA_dat=LSOLDA_dat1, nPredSubpop = row_cluster, Nodes_group = “#7570b3”)</p>
<p>LASSO_C2S2 &lt;- reformat_LASSO(c_selectID=2, s_selectID =2, LSOLDA_dat=LSOLDA_dat2, nPredSubpop = row_cluster, Nodes_group = “#1b9e77”)</p>
<p>LASSO_C3S2 &lt;- reformat_LASSO(c_selectID=3, s_selectID =2, LSOLDA_dat=LSOLDA_dat3, nPredSubpop = row_cluster, Nodes_group = “#e7298a”)</p>
<p>combined &lt;- rbind(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2 ) combined &lt;- combined[is.na(combined$Value) != TRUE,] combined_D3obj &lt;-list(Nodes=combined[,4:5], Links=combined[,c(3,2,1)])</p>
<p>library(networkD3)</p>
<p>Node_source &lt;- as.vector(sort(unique(combined_D3obj$Links$Source))) Node_target &lt;- as.vector(sort(unique(combined_D3obj$Links$Target))) Node_all &lt;-unique(c(Node_source, Node_target))</p>
</div>
<div id="assign-ids-for-source-start-from-0" class="section level1">
<h1 class="hasAnchor">
<a href="#assign-ids-for-source-start-from-0" class="anchor"></a>assign IDs for Source (start from 0)</h1>
<p>Source &lt;-combined_D3obj$Links$Source Target &lt;- combined_D3obj$Links$Target</p>
<p>for(i in 1:length(Node_all)){ Source[Source==Node_all[i]] &lt;-i-1 Target[Target==Node_all[i]] &lt;-i-1 }</p>
<p>combined_D3obj$Links$Source &lt;- as.numeric(Source) combined_D3obj$Links$Target &lt;- as.numeric(Target) combined_D3obj$Links$LinkColor &lt;- combined$NodeGroup</p>
</div>
<div id="prepare-node-info" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-node-info" class="anchor"></a>prepare node info</h1>
<p>node_df &lt;-data.frame(Node=Node_all) node_df$id &lt;-as.numeric(c(0, 1:(length(Node_all)-1)))</p>
<p>suppressMessages(library(dplyr)) Color &lt;- combined %&gt;% count(Node, color=NodeGroup) %&gt;% select(2) node_df$color &lt;- Color$color</p>
<p>suppressMessages(library(networkD3)) p1&lt;-sankeyNetwork(Links =combined_D3obj$Links, Nodes = node_df, Value = “Value”, NodeGroup =“color”, LinkGroup = “LinkColor”, NodeID=“Node”, Source=“Source”, Target=“Target”, fontSize = 22 ) p1</p>
</div>
<div id="savenetworkp1-file-paste0pathsubpopulation_net-html" class="section level1">
<h1 class="hasAnchor">
<a href="#savenetworkp1-file-paste0pathsubpopulation_net-html" class="anchor"></a>saveNetwork(p1, file = paste0(path,‘Subpopulation_Net.html’))</h1>
<div id="r-setting-information" class="section level2">
<h2 class="hasAnchor">
<a href="#r-setting-information" class="anchor"></a>R Setting Information</h2>
</div>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>sessionInfo()</h1>
</div>
<div id="rmarkdownrenderusersquan-nguyendocumentspowell_group_macquanallcodesscgpsvignettesvignette-rmdhtml_documenttoc-true-toc_depth-3" class="section level1">
<h1 class="hasAnchor">
<a href="#rmarkdownrenderusersquan-nguyendocumentspowell_group_macquanallcodesscgpsvignettesvignette-rmdhtml_documenttoc-true-toc_depth-3" class="anchor"></a>rmarkdown::render(“/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/vignettes/vignette.Rmd”,html_document(toc = TRUE, toc_depth = 3))</h1>
</div>
<div id="rmarkdownrenderusersquan-nguyendocumentspowell_group_macquanallcodesscgpsvignettesvignette-rmdpdf_documenttoc-true-toc_depth-3" class="section level1">
<h1 class="hasAnchor">
<a href="#rmarkdownrenderusersquan-nguyendocumentspowell_group_macquanallcodesscgpsvignettesvignette-rmdpdf_documenttoc-true-toc_depth-3" class="anchor"></a>rmarkdown::render(“/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/vignettes/vignette.Rmd”,pdf_document(toc = TRUE, toc_depth = 3))</h1>
<p>```</p>
<p>```{r, eval = FALSE}</p>
</div>
<div id="install-release-version-from-cran" class="section level1">
<h1 class="hasAnchor">
<a href="#install-release-version-from-cran" class="anchor"></a>Install release version from CRAN</h1>
</div>
<div id="install-packagespkgdown" class="section level1">
<h1 class="hasAnchor">
<a href="#install-packagespkgdown" class="anchor"></a>install.packages(“pkgdown”)</h1>
<p>```</p>
</div>


  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Asenbouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

