<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGPS introduction â€¢ scGPS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scGPS introduction">
<meta property="og:description" content="scGPS is a single cell RNA analysis framework to decompose a mixed population into clusters (SCORE), followed by analysing the relationship between clusters (scGPS)">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="vignette_files/htmlwidgets-1.3/htmlwidgets.js"></script><script src="vignette_files/d3-4.5.0/d3.min.js"></script><script src="vignette_files/sankey-1/sankey.js"></script><script src="vignette_files/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scGPS introduction</h1>
                        <h4 class="author">Quan Nguyen and Michael Thompson</h4>
            
            <h4 class="date">2019-03-13</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># To install scGPS from github (Depending on the configuration of the local</span>
<span class="co"># computer or HPC, possible custom C++ compilation may be required - see</span>
<span class="co"># installation trouble-shootings below)</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"IMB-Computational-Genomics-Lab/scGPS"</span>)

<span class="co"># for C++ compilation trouble-shooting, manual download and installation can be</span>
<span class="co"># done from github</span>

git clone https<span class="op">:</span><span class="er">//</span>github.com<span class="op">/</span>IMB<span class="op">-</span>Computational<span class="op">-</span>Genomics<span class="op">-</span>Lab<span class="op">/</span>scGPS

<span class="co"># then check in scGPS/src if any of the precompiled (e.g.  those with *.so and</span>
<span class="co"># *.o) files exist and delete them before recompiling</span>

<span class="co"># create a Makevars file in the scGPS/src with one line: PKG_LIBS =</span>
<span class="co"># $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)</span>

<span class="co"># then with the scGPS as the R working directory, manually recompile scGPS in R</span>
<span class="co"># using devtools to load and install functions</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/document">document</a></span>()
<span class="co">#load the package to the workspace </span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>()</code></pre></div>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: </em> <em>given a mixed population with known subpopulations, </em> <em>estimate transition scores between these subpopulation.</em></p>
<div id="create-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#create-scgps-objects" class="anchor"></a>2.1 Create scGPS objects</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load mixed population 1 (loaded from sample1 dataset, named it as day2)</span>
path_scPGS =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/getwd">getwd</a></span>()
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(path_scPGS) 

day2 &lt;-<span class="st"> </span>sample1
mixedpop1 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day2<span class="op">$</span>dat2_counts,
    <span class="dt">GeneMetadata =</span> day2<span class="op">$</span>dat2geneInfo, <span class="dt">CellMetadata =</span> day2<span class="op">$</span>dat2_clusters)

<span class="co"># load mixed population 2 (loaded from sample2 dataset, named it as day5)</span>
day5 &lt;-<span class="st"> </span>sample2
mixedpop2 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts,
    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> day5<span class="op">$</span>dat5_clusters)</code></pre></div>
</div>
<div id="run-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#run-prediction" class="anchor"></a>2.2 Run prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># select a subpopulation</span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co"># load gene list (this can be any lists of user selected genes)</span>
genes &lt;-<span class="st"> </span>GeneList
genes &lt;-<span class="st"> </span>genes<span class="op">$</span>Merged_unique
<span class="co"># load cluster information </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#run training (running nboots = 3 here, but recommend to use nboots = 50-100)</span>
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">3</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID,
    <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2, <span class="dt">trainset_ratio =</span> <span class="fl">0.7</span>)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev   Lambda
##  [1,]  0 -1.442e-15 0.163800
##  [2,]  1  1.316e-02 0.149200
##  [3,]  1  2.411e-02 0.136000
##  [4,]  1  3.324e-02 0.123900
##  [5,]  1  4.086e-02 0.112900
##  [6,]  1  4.722e-02 0.102900
##  [7,]  1  5.255e-02 0.093730
##  [8,]  1  5.700e-02 0.085400
##  [9,]  1  6.073e-02 0.077820
## [10,]  2  6.605e-02 0.070900
## [11,]  2  7.088e-02 0.064600
## [12,]  2  7.501e-02 0.058860
## [13,]  2  7.855e-02 0.053640
## [14,]  2  8.158e-02 0.048870
## [15,]  2  8.419e-02 0.044530
## [16,]  2  8.644e-02 0.040570
## [17,]  2  8.837e-02 0.036970
## [18,]  2  9.003e-02 0.033680
## [19,]  2  9.146e-02 0.030690
## [20,]  2  9.269e-02 0.027970
## [21,]  2  9.375e-02 0.025480
## [22,]  3  9.498e-02 0.023220
## [23,]  3  9.609e-02 0.021150
## [24,]  3  9.705e-02 0.019280
## [25,]  3  9.786e-02 0.017560
## [26,]  3  9.855e-02 0.016000
## [27,]  3  9.914e-02 0.014580
## [28,]  3  9.965e-02 0.013290
## [29,]  3  1.001e-01 0.012110
## [30,]  3  1.004e-01 0.011030
## [31,]  3  1.007e-01 0.010050
## [32,]  3  1.010e-01 0.009157
## [33,]  3  1.012e-01 0.008344
## [34,]  3  1.014e-01 0.007603
## [35,]  3  1.016e-01 0.006927
## [36,]  3  1.017e-01 0.006312
## [37,]  3  1.018e-01 0.005751
## [38,]  3  1.019e-01 0.005240
## [39,]  4  1.020e-01 0.004775
## [40,]  4  1.021e-01 0.004351
## [41,]  4  1.021e-01 0.003964
## [42,]  4  1.022e-01 0.003612
## [43,]  4  1.022e-01 0.003291
## [44,]  4  1.023e-01 0.002999
## [45,]  4  1.023e-01 0.002732
## [46,]  4  1.023e-01 0.002490
## [47,]  4  1.023e-01 0.002268
## [48,]  4  1.024e-01 0.002067
## [49,]  4  1.024e-01 0.001883
## [50,]  4  1.024e-01 0.001716
## [51,]  4  1.024e-01 0.001564
## [52,]  4  1.024e-01 0.001425
## [1] "lambda min is at location 28"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 212 cells and 5 genes..."
## [1] "evaluation accuracy ElasticNet 0.613207547169811"
## [1] "evaluation accuracy LDA 0.622641509433962"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 6.95187165775401"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 0"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 25"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 8.57142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 57.8947368421053"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 1.50375939849624"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 42.5"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 2.5"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev   Lambda
##  [1,]  0 3.846e-15 0.123500
##  [2,]  1 7.483e-03 0.112600
##  [3,]  1 1.370e-02 0.102600
##  [4,]  1 1.888e-02 0.093450
##  [5,]  2 2.357e-02 0.085140
##  [6,]  2 2.987e-02 0.077580
##  [7,]  2 3.520e-02 0.070690
##  [8,]  2 3.973e-02 0.064410
##  [9,]  2 4.359e-02 0.058690
## [10,]  2 4.690e-02 0.053470
## [11,]  2 4.974e-02 0.048720
## [12,]  2 5.218e-02 0.044390
## [13,]  2 5.428e-02 0.040450
## [14,]  3 5.667e-02 0.036860
## [15,]  3 5.899e-02 0.033580
## [16,]  3 6.101e-02 0.030600
## [17,]  4 6.277e-02 0.027880
## [18,]  4 6.437e-02 0.025400
## [19,]  4 6.576e-02 0.023150
## [20,]  4 6.697e-02 0.021090
## [21,]  4 6.802e-02 0.019220
## [22,]  4 6.894e-02 0.017510
## [23,]  4 6.973e-02 0.015950
## [24,]  4 7.041e-02 0.014540
## [25,]  4 7.101e-02 0.013250
## [26,]  4 7.152e-02 0.012070
## [27,]  4 7.196e-02 0.011000
## [28,]  4 7.235e-02 0.010020
## [29,]  4 7.267e-02 0.009130
## [30,]  4 7.295e-02 0.008319
## [31,]  4 7.319e-02 0.007580
## [32,]  4 7.340e-02 0.006906
## [33,]  4 7.357e-02 0.006293
## [34,]  4 7.372e-02 0.005734
## [35,]  4 7.385e-02 0.005224
## [36,]  4 7.396e-02 0.004760
## [37,]  4 7.405e-02 0.004337
## [38,]  4 7.412e-02 0.003952
## [39,]  4 7.419e-02 0.003601
## [40,]  5 7.424e-02 0.003281
## [41,]  5 7.429e-02 0.002990
## [42,]  5 7.433e-02 0.002724
## [43,]  5 7.436e-02 0.002482
## [44,]  5 7.439e-02 0.002262
## [45,]  5 7.441e-02 0.002061
## [46,]  5 7.443e-02 0.001878
## [47,]  5 7.445e-02 0.001711
## [48,]  5 7.446e-02 0.001559
## [49,]  5 7.448e-02 0.001420
## [50,]  5 7.449e-02 0.001294
## [1] "lambda min is at location 16"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 214 cells and 6 genes..."
## [1] "evaluation accuracy ElasticNet 0.663551401869159"
## [1] "evaluation accuracy LDA 0.682242990654206"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "ElasticNet for target subpop 1 has no solution"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 0"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 22.8571428571429"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 40.7142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 7.5187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 0"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 17.5"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 10"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev    Lambda
##  [1,]  0 4.485e-15 0.1674000
##  [2,]  1 1.373e-02 0.1525000
##  [3,]  1 2.516e-02 0.1390000
##  [4,]  1 3.469e-02 0.1266000
##  [5,]  1 4.264e-02 0.1154000
##  [6,]  1 4.930e-02 0.1051000
##  [7,]  1 5.486e-02 0.0957800
##  [8,]  2 6.088e-02 0.0872700
##  [9,]  2 6.762e-02 0.0795200
## [10,]  2 7.339e-02 0.0724500
## [11,]  2 7.835e-02 0.0660200
## [12,]  2 8.263e-02 0.0601500
## [13,]  3 8.652e-02 0.0548100
## [14,]  3 9.065e-02 0.0499400
## [15,]  3 9.425e-02 0.0455000
## [16,]  4 9.773e-02 0.0414600
## [17,]  4 1.009e-01 0.0377800
## [18,]  4 1.037e-01 0.0344200
## [19,]  4 1.062e-01 0.0313600
## [20,]  4 1.083e-01 0.0285800
## [21,]  4 1.102e-01 0.0260400
## [22,]  4 1.119e-01 0.0237200
## [23,]  4 1.134e-01 0.0216200
## [24,]  4 1.148e-01 0.0197000
## [25,]  4 1.159e-01 0.0179500
## [26,]  4 1.170e-01 0.0163500
## [27,]  4 1.179e-01 0.0149000
## [28,]  4 1.187e-01 0.0135800
## [29,]  4 1.195e-01 0.0123700
## [30,]  4 1.201e-01 0.0112700
## [31,]  4 1.207e-01 0.0102700
## [32,]  4 1.212e-01 0.0093570
## [33,]  4 1.216e-01 0.0085260
## [34,]  4 1.220e-01 0.0077690
## [35,]  4 1.223e-01 0.0070790
## [36,]  4 1.226e-01 0.0064500
## [37,]  4 1.229e-01 0.0058770
## [38,]  4 1.231e-01 0.0053550
## [39,]  4 1.233e-01 0.0048790
## [40,]  4 1.234e-01 0.0044460
## [41,]  5 1.236e-01 0.0040510
## [42,]  5 1.237e-01 0.0036910
## [43,]  5 1.238e-01 0.0033630
## [44,]  5 1.239e-01 0.0030640
## [45,]  5 1.240e-01 0.0027920
## [46,]  5 1.241e-01 0.0025440
## [47,]  5 1.242e-01 0.0023180
## [48,]  5 1.242e-01 0.0021120
## [49,]  5 1.243e-01 0.0019240
## [50,]  5 1.243e-01 0.0017530
## [51,]  5 1.243e-01 0.0015980
## [52,]  5 1.244e-01 0.0014560
## [53,]  5 1.244e-01 0.0013260
## [54,]  5 1.244e-01 0.0012090
## [55,]  5 1.244e-01 0.0011010
## [56,]  5 1.244e-01 0.0010030
## [57,]  5 1.244e-01 0.0009142
## [58,]  5 1.245e-01 0.0008330
## [59,]  5 1.245e-01 0.0007590
## [1] "lambda min is at location 26"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 215 cells and 6 genes..."
## [1] "evaluation accuracy ElasticNet 0.623255813953488"
## [1] "evaluation accuracy LDA 0.613953488372093"
## [1] "done training for bootstrap 3, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 28.3422459893048"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 33.6898395721925"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 25.7142857142857"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 43.5714285714286"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 36.8421052631579"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 47.3684210526316"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 27.5"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 40"</code></pre>
</div>
<div id="summarise-results" class="section level2">
<h2 class="hasAnchor">
<a href="#summarise-results" class="anchor"></a>2.3 Summarise results</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># display the list of result information in the LASOLDA_dat object </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(LSOLDA_dat)</code></pre></div>
<pre><code>## [1] "Accuracy"          "ElasticNetGenes"   "Deviance"         
## [4] "ElasticNetFit"     "LDAFit"            "predictor_S1"     
## [7] "ElasticNetPredict" "LDAPredict"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>ElasticNetPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[1]][[2]]
## [1] 6.951872
## 
## [[1]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[1]][[4]]
## [1] 25
## 
## [[1]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[1]][[6]]
## [1] 57.89474
## 
## [[1]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 42.5
## 
## 
## [[2]]
## [[2]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[2]][[2]]
## [1] "NA"
## 
## [[2]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[2]][[4]]
## [1] 22.85714
## 
## [[2]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 7.518797
## 
## [[2]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 17.5
## 
## 
## [[3]]
## [[3]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[3]][[2]]
## [1] 28.34225
## 
## [[3]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[3]][[4]]
## [1] 25.71429
## 
## [[3]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 36.84211
## 
## [[3]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 27.5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>LDAPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[1]][[2]]
## [1] 0
## 
## [[1]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[1]][[4]]
## [1] 8.571429
## 
## [[1]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[1]][[6]]
## [1] 1.503759
## 
## [[1]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 2.5
## 
## 
## [[2]]
## [[2]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[2]][[2]]
## [1] 0
## 
## [[2]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[2]][[4]]
## [1] 40.71429
## 
## [[2]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 0
## 
## [[2]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 10
## 
## 
## [[3]]
## [[3]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[3]][[2]]
## [1] 33.68984
## 
## [[3]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[3]][[4]]
## [1] 43.57143
## 
## [[3]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 47.36842
## 
## [[3]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 40</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary results LDA</span>
sum_pred_lda &lt;-<span class="st"> </span><span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># summary results Lasso to show the percent of cells</span>
<span class="co"># classified as cells belonging </span>
sum_pred_lasso &lt;-<span class="st"> </span><span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat,
    <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># plot summary results </span>
plot_sum &lt;-<span class="cf">function</span>(sum_dat){
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(sum_dat)
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/na.fail">na.omit</a></span>(sum_dat_tf)
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(sum_dat[, <span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(sum_dat)],<span class="dv">1</span>,
        <span class="cf">function</span>(x){<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(x))})
    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"for subpop "</span>,<span class="st">"sp"</span>,  sum_dat<span class="op">$</span>names )
    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"in target mixedpop"</span>,<span class="st">"in p"</span>,  sum_dat<span class="op">$</span>names) 
    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sum_dat_tf) &lt;-<span class="st"> </span>sum_dat<span class="op">$</span>names
    <span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(sum_dat_tf, <span class="dt">las=</span><span class="dv">2</span>)
}
<span class="kw">plot_sum</span>(sum_pred_lasso)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_sum</span>(sum_pred_lda)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## [1] 62.26415 68.22430 61.39535</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary maximum deviance explained by the model </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## $allDeviance
## [1] "0.09965" "0.06101" "0.117"  
## 
## $DeviMax
##          dat_DE$Dfd          Deviance           DEgenes
## 1                 0             0.117    genes_cluster1
## 2                 1             0.117    genes_cluster1
## 3                 2             0.117    genes_cluster1
## 4                 3             0.117    genes_cluster1
## 5                 4             0.117    genes_cluster1
## 6 remaining DEgenes remaining DEgenes remaining DEgenes
## 
## $LassoGenesMax
## NULL</code></pre>
</div>
</div>
<div id="a-complete-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-complete-workflow-of-the-scgps" class="anchor"></a>3. A complete workflow of the scGPS:</h1>
<ul>
<li>The purpose of this workflow is to solve the following task: given an unknown*</li>
<li>mixed population, find clusters and estimate relationship between clusters*</li>
</ul>
<div id="identify-clusters-in-a-dataset-using-core" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-core" class="anchor"></a>3.1 Identify clusters in a dataset using CORE</h2>
<p><em>(skip this step if clusters are known)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find clustering information in an expresion data using CORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, 
    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames) 

CORE_cluster &lt;-<span class="st"> </span><span class="kw">CORE_scGPS</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>)</code></pre></div>
<pre><code>## [1] "Performing 1 round of filtering"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "No more outliers detected in filtering round 1"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "writing clustering result for run 1"
## [1] "writing clustering result for run 2"
## [1] "writing clustering result for run 3"
## [1] "writing clustering result for run 4"
## [1] "writing clustering result for run 5"
## [1] "writing clustering result for run 6"
## [1] "writing clustering result for run 7"
## [1] "writing clustering result for run 8"
## [1] "writing clustering result for run 9"
## [1] "writing clustering result for run 10"
## [1] "writing clustering result for run 11"
## [1] "writing clustering result for run 12"
## [1] "writing clustering result for run 13"
## [1] "writing clustering result for run 14"
## [1] "writing clustering result for run 15"
## [1] "writing clustering result for run 16"
## [1] "writing clustering result for run 17"
## [1] "writing clustering result for run 18"
## [1] "writing clustering result for run 19"
## [1] "writing clustering result for run 20"
## [1] "writing clustering result for run 21"
## [1] "writing clustering result for run 22"
## [1] "writing clustering result for run 23"
## [1] "writing clustering result for run 24"
## [1] "writing clustering result for run 25"
## [1] "writing clustering result for run 26"
## [1] "writing clustering result for run 27"
## [1] "writing clustering result for run 28"
## [1] "writing clustering result for run 29"
## [1] "writing clustering result for run 30"
## [1] "writing clustering result for run 31"
## [1] "writing clustering result for run 32"
## [1] "writing clustering result for run 33"
## [1] "writing clustering result for run 34"
## [1] "writing clustering result for run 35"
## [1] "writing clustering result for run 36"
## [1] "writing clustering result for run 37"
## [1] "writing clustering result for run 38"
## [1] "writing clustering result for run 39"
## [1] "writing clustering result for run 40"
## [1] "Done clustering, moving to stability calculation..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done finding optimal clustering..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># to update the clustering information, users can ...</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)

clustering_after_outlier_removal &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unname">unname</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(
    CORE_cluster<span class="op">$</span>Cluster[[optimal_index]]))
corresponding_cells_after_outlier_removal &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>cellsForClustering
original_cells_before_removal &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">2</span>]
corresponding_index &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(corresponding_cells_after_outlier_removal,
    original_cells_before_removal )
<span class="co"># check the matching </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/identical">identical</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/character">as.character</a></span>(original_cells_before_removal[corresponding_index]),
    corresponding_cells_after_outlier_removal)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># create new object with the new clustering after removing outliers </span>
mixedpop2_post_clustering &lt;-<span class="st"> </span>mixedpop2[,corresponding_index]
<span class="kw">colData</span>(mixedpop2_post_clustering)[,<span class="dv">1</span>] &lt;-<span class="st"> </span>clustering_after_outlier_removal</code></pre></div>
</div>
<div id="identify-clusters-in-a-dataset-using-score" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-score" class="anchor"></a>3.2 Identify clusters in a dataset using SCORE</h2>
</div>
<div id="stable-clustering-at-optimal-resolution" class="section level2">
<h2 class="hasAnchor">
<a href="#stable-clustering-at-optimal-resolution" class="anchor"></a>(Stable Clustering at Optimal REsolution)</h2>
<p><em>(skip this step if clusters are known)</em> <em>(SCORE aims to get stable subpopulation results, </em> * by introducing bagging aggregation and bootstrapping to the CORE algorithm)*</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find clustering information in an expresion data using SCORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, 
    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames ) 

SCORE_test &lt;-<span class="st"> </span><span class="kw">CORE_scGPS_bagging</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>,
                          <span class="dt">bagging_run =</span> <span class="dv">20</span>, <span class="dt">subsample_proportion =</span> .<span class="dv">8</span>)</code></pre></div>
<pre><code>## [1] "Performing 1 round of filtering"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "No more outliers detected in filtering round 1"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "500 cells left after filtering"
## [1] "Running 20 bagging runs, with 0.8 subsampling..."
## [1] "Done clustering, moving to stability calculation..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."</code></pre>
</div>
<div id="visualise-all-cluster-results-in-all-iterations" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-all-cluster-results-in-all-iterations" class="anchor"></a>3.3 Visualise all cluster results in all iterations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/dev">dev.off</a></span>()</code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##3.2.1 plot CORE clustering 
p1 &lt;-<span class="st"> </span><span class="kw">plot_CORE</span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster, 
    <span class="dt">color_branch =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#208eb7"</span>, <span class="st">"#6ce9d3"</span>, <span class="st">"#1c5e39"</span>, <span class="st">"#8fca40"</span>, <span class="st">"#154975"</span>,
    <span class="st">"#b1c8eb"</span>))
p1 </code></pre></div>
<pre><code>## $mar
## [1] 1 5 0 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#extract optimal index identified by CORE_scGPS</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)
<span class="co">#plot one optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> CORE_cluster<span class="op">$</span>tree,
    <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(CORE_cluster<span class="op">$</span>Cluster[optimal_index]),
    <span class="dt">shift =</span> <span class="op">-</span><span class="dv">2000</span>)</code></pre></div>
<pre><code>## [1] "Ordering and assigning labels..."
## [1] 2
## [1]  24 112  NA  NA  NA  NA  NA  NA
## [1] 3
## [1]  24 112 224  NA  NA  NA  NA  NA
## [1] 4
## [1]  24 112 224 299  NA  NA  NA  NA
## [1] 5
## [1]  24 112 224 299 335  NA  NA  NA
## [1] 6
## [1]  24 112 224 299 335 367  NA  NA
## [1] 7
## [1]  24 112 224 299 335 367 414  NA
## [1] 8
## [1]  24 112 224 299 335 367 414 470
## [1] "Plotting the colored dendrogram now...."</code></pre>
<pre><code>## [1] "Plotting the bar underneath now...."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##3.2.2 plot SCORE clustering 
<span class="co">#plot all clustering bars</span>
<span class="kw">plot_CORE</span>(SCORE_test<span class="op">$</span>tree, <span class="dt">list_clusters =</span> SCORE_test<span class="op">$</span>Cluster)
<span class="co">#plot one stable optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> SCORE_test<span class="op">$</span>tree,
    <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(SCORE_test<span class="op">$</span>Cluster[SCORE_test<span class="op">$</span>optimal_index]),
    <span class="dt">shift =</span> <span class="op">-</span><span class="dv">100</span>)</code></pre></div>
<pre><code>## [1] "Ordering and assigning labels..."
## [1] 2
## [1]  24 112  NA  NA  NA  NA  NA  NA
## [1] 3
## [1]  24 112 224  NA  NA  NA  NA  NA
## [1] 4
## [1]  24 112 224 299  NA  NA  NA  NA
## [1] 5
## [1]  24 112 224 299 335  NA  NA  NA
## [1] 6
## [1]  24 112 224 299 335 367  NA  NA
## [1] 7
## [1]  24 112 224 299 335 367 414  NA
## [1] 8
## [1]  24 112 224 299 335 367 414 470
## [1] "Plotting the colored dendrogram now...."</code></pre>
<pre><code>## [1] "Plotting the bar underneath now...."</code></pre>
</div>
<div id="compare-clustering-results-with-other-dimensional-reduction-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-clustering-results-with-other-dimensional-reduction-methods" class="anchor"></a>3.4 Compare clustering results with other dimensional reduction methods</h2>
</div>
<div id="e-g--cidr" class="section level2">
<h2 class="hasAnchor">
<a href="#e-g--cidr" class="anchor"></a>(e.g., CIDR)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(cidr)
t &lt;-<span class="st"> </span><span class="kw">CIDR_scGPS</span>(<span class="dt">expression.matrix=</span><span class="kw">assay</span>(mixedpop2))</code></pre></div>
<pre><code>## [1] "building cidr object..."
## [1] "determine dropout candidates..."
## [1] "determine the imputation weighting threshold..."
## [1] "computes the _CIDR_ dissimilarity matrix..."
## [1] "PCA plot with proportion of variance explained..."</code></pre>
<p><img src="vignette_files/figure-html/compare_clustering-1.png" width="700"></p>
<pre><code>## [1] "find the number of PC..."
## [1] "perform clustering..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="kw">plotReduced_scGPS</span>(t, <span class="dt">color_fac =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]),
    <span class="dt">palletes =</span><span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])))
p2</code></pre></div>
<p><img src="vignette_files/figure-html/compare_clustering-2.png" width="700"></p>
</div>
<div id="find-gene-markers-and-annotate-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#find-gene-markers-and-annotate-clusters" class="anchor"></a>3.5 Find gene markers and annotate clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#load gene list (this can be any lists of user-selected genes)</span>
genes &lt;-GeneList
genes &lt;-genes<span class="op">$</span>Merged_unique

<span class="co">#the gene list can also be objectively identified by differential expression</span>
<span class="co">#analysis cluster information is requied for findMarkers_scGPS. Here, we use </span>
<span class="co">#CORE results. </span>

<span class="co">#colData(mixedpop2)[,1] &lt;- unlist(SCORE_test$Cluster[SCORE_test$optimal_index])</span>

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(locfit))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DESeq))

DEgenes &lt;-<span class="st"> </span><span class="kw">findMarkers_scGPS</span>(<span class="dt">expression_matrix=</span><span class="kw">assay</span>(mixedpop2), 
    <span class="dt">cluster =</span> <span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>], 
    <span class="dt">selected_cluster=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])) </code></pre></div>
<pre><code>## [1] "Start estimate dispersions for cluster 1..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 1..."
## [1] "Done nbinom test for cluster 1 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 2..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 2..."
## [1] "Done nbinom test for cluster 2 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 3..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 3..."
## [1] "Done nbinom test for cluster 3 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 4..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 4..."
## [1] "Done nbinom test for cluster 4 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#the output contains dataframes for each cluster.</span>
<span class="co">#the data frame contains all genes, sorted by p-values </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(DEgenes)</code></pre></div>
<pre><code>## [1] "DE_Subpop1vsRemaining" "DE_Subpop2vsRemaining" "DE_Subpop3vsRemaining"
## [4] "DE_Subpop4vsRemaining"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#you can annotate the identified clusters </span>
DEgeneList_1vsOthers &lt;-<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id

<span class="co">#users need to check the format of the gene input to make sure they are </span>
<span class="co">#consistent to the gene names in the expression matrix </span>

<span class="co">#the following command saves the file "PathwayEnrichment.xlsx" to the </span>
<span class="co">#working dir</span>
<span class="co">#use 500 top DE genes </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DOSE))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ReactomePA))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(clusterProfiler))
enrichment_test &lt;-<span class="st"> </span><span class="kw">annotate_scGPS</span>(DEgeneList_1vsOthers[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>], 
    <span class="dt">pvalueCutoff=</span><span class="fl">0.05</span>, <span class="dt">gene_symbol=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] "Original gene number in geneList"
## [1] 500
## [1] "Number of genes successfully converted"
## [1] 486</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#the enrichment outputs can be displayed by running</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/ReactomePA/topics/reexports">dotplot</a></span>(enrichment_test, <span class="dt">showCategory=</span><span class="dv">10</span>, <span class="dt">font.size =</span> <span class="dv">6</span>)</code></pre></div>
<p><img src="vignette_files/figure-html/find_markers-1.png" width="700"></p>
</div>
</div>
<div id="relationship-between-clusters-within-one-sample-or-between-two-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples" class="anchor"></a>4. Relationship between clusters within one sample or between two samples</h1>
<p><em>The purpose of this workflow is to solve the following task: </em> <em>given one or two unknown mixed population(s) and clusters in each mixed</em> <em>population, estimate and visualise relationship between clusters</em></p>
<div id="start-the-scgps-prediction-to-find-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#start-the-scgps-prediction-to-find-relationship-between-clusters" class="anchor"></a>4.1 Start the scGPS prediction to find relationship between clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#select a subpopulation, and input gene list </span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#note make sure the format for genes input here is the same to the format</span>
<span class="co">#for genes in the mixedpop1 and mixedpop2  </span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>]

<span class="co">#run the test bootstrap with nboots = 2 runs</span>

cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sink">sink</a></span>(<span class="st">"temp"</span>)
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID, 
     <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
     <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 153 genes for training model"
## [1] "use 153 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 4 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 2.284e-01
##  [2,]  1  2.561e-02 2.081e-01
##  [3,]  1  4.715e-02 1.896e-01
##  [4,]  1  6.545e-02 1.728e-01
##  [5,]  1  8.110e-02 1.574e-01
##  [6,]  3  1.048e-01 1.434e-01
##  [7,]  4  1.293e-01 1.307e-01
##  [8,]  5  1.550e-01 1.191e-01
##  [9,]  6  1.810e-01 1.085e-01
## [10,]  9  2.105e-01 9.886e-02
## [11,] 10  2.399e-01 9.008e-02
## [12,] 11  2.667e-01 8.208e-02
## [13,] 16  2.949e-01 7.479e-02
## [14,] 18  3.244e-01 6.814e-02
## [15,] 20  3.539e-01 6.209e-02
## [16,] 24  3.814e-01 5.657e-02
## [17,] 28  4.085e-01 5.155e-02
## [18,] 31  4.360e-01 4.697e-02
## [19,] 32  4.633e-01 4.280e-02
## [20,] 34  4.884e-01 3.899e-02
## [21,] 36  5.126e-01 3.553e-02
## [22,] 41  5.365e-01 3.237e-02
## [23,] 47  5.632e-01 2.950e-02
## [24,] 50  5.896e-01 2.688e-02
## [25,] 52  6.150e-01 2.449e-02
## [26,] 57  6.392e-01 2.231e-02
## [27,] 59  6.632e-01 2.033e-02
## [28,] 60  6.862e-01 1.853e-02
## [29,] 60  7.078e-01 1.688e-02
## [30,] 60  7.284e-01 1.538e-02
## [31,] 60  7.478e-01 1.401e-02
## [32,] 63  7.659e-01 1.277e-02
## [33,] 63  7.830e-01 1.163e-02
## [34,] 62  7.988e-01 1.060e-02
## [35,] 64  8.140e-01 9.659e-03
## [36,] 68  8.284e-01 8.801e-03
## [37,] 69  8.420e-01 8.019e-03
## [38,] 73  8.548e-01 7.307e-03
## [39,] 76  8.668e-01 6.658e-03
## [40,] 81  8.781e-01 6.066e-03
## [41,] 82  8.887e-01 5.527e-03
## [42,] 81  8.983e-01 5.036e-03
## [43,] 83  9.072e-01 4.589e-03
## [44,] 83  9.151e-01 4.181e-03
## [45,] 83  9.225e-01 3.810e-03
## [46,] 85  9.292e-01 3.471e-03
## [47,] 85  9.354e-01 3.163e-03
## [48,] 85  9.411e-01 2.882e-03
## [49,] 85  9.463e-01 2.626e-03
## [50,] 87  9.510e-01 2.393e-03
## [51,] 86  9.554e-01 2.180e-03
## [52,] 87  9.593e-01 1.986e-03
## [53,] 87  9.629e-01 1.810e-03
## [54,] 87  9.662e-01 1.649e-03
## [55,] 87  9.692e-01 1.503e-03
## [56,] 87  9.719e-01 1.369e-03
## [57,] 88  9.744e-01 1.248e-03
## [58,] 88  9.766e-01 1.137e-03
## [59,] 91  9.787e-01 1.036e-03
## [60,] 91  9.806e-01 9.437e-04
## [61,] 91  9.823e-01 8.599e-04
## [62,] 91  9.839e-01 7.835e-04
## [63,] 91  9.853e-01 7.139e-04
## [64,] 91  9.866e-01 6.505e-04
## [65,] 91  9.878e-01 5.927e-04
## [66,] 92  9.889e-01 5.400e-04
## [67,] 92  9.899e-01 4.920e-04
## [68,] 91  9.908e-01 4.483e-04
## [69,] 91  9.916e-01 4.085e-04
## [70,] 91  9.923e-01 3.722e-04
## [71,] 91  9.930e-01 3.392e-04
## [72,] 91  9.936e-01 3.090e-04
## [73,] 91  9.942e-01 2.816e-04
## [74,] 91  9.947e-01 2.566e-04
## [75,] 91  9.952e-01 2.338e-04
## [76,] 91  9.956e-01 2.130e-04
## [77,] 92  9.960e-01 1.941e-04
## [78,] 92  9.963e-01 1.768e-04
## [79,] 92  9.967e-01 1.611e-04
## [80,] 92  9.970e-01 1.468e-04
## [81,] 92  9.972e-01 1.338e-04
## [82,] 92  9.975e-01 1.219e-04
## [83,] 93  9.977e-01 1.111e-04
## [84,] 92  9.979e-01 1.012e-04
## [85,] 94  9.981e-01 9.220e-05
## [86,] 94  9.983e-01 8.401e-05
## [87,] 94  9.984e-01 7.655e-05
## [88,] 94  9.985e-01 6.975e-05
## [89,] 94  9.987e-01 6.355e-05
## [90,] 96  9.988e-01 5.790e-05
## [91,] 96  9.989e-01 5.276e-05
## [92,] 96  9.990e-01 4.807e-05
## [93,] 96  9.991e-01 4.380e-05
## [1] "lambda min is at location 80"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 149 genes..."
## [1] "evaluation accuracy ElasticNet 0.794642857142857"
## [1] "evaluation accuracy LDA 0.71875"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 187 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 62.5668449197861"
## [1] "Number of genes in the target data, but not in model genes is 4854"
## [1] "Number of genes in the model present in the target data is 146"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 187 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 68.4491978609626"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 140 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 100"
## [1] "Number of genes in the target data, but not in model genes is 4854"
## [1] "Number of genes in the model present in the target data is 146"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 140 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 72.1428571428571"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 133 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 76.6917293233083"
## [1] "Number of genes in the target data, but not in model genes is 4854"
## [1] "Number of genes in the model present in the target data is 146"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 133 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 78.1954887218045"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 40 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 82.5"
## [1] "Number of genes in the target data, but not in model genes is 4854"
## [1] "Number of genes in the model present in the target data is 146"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 40 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 67.5"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 153 genes for training model"
## [1] "use 153 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 3 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 2.352e-01
##  [2,]  1  2.716e-02 2.143e-01
##  [3,]  2  5.502e-02 1.953e-01
##  [4,]  2  8.287e-02 1.779e-01
##  [5,]  2  1.067e-01 1.621e-01
##  [6,]  3  1.356e-01 1.477e-01
##  [7,]  5  1.643e-01 1.346e-01
##  [8,]  6  1.985e-01 1.226e-01
##  [9,]  8  2.314e-01 1.117e-01
## [10,] 10  2.654e-01 1.018e-01
## [11,] 11  2.963e-01 9.277e-02
## [12,] 13  3.261e-01 8.453e-02
## [13,] 13  3.546e-01 7.702e-02
## [14,] 15  3.825e-01 7.017e-02
## [15,] 19  4.100e-01 6.394e-02
## [16,] 22  4.376e-01 5.826e-02
## [17,] 24  4.652e-01 5.308e-02
## [18,] 26  4.909e-01 4.837e-02
## [19,] 28  5.160e-01 4.407e-02
## [20,] 31  5.406e-01 4.016e-02
## [21,] 32  5.639e-01 3.659e-02
## [22,] 34  5.860e-01 3.334e-02
## [23,] 36  6.072e-01 3.038e-02
## [24,] 36  6.275e-01 2.768e-02
## [25,] 39  6.473e-01 2.522e-02
## [26,] 40  6.662e-01 2.298e-02
## [27,] 43  6.840e-01 2.094e-02
## [28,] 44  7.014e-01 1.908e-02
## [29,] 47  7.188e-01 1.738e-02
## [30,] 49  7.367e-01 1.584e-02
## [31,] 51  7.540e-01 1.443e-02
## [32,] 56  7.710e-01 1.315e-02
## [33,] 56  7.877e-01 1.198e-02
## [34,] 58  8.033e-01 1.092e-02
## [35,] 58  8.181e-01 9.947e-03
## [36,] 59  8.323e-01 9.063e-03
## [37,] 62  8.457e-01 8.258e-03
## [38,] 62  8.581e-01 7.525e-03
## [39,] 62  8.696e-01 6.856e-03
## [40,] 63  8.800e-01 6.247e-03
## [41,] 65  8.901e-01 5.692e-03
## [42,] 65  8.991e-01 5.186e-03
## [43,] 67  9.075e-01 4.726e-03
## [44,] 67  9.154e-01 4.306e-03
## [45,] 67  9.225e-01 3.923e-03
## [46,] 71  9.292e-01 3.575e-03
## [47,] 72  9.354e-01 3.257e-03
## [48,] 73  9.410e-01 2.968e-03
## [49,] 73  9.462e-01 2.704e-03
## [50,] 74  9.509e-01 2.464e-03
## [51,] 74  9.552e-01 2.245e-03
## [52,] 75  9.592e-01 2.046e-03
## [53,] 76  9.628e-01 1.864e-03
## [54,] 77  9.661e-01 1.698e-03
## [55,] 77  9.691e-01 1.547e-03
## [56,] 77  9.718e-01 1.410e-03
## [57,] 77  9.743e-01 1.285e-03
## [58,] 80  9.766e-01 1.171e-03
## [59,] 81  9.787e-01 1.067e-03
## [60,] 82  9.806e-01 9.718e-04
## [61,] 82  9.823e-01 8.855e-04
## [62,] 83  9.839e-01 8.068e-04
## [63,] 83  9.853e-01 7.352e-04
## [64,] 83  9.866e-01 6.698e-04
## [65,] 83  9.878e-01 6.103e-04
## [66,] 84  9.889e-01 5.561e-04
## [67,] 84  9.899e-01 5.067e-04
## [68,] 84  9.908e-01 4.617e-04
## [69,] 84  9.916e-01 4.207e-04
## [70,] 84  9.923e-01 3.833e-04
## [71,] 84  9.930e-01 3.493e-04
## [72,] 85  9.936e-01 3.182e-04
## [73,] 85  9.942e-01 2.900e-04
## [74,] 85  9.947e-01 2.642e-04
## [75,] 85  9.952e-01 2.407e-04
## [76,] 86  9.956e-01 2.193e-04
## [77,] 86  9.960e-01 1.999e-04
## [78,] 86  9.964e-01 1.821e-04
## [79,] 86  9.967e-01 1.659e-04
## [80,] 86  9.970e-01 1.512e-04
## [81,] 86  9.972e-01 1.378e-04
## [82,] 86  9.975e-01 1.255e-04
## [83,] 86  9.977e-01 1.144e-04
## [84,] 86  9.979e-01 1.042e-04
## [85,] 86  9.981e-01 9.495e-05
## [86,] 86  9.983e-01 8.651e-05
## [87,] 86  9.984e-01 7.883e-05
## [88,] 86  9.986e-01 7.183e-05
## [89,] 86  9.987e-01 6.544e-05
## [90,] 86  9.988e-01 5.963e-05
## [91,] 86  9.989e-01 5.433e-05
## [92,] 86  9.990e-01 4.951e-05
## [1] "lambda min is at location 32"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 150 genes..."
## [1] "evaluation accuracy ElasticNet 0.857142857142857"
## [1] "evaluation accuracy LDA 0.732142857142857"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 187 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 28.3422459893048"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 187 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 3.74331550802139"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 140 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 68.5714285714286"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 140 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 12.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 133 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 26.3157894736842"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 133 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 3.7593984962406"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 40 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 50"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 40 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 17.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sink">sink</a></span>()</code></pre></div>
</div>
<div id="display-summary-results-for-the-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#display-summary-results-for-the-prediction" class="anchor"></a>4.2 Display summary results for the prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#get the number of rows for the summary matrix </span>
row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

<span class="co">#summary results LDA to to show the percent of cells classified as cells</span>
<span class="co">#belonging by LDA classifier </span>
<span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster )</code></pre></div>
<pre><code>##                 V1               V2                                names
## 1 68.4491978609626 3.74331550802139 LDA for subpop 1 in target mixedpop2
## 2 72.1428571428571 12.8571428571429 LDA for subpop 2 in target mixedpop2
## 3 78.1954887218045  3.7593984962406 LDA for subpop 3 in target mixedpop2
## 4             67.5             17.5 LDA for subpop 4 in target mixedpop2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#summary results Lasso to show the percent of cells classified as cells</span>
<span class="co">#belonging by Lasso classifier</span>
<span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster)</code></pre></div>
<pre><code>##                 V1               V2
## 1 62.5668449197861 28.3422459893048
## 2              100 68.5714285714286
## 3 76.6917293233083 26.3157894736842
## 4             82.5               50
##                                        names
## 1 ElasticNet for subpop1 in target mixedpop2
## 2 ElasticNet for subpop2 in target mixedpop2
## 3 ElasticNet for subpop3 in target mixedpop2
## 4 ElasticNet for subpop4 in target mixedpop2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary maximum deviance explained by the model during the model training </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## $allDeviance
## [1] "0.997" "0.771"
## 
## $DeviMax
##           dat_DE$Dfd          Deviance           DEgenes
## 1                  0             0.997    genes_cluster1
## 2                  1             0.997    genes_cluster1
## 3                  3             0.997    genes_cluster1
## 4                  4             0.997    genes_cluster1
## 5                  5             0.997    genes_cluster1
## 6                  6             0.997    genes_cluster1
## 7                  9             0.997    genes_cluster1
## 8                 10             0.997    genes_cluster1
## 9                 11             0.997    genes_cluster1
## 10                16             0.997    genes_cluster1
## 11                18             0.997    genes_cluster1
## 12                20             0.997    genes_cluster1
## 13                24             0.997    genes_cluster1
## 14                28             0.997    genes_cluster1
## 15                31             0.997    genes_cluster1
## 16                32             0.997    genes_cluster1
## 17                34             0.997    genes_cluster1
## 18                36             0.997    genes_cluster1
## 19                41             0.997    genes_cluster1
## 20                47             0.997    genes_cluster1
## 21                50             0.997    genes_cluster1
## 22                52             0.997    genes_cluster1
## 23                57             0.997    genes_cluster1
## 24                59             0.997    genes_cluster1
## 25                60             0.997    genes_cluster1
## 26                62             0.997    genes_cluster1
## 27                63             0.997    genes_cluster1
## 28                64             0.997    genes_cluster1
## 29                68             0.997    genes_cluster1
## 30                69             0.997    genes_cluster1
## 31                73             0.997    genes_cluster1
## 32                76             0.997    genes_cluster1
## 33                81             0.997    genes_cluster1
## 34                82             0.997    genes_cluster1
## 35                83             0.997    genes_cluster1
## 36                85             0.997    genes_cluster1
## 37                86             0.997    genes_cluster1
## 38                87             0.997    genes_cluster1
## 39                88             0.997    genes_cluster1
## 40                91             0.997    genes_cluster1
## 41                92             0.997    genes_cluster1
## 42 remaining DEgenes remaining DEgenes remaining DEgenes
## 
## $LassoGenesMax
## NULL</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## [1] 71.87500 73.21429</code></pre>
</div>
<div id="plot-the-relationship-between-clusters-in-one-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-one-sample" class="anchor"></a>4.3 Plot the relationship between clusters in one sample</h2>
<p><em>Here we look at one example use case to find relationship between clusters #within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#cluster_mixedpop2 &lt;- as.numeric(as.vector(colData(mixedpop2)[,1]))</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#top 200 gene markers distinguishing cluster 1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] 

LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop2,
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 187 cells as source subpop"
## [1] "Total 313 cells in remaining subpops"
## [1] "subsampling 94 cells for training source subpop"
## [1] "subsampling 94 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 188 cells for testing model"
## [1] "rename remaining subpops to 2_3_4"
## [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 3.364e-15 0.305600
##   [2,]  1 2.396e-02 0.291800
##   [3,]  1 4.585e-02 0.278500
##   [4,]  1 6.589e-02 0.265800
##   [5,]  1 8.429e-02 0.253800
##   [6,]  1 1.012e-01 0.242200
##   [7,]  1 1.168e-01 0.231200
##   [8,]  1 1.312e-01 0.220700
##   [9,]  1 1.444e-01 0.210700
##  [10,]  1 1.567e-01 0.201100
##  [11,]  3 1.693e-01 0.192000
##  [12,]  3 1.880e-01 0.183200
##  [13,]  4 2.063e-01 0.174900
##  [14,]  4 2.254e-01 0.167000
##  [15,]  5 2.436e-01 0.159400
##  [16,]  7 2.618e-01 0.152100
##  [17,]  7 2.796e-01 0.145200
##  [18,]  9 2.974e-01 0.138600
##  [19,] 11 3.163e-01 0.132300
##  [20,] 12 3.350e-01 0.126300
##  [21,] 12 3.528e-01 0.120600
##  [22,] 14 3.705e-01 0.115100
##  [23,] 14 3.877e-01 0.109800
##  [24,] 14 4.039e-01 0.104900
##  [25,] 16 4.199e-01 0.100100
##  [26,] 18 4.363e-01 0.095540
##  [27,] 18 4.520e-01 0.091190
##  [28,] 20 4.672e-01 0.087050
##  [29,] 20 4.820e-01 0.083090
##  [30,] 21 4.962e-01 0.079320
##  [31,] 22 5.099e-01 0.075710
##  [32,] 21 5.228e-01 0.072270
##  [33,] 22 5.358e-01 0.068990
##  [34,] 23 5.483e-01 0.065850
##  [35,] 23 5.605e-01 0.062860
##  [36,] 24 5.727e-01 0.060000
##  [37,] 25 5.847e-01 0.057270
##  [38,] 28 5.966e-01 0.054670
##  [39,] 32 6.100e-01 0.052190
##  [40,] 32 6.236e-01 0.049810
##  [41,] 33 6.370e-01 0.047550
##  [42,] 33 6.498e-01 0.045390
##  [43,] 33 6.620e-01 0.043320
##  [44,] 33 6.737e-01 0.041360
##  [45,] 33 6.850e-01 0.039480
##  [46,] 34 6.959e-01 0.037680
##  [47,] 36 7.066e-01 0.035970
##  [48,] 37 7.171e-01 0.034330
##  [49,] 38 7.277e-01 0.032770
##  [50,] 39 7.380e-01 0.031280
##  [51,] 39 7.481e-01 0.029860
##  [52,] 39 7.577e-01 0.028500
##  [53,] 39 7.669e-01 0.027210
##  [54,] 41 7.760e-01 0.025970
##  [55,] 42 7.847e-01 0.024790
##  [56,] 43 7.933e-01 0.023670
##  [57,] 44 8.017e-01 0.022590
##  [58,] 45 8.098e-01 0.021560
##  [59,] 45 8.177e-01 0.020580
##  [60,] 45 8.251e-01 0.019650
##  [61,] 46 8.324e-01 0.018750
##  [62,] 46 8.393e-01 0.017900
##  [63,] 49 8.461e-01 0.017090
##  [64,] 49 8.526e-01 0.016310
##  [65,] 52 8.589e-01 0.015570
##  [66,] 52 8.650e-01 0.014860
##  [67,] 52 8.708e-01 0.014190
##  [68,] 53 8.763e-01 0.013540
##  [69,] 53 8.816e-01 0.012930
##  [70,] 52 8.867e-01 0.012340
##  [71,] 53 8.915e-01 0.011780
##  [72,] 53 8.962e-01 0.011240
##  [73,] 53 9.006e-01 0.010730
##  [74,] 54 9.049e-01 0.010240
##  [75,] 54 9.090e-01 0.009779
##  [76,] 55 9.130e-01 0.009334
##  [77,] 55 9.167e-01 0.008910
##  [78,] 55 9.203e-01 0.008505
##  [79,] 56 9.238e-01 0.008118
##  [80,] 56 9.271e-01 0.007749
##  [81,] 57 9.303e-01 0.007397
##  [82,] 57 9.333e-01 0.007061
##  [83,] 57 9.362e-01 0.006740
##  [84,] 57 9.390e-01 0.006434
##  [85,] 58 9.417e-01 0.006141
##  [86,] 60 9.442e-01 0.005862
##  [87,] 60 9.467e-01 0.005596
##  [88,] 61 9.490e-01 0.005341
##  [89,] 61 9.513e-01 0.005099
##  [90,] 61 9.535e-01 0.004867
##  [91,] 61 9.555e-01 0.004646
##  [92,] 60 9.575e-01 0.004434
##  [93,] 60 9.594e-01 0.004233
##  [94,] 59 9.612e-01 0.004040
##  [95,] 60 9.629e-01 0.003857
##  [96,] 60 9.646e-01 0.003682
##  [97,] 60 9.661e-01 0.003514
##  [98,] 61 9.677e-01 0.003354
##  [99,] 61 9.691e-01 0.003202
## [100,] 62 9.705e-01 0.003056
## [1] "lambda min is at location 74"
## [1] "the leave-out cells in the source subpop is 93"
## [1] "use 94 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 187 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.882352941176471"
## [1] "evaluation accuracy LDA 0.625668449197861"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 88.2352941176471"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 95.7219251336899"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 83.5714285714286"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 0.75187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 87.218045112782"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 10"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 80"
## [1] "Total 187 cells as source subpop"
## [1] "Total 313 cells in remaining subpops"
## [1] "subsampling 94 cells for training source subpop"
## [1] "subsampling 94 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 188 cells for testing model"
## [1] "rename remaining subpops to 2_3_4"
## [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 3.364e-15 0.278100
##   [2,]  1 1.983e-02 0.265400
##   [3,]  1 3.794e-02 0.253400
##   [4,]  2 5.787e-02 0.241900
##   [5,]  2 7.682e-02 0.230900
##   [6,]  2 9.425e-02 0.220400
##   [7,]  2 1.103e-01 0.210400
##   [8,]  2 1.251e-01 0.200800
##   [9,]  2 1.388e-01 0.191700
##  [10,]  2 1.515e-01 0.183000
##  [11,]  2 1.633e-01 0.174600
##  [12,]  4 1.804e-01 0.166700
##  [13,]  5 1.992e-01 0.159100
##  [14,]  6 2.187e-01 0.151900
##  [15,]  7 2.383e-01 0.145000
##  [16,]  7 2.569e-01 0.138400
##  [17,]  9 2.759e-01 0.132100
##  [18,] 11 2.976e-01 0.126100
##  [19,] 12 3.186e-01 0.120400
##  [20,] 14 3.388e-01 0.114900
##  [21,] 16 3.586e-01 0.109700
##  [22,] 17 3.788e-01 0.104700
##  [23,] 20 3.981e-01 0.099940
##  [24,] 20 4.171e-01 0.095400
##  [25,] 20 4.350e-01 0.091060
##  [26,] 21 4.525e-01 0.086920
##  [27,] 21 4.692e-01 0.082970
##  [28,] 21 4.852e-01 0.079200
##  [29,] 22 5.006e-01 0.075600
##  [30,] 22 5.152e-01 0.072160
##  [31,] 24 5.294e-01 0.068880
##  [32,] 25 5.431e-01 0.065750
##  [33,] 25 5.563e-01 0.062760
##  [34,] 25 5.689e-01 0.059910
##  [35,] 27 5.812e-01 0.057190
##  [36,] 26 5.941e-01 0.054590
##  [37,] 26 6.067e-01 0.052110
##  [38,] 26 6.187e-01 0.049740
##  [39,] 27 6.304e-01 0.047480
##  [40,] 27 6.417e-01 0.045320
##  [41,] 28 6.526e-01 0.043260
##  [42,] 28 6.631e-01 0.041290
##  [43,] 29 6.732e-01 0.039420
##  [44,] 30 6.833e-01 0.037630
##  [45,] 31 6.935e-01 0.035920
##  [46,] 34 7.035e-01 0.034280
##  [47,] 35 7.134e-01 0.032730
##  [48,] 35 7.230e-01 0.031240
##  [49,] 36 7.322e-01 0.029820
##  [50,] 37 7.410e-01 0.028460
##  [51,] 38 7.497e-01 0.027170
##  [52,] 39 7.581e-01 0.025930
##  [53,] 40 7.664e-01 0.024760
##  [54,] 42 7.747e-01 0.023630
##  [55,] 42 7.830e-01 0.022560
##  [56,] 43 7.909e-01 0.021530
##  [57,] 45 7.986e-01 0.020550
##  [58,] 46 8.062e-01 0.019620
##  [59,] 46 8.136e-01 0.018730
##  [60,] 46 8.206e-01 0.017880
##  [61,] 47 8.274e-01 0.017060
##  [62,] 47 8.340e-01 0.016290
##  [63,] 49 8.405e-01 0.015550
##  [64,] 49 8.468e-01 0.014840
##  [65,] 49 8.529e-01 0.014170
##  [66,] 51 8.588e-01 0.013520
##  [67,] 51 8.645e-01 0.012910
##  [68,] 52 8.700e-01 0.012320
##  [69,] 52 8.754e-01 0.011760
##  [70,] 53 8.805e-01 0.011230
##  [71,] 54 8.855e-01 0.010720
##  [72,] 54 8.903e-01 0.010230
##  [73,] 54 8.950e-01 0.009764
##  [74,] 55 8.994e-01 0.009320
##  [75,] 57 9.038e-01 0.008897
##  [76,] 58 9.079e-01 0.008492
##  [77,] 57 9.119e-01 0.008106
##  [78,] 59 9.158e-01 0.007738
##  [79,] 61 9.195e-01 0.007386
##  [80,] 62 9.230e-01 0.007050
##  [81,] 63 9.264e-01 0.006730
##  [82,] 63 9.297e-01 0.006424
##  [83,] 65 9.328e-01 0.006132
##  [84,] 65 9.358e-01 0.005853
##  [85,] 65 9.386e-01 0.005587
##  [86,] 68 9.414e-01 0.005333
##  [87,] 67 9.440e-01 0.005091
##  [88,] 67 9.465e-01 0.004860
##  [89,] 67 9.489e-01 0.004639
##  [90,] 66 9.512e-01 0.004428
##  [91,] 68 9.534e-01 0.004227
##  [92,] 68 9.555e-01 0.004035
##  [93,] 68 9.575e-01 0.003851
##  [94,] 68 9.594e-01 0.003676
##  [95,] 68 9.612e-01 0.003509
##  [96,] 68 9.630e-01 0.003350
##  [97,] 67 9.646e-01 0.003197
##  [98,] 67 9.662e-01 0.003052
##  [99,] 68 9.678e-01 0.002913
## [100,] 69 9.692e-01 0.002781
## [1] "lambda min is at location 47"
## [1] "the leave-out cells in the source subpop is 93"
## [1] "use 94 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 187 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.877005347593583"
## [1] "evaluation accuracy LDA 0.556149732620321"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 81.8181818181818"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 97.3262032085562"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 94.2857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 1.50375939849624"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 92.4812030075188"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 12.5"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 87.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]

LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 140 cells as source subpop"
## [1] "Total 360 cells in remaining subpops"
## [1] "subsampling 70 cells for training source subpop"
## [1] "subsampling 70 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 140 cells for testing model"
## [1] "rename remaining subpops to 1_3_4"
## [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 2.723e-15 0.386700
##   [2,]  1 3.836e-02 0.369100
##   [3,]  1 7.349e-02 0.352400
##   [4,]  1 1.058e-01 0.336300
##   [5,]  1 1.356e-01 0.321100
##   [6,]  1 1.632e-01 0.306500
##   [7,]  1 1.888e-01 0.292500
##   [8,]  3 2.189e-01 0.279200
##   [9,]  3 2.487e-01 0.266500
##  [10,]  3 2.765e-01 0.254400
##  [11,]  3 3.025e-01 0.242900
##  [12,]  3 3.269e-01 0.231800
##  [13,]  3 3.498e-01 0.221300
##  [14,]  4 3.729e-01 0.211200
##  [15,]  4 3.950e-01 0.201600
##  [16,]  5 4.158e-01 0.192500
##  [17,]  5 4.365e-01 0.183700
##  [18,]  6 4.561e-01 0.175400
##  [19,]  6 4.749e-01 0.167400
##  [20,]  6 4.927e-01 0.159800
##  [21,]  6 5.095e-01 0.152500
##  [22,]  6 5.254e-01 0.145600
##  [23,]  6 5.405e-01 0.139000
##  [24,]  6 5.548e-01 0.132700
##  [25,]  8 5.693e-01 0.126600
##  [26,]  8 5.836e-01 0.120900
##  [27,]  8 5.972e-01 0.115400
##  [28,] 10 6.107e-01 0.110100
##  [29,] 10 6.241e-01 0.105100
##  [30,] 11 6.370e-01 0.100400
##  [31,] 12 6.494e-01 0.095790
##  [32,] 12 6.614e-01 0.091440
##  [33,] 13 6.730e-01 0.087280
##  [34,] 13 6.841e-01 0.083310
##  [35,] 13 6.946e-01 0.079530
##  [36,] 13 7.047e-01 0.075910
##  [37,] 14 7.145e-01 0.072460
##  [38,] 14 7.240e-01 0.069170
##  [39,] 14 7.331e-01 0.066020
##  [40,] 14 7.417e-01 0.063020
##  [41,] 14 7.500e-01 0.060160
##  [42,] 14 7.578e-01 0.057420
##  [43,] 15 7.654e-01 0.054810
##  [44,] 16 7.730e-01 0.052320
##  [45,] 17 7.805e-01 0.049950
##  [46,] 17 7.878e-01 0.047680
##  [47,] 19 7.950e-01 0.045510
##  [48,] 20 8.021e-01 0.043440
##  [49,] 20 8.091e-01 0.041470
##  [50,] 21 8.158e-01 0.039580
##  [51,] 22 8.223e-01 0.037780
##  [52,] 23 8.286e-01 0.036060
##  [53,] 24 8.346e-01 0.034430
##  [54,] 24 8.404e-01 0.032860
##  [55,] 24 8.460e-01 0.031370
##  [56,] 25 8.514e-01 0.029940
##  [57,] 25 8.565e-01 0.028580
##  [58,] 25 8.615e-01 0.027280
##  [59,] 27 8.664e-01 0.026040
##  [60,] 27 8.713e-01 0.024860
##  [61,] 27 8.766e-01 0.023730
##  [62,] 27 8.820e-01 0.022650
##  [63,] 27 8.872e-01 0.021620
##  [64,] 28 8.921e-01 0.020640
##  [65,] 28 8.969e-01 0.019700
##  [66,] 29 9.016e-01 0.018800
##  [67,] 30 9.060e-01 0.017950
##  [68,] 30 9.103e-01 0.017130
##  [69,] 30 9.143e-01 0.016350
##  [70,] 33 9.182e-01 0.015610
##  [71,] 33 9.220e-01 0.014900
##  [72,] 32 9.256e-01 0.014220
##  [73,] 31 9.290e-01 0.013580
##  [74,] 32 9.323e-01 0.012960
##  [75,] 32 9.354e-01 0.012370
##  [76,] 32 9.383e-01 0.011810
##  [77,] 32 9.411e-01 0.011270
##  [78,] 32 9.438e-01 0.010760
##  [79,] 33 9.463e-01 0.010270
##  [80,] 33 9.487e-01 0.009804
##  [81,] 34 9.511e-01 0.009359
##  [82,] 34 9.533e-01 0.008933
##  [83,] 35 9.554e-01 0.008527
##  [84,] 35 9.574e-01 0.008140
##  [85,] 35 9.593e-01 0.007770
##  [86,] 35 9.612e-01 0.007417
##  [87,] 35 9.629e-01 0.007080
##  [88,] 35 9.646e-01 0.006758
##  [89,] 35 9.662e-01 0.006451
##  [90,] 35 9.677e-01 0.006157
##  [91,] 35 9.691e-01 0.005878
##  [92,] 35 9.705e-01 0.005610
##  [93,] 35 9.719e-01 0.005355
##  [94,] 35 9.731e-01 0.005112
##  [95,] 35 9.743e-01 0.004880
##  [96,] 35 9.755e-01 0.004658
##  [97,] 35 9.766e-01 0.004446
##  [98,] 35 9.776e-01 0.004244
##  [99,] 35 9.786e-01 0.004051
## [100,] 35 9.796e-01 0.003867
## [1] "lambda min is at location 43"
## [1] "the leave-out cells in the source subpop is 70"
## [1] "use 70 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 140 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.971428571428571"
## [1] "evaluation accuracy LDA 0.9"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 7.48663101604278"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 15.5080213903743"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 99.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 100"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 3.7593984962406"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 24.0601503759398"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 35"
## [1] "Total 140 cells as source subpop"
## [1] "Total 360 cells in remaining subpops"
## [1] "subsampling 70 cells for training source subpop"
## [1] "subsampling 70 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 140 cells for testing model"
## [1] "rename remaining subpops to 1_3_4"
## [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 2.723e-15 0.363100
##   [2,]  1 3.382e-02 0.346600
##   [3,]  1 6.478e-02 0.330800
##   [4,]  1 9.326e-02 0.315800
##   [5,]  2 1.253e-01 0.301400
##   [6,]  3 1.566e-01 0.287700
##   [7,]  5 1.929e-01 0.274600
##   [8,]  5 2.283e-01 0.262200
##   [9,]  5 2.613e-01 0.250200
##  [10,]  6 2.928e-01 0.238900
##  [11,]  6 3.228e-01 0.228000
##  [12,]  6 3.509e-01 0.217700
##  [13,]  7 3.774e-01 0.207800
##  [14,]  7 4.033e-01 0.198300
##  [15,]  8 4.283e-01 0.189300
##  [16,]  8 4.520e-01 0.180700
##  [17,]  8 4.743e-01 0.172500
##  [18,]  8 4.954e-01 0.164600
##  [19,]  8 5.154e-01 0.157200
##  [20,]  8 5.343e-01 0.150000
##  [21,]  9 5.530e-01 0.143200
##  [22,] 11 5.716e-01 0.136700
##  [23,] 12 5.893e-01 0.130500
##  [24,] 12 6.062e-01 0.124500
##  [25,] 12 6.222e-01 0.118900
##  [26,] 12 6.374e-01 0.113500
##  [27,] 13 6.519e-01 0.108300
##  [28,] 13 6.659e-01 0.103400
##  [29,] 14 6.793e-01 0.098700
##  [30,] 14 6.921e-01 0.094220
##  [31,] 14 7.044e-01 0.089930
##  [32,] 14 7.161e-01 0.085850
##  [33,] 14 7.273e-01 0.081940
##  [34,] 15 7.379e-01 0.078220
##  [35,] 16 7.483e-01 0.074660
##  [36,] 17 7.582e-01 0.071270
##  [37,] 17 7.678e-01 0.068030
##  [38,] 17 7.770e-01 0.064940
##  [39,] 17 7.858e-01 0.061990
##  [40,] 18 7.942e-01 0.059170
##  [41,] 18 8.026e-01 0.056480
##  [42,] 19 8.107e-01 0.053910
##  [43,] 19 8.186e-01 0.051460
##  [44,] 21 8.261e-01 0.049120
##  [45,] 21 8.334e-01 0.046890
##  [46,] 21 8.403e-01 0.044760
##  [47,] 23 8.470e-01 0.042730
##  [48,] 23 8.536e-01 0.040780
##  [49,] 23 8.599e-01 0.038930
##  [50,] 24 8.659e-01 0.037160
##  [51,] 24 8.717e-01 0.035470
##  [52,] 24 8.772e-01 0.033860
##  [53,] 24 8.824e-01 0.032320
##  [54,] 24 8.875e-01 0.030850
##  [55,] 24 8.923e-01 0.029450
##  [56,] 24 8.969e-01 0.028110
##  [57,] 24 9.013e-01 0.026830
##  [58,] 24 9.055e-01 0.025610
##  [59,] 24 9.095e-01 0.024450
##  [60,] 24 9.134e-01 0.023340
##  [61,] 24 9.171e-01 0.022280
##  [62,] 25 9.207e-01 0.021260
##  [63,] 25 9.241e-01 0.020300
##  [64,] 25 9.273e-01 0.019380
##  [65,] 24 9.304e-01 0.018500
##  [66,] 25 9.334e-01 0.017650
##  [67,] 26 9.362e-01 0.016850
##  [68,] 26 9.389e-01 0.016090
##  [69,] 26 9.415e-01 0.015350
##  [70,] 26 9.440e-01 0.014660
##  [71,] 26 9.464e-01 0.013990
##  [72,] 27 9.487e-01 0.013350
##  [73,] 27 9.509e-01 0.012750
##  [74,] 28 9.530e-01 0.012170
##  [75,] 28 9.550e-01 0.011620
##  [76,] 28 9.569e-01 0.011090
##  [77,] 28 9.588e-01 0.010580
##  [78,] 27 9.606e-01 0.010100
##  [79,] 27 9.622e-01 0.009643
##  [80,] 27 9.639e-01 0.009205
##  [81,] 27 9.654e-01 0.008787
##  [82,] 27 9.669e-01 0.008387
##  [83,] 27 9.683e-01 0.008006
##  [84,] 27 9.697e-01 0.007642
##  [85,] 27 9.710e-01 0.007295
##  [86,] 27 9.722e-01 0.006963
##  [87,] 27 9.734e-01 0.006647
##  [88,] 27 9.746e-01 0.006345
##  [89,] 27 9.757e-01 0.006056
##  [90,] 27 9.767e-01 0.005781
##  [91,] 27 9.777e-01 0.005518
##  [92,] 27 9.787e-01 0.005267
##  [93,] 27 9.796e-01 0.005028
##  [94,] 28 9.805e-01 0.004800
##  [95,] 28 9.813e-01 0.004581
##  [96,] 28 9.821e-01 0.004373
##  [97,] 29 9.829e-01 0.004174
##  [98,] 29 9.837e-01 0.003985
##  [99,] 29 9.844e-01 0.003804
## [100,] 29 9.851e-01 0.003631
## [1] "lambda min is at location 35"
## [1] "the leave-out cells in the source subpop is 70"
## [1] "use 70 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 140 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.971428571428571"
## [1] "evaluation accuracy LDA 0.942857142857143"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 6.95187165775401"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 6.95187165775401"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 100"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 99.2857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 35.3383458646617"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 3.00751879699248"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 52.5"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 40"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 133 cells as source subpop"
## [1] "Total 367 cells in remaining subpops"
## [1] "subsampling 66 cells for training source subpop"
## [1] "subsampling 66 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 132 cells for testing model"
## [1] "rename remaining subpops to 1_2_4"
## [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev   Lambda
##   [1,]  0 -2.403e-15 0.259300
##   [2,]  2  1.981e-02 0.247500
##   [3,]  2  4.408e-02 0.236200
##   [4,]  3  6.678e-02 0.225500
##   [5,]  4  9.521e-02 0.215200
##   [6,]  4  1.234e-01 0.205500
##   [7,]  4  1.495e-01 0.196100
##   [8,]  5  1.747e-01 0.187200
##   [9,]  6  2.000e-01 0.178700
##  [10,]  6  2.242e-01 0.170600
##  [11,]  6  2.470e-01 0.162800
##  [12,]  8  2.701e-01 0.155400
##  [13,]  8  2.934e-01 0.148400
##  [14,]  9  3.157e-01 0.141600
##  [15,]  9  3.372e-01 0.135200
##  [16,] 10  3.579e-01 0.129000
##  [17,] 10  3.776e-01 0.123200
##  [18,] 11  3.963e-01 0.117600
##  [19,] 11  4.142e-01 0.112200
##  [20,] 11  4.312e-01 0.107100
##  [21,] 12  4.477e-01 0.102300
##  [22,] 14  4.641e-01 0.097610
##  [23,] 14  4.801e-01 0.093170
##  [24,] 15  4.958e-01 0.088940
##  [25,] 15  5.108e-01 0.084890
##  [26,] 16  5.259e-01 0.081030
##  [27,] 17  5.407e-01 0.077350
##  [28,] 18  5.557e-01 0.073840
##  [29,] 19  5.704e-01 0.070480
##  [30,] 21  5.850e-01 0.067280
##  [31,] 23  5.993e-01 0.064220
##  [32,] 23  6.133e-01 0.061300
##  [33,] 25  6.275e-01 0.058510
##  [34,] 28  6.415e-01 0.055850
##  [35,] 31  6.553e-01 0.053320
##  [36,] 32  6.688e-01 0.050890
##  [37,] 33  6.818e-01 0.048580
##  [38,] 34  6.943e-01 0.046370
##  [39,] 34  7.062e-01 0.044260
##  [40,] 34  7.177e-01 0.042250
##  [41,] 34  7.289e-01 0.040330
##  [42,] 34  7.395e-01 0.038500
##  [43,] 35  7.498e-01 0.036750
##  [44,] 38  7.598e-01 0.035080
##  [45,] 38  7.694e-01 0.033480
##  [46,] 39  7.788e-01 0.031960
##  [47,] 42  7.880e-01 0.030510
##  [48,] 44  7.968e-01 0.029120
##  [49,] 43  8.054e-01 0.027800
##  [50,] 45  8.137e-01 0.026540
##  [51,] 45  8.217e-01 0.025330
##  [52,] 46  8.293e-01 0.024180
##  [53,] 47  8.367e-01 0.023080
##  [54,] 47  8.438e-01 0.022030
##  [55,] 47  8.506e-01 0.021030
##  [56,] 46  8.572e-01 0.020070
##  [57,] 46  8.634e-01 0.019160
##  [58,] 46  8.693e-01 0.018290
##  [59,] 47  8.750e-01 0.017460
##  [60,] 47  8.805e-01 0.016660
##  [61,] 48  8.857e-01 0.015910
##  [62,] 48  8.907e-01 0.015180
##  [63,] 48  8.955e-01 0.014490
##  [64,] 47  9.001e-01 0.013840
##  [65,] 47  9.045e-01 0.013210
##  [66,] 48  9.087e-01 0.012610
##  [67,] 48  9.127e-01 0.012030
##  [68,] 49  9.166e-01 0.011490
##  [69,] 48  9.203e-01 0.010960
##  [70,] 48  9.238e-01 0.010470
##  [71,] 51  9.272e-01 0.009990
##  [72,] 51  9.305e-01 0.009536
##  [73,] 52  9.336e-01 0.009103
##  [74,] 52  9.367e-01 0.008689
##  [75,] 52  9.395e-01 0.008294
##  [76,] 52  9.423e-01 0.007917
##  [77,] 52  9.449e-01 0.007557
##  [78,] 53  9.474e-01 0.007214
##  [79,] 53  9.498e-01 0.006886
##  [80,] 53  9.521e-01 0.006573
##  [81,] 53  9.543e-01 0.006274
##  [82,] 53  9.564e-01 0.005989
##  [83,] 54  9.584e-01 0.005717
##  [84,] 54  9.603e-01 0.005457
##  [85,] 52  9.621e-01 0.005209
##  [86,] 52  9.638e-01 0.004972
##  [87,] 52  9.655e-01 0.004746
##  [88,] 52  9.670e-01 0.004531
##  [89,] 53  9.685e-01 0.004325
##  [90,] 54  9.700e-01 0.004128
##  [91,] 54  9.713e-01 0.003940
##  [92,] 54  9.726e-01 0.003761
##  [93,] 54  9.739e-01 0.003590
##  [94,] 54  9.751e-01 0.003427
##  [95,] 54  9.762e-01 0.003271
##  [96,] 54  9.773e-01 0.003123
##  [97,] 54  9.783e-01 0.002981
##  [98,] 55  9.793e-01 0.002845
##  [99,] 54  9.802e-01 0.002716
## [100,] 54  9.811e-01 0.002593
## [1] "lambda min is at location 26"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 66 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.834586466165414"
## [1] "evaluation accuracy LDA 0.75187969924812"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 79.6791443850267"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 14.4385026737968"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 3.57142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 8.27067669172932"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 69.9248120300752"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 5"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 20"
## [1] "Total 133 cells as source subpop"
## [1] "Total 367 cells in remaining subpops"
## [1] "subsampling 66 cells for training source subpop"
## [1] "subsampling 66 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 132 cells for testing model"
## [1] "rename remaining subpops to 1_2_4"
## [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev   Lambda
##   [1,]  0 -2.403e-15 0.221900
##   [2,]  1  1.263e-02 0.211800
##   [3,]  2  2.519e-02 0.202200
##   [4,]  2  4.134e-02 0.193000
##   [5,]  4  5.713e-02 0.184200
##   [6,]  5  7.998e-02 0.175800
##   [7,]  6  1.036e-01 0.167800
##   [8,]  7  1.267e-01 0.160200
##   [9,]  7  1.483e-01 0.152900
##  [10,]  7  1.683e-01 0.146000
##  [11,]  8  1.885e-01 0.139300
##  [12,]  9  2.099e-01 0.133000
##  [13,]  9  2.302e-01 0.127000
##  [14,] 11  2.503e-01 0.121200
##  [15,] 12  2.709e-01 0.115700
##  [16,] 13  2.921e-01 0.110400
##  [17,] 14  3.125e-01 0.105400
##  [18,] 14  3.319e-01 0.100600
##  [19,] 15  3.509e-01 0.096050
##  [20,] 18  3.707e-01 0.091680
##  [21,] 19  3.903e-01 0.087510
##  [22,] 22  4.103e-01 0.083540
##  [23,] 24  4.304e-01 0.079740
##  [24,] 24  4.496e-01 0.076110
##  [25,] 24  4.678e-01 0.072650
##  [26,] 25  4.848e-01 0.069350
##  [27,] 26  5.009e-01 0.066200
##  [28,] 27  5.165e-01 0.063190
##  [29,] 28  5.319e-01 0.060320
##  [30,] 29  5.473e-01 0.057580
##  [31,] 29  5.620e-01 0.054960
##  [32,] 30  5.764e-01 0.052460
##  [33,] 31  5.904e-01 0.050080
##  [34,] 32  6.060e-01 0.047800
##  [35,] 32  6.210e-01 0.045630
##  [36,] 33  6.354e-01 0.043560
##  [37,] 34  6.494e-01 0.041580
##  [38,] 34  6.627e-01 0.039690
##  [39,] 34  6.754e-01 0.037880
##  [40,] 35  6.877e-01 0.036160
##  [41,] 34  6.993e-01 0.034520
##  [42,] 34  7.105e-01 0.032950
##  [43,] 36  7.214e-01 0.031450
##  [44,] 37  7.321e-01 0.030020
##  [45,] 38  7.425e-01 0.028660
##  [46,] 38  7.525e-01 0.027350
##  [47,] 38  7.622e-01 0.026110
##  [48,] 38  7.714e-01 0.024920
##  [49,] 39  7.803e-01 0.023790
##  [50,] 39  7.889e-01 0.022710
##  [51,] 40  7.972e-01 0.021680
##  [52,] 41  8.051e-01 0.020690
##  [53,] 41  8.128e-01 0.019750
##  [54,] 41  8.201e-01 0.018850
##  [55,] 42  8.273e-01 0.018000
##  [56,] 44  8.343e-01 0.017180
##  [57,] 46  8.410e-01 0.016400
##  [58,] 48  8.476e-01 0.015650
##  [59,] 48  8.539e-01 0.014940
##  [60,] 49  8.600e-01 0.014260
##  [61,] 51  8.659e-01 0.013610
##  [62,] 52  8.716e-01 0.013000
##  [63,] 52  8.771e-01 0.012400
##  [64,] 52  8.823e-01 0.011840
##  [65,] 53  8.874e-01 0.011300
##  [66,] 54  8.923e-01 0.010790
##  [67,] 55  8.969e-01 0.010300
##  [68,] 55  9.014e-01 0.009831
##  [69,] 55  9.057e-01 0.009384
##  [70,] 55  9.098e-01 0.008957
##  [71,] 55  9.138e-01 0.008550
##  [72,] 55  9.175e-01 0.008161
##  [73,] 56  9.211e-01 0.007791
##  [74,] 58  9.246e-01 0.007436
##  [75,] 57  9.280e-01 0.007098
##  [76,] 57  9.312e-01 0.006776
##  [77,] 57  9.343e-01 0.006468
##  [78,] 58  9.373e-01 0.006174
##  [79,] 58  9.401e-01 0.005893
##  [80,] 58  9.428e-01 0.005625
##  [81,] 58  9.454e-01 0.005370
##  [82,] 58  9.478e-01 0.005126
##  [83,] 58  9.502e-01 0.004893
##  [84,] 59  9.524e-01 0.004670
##  [85,] 59  9.546e-01 0.004458
##  [86,] 60  9.566e-01 0.004255
##  [87,] 60  9.586e-01 0.004062
##  [88,] 60  9.605e-01 0.003877
##  [89,] 60  9.623e-01 0.003701
##  [90,] 61  9.640e-01 0.003533
##  [91,] 61  9.656e-01 0.003372
##  [92,] 61  9.672e-01 0.003219
##  [93,] 61  9.687e-01 0.003073
##  [94,] 62  9.701e-01 0.002933
##  [95,] 62  9.715e-01 0.002800
##  [96,] 62  9.727e-01 0.002673
##  [97,] 63  9.740e-01 0.002551
##  [98,] 63  9.752e-01 0.002435
##  [99,] 63  9.763e-01 0.002324
## [100,] 63  9.774e-01 0.002219
## [1] "lambda min is at location 34"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 66 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.849624060150376"
## [1] "evaluation accuracy LDA 0.759398496240602"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 64.1711229946524"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 1.60427807486631"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 1.42857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 0"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 81.9548872180451"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 11.2781954887218"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 20"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 2.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">4</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop4vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat4 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2,
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 40 cells as source subpop"
## [1] "Total 460 cells in remaining subpops"
## [1] "subsampling 20 cells for training source subpop"
## [1] "subsampling 20 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 40 cells for testing model"
## [1] "rename remaining subpops to 1_2_3"
## [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df    %Dev   Lambda
##   [1,]  0 0.00000 0.254100
##   [2,]  1 0.01656 0.242600
##   [3,]  1 0.03170 0.231600
##   [4,]  1 0.04556 0.221000
##   [5,]  2 0.06211 0.211000
##   [6,]  2 0.08083 0.201400
##   [7,]  3 0.10050 0.192200
##   [8,]  5 0.12430 0.183500
##   [9,]  6 0.14960 0.175200
##  [10,]  8 0.17550 0.167200
##  [11,]  8 0.20060 0.159600
##  [12,]  9 0.22460 0.152300
##  [13,] 10 0.24950 0.145400
##  [14,] 10 0.27660 0.138800
##  [15,] 12 0.30390 0.132500
##  [16,] 14 0.33310 0.126500
##  [17,] 14 0.36290 0.120700
##  [18,] 14 0.39160 0.115200
##  [19,] 15 0.41870 0.110000
##  [20,] 16 0.44530 0.105000
##  [21,] 16 0.47090 0.100200
##  [22,] 16 0.49510 0.095680
##  [23,] 17 0.51860 0.091330
##  [24,] 17 0.54190 0.087180
##  [25,] 17 0.56370 0.083220
##  [26,] 17 0.58460 0.079430
##  [27,] 17 0.60430 0.075820
##  [28,] 18 0.62310 0.072380
##  [29,] 19 0.64080 0.069090
##  [30,] 19 0.65770 0.065950
##  [31,] 20 0.67370 0.062950
##  [32,] 20 0.68900 0.060090
##  [33,] 20 0.70380 0.057360
##  [34,] 20 0.71780 0.054750
##  [35,] 21 0.73120 0.052260
##  [36,] 21 0.74390 0.049890
##  [37,] 21 0.75600 0.047620
##  [38,] 20 0.76750 0.045460
##  [39,] 20 0.77830 0.043390
##  [40,] 20 0.78870 0.041420
##  [41,] 20 0.79850 0.039530
##  [42,] 21 0.80780 0.037740
##  [43,] 21 0.81680 0.036020
##  [44,] 21 0.82520 0.034390
##  [45,] 22 0.83340 0.032820
##  [46,] 23 0.84120 0.031330
##  [47,] 23 0.84860 0.029910
##  [48,] 23 0.85570 0.028550
##  [49,] 23 0.86240 0.027250
##  [50,] 23 0.86880 0.026010
##  [51,] 23 0.87490 0.024830
##  [52,] 24 0.88070 0.023700
##  [53,] 24 0.88630 0.022620
##  [54,] 24 0.89150 0.021600
##  [55,] 24 0.89660 0.020610
##  [56,] 24 0.90140 0.019680
##  [57,] 24 0.90590 0.018780
##  [58,] 24 0.91030 0.017930
##  [59,] 24 0.91440 0.017110
##  [60,] 24 0.91830 0.016340
##  [61,] 24 0.92210 0.015590
##  [62,] 24 0.92570 0.014880
##  [63,] 24 0.92910 0.014210
##  [64,] 25 0.93230 0.013560
##  [65,] 25 0.93540 0.012950
##  [66,] 24 0.93840 0.012360
##  [67,] 24 0.94120 0.011800
##  [68,] 24 0.94390 0.011260
##  [69,] 24 0.94650 0.010750
##  [70,] 25 0.94890 0.010260
##  [71,] 25 0.95120 0.009793
##  [72,] 25 0.95350 0.009348
##  [73,] 25 0.95560 0.008923
##  [74,] 25 0.95760 0.008518
##  [75,] 25 0.95960 0.008130
##  [76,] 26 0.96140 0.007761
##  [77,] 26 0.96320 0.007408
##  [78,] 26 0.96490 0.007071
##  [79,] 26 0.96650 0.006750
##  [80,] 27 0.96800 0.006443
##  [81,] 28 0.96950 0.006150
##  [82,] 28 0.97090 0.005871
##  [83,] 27 0.97220 0.005604
##  [84,] 27 0.97350 0.005349
##  [85,] 27 0.97470 0.005106
##  [86,] 27 0.97580 0.004874
##  [87,] 27 0.97690 0.004653
##  [88,] 27 0.97800 0.004441
##  [89,] 27 0.97900 0.004239
##  [90,] 27 0.97990 0.004047
##  [91,] 27 0.98080 0.003863
##  [92,] 27 0.98170 0.003687
##  [93,] 28 0.98250 0.003519
##  [94,] 28 0.98330 0.003359
##  [95,] 29 0.98410 0.003207
##  [96,] 29 0.98480 0.003061
##  [97,] 29 0.98550 0.002922
##  [98,] 29 0.98610 0.002789
##  [99,] 30 0.98680 0.002662
## [100,] 30 0.98740 0.002541
## [1] "lambda min is at location 38"
## [1] "the leave-out cells in the source subpop is 20"
## [1] "use 20 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 40 cells and 199 genes..."
## [1] "evaluation accuracy ElasticNet 0.8"
## [1] "evaluation accuracy LDA 0.75"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 199 genes and 187 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 56.1497326203209"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 199 genes and 187 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 20.3208556149733"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 199 genes and 140 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 74.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 199 genes and 140 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 7.14285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 199 genes and 133 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 84.9624060150376"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 199 genes and 133 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 25.5639097744361"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 199 genes and 40 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 97.5"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 199 genes and 40 cells. The trained model has 199 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 70"
## [1] "Total 40 cells as source subpop"
## [1] "Total 460 cells in remaining subpops"
## [1] "subsampling 20 cells for training source subpop"
## [1] "subsampling 20 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 40 cells for testing model"
## [1] "rename remaining subpops to 1_2_3"
## [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df    %Dev   Lambda
##   [1,]  0 0.00000 0.240900
##   [2,]  1 0.01488 0.229900
##   [3,]  3 0.03177 0.219500
##   [4,]  4 0.05563 0.209500
##   [5,]  5 0.08295 0.200000
##   [6,]  5 0.10990 0.190900
##   [7,]  6 0.13500 0.182200
##   [8,]  6 0.15930 0.173900
##   [9,]  6 0.18180 0.166000
##  [10,]  6 0.20290 0.158500
##  [11,]  7 0.22410 0.151300
##  [12,]  8 0.24690 0.144400
##  [13,]  9 0.26980 0.137800
##  [14,] 11 0.29340 0.131600
##  [15,] 13 0.32000 0.125600
##  [16,] 16 0.34660 0.119900
##  [17,] 17 0.37210 0.114400
##  [18,] 18 0.39780 0.109200
##  [19,] 18 0.42240 0.104300
##  [20,] 18 0.44570 0.099530
##  [21,] 19 0.46760 0.095010
##  [22,] 20 0.48900 0.090690
##  [23,] 21 0.51050 0.086570
##  [24,] 21 0.53180 0.082630
##  [25,] 22 0.55190 0.078880
##  [26,] 21 0.57100 0.075290
##  [27,] 23 0.59020 0.071870
##  [28,] 22 0.60910 0.068600
##  [29,] 22 0.62700 0.065480
##  [30,] 22 0.64390 0.062510
##  [31,] 23 0.66010 0.059670
##  [32,] 23 0.67540 0.056950
##  [33,] 24 0.69070 0.054370
##  [34,] 24 0.70550 0.051900
##  [35,] 24 0.71950 0.049540
##  [36,] 24 0.73280 0.047280
##  [37,] 24 0.74540 0.045140
##  [38,] 24 0.75740 0.043080
##  [39,] 24 0.76880 0.041130
##  [40,] 24 0.77960 0.039260
##  [41,] 24 0.78990 0.037470
##  [42,] 24 0.79960 0.035770
##  [43,] 24 0.80890 0.034140
##  [44,] 24 0.81780 0.032590
##  [45,] 24 0.82620 0.031110
##  [46,] 24 0.83420 0.029700
##  [47,] 24 0.84190 0.028350
##  [48,] 24 0.84910 0.027060
##  [49,] 24 0.85610 0.025830
##  [50,] 24 0.86270 0.024650
##  [51,] 25 0.86900 0.023530
##  [52,] 25 0.87500 0.022460
##  [53,] 25 0.88080 0.021440
##  [54,] 25 0.88630 0.020470
##  [55,] 26 0.89160 0.019540
##  [56,] 26 0.89660 0.018650
##  [57,] 26 0.90130 0.017800
##  [58,] 26 0.90590 0.016990
##  [59,] 27 0.91020 0.016220
##  [60,] 27 0.91440 0.015480
##  [61,] 27 0.91840 0.014780
##  [62,] 27 0.92220 0.014110
##  [63,] 27 0.92580 0.013470
##  [64,] 27 0.92920 0.012850
##  [65,] 27 0.93250 0.012270
##  [66,] 27 0.93560 0.011710
##  [67,] 27 0.93860 0.011180
##  [68,] 28 0.94140 0.010670
##  [69,] 28 0.94410 0.010190
##  [70,] 29 0.94670 0.009724
##  [71,] 29 0.94920 0.009282
##  [72,] 29 0.95150 0.008860
##  [73,] 29 0.95380 0.008458
##  [74,] 29 0.95590 0.008073
##  [75,] 29 0.95790 0.007706
##  [76,] 29 0.95980 0.007356
##  [77,] 29 0.96170 0.007022
##  [78,] 29 0.96340 0.006703
##  [79,] 29 0.96510 0.006398
##  [80,] 29 0.96670 0.006107
##  [81,] 29 0.96820 0.005830
##  [82,] 29 0.96970 0.005565
##  [83,] 28 0.97110 0.005312
##  [84,] 28 0.97240 0.005070
##  [85,] 28 0.97360 0.004840
##  [86,] 28 0.97480 0.004620
##  [87,] 27 0.97600 0.004410
##  [88,] 27 0.97710 0.004209
##  [89,] 27 0.97810 0.004018
##  [90,] 27 0.97910 0.003835
##  [91,] 27 0.98010 0.003661
##  [92,] 27 0.98100 0.003495
##  [93,] 27 0.98180 0.003336
##  [94,] 28 0.98270 0.003184
##  [95,] 28 0.98350 0.003040
##  [96,] 29 0.98420 0.002901
##  [97,] 29 0.98490 0.002769
##  [98,] 29 0.98560 0.002644
##  [99,] 29 0.98630 0.002523
## [100,] 29 0.98690 0.002409
## [1] "lambda min is at location 26"
## [1] "the leave-out cells in the source subpop is 20"
## [1] "use 20 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 40 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.775"
## [1] "evaluation accuracy LDA 0.675"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 44.3850267379679"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 6.41711229946524"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 2.85714285714286"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 0.714285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 0.75187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 5.26315789473684"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 62.5"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 50"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#prepare table input for sankey plot </span>

LASSO_C1S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#7570b3"</span>) 

LASSO_C2S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#1b9e77"</span>)

LASSO_C3S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#e7298a"</span>)
                          
LASSO_C4S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">4</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat4, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#00FFFF"</span>)

combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2, LASSO_C4S2 )
combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)], 
    <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))]) 

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))
Color &lt;-<span class="st"> </span>combined <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/tally.html">count</a></span>(Node, <span class="dt">color=</span>NodeGroup) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="dv">2</span>)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color<span class="op">$</span>color

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))
p1&lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/networkD3/topics/sankeyNetwork">sankeyNetwork</a></span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,
    <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>,
    <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, <span class="dt">fontSize =</span> <span class="dv">22</span> )
p1</code></pre></div>
<div id="htmlwidget-addbc122ebaf08bf6972" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-addbc122ebaf08bf6972">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2,3,3,3,3],"target":[0,1,2,3,0,1,2,3,0,1,2,3,0,1,2,3],"value":[85.0267379679144,0,1.12781954887218,11.25,7.2192513368984,99.6428571428572,19.5488721804511,46.25,71.9251336898396,0.714285714285715,45.1127819548872,12.5,50.2673796791444,38.5714285714286,42.8571428571429,80],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a","#00FFFF","#00FFFF","#00FFFF","#00FFFF"]},"nodes":{"name":["C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#7570b3","#1b9e77","#e7298a","#00FFFF"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters-in-two-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-two-samples" class="anchor"></a>4.3 Plot the relationship between clusters in two samples</h2>
<p><em>Here we look at one example use case to find relationship between clusters</em> <em>within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#top 200 gene markers distinguishing cluster 1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] 
LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 63 genes for training model"
## [1] "use 63 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 2.447e-01
##  [2,]  1  2.940e-02 2.230e-01
##  [3,]  1  5.413e-02 2.032e-01
##  [4,]  1  7.513e-02 1.851e-01
##  [5,]  2  9.580e-02 1.687e-01
##  [6,]  3  1.180e-01 1.537e-01
##  [7,]  4  1.489e-01 1.400e-01
##  [8,]  5  1.806e-01 1.276e-01
##  [9,]  7  2.161e-01 1.163e-01
## [10,]  7  2.479e-01 1.059e-01
## [11,]  7  2.760e-01 9.652e-02
## [12,]  9  3.017e-01 8.795e-02
## [13,] 11  3.287e-01 8.013e-02
## [14,] 11  3.541e-01 7.301e-02
## [15,] 11  3.767e-01 6.653e-02
## [16,] 12  3.976e-01 6.062e-02
## [17,] 15  4.205e-01 5.523e-02
## [18,] 16  4.423e-01 5.033e-02
## [19,] 16  4.622e-01 4.586e-02
## [20,] 16  4.800e-01 4.178e-02
## [21,] 16  4.959e-01 3.807e-02
## [22,] 17  5.103e-01 3.469e-02
## [23,] 20  5.246e-01 3.161e-02
## [24,] 23  5.389e-01 2.880e-02
## [25,] 24  5.522e-01 2.624e-02
## [26,] 25  5.650e-01 2.391e-02
## [27,] 26  5.769e-01 2.178e-02
## [28,] 27  5.876e-01 1.985e-02
## [29,] 29  5.983e-01 1.809e-02
## [30,] 29  6.081e-01 1.648e-02
## [31,] 30  6.169e-01 1.502e-02
## [32,] 30  6.248e-01 1.368e-02
## [33,] 32  6.321e-01 1.247e-02
## [34,] 34  6.400e-01 1.136e-02
## [35,] 36  6.481e-01 1.035e-02
## [36,] 37  6.560e-01 9.430e-03
## [37,] 41  6.636e-01 8.592e-03
## [38,] 43  6.714e-01 7.829e-03
## [39,] 45  6.785e-01 7.134e-03
## [40,] 45  6.849e-01 6.500e-03
## [41,] 47  6.907e-01 5.922e-03
## [42,] 49  6.963e-01 5.396e-03
## [43,] 49  7.015e-01 4.917e-03
## [44,] 51  7.068e-01 4.480e-03
## [45,] 54  7.119e-01 4.082e-03
## [46,] 56  7.167e-01 3.719e-03
## [47,] 57  7.214e-01 3.389e-03
## [48,] 58  7.257e-01 3.088e-03
## [49,] 58  7.296e-01 2.814e-03
## [50,] 58  7.330e-01 2.564e-03
## [51,] 58  7.362e-01 2.336e-03
## [52,] 57  7.390e-01 2.128e-03
## [53,] 58  7.416e-01 1.939e-03
## [54,] 57  7.439e-01 1.767e-03
## [55,] 57  7.460e-01 1.610e-03
## [56,] 57  7.479e-01 1.467e-03
## [57,] 58  7.496e-01 1.337e-03
## [58,] 58  7.512e-01 1.218e-03
## [59,] 58  7.525e-01 1.110e-03
## [60,] 58  7.537e-01 1.011e-03
## [61,] 59  7.548e-01 9.213e-04
## [62,] 59  7.558e-01 8.395e-04
## [63,] 60  7.568e-01 7.649e-04
## [64,] 59  7.576e-01 6.970e-04
## [65,] 59  7.583e-01 6.350e-04
## [66,] 61  7.590e-01 5.786e-04
## [67,] 61  7.596e-01 5.272e-04
## [68,] 60  7.602e-01 4.804e-04
## [69,] 60  7.607e-01 4.377e-04
## [70,] 60  7.611e-01 3.988e-04
## [71,] 60  7.615e-01 3.634e-04
## [72,] 60  7.618e-01 3.311e-04
## [73,] 60  7.621e-01 3.017e-04
## [74,] 60  7.623e-01 2.749e-04
## [75,] 60  7.625e-01 2.505e-04
## [76,] 60  7.627e-01 2.282e-04
## [77,] 61  7.629e-01 2.079e-04
## [78,] 61  7.630e-01 1.895e-04
## [79,] 60  7.631e-01 1.726e-04
## [80,] 60  7.632e-01 1.573e-04
## [81,] 60  7.633e-01 1.433e-04
## [82,] 60  7.634e-01 1.306e-04
## [83,] 60  7.635e-01 1.190e-04
## [84,] 60  7.635e-01 1.084e-04
## [85,] 61  7.636e-01 9.879e-05
## [86,] 61  7.636e-01 9.002e-05
## [87,] 61  7.637e-01 8.202e-05
## [88,] 62  7.637e-01 7.473e-05
## [89,] 62  7.637e-01 6.809e-05
## [90,] 62  7.637e-01 6.204e-05
## [91,] 62  7.638e-01 5.653e-05
## [92,] 62  7.638e-01 5.151e-05
## [1] "lambda min is at location 20"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 62 genes..."
## [1] "evaluation accuracy ElasticNet 0.808035714285714"
## [1] "evaluation accuracy LDA 0.763392857142857"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 58.8235294117647"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 63.1016042780749"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 94.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 71.4285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 72.9323308270677"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 70.6766917293233"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 95"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 67.5"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 63 genes for training model"
## [1] "use 63 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev    Lambda
##   [1,]  0 -2.563e-15 2.712e-01
##   [2,]  1  3.615e-02 2.471e-01
##   [3,]  1  6.672e-02 2.252e-01
##   [4,]  2  9.663e-02 2.052e-01
##   [5,]  2  1.276e-01 1.869e-01
##   [6,]  3  1.597e-01 1.703e-01
##   [7,]  3  1.909e-01 1.552e-01
##   [8,]  4  2.196e-01 1.414e-01
##   [9,]  6  2.503e-01 1.289e-01
##  [10,]  6  2.828e-01 1.174e-01
##  [11,]  6  3.118e-01 1.070e-01
##  [12,]  7  3.409e-01 9.747e-02
##  [13,]  7  3.683e-01 8.881e-02
##  [14,]  8  3.931e-01 8.092e-02
##  [15,]  8  4.168e-01 7.373e-02
##  [16,]  9  4.385e-01 6.718e-02
##  [17,] 10  4.606e-01 6.121e-02
##  [18,] 11  4.825e-01 5.578e-02
##  [19,] 12  5.028e-01 5.082e-02
##  [20,] 13  5.215e-01 4.631e-02
##  [21,] 15  5.395e-01 4.219e-02
##  [22,] 15  5.568e-01 3.844e-02
##  [23,] 16  5.728e-01 3.503e-02
##  [24,] 17  5.876e-01 3.192e-02
##  [25,] 19  6.020e-01 2.908e-02
##  [26,] 21  6.164e-01 2.650e-02
##  [27,] 23  6.298e-01 2.414e-02
##  [28,] 25  6.424e-01 2.200e-02
##  [29,] 26  6.545e-01 2.005e-02
##  [30,] 26  6.656e-01 1.826e-02
##  [31,] 26  6.758e-01 1.664e-02
##  [32,] 28  6.856e-01 1.516e-02
##  [33,] 29  6.955e-01 1.382e-02
##  [34,] 29  7.050e-01 1.259e-02
##  [35,] 29  7.139e-01 1.147e-02
##  [36,] 30  7.220e-01 1.045e-02
##  [37,] 34  7.305e-01 9.523e-03
##  [38,] 36  7.393e-01 8.677e-03
##  [39,] 37  7.479e-01 7.906e-03
##  [40,] 37  7.561e-01 7.204e-03
##  [41,] 39  7.642e-01 6.564e-03
##  [42,] 43  7.732e-01 5.981e-03
##  [43,] 43  7.821e-01 5.449e-03
##  [44,] 45  7.909e-01 4.965e-03
##  [45,] 45  7.992e-01 4.524e-03
##  [46,] 45  8.070e-01 4.122e-03
##  [47,] 46  8.142e-01 3.756e-03
##  [48,] 46  8.211e-01 3.422e-03
##  [49,] 46  8.275e-01 3.118e-03
##  [50,] 47  8.336e-01 2.841e-03
##  [51,] 47  8.394e-01 2.589e-03
##  [52,] 47  8.449e-01 2.359e-03
##  [53,] 48  8.503e-01 2.149e-03
##  [54,] 48  8.555e-01 1.958e-03
##  [55,] 48  8.607e-01 1.784e-03
##  [56,] 50  8.660e-01 1.626e-03
##  [57,] 53  8.716e-01 1.481e-03
##  [58,] 54  8.777e-01 1.350e-03
##  [59,] 54  8.838e-01 1.230e-03
##  [60,] 54  8.898e-01 1.121e-03
##  [61,] 53  8.956e-01 1.021e-03
##  [62,] 54  9.014e-01 9.304e-04
##  [63,] 54  9.072e-01 8.478e-04
##  [64,] 54  9.131e-01 7.724e-04
##  [65,] 54  9.190e-01 7.038e-04
##  [66,] 55  9.249e-01 6.413e-04
##  [67,] 56  9.308e-01 5.843e-04
##  [68,] 56  9.369e-01 5.324e-04
##  [69,] 56  9.425e-01 4.851e-04
##  [70,] 56  9.477e-01 4.420e-04
##  [71,] 56  9.523e-01 4.028e-04
##  [72,] 56  9.567e-01 3.670e-04
##  [73,] 55  9.607e-01 3.344e-04
##  [74,] 57  9.643e-01 3.047e-04
##  [75,] 57  9.676e-01 2.776e-04
##  [76,] 58  9.706e-01 2.529e-04
##  [77,] 59  9.733e-01 2.305e-04
##  [78,] 59  9.757e-01 2.100e-04
##  [79,] 59  9.780e-01 1.913e-04
##  [80,] 59  9.800e-01 1.743e-04
##  [81,] 59  9.818e-01 1.589e-04
##  [82,] 59  9.834e-01 1.447e-04
##  [83,] 59  9.849e-01 1.319e-04
##  [84,] 59  9.863e-01 1.202e-04
##  [85,] 60  9.875e-01 1.095e-04
##  [86,] 60  9.886e-01 9.977e-05
##  [87,] 60  9.896e-01 9.090e-05
##  [88,] 60  9.905e-01 8.283e-05
##  [89,] 60  9.914e-01 7.547e-05
##  [90,] 59  9.922e-01 6.876e-05
##  [91,] 59  9.928e-01 6.266e-05
##  [92,] 59  9.935e-01 5.709e-05
##  [93,] 59  9.940e-01 5.202e-05
##  [94,] 59  9.946e-01 4.740e-05
##  [95,] 59  9.950e-01 4.319e-05
##  [96,] 59  9.955e-01 3.935e-05
##  [97,] 59  9.959e-01 3.585e-05
##  [98,] 59  9.962e-01 3.267e-05
##  [99,] 59  9.965e-01 2.977e-05
## [100,] 59  9.968e-01 2.712e-05
## [1] "lambda min is at location 29"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 62 genes..."
## [1] "evaluation accuracy ElasticNet 0.830357142857143"
## [1] "evaluation accuracy LDA 0.767857142857143"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 41.1764705882353"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 3.74331550802139"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 97.8571428571428"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 5"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 55.6390977443609"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 3.7593984962406"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 80"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 10"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1,
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 203 cells as source subpop"
## [1] "Total 387 cells in remaining subpops"
## [1] "subsampling 102 cells for training source subpop"
## [1] "subsampling 102 cells in remaining subpops for training"
## [1] "use 60 genes for training model"
## [1] "use 60 genes 204 cells for testing model"
## [1] "rename remaining subpops to 1_3"
## [1] "there are 102 cells in class 1_3 and 102 cells in class 2"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev    Lambda
##  [1,]  0 1.281e-15 2.427e-01
##  [2,]  1 2.895e-02 2.211e-01
##  [3,]  1 5.339e-02 2.015e-01
##  [4,]  1 7.433e-02 1.836e-01
##  [5,]  2 9.739e-02 1.673e-01
##  [6,]  3 1.208e-01 1.524e-01
##  [7,]  4 1.470e-01 1.389e-01
##  [8,]  5 1.727e-01 1.265e-01
##  [9,]  5 1.957e-01 1.153e-01
## [10,]  5 2.159e-01 1.051e-01
## [11,]  8 2.373e-01 9.573e-02
## [12,] 10 2.614e-01 8.723e-02
## [13,] 10 2.833e-01 7.948e-02
## [14,] 10 3.026e-01 7.242e-02
## [15,] 11 3.197e-01 6.598e-02
## [16,] 12 3.353e-01 6.012e-02
## [17,] 14 3.512e-01 5.478e-02
## [18,] 15 3.675e-01 4.991e-02
## [19,] 16 3.828e-01 4.548e-02
## [20,] 18 3.979e-01 4.144e-02
## [21,] 18 4.122e-01 3.776e-02
## [22,] 19 4.256e-01 3.440e-02
## [23,] 21 4.384e-01 3.135e-02
## [24,] 21 4.500e-01 2.856e-02
## [25,] 21 4.604e-01 2.603e-02
## [26,] 23 4.710e-01 2.371e-02
## [27,] 24 4.813e-01 2.161e-02
## [28,] 25 4.912e-01 1.969e-02
## [29,] 26 5.004e-01 1.794e-02
## [30,] 28 5.092e-01 1.634e-02
## [31,] 29 5.178e-01 1.489e-02
## [32,] 31 5.262e-01 1.357e-02
## [33,] 32 5.342e-01 1.236e-02
## [34,] 34 5.418e-01 1.127e-02
## [35,] 35 5.498e-01 1.026e-02
## [36,] 36 5.571e-01 9.353e-03
## [37,] 38 5.642e-01 8.522e-03
## [38,] 38 5.710e-01 7.765e-03
## [39,] 39 5.772e-01 7.075e-03
## [40,] 39 5.826e-01 6.447e-03
## [41,] 40 5.877e-01 5.874e-03
## [42,] 41 5.924e-01 5.352e-03
## [43,] 42 5.968e-01 4.877e-03
## [44,] 43 6.012e-01 4.443e-03
## [45,] 44 6.053e-01 4.049e-03
## [46,] 46 6.091e-01 3.689e-03
## [47,] 46 6.124e-01 3.361e-03
## [48,] 47 6.154e-01 3.063e-03
## [49,] 48 6.182e-01 2.791e-03
## [50,] 48 6.207e-01 2.543e-03
## [51,] 48 6.229e-01 2.317e-03
## [52,] 49 6.249e-01 2.111e-03
## [53,] 50 6.266e-01 1.923e-03
## [54,] 52 6.283e-01 1.753e-03
## [55,] 52 6.298e-01 1.597e-03
## [56,] 52 6.311e-01 1.455e-03
## [57,] 53 6.323e-01 1.326e-03
## [58,] 54 6.333e-01 1.208e-03
## [59,] 55 6.342e-01 1.101e-03
## [60,] 55 6.351e-01 1.003e-03
## [61,] 55 6.359e-01 9.138e-04
## [62,] 56 6.365e-01 8.326e-04
## [63,] 56 6.370e-01 7.586e-04
## [64,] 56 6.375e-01 6.912e-04
## [65,] 57 6.379e-01 6.298e-04
## [66,] 57 6.383e-01 5.739e-04
## [67,] 57 6.387e-01 5.229e-04
## [68,] 57 6.389e-01 4.764e-04
## [69,] 57 6.392e-01 4.341e-04
## [70,] 57 6.394e-01 3.956e-04
## [71,] 58 6.396e-01 3.604e-04
## [72,] 58 6.398e-01 3.284e-04
## [73,] 58 6.399e-01 2.992e-04
## [74,] 58 6.400e-01 2.726e-04
## [75,] 58 6.401e-01 2.484e-04
## [76,] 58 6.402e-01 2.264e-04
## [77,] 58 6.402e-01 2.062e-04
## [78,] 58 6.403e-01 1.879e-04
## [79,] 58 6.404e-01 1.712e-04
## [80,] 58 6.404e-01 1.560e-04
## [81,] 58 6.404e-01 1.422e-04
## [82,] 58 6.405e-01 1.295e-04
## [83,] 58 6.405e-01 1.180e-04
## [84,] 58 6.405e-01 1.075e-04
## [85,] 58 6.405e-01 9.798e-05
## [86,] 58 6.405e-01 8.928e-05
## [87,] 58 6.406e-01 8.135e-05
## [88,] 58 6.406e-01 7.412e-05
## [1] "lambda min is at location 31"
## [1] "the leave-out cells in the source subpop is 101"
## [1] "use 102 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 203 cells and 59 genes..."
## [1] "evaluation accuracy ElasticNet 0.891625615763547"
## [1] "evaluation accuracy LDA 0.87192118226601"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 31.5508021390374"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 99.4652406417112"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 42.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 21.4285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 90.9774436090226"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 99.2481203007519"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 72.5"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 80"
## [1] "Total 203 cells as source subpop"
## [1] "Total 387 cells in remaining subpops"
## [1] "subsampling 102 cells for training source subpop"
## [1] "subsampling 102 cells in remaining subpops for training"
## [1] "use 60 genes for training model"
## [1] "use 60 genes 204 cells for testing model"
## [1] "rename remaining subpops to 1_3"
## [1] "there are 102 cells in class 1_3 and 102 cells in class 2"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 1.281e-15 2.762e-01
##   [2,]  1 3.750e-02 2.516e-01
##   [3,]  1 6.931e-02 2.293e-01
##   [4,]  2 1.021e-01 2.089e-01
##   [5,]  3 1.380e-01 1.903e-01
##   [6,]  3 1.712e-01 1.734e-01
##   [7,]  4 2.051e-01 1.580e-01
##   [8,]  4 2.371e-01 1.440e-01
##   [9,]  4 2.656e-01 1.312e-01
##  [10,]  5 2.923e-01 1.195e-01
##  [11,]  8 3.213e-01 1.089e-01
##  [12,]  8 3.495e-01 9.925e-02
##  [13,]  8 3.745e-01 9.043e-02
##  [14,] 12 4.024e-01 8.240e-02
##  [15,] 12 4.281e-01 7.508e-02
##  [16,] 13 4.521e-01 6.841e-02
##  [17,] 14 4.747e-01 6.233e-02
##  [18,] 16 4.977e-01 5.679e-02
##  [19,] 16 5.189e-01 5.175e-02
##  [20,] 16 5.381e-01 4.715e-02
##  [21,] 17 5.560e-01 4.296e-02
##  [22,] 19 5.736e-01 3.915e-02
##  [23,] 21 5.900e-01 3.567e-02
##  [24,] 21 6.063e-01 3.250e-02
##  [25,] 21 6.213e-01 2.961e-02
##  [26,] 22 6.352e-01 2.698e-02
##  [27,] 23 6.486e-01 2.458e-02
##  [28,] 23 6.612e-01 2.240e-02
##  [29,] 23 6.729e-01 2.041e-02
##  [30,] 23 6.838e-01 1.860e-02
##  [31,] 24 6.940e-01 1.695e-02
##  [32,] 26 7.039e-01 1.544e-02
##  [33,] 27 7.141e-01 1.407e-02
##  [34,] 27 7.240e-01 1.282e-02
##  [35,] 29 7.336e-01 1.168e-02
##  [36,] 31 7.429e-01 1.064e-02
##  [37,] 32 7.525e-01 9.697e-03
##  [38,] 33 7.629e-01 8.835e-03
##  [39,] 33 7.731e-01 8.050e-03
##  [40,] 34 7.830e-01 7.335e-03
##  [41,] 34 7.923e-01 6.684e-03
##  [42,] 35 8.016e-01 6.090e-03
##  [43,] 37 8.107e-01 5.549e-03
##  [44,] 37 8.203e-01 5.056e-03
##  [45,] 38 8.296e-01 4.607e-03
##  [46,] 39 8.393e-01 4.197e-03
##  [47,] 38 8.489e-01 3.825e-03
##  [48,] 38 8.580e-01 3.485e-03
##  [49,] 40 8.668e-01 3.175e-03
##  [50,] 41 8.752e-01 2.893e-03
##  [51,] 41 8.833e-01 2.636e-03
##  [52,] 41 8.909e-01 2.402e-03
##  [53,] 41 8.982e-01 2.189e-03
##  [54,] 41 9.050e-01 1.994e-03
##  [55,] 44 9.115e-01 1.817e-03
##  [56,] 43 9.180e-01 1.656e-03
##  [57,] 43 9.242e-01 1.508e-03
##  [58,] 43 9.300e-01 1.374e-03
##  [59,] 42 9.353e-01 1.252e-03
##  [60,] 45 9.406e-01 1.141e-03
##  [61,] 46 9.456e-01 1.040e-03
##  [62,] 46 9.503e-01 9.474e-04
##  [63,] 47 9.545e-01 8.632e-04
##  [64,] 49 9.586e-01 7.865e-04
##  [65,] 49 9.624e-01 7.167e-04
##  [66,] 49 9.657e-01 6.530e-04
##  [67,] 50 9.689e-01 5.950e-04
##  [68,] 50 9.716e-01 5.421e-04
##  [69,] 51 9.743e-01 4.940e-04
##  [70,] 51 9.765e-01 4.501e-04
##  [71,] 51 9.787e-01 4.101e-04
##  [72,] 51 9.806e-01 3.737e-04
##  [73,] 52 9.823e-01 3.405e-04
##  [74,] 52 9.839e-01 3.102e-04
##  [75,] 52 9.854e-01 2.827e-04
##  [76,] 52 9.867e-01 2.576e-04
##  [77,] 52 9.879e-01 2.347e-04
##  [78,] 52 9.890e-01 2.138e-04
##  [79,] 52 9.899e-01 1.948e-04
##  [80,] 52 9.908e-01 1.775e-04
##  [81,] 52 9.917e-01 1.617e-04
##  [82,] 52 9.924e-01 1.474e-04
##  [83,] 52 9.931e-01 1.343e-04
##  [84,] 52 9.937e-01 1.224e-04
##  [85,] 52 9.943e-01 1.115e-04
##  [86,] 52 9.948e-01 1.016e-04
##  [87,] 51 9.952e-01 9.256e-05
##  [88,] 51 9.956e-01 8.434e-05
##  [89,] 51 9.960e-01 7.684e-05
##  [90,] 51 9.964e-01 7.002e-05
##  [91,] 51 9.967e-01 6.380e-05
##  [92,] 51 9.970e-01 5.813e-05
##  [93,] 51 9.973e-01 5.297e-05
##  [94,] 51 9.975e-01 4.826e-05
##  [95,] 51 9.977e-01 4.397e-05
##  [96,] 51 9.979e-01 4.007e-05
##  [97,] 51 9.981e-01 3.651e-05
##  [98,] 51 9.983e-01 3.326e-05
##  [99,] 51 9.984e-01 3.031e-05
## [100,] 51 9.986e-01 2.762e-05
## [1] "lambda min is at location 20"
## [1] "the leave-out cells in the source subpop is 101"
## [1] "use 102 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 203 cells and 59 genes..."
## [1] "evaluation accuracy ElasticNet 0.822660098522167"
## [1] "evaluation accuracy LDA 0.788177339901478"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 15.5080213903743"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 3.20855614973262"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 22.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 15"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 49.6240601503759"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 12.781954887218"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 12.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 163 cells as source subpop"
## [1] "Total 427 cells in remaining subpops"
## [1] "subsampling 82 cells for training source subpop"
## [1] "subsampling 82 cells in remaining subpops for training"
## [1] "use 65 genes for training model"
## [1] "use 65 genes 164 cells for testing model"
## [1] "rename remaining subpops to 2_1"
## [1] "there are 82 cells in class 2_1 and 82 cells in class 3"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev    Lambda
##  [1,]  0 2.403e-15 1.564e-01
##  [2,]  1 1.200e-02 1.425e-01
##  [3,]  1 2.204e-02 1.299e-01
##  [4,]  1 3.046e-02 1.183e-01
##  [5,]  1 3.757e-02 1.078e-01
##  [6,]  5 5.164e-02 9.825e-02
##  [7,]  6 7.040e-02 8.952e-02
##  [8,]  8 8.723e-02 8.157e-02
##  [9,] 10 1.053e-01 7.432e-02
## [10,] 10 1.216e-01 6.772e-02
## [11,] 13 1.366e-01 6.170e-02
## [12,] 14 1.607e-01 5.622e-02
## [13,] 15 1.845e-01 5.123e-02
## [14,] 17 2.060e-01 4.668e-02
## [15,] 20 2.254e-01 4.253e-02
## [16,] 22 2.461e-01 3.875e-02
## [17,] 21 2.657e-01 3.531e-02
## [18,] 23 2.841e-01 3.217e-02
## [19,] 24 3.041e-01 2.931e-02
## [20,] 24 3.223e-01 2.671e-02
## [21,] 27 3.400e-01 2.434e-02
## [22,] 30 3.567e-01 2.218e-02
## [23,] 31 3.737e-01 2.021e-02
## [24,] 35 3.905e-01 1.841e-02
## [25,] 35 4.063e-01 1.677e-02
## [26,] 38 4.214e-01 1.528e-02
## [27,] 40 4.357e-01 1.393e-02
## [28,] 40 4.494e-01 1.269e-02
## [29,] 41 4.617e-01 1.156e-02
## [30,] 42 4.729e-01 1.054e-02
## [31,] 44 4.840e-01 9.599e-03
## [32,] 47 4.949e-01 8.746e-03
## [33,] 48 5.051e-01 7.969e-03
## [34,] 50 5.145e-01 7.261e-03
## [35,] 50 5.234e-01 6.616e-03
## [36,] 52 5.318e-01 6.029e-03
## [37,] 53 5.400e-01 5.493e-03
## [38,] 53 5.476e-01 5.005e-03
## [39,] 54 5.545e-01 4.560e-03
## [40,] 56 5.611e-01 4.155e-03
## [41,] 57 5.672e-01 3.786e-03
## [42,] 58 5.730e-01 3.450e-03
## [43,] 59 5.783e-01 3.143e-03
## [44,] 61 5.834e-01 2.864e-03
## [45,] 61 5.885e-01 2.610e-03
## [46,] 62 5.934e-01 2.378e-03
## [47,] 62 5.980e-01 2.167e-03
## [48,] 62 6.022e-01 1.974e-03
## [49,] 63 6.061e-01 1.799e-03
## [50,] 63 6.098e-01 1.639e-03
## [51,] 63 6.132e-01 1.493e-03
## [52,] 63 6.163e-01 1.361e-03
## [53,] 63 6.192e-01 1.240e-03
## [54,] 63 6.219e-01 1.130e-03
## [55,] 63 6.243e-01 1.029e-03
## [56,] 63 6.266e-01 9.378e-04
## [57,] 64 6.287e-01 8.545e-04
## [58,] 64 6.306e-01 7.786e-04
## [59,] 64 6.324e-01 7.094e-04
## [60,] 64 6.340e-01 6.464e-04
## [61,] 63 6.355e-01 5.890e-04
## [62,] 63 6.368e-01 5.367e-04
## [63,] 63 6.380e-01 4.890e-04
## [64,] 63 6.392e-01 4.456e-04
## [65,] 63 6.403e-01 4.060e-04
## [66,] 63 6.413e-01 3.699e-04
## [67,] 63 6.422e-01 3.370e-04
## [68,] 63 6.430e-01 3.071e-04
## [69,] 63 6.437e-01 2.798e-04
## [70,] 63 6.444e-01 2.550e-04
## [71,] 63 6.450e-01 2.323e-04
## [72,] 64 6.455e-01 2.117e-04
## [73,] 64 6.460e-01 1.929e-04
## [74,] 64 6.464e-01 1.757e-04
## [75,] 64 6.468e-01 1.601e-04
## [76,] 64 6.472e-01 1.459e-04
## [77,] 64 6.475e-01 1.329e-04
## [78,] 63 6.478e-01 1.211e-04
## [79,] 64 6.480e-01 1.104e-04
## [80,] 64 6.482e-01 1.006e-04
## [81,] 64 6.484e-01 9.163e-05
## [82,] 64 6.486e-01 8.349e-05
## [83,] 64 6.488e-01 7.607e-05
## [84,] 64 6.489e-01 6.931e-05
## [85,] 64 6.490e-01 6.316e-05
## [86,] 64 6.491e-01 5.755e-05
## [87,] 64 6.492e-01 5.243e-05
## [88,] 63 6.493e-01 4.778e-05
## [89,] 63 6.493e-01 4.353e-05
## [1] "lambda min is at location 40"
## [1] "the leave-out cells in the source subpop is 81"
## [1] "use 82 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 163 cells and 64 genes..."
## [1] "evaluation accuracy ElasticNet 0.656441717791411"
## [1] "evaluation accuracy LDA 0.625766871165644"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 187 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 48.1283422459893"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 187 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 93.048128342246"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 140 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 49.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 140 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 9.28571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 133 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 90.9774436090226"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 133 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 87.218045112782"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 40 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 67.5"
## [1] "Number of genes in the target data, but not in model genes is 4936"
## [1] "Number of genes in the model present in the target data is 64"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 64 genes and 40 cells. The trained model has 64 genes"
## [1] "first 10 genes in model "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "first 10 genes in target "
##  [1] "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"   "HMGN2" 
##  [8] "FKBP4"  "IGFBP7" "FSCN1" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 60"
## [1] "Total 163 cells as source subpop"
## [1] "Total 427 cells in remaining subpops"
## [1] "subsampling 82 cells for training source subpop"
## [1] "subsampling 82 cells in remaining subpops for training"
## [1] "use 65 genes for training model"
## [1] "use 65 genes 164 cells for testing model"
## [1] "rename remaining subpops to 2_1"
## [1] "there are 82 cells in class 2_1 and 82 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 2.403e-15 1.690e-01
##   [2,]  4 2.128e-02 1.540e-01
##   [3,]  5 4.864e-02 1.403e-01
##   [4,]  5 7.226e-02 1.278e-01
##   [5,]  7 9.488e-02 1.165e-01
##   [6,]  7 1.178e-01 1.061e-01
##   [7,]  8 1.383e-01 9.671e-02
##   [8,] 10 1.593e-01 8.812e-02
##   [9,] 10 1.790e-01 8.029e-02
##  [10,] 12 1.986e-01 7.316e-02
##  [11,] 12 2.177e-01 6.666e-02
##  [12,] 12 2.346e-01 6.074e-02
##  [13,] 13 2.498e-01 5.534e-02
##  [14,] 15 2.684e-01 5.043e-02
##  [15,] 16 2.875e-01 4.595e-02
##  [16,] 19 3.080e-01 4.186e-02
##  [17,] 19 3.305e-01 3.815e-02
##  [18,] 21 3.507e-01 3.476e-02
##  [19,] 21 3.694e-01 3.167e-02
##  [20,] 23 3.875e-01 2.886e-02
##  [21,] 25 4.051e-01 2.629e-02
##  [22,] 30 4.228e-01 2.396e-02
##  [23,] 30 4.404e-01 2.183e-02
##  [24,] 30 4.567e-01 1.989e-02
##  [25,] 31 4.717e-01 1.812e-02
##  [26,] 34 4.864e-01 1.651e-02
##  [27,] 36 5.030e-01 1.505e-02
##  [28,] 36 5.186e-01 1.371e-02
##  [29,] 40 5.331e-01 1.249e-02
##  [30,] 42 5.469e-01 1.138e-02
##  [31,] 44 5.601e-01 1.037e-02
##  [32,] 46 5.727e-01 9.449e-03
##  [33,] 48 5.853e-01 8.610e-03
##  [34,] 48 5.977e-01 7.845e-03
##  [35,] 48 6.091e-01 7.148e-03
##  [36,] 51 6.199e-01 6.513e-03
##  [37,] 51 6.319e-01 5.934e-03
##  [38,] 52 6.431e-01 5.407e-03
##  [39,] 52 6.536e-01 4.927e-03
##  [40,] 52 6.633e-01 4.489e-03
##  [41,] 52 6.723e-01 4.090e-03
##  [42,] 53 6.813e-01 3.727e-03
##  [43,] 54 6.902e-01 3.396e-03
##  [44,] 52 6.989e-01 3.094e-03
##  [45,] 53 7.074e-01 2.819e-03
##  [46,] 53 7.158e-01 2.569e-03
##  [47,] 54 7.239e-01 2.341e-03
##  [48,] 58 7.319e-01 2.133e-03
##  [49,] 58 7.403e-01 1.943e-03
##  [50,] 58 7.481e-01 1.771e-03
##  [51,] 56 7.557e-01 1.613e-03
##  [52,] 57 7.629e-01 1.470e-03
##  [53,] 57 7.700e-01 1.339e-03
##  [54,] 59 7.772e-01 1.220e-03
##  [55,] 59 7.844e-01 1.112e-03
##  [56,] 59 7.918e-01 1.013e-03
##  [57,] 59 7.996e-01 9.232e-04
##  [58,] 59 8.081e-01 8.412e-04
##  [59,] 61 8.176e-01 7.664e-04
##  [60,] 62 8.283e-01 6.983e-04
##  [61,] 63 8.402e-01 6.363e-04
##  [62,] 64 8.531e-01 5.798e-04
##  [63,] 62 8.649e-01 5.283e-04
##  [64,] 61 8.755e-01 4.813e-04
##  [65,] 62 8.867e-01 4.386e-04
##  [66,] 63 8.971e-01 3.996e-04
##  [67,] 63 9.067e-01 3.641e-04
##  [68,] 63 9.154e-01 3.318e-04
##  [69,] 63 9.234e-01 3.023e-04
##  [70,] 63 9.307e-01 2.754e-04
##  [71,] 62 9.372e-01 2.510e-04
##  [72,] 62 9.430e-01 2.287e-04
##  [73,] 62 9.482e-01 2.084e-04
##  [74,] 62 9.530e-01 1.899e-04
##  [75,] 62 9.572e-01 1.730e-04
##  [76,] 62 9.611e-01 1.576e-04
##  [77,] 63 9.647e-01 1.436e-04
##  [78,] 64 9.679e-01 1.309e-04
##  [79,] 64 9.708e-01 1.192e-04
##  [80,] 64 9.735e-01 1.086e-04
##  [81,] 64 9.759e-01 9.899e-05
##  [82,] 64 9.781e-01 9.019e-05
##  [83,] 64 9.800e-01 8.218e-05
##  [84,] 63 9.818e-01 7.488e-05
##  [85,] 64 9.833e-01 6.823e-05
##  [86,] 64 9.848e-01 6.217e-05
##  [87,] 64 9.861e-01 5.664e-05
##  [88,] 64 9.865e-01 5.161e-05
##  [89,] 64 9.884e-01 4.703e-05
##  [90,] 63 9.886e-01 4.285e-05
##  [91,] 64 9.903e-01 3.904e-05
##  [92,] 63 9.904e-01 3.557e-05
##  [93,] 64 9.919e-01 3.241e-05
##  [94,] 64 9.920e-01 2.953e-05
##  [95,] 64 9.932e-01 2.691e-05
##  [96,] 64 9.933e-01 2.452e-05
##  [97,] 64 9.942e-01 2.234e-05
##  [98,] 64 9.944e-01 2.036e-05
##  [99,] 64 9.951e-01 1.855e-05
## [100,] 63 9.954e-01 1.690e-05
## [1] "lambda min is at location 18"
## [1] "the leave-out cells in the source subpop is 81"
## [1] "use 82 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 163 cells and 65 genes..."
## [1] "evaluation accuracy ElasticNet 0.650306748466258"
## [1] "evaluation accuracy LDA 0.638036809815951"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 74.331550802139"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 2.67379679144385"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 4.28571428571429"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 47.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 6.76691729323308"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 10.5263157894737"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 10"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 20"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#prepare table input for sankey plot </span>

LASSO_C1S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#7570b3"</span>)

LASSO_C2S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#1b9e77"</span>)

LASSO_C3S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#e7298a"</span>)


combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S1,LASSO_C2S1,LASSO_C3S1)

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)],
    <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))]) 
combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]


<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))
n &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(node_df<span class="op">$</span>Node))
getPalette =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/colorRamp">colorRampPalette</a></span>(RColorBrewer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">9</span>, <span class="st">"Set1"</span>))
Color =<span class="st"> </span><span class="kw">getPalette</span>(n)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))
p1&lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/networkD3/topics/sankeyNetwork">sankeyNetwork</a></span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,
    <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>,
    <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, <span class="dt">fontSize =</span> <span class="dv">22</span>)
p1</code></pre></div>
<div id="htmlwidget-d4a8084d524b61b1be47" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-d4a8084d524b61b1be47">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2],"target":[3,4,5,6,3,4,5,6,3,4,5,6],"value":[50,96.0714285714286,64.2857142857143,87.5,23.5294117647058,32.1428571428571,70.3007518796992,56.25,61.2299465240642,26.7857142857143,48.8721804511278,38.75],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a"]},"nodes":{"name":["C1_MP1","C2_MP1","C3_MP1","C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#E41A1C","#3E8E93","#7E6E85","#FF7F00","#E1C62F","#DB728C","#999999"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></code></pre></div>
</div>
<div id="annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with" class="anchor"></a>4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with</h2>
</div>
<div id="a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="anchor"></a>a reference dataset to see which cluster is most similar to the reference</h2>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation-instruction">1. Installation instruction</a></li>
      <li>
<a href="#a-simple-workflow-of-the-scgps">2. A simple workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#create-scgps-objects">2.1 Create scGPS objects</a></li>
      <li><a href="#run-prediction">2.2 Run prediction</a></li>
      <li><a href="#summarise-results">2.3 Summarise results</a></li>
      </ul>
</li>
      <li>
<a href="#a-complete-workflow-of-the-scgps">3. A complete workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#identify-clusters-in-a-dataset-using-core">3.1 Identify clusters in a dataset using CORE</a></li>
      <li><a href="#identify-clusters-in-a-dataset-using-score">3.2 Identify clusters in a dataset using SCORE</a></li>
      <li><a href="#stable-clustering-at-optimal-resolution">(Stable Clustering at Optimal REsolution)</a></li>
      <li><a href="#visualise-all-cluster-results-in-all-iterations">3.3 Visualise all cluster results in all iterations</a></li>
      <li><a href="#compare-clustering-results-with-other-dimensional-reduction-methods">3.4 Compare clustering results with other dimensional reduction methods</a></li>
      <li><a href="#e-g--cidr">(e.g., CIDR)</a></li>
      <li><a href="#find-gene-markers-and-annotate-clusters">3.5 Find gene markers and annotate clusters</a></li>
      </ul>
</li>
      <li>
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples">4. Relationship between clusters within one sample or between two samples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#start-the-scgps-prediction-to-find-relationship-between-clusters">4.1 Start the scGPS prediction to find relationship between clusters</a></li>
      <li><a href="#display-summary-results-for-the-prediction">4.2 Display summary results for the prediction</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-one-sample">4.3 Plot the relationship between clusters in one sample</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-two-samples">4.3 Plot the relationship between clusters in two samples</a></li>
      <li><a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with">4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with</a></li>
      <li><a href="#a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference">a reference dataset to see which cluster is most similar to the reference</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Asenbouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
