<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGPS introduction â€¢ scGPS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scGPS introduction">
<meta property="og:description" content="scGPS is a single cell RNA analysis framework to decompose a mixed population into clusters (SCORE), followed by analysing the relationship between clusters (scGPS)">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="vignette_files/htmlwidgets-1.3/htmlwidgets.js"></script><script src="vignette_files/d3-4.5.0/d3.min.js"></script><script src="vignette_files/sankey-1/sankey.js"></script><script src="vignette_files/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scGPS introduction</h1>
                        <h4 class="author">Quan Nguyen and Michael Thompson</h4>
            
            <h4 class="date">2019-03-13</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># To install scGPS from github (Depending on the configuration of the local</span>
<span class="co"># computer or HPC, possible custom C++ compilation may be required - see</span>
<span class="co"># installation trouble-shootings below)</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"IMB-Computational-Genomics-Lab/scGPS"</span>)

<span class="co"># for C++ compilation trouble-shooting, manual download and installation can be</span>
<span class="co"># done from github</span>

git clone https<span class="op">:</span><span class="er">//</span>github.com<span class="op">/</span>IMB<span class="op">-</span>Computational<span class="op">-</span>Genomics<span class="op">-</span>Lab<span class="op">/</span>scGPS

<span class="co"># then check in scGPS/src if any of the precompiled (e.g.  those with *.so and</span>
<span class="co"># *.o) files exist and delete them before recompiling</span>

<span class="co"># create a Makevars file in the scGPS/src with one line: PKG_LIBS =</span>
<span class="co"># $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)</span>

<span class="co"># then with the scGPS as the R working directory, manually recompile scGPS in R</span>
<span class="co"># using devtools to load and install functions</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/document">document</a></span>()
<span class="co">#load the package to the workspace </span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>()</code></pre></div>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: </em> <em>given a mixed population with known subpopulations, </em> <em>estimate transition scores between these subpopulation.</em></p>
<div id="create-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#create-scgps-objects" class="anchor"></a>2.1 Create scGPS objects</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load mixed population 1 (loaded from sample1 dataset, named it as day2)</span>
path_scPGS =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/getwd">getwd</a></span>()
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(path_scPGS) 

day2 &lt;-<span class="st"> </span>sample1
mixedpop1 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day2<span class="op">$</span>dat2_counts,
    <span class="dt">GeneMetadata =</span> day2<span class="op">$</span>dat2geneInfo, <span class="dt">CellMetadata =</span> day2<span class="op">$</span>dat2_clusters)

<span class="co"># load mixed population 2 (loaded from sample2 dataset, named it as day5)</span>
day5 &lt;-<span class="st"> </span>sample2
mixedpop2 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts,
    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> day5<span class="op">$</span>dat5_clusters)</code></pre></div>
</div>
<div id="run-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#run-prediction" class="anchor"></a>2.2 Run prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># select a subpopulation</span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co"># load gene list (this can be any lists of user selected genes)</span>
genes &lt;-<span class="st"> </span>GeneList
genes &lt;-<span class="st"> </span>genes<span class="op">$</span>Merged_unique
<span class="co"># load cluster information </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#run training (running nboots = 3 here, but recommend to use nboots = 50-100)</span>
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">3</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID,
    <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2, <span class="dt">trainset_ratio =</span> <span class="fl">0.7</span>)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev   Lambda
##  [1,]  0 1.281e-15 0.138600
##  [2,]  1 9.421e-03 0.126300
##  [3,]  1 1.726e-02 0.115100
##  [4,]  1 2.378e-02 0.104900
##  [5,]  1 2.923e-02 0.095560
##  [6,]  1 3.377e-02 0.087070
##  [7,]  1 3.756e-02 0.079340
##  [8,]  2 4.269e-02 0.072290
##  [9,]  2 4.755e-02 0.065870
## [10,]  2 5.167e-02 0.060020
## [11,]  2 5.519e-02 0.054680
## [12,]  2 5.819e-02 0.049830
## [13,]  2 6.076e-02 0.045400
## [14,]  3 6.303e-02 0.041370
## [15,]  3 6.577e-02 0.037690
## [16,]  3 6.811e-02 0.034340
## [17,]  3 7.013e-02 0.031290
## [18,]  3 7.185e-02 0.028510
## [19,]  3 7.334e-02 0.025980
## [20,]  3 7.461e-02 0.023670
## [21,]  3 7.570e-02 0.021570
## [22,]  3 7.664e-02 0.019650
## [23,]  3 7.744e-02 0.017910
## [24,]  3 7.813e-02 0.016320
## [25,]  3 7.871e-02 0.014870
## [26,]  3 7.921e-02 0.013550
## [27,]  3 7.963e-02 0.012340
## [28,]  3 7.999e-02 0.011250
## [29,]  3 8.029e-02 0.010250
## [30,]  3 8.055e-02 0.009337
## [31,]  3 8.077e-02 0.008507
## [32,]  3 8.095e-02 0.007751
## [33,]  3 8.111e-02 0.007063
## [34,]  3 8.124e-02 0.006435
## [35,]  3 8.135e-02 0.005864
## [36,]  3 8.144e-02 0.005343
## [37,]  4 8.153e-02 0.004868
## [38,]  4 8.161e-02 0.004436
## [39,]  4 8.167e-02 0.004042
## [40,]  4 8.173e-02 0.003683
## [41,]  4 8.178e-02 0.003355
## [42,]  4 8.182e-02 0.003057
## [43,]  4 8.185e-02 0.002786
## [44,]  4 8.188e-02 0.002538
## [45,]  4 8.190e-02 0.002313
## [46,]  4 8.192e-02 0.002107
## [47,]  4 8.193e-02 0.001920
## [48,]  4 8.195e-02 0.001750
## [49,]  4 8.196e-02 0.001594
## [50,]  4 8.197e-02 0.001452
## [1] "lambda min is at location 30"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 213 cells and 5 genes..."
## [1] "evaluation accuracy ElasticNet 0.657276995305164"
## [1] "evaluation accuracy LDA 0.661971830985915"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 12.2994652406417"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 4.27807486631016"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 25.7142857142857"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 7.85714285714286"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 86.4661654135338"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 9.77443609022556"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 70"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 7.5"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev   Lambda
##  [1,]  0 2.884e-15 0.151400
##  [2,]  1 1.124e-02 0.138000
##  [3,]  1 2.059e-02 0.125700
##  [4,]  1 2.838e-02 0.114500
##  [5,]  1 3.488e-02 0.104400
##  [6,]  1 4.031e-02 0.095100
##  [7,]  1 4.485e-02 0.086650
##  [8,]  2 5.073e-02 0.078950
##  [9,]  2 5.664e-02 0.071940
## [10,]  2 6.165e-02 0.065550
## [11,]  2 6.593e-02 0.059730
## [12,]  2 6.958e-02 0.054420
## [13,]  3 7.337e-02 0.049590
## [14,]  3 7.739e-02 0.045180
## [15,]  3 8.086e-02 0.041170
## [16,]  3 8.384e-02 0.037510
## [17,]  3 8.642e-02 0.034180
## [18,]  3 8.865e-02 0.031140
## [19,]  3 9.057e-02 0.028370
## [20,]  3 9.224e-02 0.025850
## [21,]  3 9.368e-02 0.023560
## [22,]  3 9.493e-02 0.021460
## [23,]  3 9.601e-02 0.019560
## [24,]  3 9.694e-02 0.017820
## [25,]  3 9.774e-02 0.016240
## [26,]  3 9.843e-02 0.014790
## [27,]  3 9.902e-02 0.013480
## [28,]  3 9.953e-02 0.012280
## [29,]  3 9.996e-02 0.011190
## [30,]  3 1.003e-01 0.010200
## [31,]  3 1.007e-01 0.009291
## [32,]  3 1.009e-01 0.008466
## [33,]  3 1.012e-01 0.007714
## [34,]  3 1.014e-01 0.007029
## [35,]  3 1.015e-01 0.006404
## [36,]  4 1.017e-01 0.005835
## [37,]  4 1.018e-01 0.005317
## [38,]  4 1.019e-01 0.004845
## [39,]  4 1.020e-01 0.004414
## [40,]  4 1.021e-01 0.004022
## [41,]  4 1.022e-01 0.003665
## [42,]  4 1.022e-01 0.003339
## [43,]  4 1.023e-01 0.003042
## [44,]  4 1.023e-01 0.002772
## [45,]  4 1.024e-01 0.002526
## [46,]  4 1.024e-01 0.002302
## [47,]  4 1.024e-01 0.002097
## [48,]  4 1.024e-01 0.001911
## [49,]  4 1.024e-01 0.001741
## [50,]  4 1.025e-01 0.001586
## [51,]  4 1.025e-01 0.001445
## [52,]  4 1.025e-01 0.001317
## [1] "lambda min is at location 30"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 214 cells and 5 genes..."
## [1] "evaluation accuracy ElasticNet 0.607476635514019"
## [1] "evaluation accuracy LDA 0.607476635514019"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 27.2727272727273"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 28.8770053475936"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 42.8571428571429"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 26.4285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 94.7368421052632"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 39.8496240601504"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 75"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 35"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev    Lambda
##  [1,]  0 3.044e-15 0.1378000
##  [2,]  1 9.302e-03 0.1255000
##  [3,]  1 1.704e-02 0.1144000
##  [4,]  1 2.348e-02 0.1042000
##  [5,]  1 2.885e-02 0.0949500
##  [6,]  2 3.677e-02 0.0865100
##  [7,]  2 4.370e-02 0.0788300
##  [8,]  2 4.958e-02 0.0718300
##  [9,]  3 5.534e-02 0.0654400
## [10,]  3 6.143e-02 0.0596300
## [11,]  3 6.668e-02 0.0543300
## [12,]  4 7.137e-02 0.0495100
## [13,]  4 7.591e-02 0.0451100
## [14,]  4 7.991e-02 0.0411000
## [15,]  4 8.344e-02 0.0374500
## [16,]  4 8.657e-02 0.0341200
## [17,]  4 8.935e-02 0.0310900
## [18,]  4 9.181e-02 0.0283300
## [19,]  4 9.401e-02 0.0258100
## [20,]  4 9.596e-02 0.0235200
## [21,]  4 9.770e-02 0.0214300
## [22,]  4 9.925e-02 0.0195300
## [23,]  4 1.006e-01 0.0177900
## [24,]  4 1.018e-01 0.0162100
## [25,]  4 1.029e-01 0.0147700
## [26,]  4 1.039e-01 0.0134600
## [27,]  4 1.047e-01 0.0122600
## [28,]  4 1.055e-01 0.0111700
## [29,]  5 1.062e-01 0.0101800
## [30,]  5 1.068e-01 0.0092770
## [31,]  5 1.074e-01 0.0084520
## [32,]  5 1.078e-01 0.0077020
## [33,]  5 1.083e-01 0.0070170
## [34,]  5 1.086e-01 0.0063940
## [35,]  5 1.089e-01 0.0058260
## [36,]  5 1.092e-01 0.0053080
## [37,]  5 1.094e-01 0.0048370
## [38,]  5 1.096e-01 0.0044070
## [39,]  5 1.098e-01 0.0040160
## [40,]  5 1.100e-01 0.0036590
## [41,]  5 1.101e-01 0.0033340
## [42,]  5 1.102e-01 0.0030380
## [43,]  5 1.103e-01 0.0027680
## [44,]  5 1.104e-01 0.0025220
## [45,]  5 1.104e-01 0.0022980
## [46,]  5 1.105e-01 0.0020940
## [47,]  5 1.105e-01 0.0019080
## [48,]  5 1.106e-01 0.0017380
## [49,]  5 1.106e-01 0.0015840
## [50,]  5 1.106e-01 0.0014430
## [51,]  5 1.107e-01 0.0013150
## [52,]  5 1.107e-01 0.0011980
## [53,]  5 1.107e-01 0.0010920
## [54,]  5 1.107e-01 0.0009947
## [55,]  5 1.107e-01 0.0009063
## [56,]  5 1.107e-01 0.0008258
## [1] "lambda min is at location 19"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 214 cells and 6 genes..."
## [1] "evaluation accuracy ElasticNet 0.672897196261682"
## [1] "evaluation accuracy LDA 0.672897196261682"
## [1] "done training for bootstrap 3, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "ElasticNet for target subpop 1 has no solution"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 34.7593582887701"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 62.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 48.5714285714286"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 94.7368421052632"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 48.1203007518797"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 77.5"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 45"</code></pre>
</div>
<div id="summarise-results" class="section level2">
<h2 class="hasAnchor">
<a href="#summarise-results" class="anchor"></a>2.3 Summarise results</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># display the list of result information in the LASOLDA_dat object </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(LSOLDA_dat)</code></pre></div>
<pre><code>## [1] "Accuracy"          "ElasticNetGenes"   "Deviance"         
## [4] "ElasticNetFit"     "LDAFit"            "predictor_S1"     
## [7] "ElasticNetPredict" "LDAPredict"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>ElasticNetPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[1]][[2]]
## [1] 12.29947
## 
## [[1]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[1]][[4]]
## [1] 25.71429
## 
## [[1]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[1]][[6]]
## [1] 86.46617
## 
## [[1]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 70
## 
## 
## [[2]]
## [[2]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[2]][[2]]
## [1] 27.27273
## 
## [[2]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[2]][[4]]
## [1] 42.85714
## 
## [[2]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 94.73684
## 
## [[2]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 75
## 
## 
## [[3]]
## [[3]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[3]][[2]]
## [1] "NA"
## 
## [[3]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[3]][[4]]
## [1] 62.14286
## 
## [[3]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 94.73684
## 
## [[3]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 77.5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>LDAPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[1]][[2]]
## [1] 4.278075
## 
## [[1]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[1]][[4]]
## [1] 7.857143
## 
## [[1]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[1]][[6]]
## [1] 9.774436
## 
## [[1]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 7.5
## 
## 
## [[2]]
## [[2]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[2]][[2]]
## [1] 28.87701
## 
## [[2]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[2]][[4]]
## [1] 26.42857
## 
## [[2]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 39.84962
## 
## [[2]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 35
## 
## 
## [[3]]
## [[3]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[3]][[2]]
## [1] 34.75936
## 
## [[3]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[3]][[4]]
## [1] 48.57143
## 
## [[3]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 48.1203
## 
## [[3]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 45</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary results LDA</span>
sum_pred_lda &lt;-<span class="st"> </span><span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># summary results Lasso to show the percent of cells</span>
<span class="co"># classified as cells belonging </span>
sum_pred_lasso &lt;-<span class="st"> </span><span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat,
    <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># plot summary results </span>
plot_sum &lt;-<span class="cf">function</span>(sum_dat){
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(sum_dat)
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/na.fail">na.omit</a></span>(sum_dat_tf)
    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(sum_dat[, <span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(sum_dat)],<span class="dv">1</span>,
        <span class="cf">function</span>(x){<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(x))})
    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"for subpop "</span>,<span class="st">"sp"</span>,  sum_dat<span class="op">$</span>names )
    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"in target mixedpop"</span>,<span class="st">"in p"</span>,  sum_dat<span class="op">$</span>names) 
    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sum_dat_tf) &lt;-<span class="st"> </span>sum_dat<span class="op">$</span>names
    <span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(sum_dat_tf, <span class="dt">las=</span><span class="dv">2</span>)
}
<span class="kw">plot_sum</span>(sum_pred_lasso)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_sum</span>(sum_pred_lda)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## [1] 66.19718 60.74766 67.28972</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary maximum deviance explained by the model </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## $allDeviance
## [1] "0.08055" "0.1003"  "0.09401"
## 
## $DeviMax
##          dat_DE$Dfd          Deviance           DEgenes
## 1                 0            0.1003    genes_cluster1
## 2                 1            0.1003    genes_cluster1
## 3                 2            0.1003    genes_cluster1
## 4                 3            0.1003    genes_cluster1
## 5 remaining DEgenes remaining DEgenes remaining DEgenes
## 
## $LassoGenesMax
## NULL</code></pre>
</div>
</div>
<div id="a-complete-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-complete-workflow-of-the-scgps" class="anchor"></a>3. A complete workflow of the scGPS:</h1>
<ul>
<li>The purpose of this workflow is to solve the following task: given an unknown*</li>
<li>mixed population, find clusters and estimate relationship between clusters*</li>
</ul>
<div id="identify-clusters-in-a-dataset-using-core" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-core" class="anchor"></a>3.1 Identify clusters in a dataset using CORE</h2>
<p><em>(skip this step if clusters are known)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find clustering information in an expresion data using CORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, 
    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames) 

CORE_cluster &lt;-<span class="st"> </span><span class="kw">CORE_scGPS</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>)</code></pre></div>
<pre><code>## [1] "Performing 1 round of filtering"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "No more outliers detected in filtering round 1"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "writing clustering result for run 1"
## [1] "writing clustering result for run 2"
## [1] "writing clustering result for run 3"
## [1] "writing clustering result for run 4"
## [1] "writing clustering result for run 5"
## [1] "writing clustering result for run 6"
## [1] "writing clustering result for run 7"
## [1] "writing clustering result for run 8"
## [1] "writing clustering result for run 9"
## [1] "writing clustering result for run 10"
## [1] "writing clustering result for run 11"
## [1] "writing clustering result for run 12"
## [1] "writing clustering result for run 13"
## [1] "writing clustering result for run 14"
## [1] "writing clustering result for run 15"
## [1] "writing clustering result for run 16"
## [1] "writing clustering result for run 17"
## [1] "writing clustering result for run 18"
## [1] "writing clustering result for run 19"
## [1] "writing clustering result for run 20"
## [1] "writing clustering result for run 21"
## [1] "writing clustering result for run 22"
## [1] "writing clustering result for run 23"
## [1] "writing clustering result for run 24"
## [1] "writing clustering result for run 25"
## [1] "writing clustering result for run 26"
## [1] "writing clustering result for run 27"
## [1] "writing clustering result for run 28"
## [1] "writing clustering result for run 29"
## [1] "writing clustering result for run 30"
## [1] "writing clustering result for run 31"
## [1] "writing clustering result for run 32"
## [1] "writing clustering result for run 33"
## [1] "writing clustering result for run 34"
## [1] "writing clustering result for run 35"
## [1] "writing clustering result for run 36"
## [1] "writing clustering result for run 37"
## [1] "writing clustering result for run 38"
## [1] "writing clustering result for run 39"
## [1] "writing clustering result for run 40"
## [1] "Done clustering, moving to stability calculation..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done finding optimal clustering..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># to update the clustering information, users can ...</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)

clustering_after_outlier_removal &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unname">unname</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(
    CORE_cluster<span class="op">$</span>Cluster[[optimal_index]]))
corresponding_cells_after_outlier_removal &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>cellsForClustering
original_cells_before_removal &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">2</span>]
corresponding_index &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(corresponding_cells_after_outlier_removal,
    original_cells_before_removal )
<span class="co"># check the matching </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/identical">identical</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/character">as.character</a></span>(original_cells_before_removal[corresponding_index]),
    corresponding_cells_after_outlier_removal)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># create new object with the new clustering after removing outliers </span>
mixedpop2_post_clustering &lt;-<span class="st"> </span>mixedpop2[,corresponding_index]
<span class="kw">colData</span>(mixedpop2_post_clustering)[,<span class="dv">1</span>] &lt;-<span class="st"> </span>clustering_after_outlier_removal</code></pre></div>
</div>
<div id="identify-clusters-in-a-dataset-using-score" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-score" class="anchor"></a>3.2 Identify clusters in a dataset using SCORE</h2>
</div>
<div id="stable-clustering-at-optimal-resolution" class="section level2">
<h2 class="hasAnchor">
<a href="#stable-clustering-at-optimal-resolution" class="anchor"></a>(Stable Clustering at Optimal REsolution)</h2>
<p><em>(skip this step if clusters are known)</em> <em>(SCORE aims to get stable subpopulation results, </em> * by introducing bagging aggregation and bootstrapping to the CORE algorithm)*</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find clustering information in an expresion data using SCORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, 
    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames ) 

SCORE_test &lt;-<span class="st"> </span><span class="kw">CORE_scGPS_bagging</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>,
                          <span class="dt">bagging_run =</span> <span class="dv">20</span>, <span class="dt">subsample_proportion =</span> .<span class="dv">8</span>)</code></pre></div>
<pre><code>## [1] "Performing 1 round of filtering"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "No more outliers detected in filtering round 1"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "500 cells left after filtering"
## [1] "Running 20 bagging runs, with 0.8 subsampling..."
## [1] "Done clustering, moving to stability calculation..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."</code></pre>
</div>
<div id="visualise-all-cluster-results-in-all-iterations" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-all-cluster-results-in-all-iterations" class="anchor"></a>3.3 Visualise all cluster results in all iterations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/dev">dev.off</a></span>()</code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##3.2.1 plot CORE clustering 
p1 &lt;-<span class="st"> </span><span class="kw">plot_CORE</span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster, 
    <span class="dt">color_branch =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#208eb7"</span>, <span class="st">"#6ce9d3"</span>, <span class="st">"#1c5e39"</span>, <span class="st">"#8fca40"</span>, <span class="st">"#154975"</span>,
    <span class="st">"#b1c8eb"</span>))
p1 </code></pre></div>
<pre><code>## $mar
## [1] 1 5 0 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#extract optimal index identified by CORE_scGPS</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)
<span class="co">#plot one optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> CORE_cluster<span class="op">$</span>tree,
    <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(CORE_cluster<span class="op">$</span>Cluster[optimal_index]),
    <span class="dt">shift =</span> <span class="op">-</span><span class="dv">2000</span>)</code></pre></div>
<pre><code>## [1] "Ordering and assigning labels..."
## [1] 2
## [1]  24 112  NA  NA  NA  NA  NA  NA
## [1] 3
## [1]  24 112 224  NA  NA  NA  NA  NA
## [1] 4
## [1]  24 112 224 299  NA  NA  NA  NA
## [1] 5
## [1]  24 112 224 299 335  NA  NA  NA
## [1] 6
## [1]  24 112 224 299 335 367  NA  NA
## [1] 7
## [1]  24 112 224 299 335 367 414  NA
## [1] 8
## [1]  24 112 224 299 335 367 414 470
## [1] "Plotting the colored dendrogram now...."</code></pre>
<pre><code>## [1] "Plotting the bar underneath now...."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##3.2.2 plot SCORE clustering 
<span class="co">#plot all clustering bars</span>
<span class="kw">plot_CORE</span>(SCORE_test<span class="op">$</span>tree, <span class="dt">list_clusters =</span> SCORE_test<span class="op">$</span>Cluster)
<span class="co">#plot one stable optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> SCORE_test<span class="op">$</span>tree,
    <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(SCORE_test<span class="op">$</span>Cluster[SCORE_test<span class="op">$</span>optimal_index]),
    <span class="dt">shift =</span> <span class="op">-</span><span class="dv">100</span>)</code></pre></div>
<pre><code>## [1] "Ordering and assigning labels..."
## [1] 2
## [1]  24 112  NA  NA  NA  NA  NA
## [1] 3
## [1]  24 112 250  NA  NA  NA  NA
## [1] 4
## [1]  24 112 250 335  NA  NA  NA
## [1] 5
## [1]  24 112 250 335 367  NA  NA
## [1] 6
## [1]  24 112 250 335 367 414  NA
## [1] 7
## [1]  24 112 250 335 367 414 470
## [1] "Plotting the colored dendrogram now...."</code></pre>
<pre><code>## [1] "Plotting the bar underneath now...."</code></pre>
</div>
<div id="compare-clustering-results-with-other-dimensional-reduction-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-clustering-results-with-other-dimensional-reduction-methods" class="anchor"></a>3.4 Compare clustering results with other dimensional reduction methods</h2>
</div>
<div id="e-g--cidr" class="section level2">
<h2 class="hasAnchor">
<a href="#e-g--cidr" class="anchor"></a>(e.g., CIDR)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(cidr)
t &lt;-<span class="st"> </span><span class="kw">CIDR_scGPS</span>(<span class="dt">expression.matrix=</span><span class="kw">assay</span>(mixedpop2))</code></pre></div>
<pre><code>## [1] "building cidr object..."
## [1] "determine dropout candidates..."
## [1] "determine the imputation weighting threshold..."
## [1] "computes the _CIDR_ dissimilarity matrix..."
## [1] "PCA plot with proportion of variance explained..."</code></pre>
<p><img src="vignette_files/figure-html/compare_clustering-1.png" width="700"></p>
<pre><code>## [1] "find the number of PC..."
## [1] "perform clustering..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="kw">plotReduced_scGPS</span>(t, <span class="dt">color_fac =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]),
    <span class="dt">palletes =</span><span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])))
p2</code></pre></div>
<p><img src="vignette_files/figure-html/compare_clustering-2.png" width="700"></p>
</div>
<div id="find-gene-markers-and-annotate-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#find-gene-markers-and-annotate-clusters" class="anchor"></a>3.5 Find gene markers and annotate clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#load gene list (this can be any lists of user-selected genes)</span>
genes &lt;-GeneList
genes &lt;-genes<span class="op">$</span>Merged_unique

<span class="co">#the gene list can also be objectively identified by differential expression</span>
<span class="co">#analysis cluster information is requied for findMarkers_scGPS. Here, we use </span>
<span class="co">#CORE results. </span>

<span class="co">#colData(mixedpop2)[,1] &lt;- unlist(SCORE_test$Cluster[SCORE_test$optimal_index])</span>

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(locfit))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DESeq))

DEgenes &lt;-<span class="st"> </span><span class="kw">findMarkers_scGPS</span>(<span class="dt">expression_matrix=</span><span class="kw">assay</span>(mixedpop2), 
    <span class="dt">cluster =</span> <span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>], 
    <span class="dt">selected_cluster=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])) </code></pre></div>
<pre><code>## [1] "Start estimate dispersions for cluster 1..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 1..."
## [1] "Done nbinom test for cluster 1 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 2..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 2..."
## [1] "Done nbinom test for cluster 2 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 3..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 3..."
## [1] "Done nbinom test for cluster 3 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 4..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 4..."
## [1] "Done nbinom test for cluster 4 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#the output contains dataframes for each cluster.</span>
<span class="co">#the data frame contains all genes, sorted by p-values </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(DEgenes)</code></pre></div>
<pre><code>## [1] "DE_Subpop1vsRemaining" "DE_Subpop2vsRemaining" "DE_Subpop3vsRemaining"
## [4] "DE_Subpop4vsRemaining"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#you can annotate the identified clusters </span>
DEgeneList_1vsOthers &lt;-<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id

<span class="co">#users need to check the format of the gene input to make sure they are </span>
<span class="co">#consistent to the gene names in the expression matrix </span>

<span class="co">#the following command saves the file "PathwayEnrichment.xlsx" to the </span>
<span class="co">#working dir</span>
<span class="co">#use 500 top DE genes </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DOSE))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ReactomePA))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(clusterProfiler))
enrichment_test &lt;-<span class="st"> </span><span class="kw">annotate_scGPS</span>(DEgeneList_1vsOthers[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>], 
    <span class="dt">pvalueCutoff=</span><span class="fl">0.05</span>, <span class="dt">gene_symbol=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] "Original gene number in geneList"
## [1] 500
## [1] "Number of genes successfully converted"
## [1] 486</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#the enrichment outputs can be displayed by running</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/graphics-defunct">dotplot</a></span>(enrichment_test, <span class="dt">showCategory=</span><span class="dv">10</span>, <span class="dt">font.size =</span> <span class="dv">6</span>)</code></pre></div>
<p><img src="vignette_files/figure-html/find_markers-1.png" width="700"></p>
</div>
</div>
<div id="relationship-between-clusters-within-one-sample-or-between-two-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples" class="anchor"></a>4. Relationship between clusters within one sample or between two samples</h1>
<p><em>The purpose of this workflow is to solve the following task: </em> <em>given one or two unknown mixed population(s) and clusters in each mixed</em> <em>population, estimate and visualise relationship between clusters</em></p>
<div id="start-the-scgps-prediction-to-find-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#start-the-scgps-prediction-to-find-relationship-between-clusters" class="anchor"></a>4.1 Start the scGPS prediction to find relationship between clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#select a subpopulation, and input gene list </span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#note make sure the format for genes input here is the same to the format</span>
<span class="co">#for genes in the mixedpop1 and mixedpop2  </span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>]

<span class="co">#run the test bootstrap with nboots = 2 runs</span>

cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sink">sink</a></span>(<span class="st">"temp"</span>)
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID, 
     <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
     <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 153 genes for training model"
## [1] "use 153 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 3 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 2.369e-01
##  [2,]  1  2.757e-02 2.159e-01
##  [3,]  1  5.078e-02 1.967e-01
##  [4,]  2  7.607e-02 1.792e-01
##  [5,]  3  1.038e-01 1.633e-01
##  [6,]  3  1.305e-01 1.488e-01
##  [7,]  6  1.640e-01 1.356e-01
##  [8,]  6  1.982e-01 1.235e-01
##  [9,]  6  2.281e-01 1.126e-01
## [10,]  7  2.570e-01 1.026e-01
## [11,]  9  2.863e-01 9.346e-02
## [12,] 10  3.136e-01 8.515e-02
## [13,] 11  3.400e-01 7.759e-02
## [14,] 13  3.652e-01 7.070e-02
## [15,] 16  3.903e-01 6.442e-02
## [16,] 19  4.152e-01 5.869e-02
## [17,] 24  4.416e-01 5.348e-02
## [18,] 25  4.669e-01 4.873e-02
## [19,] 25  4.899e-01 4.440e-02
## [20,] 28  5.112e-01 4.046e-02
## [21,] 33  5.342e-01 3.686e-02
## [22,] 34  5.563e-01 3.359e-02
## [23,] 39  5.778e-01 3.060e-02
## [24,] 45  6.007e-01 2.788e-02
## [25,] 49  6.241e-01 2.541e-02
## [26,] 52  6.460e-01 2.315e-02
## [27,] 52  6.670e-01 2.109e-02
## [28,] 54  6.877e-01 1.922e-02
## [29,] 55  7.072e-01 1.751e-02
## [30,] 59  7.264e-01 1.596e-02
## [31,] 63  7.455e-01 1.454e-02
## [32,] 63  7.636e-01 1.325e-02
## [33,] 64  7.805e-01 1.207e-02
## [34,] 65  7.965e-01 1.100e-02
## [35,] 68  8.115e-01 1.002e-02
## [36,] 67  8.255e-01 9.131e-03
## [37,] 66  8.384e-01 8.320e-03
## [38,] 68  8.507e-01 7.581e-03
## [39,] 69  8.621e-01 6.907e-03
## [40,] 72  8.727e-01 6.294e-03
## [41,] 76  8.828e-01 5.734e-03
## [42,] 76  8.926e-01 5.225e-03
## [43,] 77  9.016e-01 4.761e-03
## [44,] 78  9.101e-01 4.338e-03
## [45,] 79  9.177e-01 3.953e-03
## [46,] 80  9.247e-01 3.601e-03
## [47,] 83  9.313e-01 3.281e-03
## [48,] 83  9.374e-01 2.990e-03
## [49,] 84  9.430e-01 2.724e-03
## [50,] 84  9.480e-01 2.482e-03
## [51,] 88  9.526e-01 2.262e-03
## [52,] 90  9.569e-01 2.061e-03
## [53,] 90  9.607e-01 1.878e-03
## [54,] 91  9.642e-01 1.711e-03
## [55,] 91  9.674e-01 1.559e-03
## [56,] 91  9.703e-01 1.420e-03
## [57,] 91  9.730e-01 1.294e-03
## [58,] 90  9.754e-01 1.179e-03
## [59,] 90  9.776e-01 1.075e-03
## [60,] 91  9.796e-01 9.791e-04
## [61,] 91  9.814e-01 8.921e-04
## [62,] 91  9.830e-01 8.128e-04
## [63,] 91  9.845e-01 7.406e-04
## [64,] 91  9.859e-01 6.748e-04
## [65,] 91  9.872e-01 6.149e-04
## [66,] 91  9.883e-01 5.603e-04
## [67,] 92  9.893e-01 5.105e-04
## [68,] 92  9.903e-01 4.651e-04
## [69,] 92  9.911e-01 4.238e-04
## [70,] 92  9.919e-01 3.862e-04
## [71,] 93  9.926e-01 3.519e-04
## [72,] 93  9.933e-01 3.206e-04
## [73,] 93  9.939e-01 2.921e-04
## [74,] 93  9.944e-01 2.662e-04
## [75,] 93  9.949e-01 2.425e-04
## [76,] 93  9.954e-01 2.210e-04
## [77,] 93  9.958e-01 2.013e-04
## [78,] 94  9.961e-01 1.835e-04
## [79,] 94  9.965e-01 1.672e-04
## [80,] 93  9.968e-01 1.523e-04
## [81,] 93  9.971e-01 1.388e-04
## [82,] 93  9.973e-01 1.265e-04
## [83,] 93  9.976e-01 1.152e-04
## [84,] 94  9.978e-01 1.050e-04
## [85,] 94  9.980e-01 9.566e-05
## [86,] 93  9.982e-01 8.716e-05
## [87,] 94  9.983e-01 7.942e-05
## [88,] 94  9.985e-01 7.236e-05
## [89,] 94  9.986e-01 6.593e-05
## [90,] 94  9.987e-01 6.007e-05
## [91,] 94  9.988e-01 5.474e-05
## [92,] 94  9.989e-01 4.988e-05
## [93,] 94  9.990e-01 4.544e-05
## [1] "lambda min is at location 32"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 150 genes..."
## [1] "evaluation accuracy ElasticNet 0.852678571428571"
## [1] "evaluation accuracy LDA 0.71875"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 187 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 79.6791443850267"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 187 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 92.5133689839572"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 140 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 99.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 140 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 80.7142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 133 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 100"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 133 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 85.7142857142857"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4850"
## [1] "Number of genes in the model present in the target data is 150"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 150 genes and 40 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 100"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 150 genes and 40 cells. The trained model has 150 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 82.5"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 153 genes for training model"
## [1] "use 153 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 4 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 2.596e-01
##  [2,]  1  3.308e-02 2.365e-01
##  [3,]  1  6.089e-02 2.155e-01
##  [4,]  3  9.258e-02 1.964e-01
##  [5,]  3  1.288e-01 1.789e-01
##  [6,]  4  1.608e-01 1.630e-01
##  [7,]  4  1.944e-01 1.485e-01
##  [8,]  4  2.242e-01 1.353e-01
##  [9,]  4  2.508e-01 1.233e-01
## [10,]  6  2.806e-01 1.124e-01
## [11,]  6  3.092e-01 1.024e-01
## [12,]  8  3.363e-01 9.328e-02
## [13,]  9  3.617e-01 8.500e-02
## [14,] 11  3.883e-01 7.745e-02
## [15,] 14  4.149e-01 7.057e-02
## [16,] 15  4.420e-01 6.430e-02
## [17,] 16  4.680e-01 5.858e-02
## [18,] 19  4.938e-01 5.338e-02
## [19,] 22  5.187e-01 4.864e-02
## [20,] 25  5.441e-01 4.432e-02
## [21,] 26  5.679e-01 4.038e-02
## [22,] 27  5.901e-01 3.679e-02
## [23,] 26  6.106e-01 3.352e-02
## [24,] 31  6.306e-01 3.055e-02
## [25,] 33  6.505e-01 2.783e-02
## [26,] 39  6.706e-01 2.536e-02
## [27,] 39  6.916e-01 2.311e-02
## [28,] 41  7.112e-01 2.105e-02
## [29,] 41  7.306e-01 1.918e-02
## [30,] 41  7.486e-01 1.748e-02
## [31,] 45  7.658e-01 1.593e-02
## [32,] 48  7.828e-01 1.451e-02
## [33,] 49  7.993e-01 1.322e-02
## [34,] 50  8.148e-01 1.205e-02
## [35,] 52  8.293e-01 1.098e-02
## [36,] 58  8.431e-01 1.000e-02
## [37,] 59  8.562e-01 9.114e-03
## [38,] 59  8.682e-01 8.304e-03
## [39,] 63  8.794e-01 7.567e-03
## [40,] 64  8.897e-01 6.894e-03
## [41,] 64  8.992e-01 6.282e-03
## [42,] 65  9.080e-01 5.724e-03
## [43,] 67  9.160e-01 5.215e-03
## [44,] 67  9.233e-01 4.752e-03
## [45,] 68  9.300e-01 4.330e-03
## [46,] 68  9.362e-01 3.945e-03
## [47,] 68  9.417e-01 3.595e-03
## [48,] 69  9.468e-01 3.275e-03
## [49,] 70  9.515e-01 2.984e-03
## [50,] 70  9.558e-01 2.719e-03
## [51,] 71  9.597e-01 2.478e-03
## [52,] 71  9.633e-01 2.258e-03
## [53,] 71  9.665e-01 2.057e-03
## [54,] 72  9.695e-01 1.874e-03
## [55,] 72  9.722e-01 1.708e-03
## [56,] 72  9.746e-01 1.556e-03
## [57,] 75  9.769e-01 1.418e-03
## [58,] 75  9.789e-01 1.292e-03
## [59,] 74  9.808e-01 1.177e-03
## [60,] 73  9.825e-01 1.073e-03
## [61,] 74  9.840e-01 9.773e-04
## [62,] 74  9.854e-01 8.904e-04
## [63,] 74  9.867e-01 8.113e-04
## [64,] 75  9.879e-01 7.393e-04
## [65,] 75  9.890e-01 6.736e-04
## [66,] 75  9.899e-01 6.137e-04
## [67,] 74  9.908e-01 5.592e-04
## [68,] 74  9.916e-01 5.095e-04
## [69,] 74  9.924e-01 4.643e-04
## [70,] 75  9.931e-01 4.230e-04
## [71,] 77  9.937e-01 3.854e-04
## [72,] 78  9.942e-01 3.512e-04
## [73,] 79  9.947e-01 3.200e-04
## [74,] 79  9.952e-01 2.916e-04
## [75,] 79  9.956e-01 2.657e-04
## [76,] 79  9.960e-01 2.421e-04
## [77,] 79  9.964e-01 2.206e-04
## [78,] 80  9.967e-01 2.010e-04
## [79,] 80  9.970e-01 1.831e-04
## [80,] 80  9.972e-01 1.669e-04
## [81,] 81  9.975e-01 1.520e-04
## [82,] 81  9.977e-01 1.385e-04
## [83,] 81  9.979e-01 1.262e-04
## [84,] 82  9.981e-01 1.150e-04
## [85,] 82  9.983e-01 1.048e-04
## [86,] 82  9.984e-01 9.548e-05
## [87,] 82  9.986e-01 8.700e-05
## [88,] 82  9.987e-01 7.927e-05
## [89,] 83  9.988e-01 7.223e-05
## [90,] 83  9.989e-01 6.581e-05
## [91,] 83  9.990e-01 5.996e-05
## [1] "lambda min is at location 19"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 149 genes..."
## [1] "evaluation accuracy ElasticNet 0.857142857142857"
## [1] "evaluation accuracy LDA 0.714285714285714"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 187 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 42.2459893048128"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are THUMPD3-AS1"
## [2] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 187 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 17.1122994652406"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 140 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 97.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are THUMPD3-AS1"
## [2] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 140 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 52.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 133 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 62.406015037594"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are THUMPD3-AS1"
## [2] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 133 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 10.5263157894737"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4851"
## [1] "Number of genes in the model present in the target data is 149"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 149 genes and 40 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 92.5"
## [1] "Number of genes in the target data, but not in model genes is 4853"
## [1] "Number of genes in the model present in the target data is 147"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are THUMPD3-AS1"
## [2] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 149 genes and 40 cells. The trained model has 149 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 25"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sink">sink</a></span>()</code></pre></div>
</div>
<div id="display-summary-results-for-the-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#display-summary-results-for-the-prediction" class="anchor"></a>4.2 Display summary results for the prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#get the number of rows for the summary matrix </span>
row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

<span class="co">#summary results LDA to to show the percent of cells classified as cells</span>
<span class="co">#belonging by LDA classifier </span>
<span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster )</code></pre></div>
<pre><code>##                 V1               V2                                names
## 1 92.5133689839572 17.1122994652406 LDA for subpop 1 in target mixedpop2
## 2 80.7142857142857 52.8571428571429 LDA for subpop 2 in target mixedpop2
## 3 85.7142857142857 10.5263157894737 LDA for subpop 3 in target mixedpop2
## 4             82.5               25 LDA for subpop 4 in target mixedpop2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#summary results Lasso to show the percent of cells classified as cells</span>
<span class="co">#belonging by Lasso classifier</span>
<span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster)</code></pre></div>
<pre><code>##                 V1               V2
## 1 79.6791443850267 42.2459893048128
## 2 99.2857142857143 97.1428571428571
## 3              100  62.406015037594
## 4              100             92.5
##                                        names
## 1 ElasticNet for subpop1 in target mixedpop2
## 2 ElasticNet for subpop2 in target mixedpop2
## 3 ElasticNet for subpop3 in target mixedpop2
## 4 ElasticNet for subpop4 in target mixedpop2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary maximum deviance explained by the model during the model training </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## $allDeviance
## [1] "0.7636" "0.5187"
## 
## $DeviMax
##           dat_DE$Dfd          Deviance           DEgenes
## 1                  0            0.7636    genes_cluster1
## 2                  1            0.7636    genes_cluster1
## 3                  2            0.7636    genes_cluster1
## 4                  3            0.7636    genes_cluster1
## 5                  6            0.7636    genes_cluster1
## 6                  7            0.7636    genes_cluster1
## 7                  9            0.7636    genes_cluster1
## 8                 10            0.7636    genes_cluster1
## 9                 11            0.7636    genes_cluster1
## 10                13            0.7636    genes_cluster1
## 11                16            0.7636    genes_cluster1
## 12                19            0.7636    genes_cluster1
## 13                24            0.7636    genes_cluster1
## 14                25            0.7636    genes_cluster1
## 15                28            0.7636    genes_cluster1
## 16                33            0.7636    genes_cluster1
## 17                34            0.7636    genes_cluster1
## 18                39            0.7636    genes_cluster1
## 19                45            0.7636    genes_cluster1
## 20                49            0.7636    genes_cluster1
## 21                52            0.7636    genes_cluster1
## 22                54            0.7636    genes_cluster1
## 23                55            0.7636    genes_cluster1
## 24                59            0.7636    genes_cluster1
## 25                63            0.7636    genes_cluster1
## 26 remaining DEgenes remaining DEgenes remaining DEgenes
## 
## $LassoGenesMax
## NULL</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## [1] 71.87500 71.42857</code></pre>
</div>
<div id="plot-the-relationship-between-clusters-in-one-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-one-sample" class="anchor"></a>4.3 Plot the relationship between clusters in one sample</h2>
<p><em>Here we look at one example use case to find relationship between clusters #within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#cluster_mixedpop2 &lt;- as.numeric(as.vector(colData(mixedpop2)[,1]))</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#top 200 gene markers distinguishing cluster 1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] 

LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop2,
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 187 cells as source subpop"
## [1] "Total 313 cells in remaining subpops"
## [1] "subsampling 94 cells for training source subpop"
## [1] "subsampling 94 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 188 cells for testing model"
## [1] "rename remaining subpops to 2_3_4"
## [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 3.364e-15 0.264600
##   [2,]  1 1.795e-02 0.252600
##   [3,]  1 3.434e-02 0.241100
##   [4,]  1 4.934e-02 0.230100
##   [5,]  2 6.494e-02 0.219700
##   [6,]  2 8.135e-02 0.209700
##   [7,]  2 9.644e-02 0.200200
##   [8,]  2 1.104e-01 0.191100
##   [9,]  2 1.232e-01 0.182400
##  [10,]  3 1.399e-01 0.174100
##  [11,]  6 1.605e-01 0.166200
##  [12,]  6 1.827e-01 0.158600
##  [13,]  7 2.037e-01 0.151400
##  [14,]  8 2.258e-01 0.144500
##  [15,]  9 2.481e-01 0.138000
##  [16,] 10 2.702e-01 0.131700
##  [17,] 11 2.926e-01 0.125700
##  [18,] 11 3.144e-01 0.120000
##  [19,] 13 3.352e-01 0.114500
##  [20,] 14 3.560e-01 0.109300
##  [21,] 14 3.757e-01 0.104400
##  [22,] 15 3.945e-01 0.099620
##  [23,] 16 4.123e-01 0.095090
##  [24,] 16 4.294e-01 0.090770
##  [25,] 16 4.456e-01 0.086640
##  [26,] 17 4.614e-01 0.082700
##  [27,] 17 4.764e-01 0.078950
##  [28,] 18 4.915e-01 0.075360
##  [29,] 18 5.060e-01 0.071930
##  [30,] 19 5.198e-01 0.068660
##  [31,] 20 5.331e-01 0.065540
##  [32,] 21 5.461e-01 0.062560
##  [33,] 22 5.585e-01 0.059720
##  [34,] 22 5.707e-01 0.057010
##  [35,] 22 5.832e-01 0.054410
##  [36,] 23 5.954e-01 0.051940
##  [37,] 23 6.076e-01 0.049580
##  [38,] 24 6.193e-01 0.047330
##  [39,] 26 6.307e-01 0.045180
##  [40,] 27 6.417e-01 0.043120
##  [41,] 29 6.524e-01 0.041160
##  [42,] 32 6.631e-01 0.039290
##  [43,] 35 6.741e-01 0.037510
##  [44,] 37 6.850e-01 0.035800
##  [45,] 38 6.955e-01 0.034170
##  [46,] 40 7.058e-01 0.032620
##  [47,] 42 7.159e-01 0.031140
##  [48,] 44 7.258e-01 0.029720
##  [49,] 46 7.360e-01 0.028370
##  [50,] 48 7.467e-01 0.027080
##  [51,] 50 7.574e-01 0.025850
##  [52,] 51 7.677e-01 0.024680
##  [53,] 53 7.775e-01 0.023550
##  [54,] 53 7.868e-01 0.022480
##  [55,] 54 7.958e-01 0.021460
##  [56,] 54 8.043e-01 0.020490
##  [57,] 54 8.125e-01 0.019560
##  [58,] 55 8.204e-01 0.018670
##  [59,] 57 8.279e-01 0.017820
##  [60,] 59 8.353e-01 0.017010
##  [61,] 59 8.425e-01 0.016240
##  [62,] 59 8.494e-01 0.015500
##  [63,] 59 8.560e-01 0.014790
##  [64,] 58 8.623e-01 0.014120
##  [65,] 59 8.683e-01 0.013480
##  [66,] 60 8.741e-01 0.012870
##  [67,] 60 8.796e-01 0.012280
##  [68,] 60 8.848e-01 0.011720
##  [69,] 61 8.899e-01 0.011190
##  [70,] 62 8.947e-01 0.010680
##  [71,] 62 8.993e-01 0.010200
##  [72,] 61 9.038e-01 0.009733
##  [73,] 62 9.080e-01 0.009290
##  [74,] 63 9.121e-01 0.008868
##  [75,] 63 9.159e-01 0.008465
##  [76,] 63 9.197e-01 0.008080
##  [77,] 63 9.232e-01 0.007713
##  [78,] 63 9.266e-01 0.007363
##  [79,] 64 9.299e-01 0.007028
##  [80,] 64 9.330e-01 0.006708
##  [81,] 63 9.360e-01 0.006404
##  [82,] 64 9.389e-01 0.006112
##  [83,] 64 9.416e-01 0.005835
##  [84,] 64 9.443e-01 0.005569
##  [85,] 64 9.468e-01 0.005316
##  [86,] 64 9.492e-01 0.005075
##  [87,] 64 9.515e-01 0.004844
##  [88,] 64 9.536e-01 0.004624
##  [89,] 64 9.557e-01 0.004414
##  [90,] 64 9.577e-01 0.004213
##  [91,] 64 9.596e-01 0.004022
##  [92,] 64 9.614e-01 0.003839
##  [93,] 64 9.632e-01 0.003664
##  [94,] 65 9.648e-01 0.003498
##  [95,] 65 9.664e-01 0.003339
##  [96,] 64 9.679e-01 0.003187
##  [97,] 64 9.693e-01 0.003042
##  [98,] 65 9.707e-01 0.002904
##  [99,] 65 9.720e-01 0.002772
## [100,] 65 9.733e-01 0.002646
## [1] "lambda min is at location 93"
## [1] "the leave-out cells in the source subpop is 93"
## [1] "use 94 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 187 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.850267379679144"
## [1] "evaluation accuracy LDA 0.625668449197861"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 80.2139037433155"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 49.7326203208556"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 1.42857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 6.42857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 14.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 11.2781954887218"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 27.5"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 22.5"
## [1] "Total 187 cells as source subpop"
## [1] "Total 313 cells in remaining subpops"
## [1] "subsampling 94 cells for training source subpop"
## [1] "subsampling 94 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 188 cells for testing model"
## [1] "rename remaining subpops to 2_3_4"
## [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 3.364e-15 0.316000
##   [2,]  1 2.561e-02 0.301600
##   [3,]  1 4.902e-02 0.287900
##   [4,]  1 7.048e-02 0.274900
##   [5,]  1 9.019e-02 0.262400
##   [6,]  1 1.083e-01 0.250400
##   [7,]  1 1.251e-01 0.239100
##   [8,]  1 1.406e-01 0.228200
##   [9,]  1 1.549e-01 0.217800
##  [10,]  1 1.681e-01 0.207900
##  [11,]  1 1.804e-01 0.198500
##  [12,]  2 1.938e-01 0.189400
##  [13,]  2 2.069e-01 0.180800
##  [14,]  2 2.191e-01 0.172600
##  [15,]  2 2.305e-01 0.164800
##  [16,]  2 2.410e-01 0.157300
##  [17,]  3 2.540e-01 0.150100
##  [18,]  5 2.677e-01 0.143300
##  [19,]  5 2.828e-01 0.136800
##  [20,]  5 2.970e-01 0.130600
##  [21,]  5 3.103e-01 0.124600
##  [22,]  6 3.235e-01 0.119000
##  [23,]  7 3.366e-01 0.113600
##  [24,]  8 3.493e-01 0.108400
##  [25,] 10 3.619e-01 0.103500
##  [26,] 12 3.760e-01 0.098780
##  [27,] 13 3.906e-01 0.094290
##  [28,] 13 4.049e-01 0.090000
##  [29,] 15 4.191e-01 0.085910
##  [30,] 16 4.335e-01 0.082010
##  [31,] 17 4.473e-01 0.078280
##  [32,] 18 4.610e-01 0.074720
##  [33,] 18 4.745e-01 0.071320
##  [34,] 18 4.874e-01 0.068080
##  [35,] 20 5.000e-01 0.064990
##  [36,] 21 5.124e-01 0.062030
##  [37,] 22 5.246e-01 0.059220
##  [38,] 25 5.366e-01 0.056520
##  [39,] 25 5.490e-01 0.053950
##  [40,] 27 5.612e-01 0.051500
##  [41,] 27 5.729e-01 0.049160
##  [42,] 27 5.842e-01 0.046930
##  [43,] 30 5.952e-01 0.044790
##  [44,] 32 6.064e-01 0.042760
##  [45,] 33 6.174e-01 0.040810
##  [46,] 32 6.277e-01 0.038960
##  [47,] 32 6.376e-01 0.037190
##  [48,] 33 6.472e-01 0.035500
##  [49,] 36 6.567e-01 0.033890
##  [50,] 36 6.662e-01 0.032340
##  [51,] 39 6.753e-01 0.030870
##  [52,] 41 6.849e-01 0.029470
##  [53,] 44 6.946e-01 0.028130
##  [54,] 46 7.047e-01 0.026850
##  [55,] 48 7.151e-01 0.025630
##  [56,] 49 7.252e-01 0.024470
##  [57,] 50 7.353e-01 0.023360
##  [58,] 52 7.451e-01 0.022290
##  [59,] 53 7.548e-01 0.021280
##  [60,] 54 7.641e-01 0.020310
##  [61,] 53 7.730e-01 0.019390
##  [62,] 52 7.815e-01 0.018510
##  [63,] 52 7.896e-01 0.017670
##  [64,] 51 7.974e-01 0.016860
##  [65,] 51 8.050e-01 0.016100
##  [66,] 51 8.122e-01 0.015370
##  [67,] 55 8.193e-01 0.014670
##  [68,] 57 8.267e-01 0.014000
##  [69,] 60 8.339e-01 0.013360
##  [70,] 60 8.409e-01 0.012760
##  [71,] 61 8.476e-01 0.012180
##  [72,] 61 8.541e-01 0.011620
##  [73,] 62 8.603e-01 0.011100
##  [74,] 65 8.664e-01 0.010590
##  [75,] 67 8.722e-01 0.010110
##  [76,] 68 8.778e-01 0.009651
##  [77,] 69 8.832e-01 0.009212
##  [78,] 70 8.884e-01 0.008793
##  [79,] 71 8.934e-01 0.008394
##  [80,] 71 8.982e-01 0.008012
##  [81,] 72 9.027e-01 0.007648
##  [82,] 74 9.071e-01 0.007300
##  [83,] 74 9.113e-01 0.006968
##  [84,] 75 9.154e-01 0.006652
##  [85,] 75 9.192e-01 0.006349
##  [86,] 75 9.229e-01 0.006061
##  [87,] 75 9.264e-01 0.005785
##  [88,] 76 9.297e-01 0.005522
##  [89,] 76 9.329e-01 0.005271
##  [90,] 76 9.359e-01 0.005032
##  [91,] 75 9.389e-01 0.004803
##  [92,] 75 9.416e-01 0.004585
##  [93,] 75 9.443e-01 0.004376
##  [94,] 76 9.468e-01 0.004178
##  [95,] 76 9.492e-01 0.003988
##  [96,] 76 9.515e-01 0.003806
##  [97,] 76 9.537e-01 0.003633
##  [98,] 76 9.558e-01 0.003468
##  [99,] 76 9.578e-01 0.003311
## [100,] 76 9.597e-01 0.003160
## [1] "lambda min is at location 85"
## [1] "the leave-out cells in the source subpop is 93"
## [1] "use 94 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 187 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.855614973262032"
## [1] "evaluation accuracy LDA 0.620320855614973"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 84.4919786096257"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 83.4224598930481"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 16.4285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 2.25563909774436"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 37.593984962406"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 10"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 42.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]

LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 140 cells as source subpop"
## [1] "Total 360 cells in remaining subpops"
## [1] "subsampling 70 cells for training source subpop"
## [1] "subsampling 70 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 140 cells for testing model"
## [1] "rename remaining subpops to 1_3_4"
## [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 2.723e-15 0.406200
##   [2,]  1 4.234e-02 0.387800
##   [3,]  1 8.112e-02 0.370100
##   [4,]  1 1.168e-01 0.353300
##   [5,]  1 1.498e-01 0.337300
##   [6,]  1 1.804e-01 0.321900
##   [7,]  1 2.088e-01 0.307300
##   [8,]  1 2.354e-01 0.293300
##   [9,]  1 2.603e-01 0.280000
##  [10,]  1 2.836e-01 0.267300
##  [11,]  1 3.054e-01 0.255100
##  [12,]  1 3.261e-01 0.243500
##  [13,]  1 3.455e-01 0.232500
##  [14,]  1 3.638e-01 0.221900
##  [15,]  2 3.828e-01 0.211800
##  [16,]  2 4.015e-01 0.202200
##  [17,]  3 4.216e-01 0.193000
##  [18,]  3 4.414e-01 0.184200
##  [19,]  4 4.603e-01 0.175800
##  [20,]  4 4.792e-01 0.167900
##  [21,]  5 4.970e-01 0.160200
##  [22,]  6 5.143e-01 0.152900
##  [23,]  6 5.307e-01 0.146000
##  [24,]  8 5.469e-01 0.139400
##  [25,]  9 5.630e-01 0.133000
##  [26,] 10 5.788e-01 0.127000
##  [27,] 10 5.939e-01 0.121200
##  [28,] 11 6.084e-01 0.115700
##  [29,] 11 6.222e-01 0.110400
##  [30,] 11 6.353e-01 0.105400
##  [31,] 11 6.476e-01 0.100600
##  [32,] 11 6.593e-01 0.096050
##  [33,] 14 6.715e-01 0.091690
##  [34,] 14 6.834e-01 0.087520
##  [35,] 14 6.947e-01 0.083540
##  [36,] 14 7.054e-01 0.079740
##  [37,] 15 7.158e-01 0.076120
##  [38,] 15 7.257e-01 0.072660
##  [39,] 15 7.351e-01 0.069360
##  [40,] 16 7.441e-01 0.066200
##  [41,] 16 7.527e-01 0.063200
##  [42,] 16 7.609e-01 0.060320
##  [43,] 16 7.687e-01 0.057580
##  [44,] 16 7.762e-01 0.054960
##  [45,] 16 7.833e-01 0.052470
##  [46,] 18 7.902e-01 0.050080
##  [47,] 19 7.970e-01 0.047810
##  [48,] 19 8.036e-01 0.045630
##  [49,] 19 8.098e-01 0.043560
##  [50,] 20 8.160e-01 0.041580
##  [51,] 21 8.220e-01 0.039690
##  [52,] 21 8.278e-01 0.037880
##  [53,] 22 8.334e-01 0.036160
##  [54,] 24 8.391e-01 0.034520
##  [55,] 24 8.453e-01 0.032950
##  [56,] 23 8.513e-01 0.031450
##  [57,] 23 8.572e-01 0.030020
##  [58,] 23 8.628e-01 0.028660
##  [59,] 24 8.683e-01 0.027360
##  [60,] 24 8.737e-01 0.026110
##  [61,] 26 8.791e-01 0.024930
##  [62,] 26 8.842e-01 0.023790
##  [63,] 26 8.892e-01 0.022710
##  [64,] 26 8.939e-01 0.021680
##  [65,] 27 8.985e-01 0.020690
##  [66,] 27 9.031e-01 0.019750
##  [67,] 27 9.075e-01 0.018860
##  [68,] 27 9.116e-01 0.018000
##  [69,] 27 9.155e-01 0.017180
##  [70,] 27 9.193e-01 0.016400
##  [71,] 27 9.229e-01 0.015650
##  [72,] 28 9.263e-01 0.014940
##  [73,] 28 9.296e-01 0.014260
##  [74,] 28 9.327e-01 0.013620
##  [75,] 28 9.357e-01 0.013000
##  [76,] 28 9.386e-01 0.012410
##  [77,] 28 9.413e-01 0.011840
##  [78,] 28 9.439e-01 0.011300
##  [79,] 28 9.463e-01 0.010790
##  [80,] 28 9.487e-01 0.010300
##  [81,] 28 9.510e-01 0.009831
##  [82,] 28 9.531e-01 0.009384
##  [83,] 28 9.552e-01 0.008958
##  [84,] 28 9.571e-01 0.008551
##  [85,] 28 9.590e-01 0.008162
##  [86,] 29 9.608e-01 0.007791
##  [87,] 29 9.625e-01 0.007437
##  [88,] 29 9.642e-01 0.007099
##  [89,] 29 9.658e-01 0.006776
##  [90,] 28 9.673e-01 0.006468
##  [91,] 29 9.687e-01 0.006174
##  [92,] 29 9.700e-01 0.005894
##  [93,] 29 9.714e-01 0.005626
##  [94,] 29 9.726e-01 0.005370
##  [95,] 29 9.738e-01 0.005126
##  [96,] 29 9.749e-01 0.004893
##  [97,] 29 9.760e-01 0.004671
##  [98,] 29 9.771e-01 0.004458
##  [99,] 29 9.781e-01 0.004256
## [100,] 29 9.790e-01 0.004062
## [1] "lambda min is at location 46"
## [1] "the leave-out cells in the source subpop is 70"
## [1] "use 70 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 140 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.964285714285714"
## [1] "evaluation accuracy LDA 0.914285714285714"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 1.60427807486631"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 49.1978609625668"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 98.5714285714286"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 100"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 15.0375939849624"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 57.8947368421053"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 35"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 55"
## [1] "Total 140 cells as source subpop"
## [1] "Total 360 cells in remaining subpops"
## [1] "subsampling 70 cells for training source subpop"
## [1] "subsampling 70 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 140 cells for testing model"
## [1] "rename remaining subpops to 1_3_4"
## [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 2.723e-15 0.395400
##   [2,]  1 4.011e-02 0.377400
##   [3,]  1 7.685e-02 0.360300
##   [4,]  1 1.107e-01 0.343900
##   [5,]  1 1.419e-01 0.328300
##   [6,]  1 1.709e-01 0.313300
##   [7,]  1 1.978e-01 0.299100
##   [8,]  1 2.230e-01 0.285500
##   [9,]  2 2.504e-01 0.272500
##  [10,]  2 2.775e-01 0.260100
##  [11,]  2 3.029e-01 0.248300
##  [12,]  3 3.284e-01 0.237000
##  [13,]  4 3.548e-01 0.226300
##  [14,]  5 3.805e-01 0.216000
##  [15,]  6 4.057e-01 0.206200
##  [16,]  6 4.296e-01 0.196800
##  [17,]  6 4.521e-01 0.187800
##  [18,]  6 4.734e-01 0.179300
##  [19,]  6 4.935e-01 0.171200
##  [20,]  7 5.133e-01 0.163400
##  [21,]  7 5.326e-01 0.156000
##  [22,]  7 5.509e-01 0.148900
##  [23,]  8 5.687e-01 0.142100
##  [24,]  8 5.857e-01 0.135600
##  [25,]  8 6.020e-01 0.129500
##  [26,]  8 6.174e-01 0.123600
##  [27,]  8 6.321e-01 0.118000
##  [28,]  8 6.462e-01 0.112600
##  [29,]  8 6.596e-01 0.107500
##  [30,]  8 6.724e-01 0.102600
##  [31,]  8 6.847e-01 0.097940
##  [32,]  8 6.964e-01 0.093490
##  [33,]  8 7.076e-01 0.089240
##  [34,]  8 7.183e-01 0.085180
##  [35,]  8 7.286e-01 0.081310
##  [36,]  8 7.384e-01 0.077620
##  [37,]  8 7.479e-01 0.074090
##  [38,]  9 7.570e-01 0.070720
##  [39,] 10 7.658e-01 0.067510
##  [40,] 11 7.743e-01 0.064440
##  [41,] 14 7.826e-01 0.061510
##  [42,] 14 7.909e-01 0.058710
##  [43,] 14 7.988e-01 0.056050
##  [44,] 14 8.063e-01 0.053500
##  [45,] 15 8.136e-01 0.051070
##  [46,] 15 8.206e-01 0.048750
##  [47,] 16 8.273e-01 0.046530
##  [48,] 16 8.338e-01 0.044420
##  [49,] 16 8.401e-01 0.042400
##  [50,] 19 8.463e-01 0.040470
##  [51,] 19 8.523e-01 0.038630
##  [52,] 21 8.582e-01 0.036870
##  [53,] 21 8.638e-01 0.035200
##  [54,] 21 8.692e-01 0.033600
##  [55,] 22 8.744e-01 0.032070
##  [56,] 23 8.794e-01 0.030610
##  [57,] 23 8.843e-01 0.029220
##  [58,] 24 8.890e-01 0.027890
##  [59,] 24 8.935e-01 0.026630
##  [60,] 24 8.978e-01 0.025420
##  [61,] 24 9.020e-01 0.024260
##  [62,] 24 9.060e-01 0.023160
##  [63,] 24 9.098e-01 0.022110
##  [64,] 24 9.135e-01 0.021100
##  [65,] 24 9.171e-01 0.020140
##  [66,] 25 9.205e-01 0.019230
##  [67,] 25 9.238e-01 0.018350
##  [68,] 26 9.269e-01 0.017520
##  [69,] 26 9.299e-01 0.016720
##  [70,] 26 9.328e-01 0.015960
##  [71,] 26 9.357e-01 0.015240
##  [72,] 26 9.384e-01 0.014540
##  [73,] 26 9.410e-01 0.013880
##  [74,] 26 9.435e-01 0.013250
##  [75,] 26 9.459e-01 0.012650
##  [76,] 27 9.482e-01 0.012070
##  [77,] 27 9.504e-01 0.011530
##  [78,] 27 9.526e-01 0.011000
##  [79,] 27 9.546e-01 0.010500
##  [80,] 27 9.566e-01 0.010020
##  [81,] 27 9.584e-01 0.009569
##  [82,] 27 9.602e-01 0.009134
##  [83,] 28 9.619e-01 0.008719
##  [84,] 28 9.636e-01 0.008323
##  [85,] 28 9.652e-01 0.007944
##  [86,] 28 9.667e-01 0.007583
##  [87,] 28 9.681e-01 0.007239
##  [88,] 29 9.695e-01 0.006910
##  [89,] 29 9.708e-01 0.006596
##  [90,] 29 9.721e-01 0.006296
##  [91,] 29 9.733e-01 0.006010
##  [92,] 29 9.745e-01 0.005736
##  [93,] 29 9.756e-01 0.005476
##  [94,] 29 9.767e-01 0.005227
##  [95,] 29 9.777e-01 0.004989
##  [96,] 29 9.787e-01 0.004763
##  [97,] 29 9.796e-01 0.004546
##  [98,] 29 9.805e-01 0.004339
##  [99,] 29 9.814e-01 0.004142
## [100,] 30 9.822e-01 0.003954
## [1] "lambda min is at location 35"
## [1] "the leave-out cells in the source subpop is 70"
## [1] "use 70 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 140 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.957142857142857"
## [1] "evaluation accuracy LDA 0.921428571428571"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 10.1604278074866"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 19.7860962566845"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 97.8571428571428"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 100"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 0.75187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 32.3308270676692"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 32.5"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 52.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 133 cells as source subpop"
## [1] "Total 367 cells in remaining subpops"
## [1] "subsampling 66 cells for training source subpop"
## [1] "subsampling 66 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 132 cells for testing model"
## [1] "rename remaining subpops to 1_2_4"
## [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev   Lambda
##   [1,]  0 -2.403e-15 0.219300
##   [2,]  3  1.934e-02 0.209300
##   [3,]  4  4.345e-02 0.199800
##   [4,]  5  6.890e-02 0.190700
##   [5,]  6  9.313e-02 0.182100
##   [6,]  6  1.157e-01 0.173800
##   [7,]  7  1.367e-01 0.165900
##   [8,]  8  1.578e-01 0.158300
##   [9,]  9  1.782e-01 0.151200
##  [10,] 10  1.993e-01 0.144300
##  [11,] 11  2.200e-01 0.137700
##  [12,] 12  2.405e-01 0.131500
##  [13,] 13  2.614e-01 0.125500
##  [14,] 14  2.817e-01 0.119800
##  [15,] 15  3.013e-01 0.114300
##  [16,] 15  3.207e-01 0.109100
##  [17,] 16  3.393e-01 0.104200
##  [18,] 16  3.592e-01 0.099450
##  [19,] 16  3.780e-01 0.094930
##  [20,] 17  3.963e-01 0.090610
##  [21,] 18  4.142e-01 0.086490
##  [22,] 19  4.315e-01 0.082560
##  [23,] 23  4.496e-01 0.078810
##  [24,] 24  4.681e-01 0.075230
##  [25,] 25  4.858e-01 0.071810
##  [26,] 27  5.038e-01 0.068550
##  [27,] 27  5.215e-01 0.065430
##  [28,] 27  5.383e-01 0.062460
##  [29,] 28  5.548e-01 0.059620
##  [30,] 30  5.706e-01 0.056910
##  [31,] 30  5.867e-01 0.054320
##  [32,] 31  6.023e-01 0.051850
##  [33,] 32  6.173e-01 0.049500
##  [34,] 36  6.319e-01 0.047250
##  [35,] 37  6.462e-01 0.045100
##  [36,] 37  6.600e-01 0.043050
##  [37,] 38  6.733e-01 0.041090
##  [38,] 38  6.862e-01 0.039220
##  [39,] 39  6.985e-01 0.037440
##  [40,] 40  7.105e-01 0.035740
##  [41,] 40  7.220e-01 0.034120
##  [42,] 41  7.330e-01 0.032560
##  [43,] 41  7.437e-01 0.031080
##  [44,] 42  7.540e-01 0.029670
##  [45,] 44  7.643e-01 0.028320
##  [46,] 45  7.744e-01 0.027040
##  [47,] 46  7.841e-01 0.025810
##  [48,] 47  7.935e-01 0.024630
##  [49,] 50  8.024e-01 0.023510
##  [50,] 50  8.112e-01 0.022450
##  [51,] 50  8.195e-01 0.021430
##  [52,] 50  8.276e-01 0.020450
##  [53,] 51  8.352e-01 0.019520
##  [54,] 51  8.425e-01 0.018630
##  [55,] 50  8.495e-01 0.017790
##  [56,] 50  8.561e-01 0.016980
##  [57,] 51  8.624e-01 0.016210
##  [58,] 50  8.685e-01 0.015470
##  [59,] 50  8.742e-01 0.014770
##  [60,] 50  8.798e-01 0.014100
##  [61,] 50  8.850e-01 0.013460
##  [62,] 51  8.901e-01 0.012840
##  [63,] 50  8.950e-01 0.012260
##  [64,] 50  8.997e-01 0.011700
##  [65,] 51  9.041e-01 0.011170
##  [66,] 51  9.084e-01 0.010660
##  [67,] 51  9.124e-01 0.010180
##  [68,] 51  9.163e-01 0.009716
##  [69,] 51  9.201e-01 0.009275
##  [70,] 51  9.236e-01 0.008853
##  [71,] 51  9.270e-01 0.008451
##  [72,] 51  9.302e-01 0.008067
##  [73,] 52  9.334e-01 0.007700
##  [74,] 54  9.364e-01 0.007350
##  [75,] 54  9.392e-01 0.007016
##  [76,] 55  9.420e-01 0.006697
##  [77,] 55  9.446e-01 0.006393
##  [78,] 56  9.471e-01 0.006102
##  [79,] 57  9.495e-01 0.005825
##  [80,] 57  9.517e-01 0.005560
##  [81,] 57  9.539e-01 0.005307
##  [82,] 57  9.560e-01 0.005066
##  [83,] 56  9.580e-01 0.004836
##  [84,] 57  9.599e-01 0.004616
##  [85,] 58  9.617e-01 0.004406
##  [86,] 59  9.634e-01 0.004206
##  [87,] 60  9.651e-01 0.004015
##  [88,] 60  9.667e-01 0.003832
##  [89,] 61  9.682e-01 0.003658
##  [90,] 62  9.696e-01 0.003492
##  [91,] 61  9.710e-01 0.003333
##  [92,] 61  9.723e-01 0.003182
##  [93,] 61  9.736e-01 0.003037
##  [94,] 61  9.748e-01 0.002899
##  [95,] 61  9.759e-01 0.002767
##  [96,] 61  9.770e-01 0.002641
##  [97,] 62  9.780e-01 0.002521
##  [98,] 62  9.790e-01 0.002407
##  [99,] 62  9.800e-01 0.002297
## [100,] 62  9.809e-01 0.002193
## [1] "lambda min is at location 95"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 66 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.849624060150376"
## [1] "evaluation accuracy LDA 0.714285714285714"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 50.2673796791444"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 43.3155080213904"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 3.57142857142857"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 42.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 88.7218045112782"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 88.7218045112782"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 17.5"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 52.5"
## [1] "Total 133 cells as source subpop"
## [1] "Total 367 cells in remaining subpops"
## [1] "subsampling 66 cells for training source subpop"
## [1] "subsampling 66 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 132 cells for testing model"
## [1] "rename remaining subpops to 1_2_4"
## [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev   Lambda
##   [1,]  0 -2.403e-15 0.221100
##   [2,]  1  1.253e-02 0.211000
##   [3,]  3  3.408e-02 0.201400
##   [4,]  3  5.563e-02 0.192300
##   [5,]  4  7.796e-02 0.183500
##   [6,]  4  9.997e-02 0.175200
##   [7,]  4  1.203e-01 0.167200
##   [8,]  4  1.392e-01 0.159600
##   [9,]  5  1.574e-01 0.152400
##  [10,]  7  1.789e-01 0.145400
##  [11,]  7  2.003e-01 0.138800
##  [12,]  8  2.211e-01 0.132500
##  [13,]  8  2.407e-01 0.126500
##  [14,]  9  2.592e-01 0.120700
##  [15,]  9  2.773e-01 0.115300
##  [16,] 10  2.943e-01 0.110000
##  [17,] 10  3.103e-01 0.105000
##  [18,] 12  3.282e-01 0.100200
##  [19,] 14  3.464e-01 0.095690
##  [20,] 14  3.639e-01 0.091340
##  [21,] 15  3.813e-01 0.087190
##  [22,] 16  3.979e-01 0.083230
##  [23,] 17  4.154e-01 0.079440
##  [24,] 17  4.323e-01 0.075830
##  [25,] 17  4.483e-01 0.072390
##  [26,] 19  4.636e-01 0.069100
##  [27,] 20  4.802e-01 0.065960
##  [28,] 22  4.964e-01 0.062960
##  [29,] 23  5.123e-01 0.060100
##  [30,] 25  5.285e-01 0.057360
##  [31,] 26  5.445e-01 0.054760
##  [32,] 27  5.603e-01 0.052270
##  [33,] 28  5.755e-01 0.049890
##  [34,] 28  5.901e-01 0.047630
##  [35,] 28  6.041e-01 0.045460
##  [36,] 30  6.180e-01 0.043390
##  [37,] 31  6.316e-01 0.041420
##  [38,] 32  6.453e-01 0.039540
##  [39,] 32  6.586e-01 0.037740
##  [40,] 33  6.718e-01 0.036030
##  [41,] 33  6.846e-01 0.034390
##  [42,] 33  6.968e-01 0.032830
##  [43,] 35  7.089e-01 0.031330
##  [44,] 36  7.207e-01 0.029910
##  [45,] 36  7.321e-01 0.028550
##  [46,] 36  7.429e-01 0.027250
##  [47,] 36  7.533e-01 0.026010
##  [48,] 39  7.636e-01 0.024830
##  [49,] 39  7.736e-01 0.023700
##  [50,] 40  7.831e-01 0.022630
##  [51,] 40  7.923e-01 0.021600
##  [52,] 41  8.011e-01 0.020620
##  [53,] 42  8.096e-01 0.019680
##  [54,] 42  8.179e-01 0.018780
##  [55,] 43  8.259e-01 0.017930
##  [56,] 46  8.336e-01 0.017120
##  [57,] 46  8.409e-01 0.016340
##  [58,] 46  8.480e-01 0.015600
##  [59,] 47  8.547e-01 0.014890
##  [60,] 47  8.612e-01 0.014210
##  [61,] 46  8.673e-01 0.013560
##  [62,] 46  8.731e-01 0.012950
##  [63,] 47  8.787e-01 0.012360
##  [64,] 47  8.841e-01 0.011800
##  [65,] 47  8.891e-01 0.011260
##  [66,] 47  8.940e-01 0.010750
##  [67,] 48  8.987e-01 0.010260
##  [68,] 48  9.032e-01 0.009794
##  [69,] 48  9.074e-01 0.009349
##  [70,] 48  9.115e-01 0.008924
##  [71,] 49  9.154e-01 0.008518
##  [72,] 49  9.191e-01 0.008131
##  [73,] 49  9.227e-01 0.007762
##  [74,] 49  9.261e-01 0.007409
##  [75,] 49  9.294e-01 0.007072
##  [76,] 50  9.325e-01 0.006751
##  [77,] 50  9.354e-01 0.006444
##  [78,] 50  9.383e-01 0.006151
##  [79,] 51  9.410e-01 0.005871
##  [80,] 51  9.437e-01 0.005605
##  [81,] 51  9.462e-01 0.005350
##  [82,] 51  9.486e-01 0.005107
##  [83,] 52  9.508e-01 0.004875
##  [84,] 51  9.530e-01 0.004653
##  [85,] 51  9.551e-01 0.004442
##  [86,] 51  9.571e-01 0.004240
##  [87,] 51  9.590e-01 0.004047
##  [88,] 51  9.608e-01 0.003863
##  [89,] 51  9.626e-01 0.003687
##  [90,] 51  9.642e-01 0.003520
##  [91,] 51  9.658e-01 0.003360
##  [92,] 51  9.674e-01 0.003207
##  [93,] 50  9.688e-01 0.003061
##  [94,] 50  9.702e-01 0.002922
##  [95,] 50  9.715e-01 0.002789
##  [96,] 50  9.728e-01 0.002663
##  [97,] 51  9.740e-01 0.002542
##  [98,] 51  9.752e-01 0.002426
##  [99,] 52  9.763e-01 0.002316
## [100,] 52  9.773e-01 0.002211
## [1] "lambda min is at location 71"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 66 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.789473684210526"
## [1] "evaluation accuracy LDA 0.75187969924812"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 35.2941176470588"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 28.8770053475936"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0.714285714285714"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 23.5714285714286"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 82.7067669172932"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 89.4736842105263"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 10"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 47.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">4</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop4vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat4 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2,
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, 
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 40 cells as source subpop"
## [1] "Total 460 cells in remaining subpops"
## [1] "subsampling 20 cells for training source subpop"
## [1] "subsampling 20 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 40 cells for testing model"
## [1] "rename remaining subpops to 1_2_3"
## [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df    %Dev   Lambda
##   [1,]  0 0.00000 0.273100
##   [2,]  1 0.01913 0.260700
##   [3,]  2 0.04428 0.248900
##   [4,]  2 0.06788 0.237500
##   [5,]  2 0.08959 0.226700
##   [6,]  3 0.11000 0.216400
##   [7,]  3 0.13340 0.206600
##   [8,]  4 0.16410 0.197200
##   [9,]  4 0.19280 0.188200
##  [10,]  5 0.21960 0.179700
##  [11,]  6 0.25170 0.171500
##  [12,]  6 0.28210 0.163700
##  [13,]  7 0.31140 0.156300
##  [14,]  7 0.33890 0.149200
##  [15,]  7 0.36480 0.142400
##  [16,]  7 0.38900 0.135900
##  [17,]  8 0.41200 0.129800
##  [18,]  8 0.43450 0.123900
##  [19,]  8 0.45570 0.118200
##  [20,]  8 0.47570 0.112900
##  [21,]  9 0.49510 0.107700
##  [22,] 10 0.51490 0.102800
##  [23,] 12 0.53410 0.098150
##  [24,] 12 0.55320 0.093690
##  [25,] 14 0.57230 0.089430
##  [26,] 14 0.59210 0.085370
##  [27,] 14 0.61080 0.081490
##  [28,] 14 0.62850 0.077790
##  [29,] 14 0.64540 0.074250
##  [30,] 14 0.66130 0.070870
##  [31,] 14 0.67650 0.067650
##  [32,] 13 0.69070 0.064580
##  [33,] 13 0.70430 0.061640
##  [34,] 15 0.71740 0.058840
##  [35,] 16 0.73030 0.056170
##  [36,] 16 0.74250 0.053610
##  [37,] 17 0.75420 0.051180
##  [38,] 17 0.76530 0.048850
##  [39,] 17 0.77590 0.046630
##  [40,] 17 0.78600 0.044510
##  [41,] 17 0.79560 0.042490
##  [42,] 17 0.80470 0.040560
##  [43,] 17 0.81350 0.038710
##  [44,] 17 0.82180 0.036950
##  [45,] 17 0.82980 0.035270
##  [46,] 17 0.83740 0.033670
##  [47,] 16 0.84470 0.032140
##  [48,] 18 0.85170 0.030680
##  [49,] 19 0.85830 0.029290
##  [50,] 20 0.86470 0.027950
##  [51,] 20 0.87080 0.026680
##  [52,] 20 0.87660 0.025470
##  [53,] 20 0.88210 0.024310
##  [54,] 20 0.88740 0.023210
##  [55,] 20 0.89240 0.022150
##  [56,] 20 0.89720 0.021150
##  [57,] 21 0.90190 0.020190
##  [58,] 21 0.90630 0.019270
##  [59,] 23 0.91050 0.018390
##  [60,] 23 0.91460 0.017560
##  [61,] 23 0.91850 0.016760
##  [62,] 24 0.92230 0.016000
##  [63,] 24 0.92590 0.015270
##  [64,] 24 0.92930 0.014580
##  [65,] 24 0.93260 0.013910
##  [66,] 24 0.93570 0.013280
##  [67,] 24 0.93860 0.012680
##  [68,] 24 0.94140 0.012100
##  [69,] 25 0.94410 0.011550
##  [70,] 25 0.94670 0.011030
##  [71,] 25 0.94910 0.010520
##  [72,] 25 0.95150 0.010050
##  [73,] 25 0.95370 0.009590
##  [74,] 26 0.95580 0.009154
##  [75,] 26 0.95790 0.008738
##  [76,] 26 0.95980 0.008341
##  [77,] 26 0.96170 0.007962
##  [78,] 26 0.96340 0.007600
##  [79,] 26 0.96510 0.007254
##  [80,] 25 0.96670 0.006925
##  [81,] 25 0.96820 0.006610
##  [82,] 26 0.96970 0.006309
##  [83,] 26 0.97110 0.006023
##  [84,] 26 0.97240 0.005749
##  [85,] 26 0.97370 0.005488
##  [86,] 26 0.97490 0.005238
##  [87,] 26 0.97600 0.005000
##  [88,] 26 0.97710 0.004773
##  [89,] 26 0.97820 0.004556
##  [90,] 26 0.97920 0.004349
##  [91,] 26 0.98010 0.004151
##  [92,] 26 0.98100 0.003962
##  [93,] 26 0.98190 0.003782
##  [94,] 26 0.98270 0.003610
##  [95,] 26 0.98350 0.003446
##  [96,] 26 0.98430 0.003290
##  [97,] 26 0.98500 0.003140
##  [98,] 26 0.98570 0.002997
##  [99,] 26 0.98630 0.002861
## [100,] 27 0.98690 0.002731
## [1] "lambda min is at location 29"
## [1] "the leave-out cells in the source subpop is 20"
## [1] "use 20 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 40 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.725"
## [1] "evaluation accuracy LDA 0.675"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 50.8021390374332"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 19.7860962566845"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 3.57142857142857"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 10"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 7.5187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 29.3233082706767"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 75"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 72.5"
## [1] "Total 40 cells as source subpop"
## [1] "Total 460 cells in remaining subpops"
## [1] "subsampling 20 cells for training source subpop"
## [1] "subsampling 20 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 40 cells for testing model"
## [1] "rename remaining subpops to 1_2_3"
## [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df    %Dev   Lambda
##   [1,]  0 0.00000 0.233200
##   [2,]  1 0.01395 0.222600
##   [3,]  2 0.03438 0.212500
##   [4,]  2 0.05396 0.202800
##   [5,]  3 0.07391 0.193600
##   [6,]  3 0.09641 0.184800
##   [7,]  3 0.11730 0.176400
##   [8,]  6 0.13830 0.168400
##   [9,]  7 0.16820 0.160700
##  [10,]  9 0.20010 0.153400
##  [11,] 10 0.23380 0.146500
##  [12,] 10 0.26700 0.139800
##  [13,] 10 0.29810 0.133400
##  [14,] 10 0.32710 0.127400
##  [15,] 11 0.35500 0.121600
##  [16,] 11 0.38190 0.116100
##  [17,] 12 0.40820 0.110800
##  [18,] 12 0.43350 0.105800
##  [19,] 13 0.45740 0.100900
##  [20,] 15 0.48100 0.096360
##  [21,] 15 0.50350 0.091980
##  [22,] 16 0.52490 0.087800
##  [23,] 16 0.54560 0.083810
##  [24,] 15 0.56520 0.080000
##  [25,] 16 0.58380 0.076360
##  [26,] 18 0.60260 0.072890
##  [27,] 18 0.62130 0.069580
##  [28,] 19 0.63910 0.066420
##  [29,] 19 0.65600 0.063400
##  [30,] 19 0.67210 0.060520
##  [31,] 19 0.68740 0.057760
##  [32,] 20 0.70210 0.055140
##  [33,] 20 0.71620 0.052630
##  [34,] 21 0.72980 0.050240
##  [35,] 21 0.74290 0.047960
##  [36,] 21 0.75530 0.045780
##  [37,] 21 0.76700 0.043700
##  [38,] 23 0.77830 0.041710
##  [39,] 23 0.78890 0.039820
##  [40,] 24 0.79910 0.038010
##  [41,] 24 0.80870 0.036280
##  [42,] 25 0.81790 0.034630
##  [43,] 25 0.82670 0.033060
##  [44,] 25 0.83500 0.031550
##  [45,] 25 0.84280 0.030120
##  [46,] 25 0.85030 0.028750
##  [47,] 26 0.85740 0.027440
##  [48,] 26 0.86420 0.026200
##  [49,] 26 0.87070 0.025010
##  [50,] 26 0.87680 0.023870
##  [51,] 26 0.88270 0.022780
##  [52,] 27 0.88820 0.021750
##  [53,] 28 0.89350 0.020760
##  [54,] 28 0.89850 0.019820
##  [55,] 28 0.90330 0.018920
##  [56,] 28 0.90790 0.018060
##  [57,] 29 0.91220 0.017230
##  [58,] 29 0.91630 0.016450
##  [59,] 29 0.92030 0.015700
##  [60,] 29 0.92400 0.014990
##  [61,] 29 0.92750 0.014310
##  [62,] 29 0.93090 0.013660
##  [63,] 29 0.93410 0.013040
##  [64,] 29 0.93720 0.012450
##  [65,] 29 0.94010 0.011880
##  [66,] 29 0.94290 0.011340
##  [67,] 29 0.94550 0.010820
##  [68,] 29 0.94810 0.010330
##  [69,] 29 0.95050 0.009862
##  [70,] 29 0.95270 0.009414
##  [71,] 29 0.95490 0.008986
##  [72,] 29 0.95700 0.008578
##  [73,] 28 0.95900 0.008188
##  [74,] 28 0.96090 0.007816
##  [75,] 28 0.96270 0.007461
##  [76,] 28 0.96440 0.007122
##  [77,] 28 0.96600 0.006798
##  [78,] 28 0.96760 0.006489
##  [79,] 28 0.96910 0.006194
##  [80,] 28 0.97050 0.005912
##  [81,] 28 0.97180 0.005644
##  [82,] 29 0.97310 0.005387
##  [83,] 29 0.97440 0.005142
##  [84,] 29 0.97550 0.004909
##  [85,] 29 0.97660 0.004685
##  [86,] 29 0.97770 0.004473
##  [87,] 29 0.97870 0.004269
##  [88,] 29 0.97970 0.004075
##  [89,] 29 0.98060 0.003890
##  [90,] 29 0.98150 0.003713
##  [91,] 29 0.98240 0.003544
##  [92,] 29 0.98320 0.003383
##  [93,] 29 0.98390 0.003230
##  [94,] 29 0.98470 0.003083
##  [95,] 29 0.98540 0.002943
##  [96,] 28 0.98600 0.002809
##  [97,] 29 0.98670 0.002681
##  [98,] 29 0.98730 0.002559
##  [99,] 29 0.98790 0.002443
## [100,] 28 0.98840 0.002332
## [1] "lambda min is at location 30"
## [1] "the leave-out cells in the source subpop is 20"
## [1] "use 20 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 40 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.775"
## [1] "evaluation accuracy LDA 0.825"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 54.0106951871658"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 55.0802139037433"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 22.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 81.4285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 12.0300751879699"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 61.6541353383459"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 82.5"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 92.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#prepare table input for sankey plot </span>

LASSO_C1S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#7570b3"</span>) 

LASSO_C2S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#1b9e77"</span>)

LASSO_C3S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#e7298a"</span>)
                          
LASSO_C4S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">4</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, 
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat4, 
    <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
    <span class="dt">Nodes_group =</span><span class="st">"#00FFFF"</span>)

combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2, LASSO_C4S2 )
combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)], 
    <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))]) 

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))
Color &lt;-<span class="st"> </span>combined <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(Node, <span class="dt">color=</span>NodeGroup) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="dv">2</span>)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color<span class="op">$</span>color

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))
p1&lt;-<span class="kw">sankeyNetwork</span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,
    <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>,
    <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, <span class="dt">fontSize =</span> <span class="dv">22</span> )
p1</code></pre></div>
<div id="htmlwidget-d26c809d65fad1f6617e" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-d26c809d65fad1f6617e">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2,3,3,3,3],"target":[0,1,2,3,0,1,2,3,0,1,2,3,0,1,2,3],"value":[82.3529411764706,0.714285714285715,8.27067669172933,18.75,5.88235294117646,98.2142857142857,7.89473684210526,33.75,42.7807486631016,2.14285714285714,85.7142857142857,13.75,52.4064171122995,12.8571428571428,9.77443609022555,78.75],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a","#00FFFF","#00FFFF","#00FFFF","#00FFFF"]},"nodes":{"name":["C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#7570b3","#1b9e77","#e7298a","#00FFFF"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters-in-two-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-two-samples" class="anchor"></a>4.3 Plot the relationship between clusters in two samples</h2>
<p><em>Here we look at one example use case to find relationship between clusters</em> <em>within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#top 200 gene markers distinguishing cluster 1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] 
LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 63 genes for training model"
## [1] "use 63 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 0.2052000
##  [2,]  1  2.067e-02 0.1870000
##  [3,]  1  3.801e-02 0.1704000
##  [4,]  4  6.757e-02 0.1553000
##  [5,]  5  1.040e-01 0.1415000
##  [6,]  5  1.386e-01 0.1289000
##  [7,]  6  1.701e-01 0.1175000
##  [8,]  6  1.991e-01 0.1070000
##  [9,]  6  2.244e-01 0.0975100
## [10,]  6  2.468e-01 0.0888500
## [11,]  6  2.664e-01 0.0809500
## [12,]  6  2.839e-01 0.0737600
## [13,]  7  2.998e-01 0.0672100
## [14,]  9  3.170e-01 0.0612400
## [15,] 11  3.336e-01 0.0558000
## [16,] 13  3.520e-01 0.0508400
## [17,] 15  3.700e-01 0.0463200
## [18,] 16  3.872e-01 0.0422100
## [19,] 17  4.033e-01 0.0384600
## [20,] 19  4.193e-01 0.0350400
## [21,] 21  4.344e-01 0.0319300
## [22,] 23  4.486e-01 0.0290900
## [23,] 23  4.614e-01 0.0265100
## [24,] 25  4.742e-01 0.0241500
## [25,] 27  4.869e-01 0.0220100
## [26,] 29  4.997e-01 0.0200500
## [27,] 31  5.128e-01 0.0182700
## [28,] 31  5.252e-01 0.0166500
## [29,] 31  5.367e-01 0.0151700
## [30,] 32  5.475e-01 0.0138200
## [31,] 34  5.582e-01 0.0125900
## [32,] 36  5.687e-01 0.0114800
## [33,] 38  5.785e-01 0.0104600
## [34,] 40  5.881e-01 0.0095270
## [35,] 41  5.971e-01 0.0086800
## [36,] 42  6.054e-01 0.0079090
## [37,] 42  6.130e-01 0.0072070
## [38,] 42  6.199e-01 0.0065660
## [39,] 42  6.264e-01 0.0059830
## [40,] 45  6.326e-01 0.0054520
## [41,] 48  6.386e-01 0.0049670
## [42,] 49  6.444e-01 0.0045260
## [43,] 50  6.498e-01 0.0041240
## [44,] 51  6.550e-01 0.0037580
## [45,] 51  6.600e-01 0.0034240
## [46,] 51  6.645e-01 0.0031200
## [47,] 52  6.684e-01 0.0028420
## [48,] 53  6.722e-01 0.0025900
## [49,] 53  6.759e-01 0.0023600
## [50,] 54  6.792e-01 0.0021500
## [51,] 55  6.823e-01 0.0019590
## [52,] 56  6.852e-01 0.0017850
## [53,] 57  6.879e-01 0.0016270
## [54,] 57  6.902e-01 0.0014820
## [55,] 59  6.924e-01 0.0013500
## [56,] 59  6.945e-01 0.0012300
## [57,] 59  6.964e-01 0.0011210
## [58,] 59  6.981e-01 0.0010220
## [59,] 59  6.996e-01 0.0009308
## [60,] 59  7.009e-01 0.0008481
## [61,] 59  7.020e-01 0.0007727
## [62,] 59  7.031e-01 0.0007041
## [63,] 58  7.040e-01 0.0006415
## [64,] 58  7.047e-01 0.0005846
## [65,] 58  7.054e-01 0.0005326
## [66,] 59  7.060e-01 0.0004853
## [67,] 59  7.065e-01 0.0004422
## [68,] 59  7.070e-01 0.0004029
## [69,] 59  7.074e-01 0.0003671
## [70,] 60  7.077e-01 0.0003345
## [71,] 60  7.080e-01 0.0003048
## [72,] 61  7.082e-01 0.0002777
## [73,] 61  7.084e-01 0.0002530
## [74,] 61  7.086e-01 0.0002306
## [75,] 61  7.088e-01 0.0002101
## [76,] 61  7.089e-01 0.0001914
## [77,] 61  7.090e-01 0.0001744
## [78,] 61  7.091e-01 0.0001589
## [79,] 61  7.092e-01 0.0001448
## [80,] 61  7.092e-01 0.0001319
## [1] "lambda min is at location 11"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 62 genes..."
## [1] "evaluation accuracy ElasticNet 0.84375"
## [1] "evaluation accuracy LDA 0.816964285714286"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 27.2727272727273"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 70.0534759358289"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 84.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 55"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 80.4511278195489"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 66.1654135338346"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 80"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 67.5"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 63 genes for training model"
## [1] "use 63 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev    Lambda
##   [1,]  0 -2.563e-15 2.690e-01
##   [2,]  1  3.556e-02 2.451e-01
##   [3,]  1  6.561e-02 2.233e-01
##   [4,]  1  9.135e-02 2.035e-01
##   [5,]  1  1.136e-01 1.854e-01
##   [6,]  2  1.386e-01 1.689e-01
##   [7,]  3  1.700e-01 1.539e-01
##   [8,]  4  2.030e-01 1.403e-01
##   [9,]  5  2.350e-01 1.278e-01
##  [10,]  6  2.667e-01 1.165e-01
##  [11,]  6  2.950e-01 1.061e-01
##  [12,]  6  3.201e-01 9.668e-02
##  [13,]  6  3.424e-01 8.809e-02
##  [14,]  6  3.624e-01 8.026e-02
##  [15,]  8  3.805e-01 7.313e-02
##  [16,]  9  3.984e-01 6.664e-02
##  [17,] 10  4.150e-01 6.072e-02
##  [18,] 11  4.308e-01 5.532e-02
##  [19,] 13  4.474e-01 5.041e-02
##  [20,] 14  4.636e-01 4.593e-02
##  [21,] 16  4.793e-01 4.185e-02
##  [22,] 18  4.948e-01 3.813e-02
##  [23,] 18  5.090e-01 3.474e-02
##  [24,] 23  5.238e-01 3.166e-02
##  [25,] 24  5.401e-01 2.885e-02
##  [26,] 25  5.554e-01 2.628e-02
##  [27,] 27  5.699e-01 2.395e-02
##  [28,] 28  5.837e-01 2.182e-02
##  [29,] 28  5.970e-01 1.988e-02
##  [30,] 30  6.093e-01 1.812e-02
##  [31,] 32  6.226e-01 1.651e-02
##  [32,] 34  6.350e-01 1.504e-02
##  [33,] 36  6.477e-01 1.370e-02
##  [34,] 38  6.611e-01 1.249e-02
##  [35,] 39  6.737e-01 1.138e-02
##  [36,] 39  6.857e-01 1.037e-02
##  [37,] 40  6.967e-01 9.446e-03
##  [38,] 41  7.079e-01 8.606e-03
##  [39,] 44  7.189e-01 7.842e-03
##  [40,] 44  7.294e-01 7.145e-03
##  [41,] 44  7.391e-01 6.510e-03
##  [42,] 44  7.482e-01 5.932e-03
##  [43,] 44  7.566e-01 5.405e-03
##  [44,] 44  7.645e-01 4.925e-03
##  [45,] 45  7.719e-01 4.487e-03
##  [46,] 46  7.791e-01 4.089e-03
##  [47,] 47  7.858e-01 3.726e-03
##  [48,] 48  7.922e-01 3.395e-03
##  [49,] 48  7.981e-01 3.093e-03
##  [50,] 50  8.039e-01 2.818e-03
##  [51,] 51  8.098e-01 2.568e-03
##  [52,] 51  8.156e-01 2.340e-03
##  [53,] 51  8.213e-01 2.132e-03
##  [54,] 53  8.271e-01 1.943e-03
##  [55,] 54  8.330e-01 1.770e-03
##  [56,] 53  8.389e-01 1.613e-03
##  [57,] 52  8.443e-01 1.469e-03
##  [58,] 52  8.495e-01 1.339e-03
##  [59,] 52  8.545e-01 1.220e-03
##  [60,] 53  8.595e-01 1.112e-03
##  [61,] 55  8.646e-01 1.013e-03
##  [62,] 57  8.707e-01 9.228e-04
##  [63,] 57  8.773e-01 8.409e-04
##  [64,] 55  8.840e-01 7.662e-04
##  [65,] 56  8.908e-01 6.981e-04
##  [66,] 59  8.987e-01 6.361e-04
##  [67,] 59  9.072e-01 5.796e-04
##  [68,] 59  9.156e-01 5.281e-04
##  [69,] 59  9.234e-01 4.812e-04
##  [70,] 59  9.305e-01 4.384e-04
##  [71,] 59  9.368e-01 3.995e-04
##  [72,] 59  9.426e-01 3.640e-04
##  [73,] 58  9.479e-01 3.317e-04
##  [74,] 58  9.527e-01 3.022e-04
##  [75,] 58  9.570e-01 2.753e-04
##  [76,] 58  9.610e-01 2.509e-04
##  [77,] 57  9.645e-01 2.286e-04
##  [78,] 57  9.677e-01 2.083e-04
##  [79,] 57  9.706e-01 1.898e-04
##  [80,] 57  9.733e-01 1.729e-04
##  [81,] 57  9.757e-01 1.576e-04
##  [82,] 57  9.779e-01 1.436e-04
##  [83,] 57  9.799e-01 1.308e-04
##  [84,] 57  9.817e-01 1.192e-04
##  [85,] 57  9.833e-01 1.086e-04
##  [86,] 57  9.848e-01 9.895e-05
##  [87,] 57  9.862e-01 9.016e-05
##  [88,] 57  9.874e-01 8.215e-05
##  [89,] 57  9.885e-01 7.485e-05
##  [90,] 58  9.895e-01 6.821e-05
##  [91,] 58  9.904e-01 6.215e-05
##  [92,] 58  9.913e-01 5.662e-05
##  [93,] 58  9.920e-01 5.159e-05
##  [94,] 58  9.927e-01 4.701e-05
##  [95,] 58  9.934e-01 4.283e-05
##  [96,] 58  9.939e-01 3.903e-05
##  [97,] 58  9.945e-01 3.556e-05
##  [98,] 58  9.949e-01 3.240e-05
##  [99,] 58  9.954e-01 2.952e-05
## [100,] 58  9.958e-01 2.690e-05
## [1] "lambda min is at location 30"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 62 genes..."
## [1] "evaluation accuracy ElasticNet 0.816964285714286"
## [1] "evaluation accuracy LDA 0.816964285714286"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 16.0427807486631"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 85.0267379679144"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 72.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 83.5714285714286"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 16.5413533834586"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 81.203007518797"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 45"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 85"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1,
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), 
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 203 cells as source subpop"
## [1] "Total 387 cells in remaining subpops"
## [1] "subsampling 102 cells for training source subpop"
## [1] "subsampling 102 cells in remaining subpops for training"
## [1] "use 60 genes for training model"
## [1] "use 60 genes 204 cells for testing model"
## [1] "rename remaining subpops to 1_3"
## [1] "there are 102 cells in class 1_3 and 102 cells in class 2"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 1.281e-15 2.476e-01
##   [2,]  2 3.337e-02 2.256e-01
##   [3,]  2 7.124e-02 2.056e-01
##   [4,]  2 1.037e-01 1.873e-01
##   [5,]  3 1.321e-01 1.707e-01
##   [6,]  4 1.623e-01 1.555e-01
##   [7,]  4 1.902e-01 1.417e-01
##   [8,]  4 2.148e-01 1.291e-01
##   [9,]  4 2.365e-01 1.176e-01
##  [10,]  6 2.571e-01 1.072e-01
##  [11,]  8 2.787e-01 9.766e-02
##  [12,]  9 3.015e-01 8.898e-02
##  [13,] 10 3.251e-01 8.108e-02
##  [14,] 11 3.472e-01 7.387e-02
##  [15,] 13 3.700e-01 6.731e-02
##  [16,] 15 3.925e-01 6.133e-02
##  [17,] 17 4.138e-01 5.588e-02
##  [18,] 19 4.342e-01 5.092e-02
##  [19,] 19 4.531e-01 4.640e-02
##  [20,] 19 4.701e-01 4.227e-02
##  [21,] 20 4.856e-01 3.852e-02
##  [22,] 20 4.997e-01 3.510e-02
##  [23,] 21 5.129e-01 3.198e-02
##  [24,] 22 5.252e-01 2.914e-02
##  [25,] 23 5.369e-01 2.655e-02
##  [26,] 25 5.483e-01 2.419e-02
##  [27,] 25 5.592e-01 2.204e-02
##  [28,] 26 5.692e-01 2.008e-02
##  [29,] 27 5.791e-01 1.830e-02
##  [30,] 28 5.887e-01 1.667e-02
##  [31,] 29 5.977e-01 1.519e-02
##  [32,] 31 6.062e-01 1.384e-02
##  [33,] 31 6.153e-01 1.261e-02
##  [34,] 33 6.243e-01 1.149e-02
##  [35,] 36 6.332e-01 1.047e-02
##  [36,] 38 6.420e-01 9.541e-03
##  [37,] 39 6.508e-01 8.694e-03
##  [38,] 40 6.592e-01 7.921e-03
##  [39,] 41 6.670e-01 7.218e-03
##  [40,] 41 6.745e-01 6.576e-03
##  [41,] 43 6.818e-01 5.992e-03
##  [42,] 43 6.889e-01 5.460e-03
##  [43,] 43 6.955e-01 4.975e-03
##  [44,] 45 7.020e-01 4.533e-03
##  [45,] 48 7.086e-01 4.130e-03
##  [46,] 48 7.158e-01 3.763e-03
##  [47,] 49 7.231e-01 3.429e-03
##  [48,] 49 7.301e-01 3.124e-03
##  [49,] 50 7.370e-01 2.847e-03
##  [50,] 50 7.437e-01 2.594e-03
##  [51,] 51 7.502e-01 2.363e-03
##  [52,] 52 7.571e-01 2.153e-03
##  [53,] 53 7.643e-01 1.962e-03
##  [54,] 53 7.723e-01 1.788e-03
##  [55,] 56 7.811e-01 1.629e-03
##  [56,] 56 7.900e-01 1.484e-03
##  [57,] 56 7.992e-01 1.352e-03
##  [58,] 55 8.082e-01 1.232e-03
##  [59,] 55 8.176e-01 1.123e-03
##  [60,] 54 8.268e-01 1.023e-03
##  [61,] 53 8.367e-01 9.322e-04
##  [62,] 53 8.456e-01 8.494e-04
##  [63,] 53 8.550e-01 7.739e-04
##  [64,] 55 8.655e-01 7.052e-04
##  [65,] 56 8.766e-01 6.425e-04
##  [66,] 56 8.870e-01 5.854e-04
##  [67,] 56 8.966e-01 5.334e-04
##  [68,] 56 9.054e-01 4.860e-04
##  [69,] 56 9.140e-01 4.429e-04
##  [70,] 57 9.213e-01 4.035e-04
##  [71,] 57 9.282e-01 3.677e-04
##  [72,] 57 9.347e-01 3.350e-04
##  [73,] 57 9.405e-01 3.053e-04
##  [74,] 57 9.459e-01 2.781e-04
##  [75,] 57 9.508e-01 2.534e-04
##  [76,] 57 9.552e-01 2.309e-04
##  [77,] 57 9.593e-01 2.104e-04
##  [78,] 57 9.629e-01 1.917e-04
##  [79,] 57 9.663e-01 1.747e-04
##  [80,] 57 9.693e-01 1.592e-04
##  [81,] 57 9.721e-01 1.450e-04
##  [82,] 57 9.746e-01 1.321e-04
##  [83,] 57 9.768e-01 1.204e-04
##  [84,] 57 9.789e-01 1.097e-04
##  [85,] 57 9.808e-01 9.996e-05
##  [86,] 57 9.825e-01 9.108e-05
##  [87,] 57 9.840e-01 8.299e-05
##  [88,] 57 9.854e-01 7.561e-05
##  [89,] 57 9.867e-01 6.890e-05
##  [90,] 57 9.879e-01 6.278e-05
##  [91,] 57 9.889e-01 5.720e-05
##  [92,] 57 9.899e-01 5.212e-05
##  [93,] 57 9.908e-01 4.749e-05
##  [94,] 57 9.916e-01 4.327e-05
##  [95,] 57 9.923e-01 3.942e-05
##  [96,] 57 9.930e-01 3.592e-05
##  [97,] 57 9.936e-01 3.273e-05
##  [98,] 57 9.941e-01 2.982e-05
##  [99,] 57 9.946e-01 2.717e-05
## [100,] 57 9.951e-01 2.476e-05
## [1] "lambda min is at location 19"
## [1] "the leave-out cells in the source subpop is 101"
## [1] "use 102 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 203 cells and 59 genes..."
## [1] "evaluation accuracy ElasticNet 0.847290640394089"
## [1] "evaluation accuracy LDA 0.807881773399015"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 6.41711229946524"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 5.3475935828877"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 17.8571428571429"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 37.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 21.8045112781955"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 10.5263157894737"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 12.5"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 15"
## [1] "Total 203 cells as source subpop"
## [1] "Total 387 cells in remaining subpops"
## [1] "subsampling 102 cells for training source subpop"
## [1] "subsampling 102 cells in remaining subpops for training"
## [1] "use 60 genes for training model"
## [1] "use 60 genes 204 cells for testing model"
## [1] "rename remaining subpops to 1_3"
## [1] "there are 102 cells in class 1_3 and 102 cells in class 2"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev    Lambda
##  [1,]  0 1.281e-15 2.493e-01
##  [2,]  1 3.056e-02 2.272e-01
##  [3,]  1 5.639e-02 2.070e-01
##  [4,]  2 8.408e-02 1.886e-01
##  [5,]  2 1.101e-01 1.719e-01
##  [6,]  2 1.327e-01 1.566e-01
##  [7,]  4 1.599e-01 1.427e-01
##  [8,]  4 1.866e-01 1.300e-01
##  [9,]  6 2.115e-01 1.185e-01
## [10,]  6 2.346e-01 1.079e-01
## [11,]  6 2.550e-01 9.835e-02
## [12,]  6 2.731e-01 8.961e-02
## [13,]  7 2.903e-01 8.165e-02
## [14,]  9 3.106e-01 7.440e-02
## [15,] 10 3.293e-01 6.779e-02
## [16,] 10 3.472e-01 6.176e-02
## [17,] 11 3.636e-01 5.628e-02
## [18,] 13 3.792e-01 5.128e-02
## [19,] 14 3.942e-01 4.672e-02
## [20,] 17 4.083e-01 4.257e-02
## [21,] 17 4.222e-01 3.879e-02
## [22,] 18 4.348e-01 3.534e-02
## [23,] 19 4.467e-01 3.220e-02
## [24,] 20 4.577e-01 2.934e-02
## [25,] 22 4.693e-01 2.674e-02
## [26,] 23 4.800e-01 2.436e-02
## [27,] 25 4.902e-01 2.220e-02
## [28,] 26 4.998e-01 2.022e-02
## [29,] 26 5.087e-01 1.843e-02
## [30,] 28 5.186e-01 1.679e-02
## [31,] 30 5.296e-01 1.530e-02
## [32,] 31 5.403e-01 1.394e-02
## [33,] 33 5.508e-01 1.270e-02
## [34,] 36 5.614e-01 1.157e-02
## [35,] 38 5.724e-01 1.055e-02
## [36,] 38 5.831e-01 9.608e-03
## [37,] 39 5.930e-01 8.755e-03
## [38,] 40 6.023e-01 7.977e-03
## [39,] 40 6.109e-01 7.268e-03
## [40,] 40 6.188e-01 6.623e-03
## [41,] 42 6.264e-01 6.034e-03
## [42,] 42 6.334e-01 5.498e-03
## [43,] 42 6.399e-01 5.010e-03
## [44,] 43 6.462e-01 4.565e-03
## [45,] 45 6.523e-01 4.159e-03
## [46,] 47 6.587e-01 3.790e-03
## [47,] 47 6.648e-01 3.453e-03
## [48,] 48 6.706e-01 3.146e-03
## [49,] 48 6.761e-01 2.867e-03
## [50,] 48 6.811e-01 2.612e-03
## [51,] 48 6.860e-01 2.380e-03
## [52,] 48 6.904e-01 2.169e-03
## [53,] 48 6.945e-01 1.976e-03
## [54,] 49 6.986e-01 1.800e-03
## [55,] 50 7.029e-01 1.641e-03
## [56,] 50 7.063e-01 1.495e-03
## [57,] 52 7.095e-01 1.362e-03
## [58,] 55 7.126e-01 1.241e-03
## [59,] 56 7.168e-01 1.131e-03
## [60,] 55 7.204e-01 1.030e-03
## [61,] 55 7.235e-01 9.388e-04
## [62,] 56 7.264e-01 8.554e-04
## [63,] 57 7.294e-01 7.794e-04
## [64,] 57 7.324e-01 7.101e-04
## [65,] 57 7.353e-01 6.470e-04
## [66,] 57 7.381e-01 5.896e-04
## [67,] 57 7.408e-01 5.372e-04
## [68,] 58 7.434e-01 4.895e-04
## [69,] 58 7.459e-01 4.460e-04
## [70,] 57 7.483e-01 4.064e-04
## [71,] 58 7.507e-01 3.703e-04
## [72,] 57 7.528e-01 3.374e-04
## [73,] 57 7.548e-01 3.074e-04
## [74,] 57 7.569e-01 2.801e-04
## [75,] 57 7.588e-01 2.552e-04
## [76,] 58 7.605e-01 2.325e-04
## [77,] 56 7.622e-01 2.119e-04
## [78,] 58 7.636e-01 1.931e-04
## [79,] 57 7.649e-01 1.759e-04
## [80,] 57 7.663e-01 1.603e-04
## [81,] 58 7.675e-01 1.460e-04
## [82,] 58 7.686e-01 1.331e-04
## [83,] 58 7.697e-01 1.212e-04
## [84,] 58 7.706e-01 1.105e-04
## [85,] 58 7.715e-01 1.007e-04
## [86,] 58 7.723e-01 9.172e-05
## [87,] 59 7.730e-01 8.357e-05
## [88,] 59 7.736e-01 7.615e-05
## [89,] 59 7.742e-01 6.938e-05
## [90,] 58 7.747e-01 6.322e-05
## [91,] 59 7.751e-01 5.760e-05
## [92,] 59 7.755e-01 5.248e-05
## [93,] 59 7.759e-01 4.782e-05
## [94,] 59 7.759e-01 4.357e-05
## [1] "lambda min is at location 18"
## [1] "the leave-out cells in the source subpop is 101"
## [1] "use 102 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 203 cells and 59 genes..."
## [1] "evaluation accuracy ElasticNet 0.87192118226601"
## [1] "evaluation accuracy LDA 0.783251231527094"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 31.0160427807487"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 16.0427807486631"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 23.5714285714286"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 52.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 61.6541353383459"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 19.5488721804511"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 52.5"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 47.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
    <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 163 cells as source subpop"
## [1] "Total 427 cells in remaining subpops"
## [1] "subsampling 82 cells for training source subpop"
## [1] "subsampling 82 cells in remaining subpops for training"
## [1] "use 65 genes for training model"
## [1] "use 65 genes 164 cells for testing model"
## [1] "rename remaining subpops to 2_1"
## [1] "there are 82 cells in class 2_1 and 82 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 2.403e-15 1.729e-01
##   [2,]  1 1.465e-02 1.575e-01
##   [3,]  3 3.212e-02 1.435e-01
##   [4,]  3 5.555e-02 1.308e-01
##   [5,]  4 7.633e-02 1.191e-01
##   [6,]  4 9.545e-02 1.086e-01
##   [7,]  5 1.131e-01 9.892e-02
##   [8,]  5 1.313e-01 9.013e-02
##   [9,]  5 1.470e-01 8.212e-02
##  [10,]  9 1.641e-01 7.483e-02
##  [11,]  9 1.858e-01 6.818e-02
##  [12,]  9 2.047e-01 6.212e-02
##  [13,] 10 2.225e-01 5.660e-02
##  [14,] 11 2.402e-01 5.157e-02
##  [15,] 15 2.589e-01 4.699e-02
##  [16,] 19 2.813e-01 4.282e-02
##  [17,] 19 3.042e-01 3.901e-02
##  [18,] 19 3.249e-01 3.555e-02
##  [19,] 23 3.460e-01 3.239e-02
##  [20,] 26 3.667e-01 2.951e-02
##  [21,] 26 3.867e-01 2.689e-02
##  [22,] 28 4.060e-01 2.450e-02
##  [23,] 30 4.248e-01 2.233e-02
##  [24,] 31 4.430e-01 2.034e-02
##  [25,] 33 4.609e-01 1.854e-02
##  [26,] 34 4.778e-01 1.689e-02
##  [27,] 37 4.934e-01 1.539e-02
##  [28,] 38 5.081e-01 1.402e-02
##  [29,] 39 5.216e-01 1.278e-02
##  [30,] 40 5.344e-01 1.164e-02
##  [31,] 41 5.465e-01 1.061e-02
##  [32,] 41 5.580e-01 9.664e-03
##  [33,] 42 5.686e-01 8.806e-03
##  [34,] 43 5.789e-01 8.023e-03
##  [35,] 44 5.884e-01 7.311e-03
##  [36,] 44 5.972e-01 6.661e-03
##  [37,] 42 6.053e-01 6.069e-03
##  [38,] 43 6.131e-01 5.530e-03
##  [39,] 46 6.208e-01 5.039e-03
##  [40,] 50 6.298e-01 4.591e-03
##  [41,] 52 6.387e-01 4.183e-03
##  [42,] 54 6.479e-01 3.812e-03
##  [43,] 52 6.568e-01 3.473e-03
##  [44,] 53 6.654e-01 3.165e-03
##  [45,] 53 6.736e-01 2.883e-03
##  [46,] 56 6.816e-01 2.627e-03
##  [47,] 57 6.901e-01 2.394e-03
##  [48,] 57 6.987e-01 2.181e-03
##  [49,] 58 7.073e-01 1.987e-03
##  [50,] 59 7.157e-01 1.811e-03
##  [51,] 59 7.237e-01 1.650e-03
##  [52,] 59 7.316e-01 1.503e-03
##  [53,] 60 7.396e-01 1.370e-03
##  [54,] 61 7.478e-01 1.248e-03
##  [55,] 61 7.568e-01 1.137e-03
##  [56,] 60 7.652e-01 1.036e-03
##  [57,] 60 7.753e-01 9.442e-04
##  [58,] 60 7.847e-01 8.603e-04
##  [59,] 63 7.982e-01 7.839e-04
##  [60,] 61 8.101e-01 7.143e-04
##  [61,] 62 8.214e-01 6.508e-04
##  [62,] 61 8.342e-01 5.930e-04
##  [63,] 61 8.469e-01 5.403e-04
##  [64,] 61 8.590e-01 4.923e-04
##  [65,] 62 8.706e-01 4.486e-04
##  [66,] 62 8.819e-01 4.087e-04
##  [67,] 63 8.927e-01 3.724e-04
##  [68,] 63 9.026e-01 3.393e-04
##  [69,] 63 9.116e-01 3.092e-04
##  [70,] 63 9.198e-01 2.817e-04
##  [71,] 60 9.269e-01 2.567e-04
##  [72,] 61 9.335e-01 2.339e-04
##  [73,] 61 9.398e-01 2.131e-04
##  [74,] 61 9.452e-01 1.942e-04
##  [75,] 61 9.503e-01 1.769e-04
##  [76,] 61 9.549e-01 1.612e-04
##  [77,] 62 9.590e-01 1.469e-04
##  [78,] 62 9.628e-01 1.338e-04
##  [79,] 62 9.662e-01 1.219e-04
##  [80,] 62 9.692e-01 1.111e-04
##  [81,] 62 9.720e-01 1.012e-04
##  [82,] 62 9.745e-01 9.225e-05
##  [83,] 62 9.767e-01 8.405e-05
##  [84,] 62 9.788e-01 7.659e-05
##  [85,] 62 9.806e-01 6.978e-05
##  [86,] 62 9.823e-01 6.358e-05
##  [87,] 63 9.838e-01 5.794e-05
##  [88,] 63 9.853e-01 5.279e-05
##  [89,] 64 9.865e-01 4.810e-05
##  [90,] 64 9.876e-01 4.383e-05
##  [91,] 64 9.887e-01 3.993e-05
##  [92,] 64 9.897e-01 3.639e-05
##  [93,] 63 9.906e-01 3.315e-05
##  [94,] 63 9.914e-01 3.021e-05
##  [95,] 63 9.921e-01 2.752e-05
##  [96,] 64 9.927e-01 2.508e-05
##  [97,] 64 9.933e-01 2.285e-05
##  [98,] 64 9.939e-01 2.082e-05
##  [99,] 64 9.944e-01 1.897e-05
## [100,] 64 9.949e-01 1.729e-05
## [1] "lambda min is at location 30"
## [1] "the leave-out cells in the source subpop is 81"
## [1] "use 82 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 163 cells and 65 genes..."
## [1] "evaluation accuracy ElasticNet 0.680981595092024"
## [1] "evaluation accuracy LDA 0.631901840490798"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 27.2727272727273"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 73.2620320855615"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0.714285714285714"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 3.57142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 6.01503759398496"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 57.1428571428571"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 5"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 35"
## [1] "Total 163 cells as source subpop"
## [1] "Total 427 cells in remaining subpops"
## [1] "subsampling 82 cells for training source subpop"
## [1] "subsampling 82 cells in remaining subpops for training"
## [1] "use 65 genes for training model"
## [1] "use 65 genes 164 cells for testing model"
## [1] "rename remaining subpops to 2_1"
## [1] "there are 82 cells in class 2_1 and 82 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 2.403e-15 1.486e-01
##   [2,]  1 1.084e-02 1.354e-01
##   [3,]  2 2.407e-02 1.234e-01
##   [4,]  4 3.978e-02 1.124e-01
##   [5,]  4 5.790e-02 1.024e-01
##   [6,]  7 7.651e-02 9.332e-02
##   [7,]  7 9.336e-02 8.503e-02
##   [8,]  8 1.084e-01 7.748e-02
##   [9,] 12 1.258e-01 7.060e-02
##  [10,] 14 1.515e-01 6.433e-02
##  [11,] 17 1.769e-01 5.861e-02
##  [12,] 19 2.061e-01 5.340e-02
##  [13,] 22 2.337e-01 4.866e-02
##  [14,] 23 2.591e-01 4.434e-02
##  [15,] 24 2.819e-01 4.040e-02
##  [16,] 25 3.030e-01 3.681e-02
##  [17,] 26 3.223e-01 3.354e-02
##  [18,] 27 3.418e-01 3.056e-02
##  [19,] 28 3.603e-01 2.784e-02
##  [20,] 30 3.776e-01 2.537e-02
##  [21,] 31 3.939e-01 2.312e-02
##  [22,] 32 4.094e-01 2.106e-02
##  [23,] 35 4.259e-01 1.919e-02
##  [24,] 39 4.416e-01 1.749e-02
##  [25,] 41 4.572e-01 1.593e-02
##  [26,] 43 4.722e-01 1.452e-02
##  [27,] 44 4.871e-01 1.323e-02
##  [28,] 46 5.012e-01 1.205e-02
##  [29,] 47 5.145e-01 1.098e-02
##  [30,] 48 5.273e-01 1.001e-02
##  [31,] 49 5.391e-01 9.118e-03
##  [32,] 49 5.504e-01 8.308e-03
##  [33,] 49 5.609e-01 7.570e-03
##  [34,] 51 5.706e-01 6.897e-03
##  [35,] 52 5.801e-01 6.285e-03
##  [36,] 54 5.896e-01 5.726e-03
##  [37,] 55 5.994e-01 5.218e-03
##  [38,] 57 6.088e-01 4.754e-03
##  [39,] 54 6.181e-01 4.332e-03
##  [40,] 55 6.269e-01 3.947e-03
##  [41,] 56 6.355e-01 3.596e-03
##  [42,] 57 6.436e-01 3.277e-03
##  [43,] 55 6.514e-01 2.986e-03
##  [44,] 55 6.588e-01 2.720e-03
##  [45,] 56 6.659e-01 2.479e-03
##  [46,] 56 6.728e-01 2.259e-03
##  [47,] 57 6.800e-01 2.058e-03
##  [48,] 59 6.873e-01 1.875e-03
##  [49,] 59 6.945e-01 1.709e-03
##  [50,] 59 7.018e-01 1.557e-03
##  [51,] 60 7.087e-01 1.418e-03
##  [52,] 60 7.158e-01 1.292e-03
##  [53,] 60 7.236e-01 1.178e-03
##  [54,] 60 7.311e-01 1.073e-03
##  [55,] 60 7.391e-01 9.777e-04
##  [56,] 60 7.470e-01 8.908e-04
##  [57,] 60 7.552e-01 8.117e-04
##  [58,] 61 7.635e-01 7.396e-04
##  [59,] 62 7.733e-01 6.739e-04
##  [60,] 62 7.861e-01 6.140e-04
##  [61,] 61 7.984e-01 5.595e-04
##  [62,] 63 8.124e-01 5.098e-04
##  [63,] 63 8.272e-01 4.645e-04
##  [64,] 63 8.410e-01 4.232e-04
##  [65,] 63 8.542e-01 3.856e-04
##  [66,] 63 8.665e-01 3.514e-04
##  [67,] 64 8.781e-01 3.201e-04
##  [68,] 63 8.887e-01 2.917e-04
##  [69,] 63 8.984e-01 2.658e-04
##  [70,] 64 9.078e-01 2.422e-04
##  [71,] 64 9.161e-01 2.207e-04
##  [72,] 62 9.236e-01 2.011e-04
##  [73,] 62 9.305e-01 1.832e-04
##  [74,] 63 9.368e-01 1.669e-04
##  [75,] 63 9.424e-01 1.521e-04
##  [76,] 63 9.476e-01 1.386e-04
##  [77,] 63 9.523e-01 1.263e-04
##  [78,] 63 9.565e-01 1.151e-04
##  [79,] 62 9.603e-01 1.048e-04
##  [80,] 63 9.639e-01 9.552e-05
##  [81,] 63 9.671e-01 8.704e-05
##  [82,] 63 9.700e-01 7.930e-05
##  [83,] 63 9.726e-01 7.226e-05
##  [84,] 63 9.750e-01 6.584e-05
##  [85,] 63 9.772e-01 5.999e-05
##  [86,] 63 9.792e-01 5.466e-05
##  [87,] 63 9.810e-01 4.980e-05
##  [88,] 63 9.826e-01 4.538e-05
##  [89,] 63 9.841e-01 4.135e-05
##  [90,] 63 9.854e-01 3.768e-05
##  [91,] 64 9.866e-01 3.433e-05
##  [92,] 63 9.877e-01 3.128e-05
##  [93,] 64 9.888e-01 2.850e-05
##  [94,] 62 9.896e-01 2.597e-05
##  [95,] 63 9.906e-01 2.366e-05
##  [96,] 62 9.913e-01 2.156e-05
##  [97,] 63 9.921e-01 1.964e-05
##  [98,] 62 9.927e-01 1.790e-05
##  [99,] 62 9.932e-01 1.631e-05
## [100,] 62 9.938e-01 1.486e-05
## [1] "lambda min is at location 16"
## [1] "the leave-out cells in the source subpop is 81"
## [1] "use 82 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 163 cells and 65 genes..."
## [1] "evaluation accuracy ElasticNet 0.680981595092024"
## [1] "evaluation accuracy LDA 0.674846625766871"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 35.2941176470588"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 83.4224598930481"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 19.2857142857143"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 5.71428571428571"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 48.1203007518797"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 63.9097744360902"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 25"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 40"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#prepare table input for sankey plot </span>

LASSO_C1S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#7570b3"</span>)

LASSO_C2S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#1b9e77"</span>)

LASSO_C3S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,
    <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#e7298a"</span>)


combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S1,LASSO_C2S1,LASSO_C3S1)

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)],
    <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))]) 
combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]


<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))
n &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(node_df<span class="op">$</span>Node))
getPalette =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/colorRamp">colorRampPalette</a></span>(RColorBrewer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">9</span>, <span class="st">"Set1"</span>))
Color =<span class="st"> </span><span class="kw">getPalette</span>(n)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))
p1&lt;-<span class="kw">sankeyNetwork</span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,
    <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>,
    <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, <span class="dt">fontSize =</span> <span class="dv">22</span>)
p1</code></pre></div>
<div id="htmlwidget-2d568155671e0d1e03cd" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-2d568155671e0d1e03cd">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2],"target":[3,4,5,6,3,4,5,6,3,4,5,6],"value":[21.6577540106952,78.2142857142857,48.4962406015038,62.5,18.716577540107,20.7142857142858,41.7293233082707,32.5,31.283422459893,10,27.0676691729323,15],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a"]},"nodes":{"name":["C1_MP1","C2_MP1","C3_MP1","C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#E41A1C","#3E8E93","#7E6E85","#FF7F00","#E1C62F","#DB728C","#999999"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></code></pre></div>
</div>
<div id="annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with" class="anchor"></a>4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with</h2>
</div>
<div id="a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="anchor"></a>a reference dataset to see which cluster is most similar to the reference</h2>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation-instruction">1. Installation instruction</a></li>
      <li>
<a href="#a-simple-workflow-of-the-scgps">2. A simple workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#create-scgps-objects">2.1 Create scGPS objects</a></li>
      <li><a href="#run-prediction">2.2 Run prediction</a></li>
      <li><a href="#summarise-results">2.3 Summarise results</a></li>
      </ul>
</li>
      <li>
<a href="#a-complete-workflow-of-the-scgps">3. A complete workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#identify-clusters-in-a-dataset-using-core">3.1 Identify clusters in a dataset using CORE</a></li>
      <li><a href="#identify-clusters-in-a-dataset-using-score">3.2 Identify clusters in a dataset using SCORE</a></li>
      <li><a href="#stable-clustering-at-optimal-resolution">(Stable Clustering at Optimal REsolution)</a></li>
      <li><a href="#visualise-all-cluster-results-in-all-iterations">3.3 Visualise all cluster results in all iterations</a></li>
      <li><a href="#compare-clustering-results-with-other-dimensional-reduction-methods">3.4 Compare clustering results with other dimensional reduction methods</a></li>
      <li><a href="#e-g--cidr">(e.g., CIDR)</a></li>
      <li><a href="#find-gene-markers-and-annotate-clusters">3.5 Find gene markers and annotate clusters</a></li>
      </ul>
</li>
      <li>
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples">4. Relationship between clusters within one sample or between two samples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#start-the-scgps-prediction-to-find-relationship-between-clusters">4.1 Start the scGPS prediction to find relationship between clusters</a></li>
      <li><a href="#display-summary-results-for-the-prediction">4.2 Display summary results for the prediction</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-one-sample">4.3 Plot the relationship between clusters in one sample</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-two-samples">4.3 Plot the relationship between clusters in two samples</a></li>
      <li><a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with">4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with</a></li>
      <li><a href="#a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference">a reference dataset to see which cluster is most similar to the reference</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
