<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGPS introduction â€¢ scGPS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scGPS introduction">
<meta property="og:description" content="scGPS is a single cell RNA analysis framework to decompose a mixed population into clusters (SCORE), followed by analysing the relationship between clusters (scGPS)">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="vignette_files/htmlwidgets-1.3/htmlwidgets.js"></script><script src="vignette_files/d3-4.5.0/d3.min.js"></script><script src="vignette_files/sankey-1/sankey.js"></script><script src="vignette_files/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scGPS introduction</h1>
                        <h4 class="author">Quan Nguyen and Michael Thompson</h4>
            
            <h4 class="date">2019-02-24</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># To install scGPS from github (Depending on the configuration of the local</span>
<span class="co"># computer or HPC, possible custom C++ compilation may be required - see</span>
<span class="co"># installation trouble-shootings below)</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"IMB-Computational-Genomics-Lab/scGPS"</span>)

<span class="co"># for C++ compilation trouble-shooting, manual download and installation can be</span>
<span class="co"># done from github</span>

git clone https<span class="op">:</span><span class="er">//</span>github.com<span class="op">/</span>IMB<span class="op">-</span>Computational<span class="op">-</span>Genomics<span class="op">-</span>Lab<span class="op">/</span>scGPS

<span class="co"># then check in scGPS/src if any of the precompiled (e.g.  those with *.so and</span>
<span class="co"># *.o) files exist and delete them before recompiling</span>

<span class="co"># create a Makevars file in the scGPS/src with one line: PKG_LIBS =</span>
<span class="co"># $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)</span>

<span class="co"># then with the scGPS as the R working directory, manually recompile scGPS in R</span>
<span class="co"># using devtools to load and install functions</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/document">document</a></span>()
<span class="co">#load the package to the workspace </span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>()</code></pre></div>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given a mixed population with known subpopulations, estimate transition scores between these subpopulation</em></p>
<div id="create-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#create-scgps-objects" class="anchor"></a>2.1 Create scGPS objects</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load mixed population 1 (loaded from sample1 dataset, named it as day2)</span>

devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">"/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS"</span>) 

<span class="co">#devtools::load_all("C:/Users/s5125601/Desktop/new2/scGPS") </span>

day2 &lt;-<span class="st"> </span>sample1
mixedpop1 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day2<span class="op">$</span>dat2_counts, <span class="dt">GeneMetadata =</span> day2<span class="op">$</span>dat2geneInfo, 
    <span class="dt">CellMetadata =</span> day2<span class="op">$</span>dat2_clusters)

<span class="co"># load mixed population 2 (loaded from sample2 dataset, named it as day5)</span>
day5 &lt;-<span class="st"> </span>sample2
mixedpop2 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo,
    <span class="dt">CellMetadata =</span> day5<span class="op">$</span>dat5_clusters)</code></pre></div>
</div>
<div id="run-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#run-prediction" class="anchor"></a>2.2 Run prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># select a subpopulation</span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co"># load gene list (this can be any lists of user selected genes)</span>
genes &lt;-<span class="st"> </span>GeneList
genes &lt;-<span class="st"> </span>genes<span class="op">$</span>Merged_unique
<span class="co"># load cluster information </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#run training (running nboots = 3 here, but recommend to use nboots = 50-100)</span>
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">3</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID, <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
     <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2,
     <span class="dt">trainset_ratio =</span> <span class="fl">0.7</span>)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev   Lambda
##  [1,]  0 1.281e-15 0.128500
##  [2,]  1 8.088e-03 0.117100
##  [3,]  1 1.481e-02 0.106700
##  [4,]  1 2.041e-02 0.097180
##  [5,]  1 2.507e-02 0.088550
##  [6,]  1 2.896e-02 0.080680
##  [7,]  1 3.220e-02 0.073520
##  [8,]  2 3.646e-02 0.066980
##  [9,]  2 4.083e-02 0.061030
## [10,]  2 4.452e-02 0.055610
## [11,]  2 4.765e-02 0.050670
## [12,]  3 5.125e-02 0.046170
## [13,]  3 5.465e-02 0.042070
## [14,]  3 5.756e-02 0.038330
## [15,]  3 6.007e-02 0.034930
## [16,]  3 6.222e-02 0.031820
## [17,]  3 6.408e-02 0.029000
## [18,]  3 6.568e-02 0.026420
## [19,]  3 6.706e-02 0.024070
## [20,]  3 6.825e-02 0.021930
## [21,]  3 6.926e-02 0.019990
## [22,]  3 7.014e-02 0.018210
## [23,]  3 7.089e-02 0.016590
## [24,]  3 7.153e-02 0.015120
## [25,]  3 7.208e-02 0.013780
## [26,]  3 7.255e-02 0.012550
## [27,]  3 7.295e-02 0.011440
## [28,]  3 7.329e-02 0.010420
## [29,]  3 7.358e-02 0.009495
## [30,]  3 7.382e-02 0.008651
## [31,]  3 7.403e-02 0.007883
## [32,]  3 7.420e-02 0.007183
## [33,]  4 7.436e-02 0.006544
## [34,]  4 7.451e-02 0.005963
## [35,]  4 7.463e-02 0.005433
## [36,]  4 7.473e-02 0.004951
## [37,]  4 7.482e-02 0.004511
## [38,]  4 7.489e-02 0.004110
## [39,]  4 7.495e-02 0.003745
## [40,]  4 7.500e-02 0.003412
## [41,]  4 7.504e-02 0.003109
## [42,]  4 7.508e-02 0.002833
## [43,]  4 7.511e-02 0.002581
## [44,]  4 7.514e-02 0.002352
## [45,]  4 7.516e-02 0.002143
## [46,]  4 7.517e-02 0.001953
## [47,]  4 7.519e-02 0.001779
## [48,]  4 7.520e-02 0.001621
## [49,]  4 7.521e-02 0.001477
## [50,]  4 7.522e-02 0.001346
## [1] "lambda min is at location 21"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 212 cells and 5 genes..."
## [1] "evaluation accuracy ElasticNet 0.660377358490566"
## [1] "evaluation accuracy LDA 0.665094339622642"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 17.1122994652406"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 15.5080213903743"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 32.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 10.7142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 89.4736842105263"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 27.0676691729323"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 72.5"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 20"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev   Lambda
##  [1,]  0 3.044e-15 0.156400
##  [2,]  1 1.200e-02 0.142500
##  [3,]  1 2.198e-02 0.129900
##  [4,]  1 3.030e-02 0.118300
##  [5,]  1 3.724e-02 0.107800
##  [6,]  2 4.563e-02 0.098250
##  [7,]  2 5.421e-02 0.089520
##  [8,]  2 6.152e-02 0.081570
##  [9,]  2 6.778e-02 0.074320
## [10,]  2 7.317e-02 0.067720
## [11,]  2 7.781e-02 0.061700
## [12,]  2 8.183e-02 0.056220
## [13,]  2 8.530e-02 0.051230
## [14,]  2 8.831e-02 0.046680
## [15,]  3 9.136e-02 0.042530
## [16,]  3 9.465e-02 0.038750
## [17,]  3 9.751e-02 0.035310
## [18,]  3 1.000e-01 0.032170
## [19,]  3 1.022e-01 0.029310
## [20,]  3 1.040e-01 0.026710
## [21,]  4 1.057e-01 0.024340
## [22,]  4 1.072e-01 0.022180
## [23,]  4 1.085e-01 0.020210
## [24,]  4 1.096e-01 0.018410
## [25,]  4 1.105e-01 0.016770
## [26,]  4 1.113e-01 0.015280
## [27,]  4 1.120e-01 0.013930
## [28,]  4 1.127e-01 0.012690
## [29,]  4 1.132e-01 0.011560
## [30,]  4 1.136e-01 0.010540
## [31,]  4 1.140e-01 0.009599
## [32,]  4 1.143e-01 0.008746
## [33,]  4 1.146e-01 0.007969
## [34,]  4 1.148e-01 0.007261
## [35,]  4 1.150e-01 0.006616
## [36,]  4 1.152e-01 0.006029
## [37,]  4 1.154e-01 0.005493
## [38,]  4 1.155e-01 0.005005
## [39,]  4 1.156e-01 0.004560
## [40,]  4 1.157e-01 0.004155
## [41,]  4 1.157e-01 0.003786
## [42,]  4 1.158e-01 0.003450
## [43,]  4 1.159e-01 0.003143
## [44,]  4 1.159e-01 0.002864
## [45,]  4 1.159e-01 0.002610
## [46,]  4 1.160e-01 0.002378
## [47,]  4 1.160e-01 0.002167
## [48,]  4 1.160e-01 0.001974
## [49,]  4 1.160e-01 0.001799
## [50,]  4 1.160e-01 0.001639
## [51,]  4 1.161e-01 0.001493
## [52,]  4 1.161e-01 0.001361
## [53,]  4 1.161e-01 0.001240
## [1] "lambda min is at location 12"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 215 cells and 5 genes..."
## [1] "evaluation accuracy ElasticNet 0.604651162790698"
## [1] "evaluation accuracy LDA 0.632558139534884"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 4.27807486631016"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 187 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 8.02139037433155"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 22.8571428571429"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 140 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 22.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 7.5187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 133 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 17.2932330827068"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 17.5"
## [1] "Number of genes in the target data, but not in model genes is 4995"
## [1] "Number of genes in the model present in the target data is 5"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 5 genes and 40 cells. The trained model has 5 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 10"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 157 cells for training source subpop"
## [1] "subsampling 157 cells in remaining subpops for training"
## [1] "use 6 genes for training model"
## [1] "use 6 genes 314 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 157 cells in class 2_3 and 157 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev   Lambda
##  [1,]  0 -8.009e-16 0.119900
##  [2,]  2  1.224e-02 0.109200
##  [3,]  2  2.271e-02 0.099500
##  [4,]  2  3.160e-02 0.090660
##  [5,]  2  3.919e-02 0.082610
##  [6,]  2  4.573e-02 0.075270
##  [7,]  2  5.139e-02 0.068580
##  [8,]  2  5.631e-02 0.062490
##  [9,]  2  6.061e-02 0.056940
## [10,]  2  6.436e-02 0.051880
## [11,]  2  6.766e-02 0.047270
## [12,]  2  7.055e-02 0.043070
## [13,]  2  7.310e-02 0.039250
## [14,]  2  7.534e-02 0.035760
## [15,]  2  7.730e-02 0.032580
## [16,]  2  7.903e-02 0.029690
## [17,]  2  8.054e-02 0.027050
## [18,]  3  8.219e-02 0.024650
## [19,]  3  8.366e-02 0.022460
## [20,]  4  8.496e-02 0.020460
## [21,]  4  8.612e-02 0.018650
## [22,]  4  8.712e-02 0.016990
## [23,]  4  8.798e-02 0.015480
## [24,]  4  8.873e-02 0.014100
## [25,]  4  8.937e-02 0.012850
## [26,]  4  8.992e-02 0.011710
## [27,]  4  9.040e-02 0.010670
## [28,]  4  9.081e-02 0.009722
## [29,]  5  9.117e-02 0.008858
## [30,]  5  9.151e-02 0.008071
## [31,]  5  9.180e-02 0.007354
## [32,]  5  9.204e-02 0.006701
## [33,]  5  9.224e-02 0.006105
## [34,]  5  9.241e-02 0.005563
## [35,]  5  9.255e-02 0.005069
## [36,]  5  9.267e-02 0.004619
## [37,]  5  9.278e-02 0.004208
## [38,]  5  9.286e-02 0.003834
## [39,]  5  9.293e-02 0.003494
## [40,]  5  9.299e-02 0.003183
## [41,]  5  9.304e-02 0.002901
## [42,]  5  9.308e-02 0.002643
## [43,]  5  9.312e-02 0.002408
## [44,]  5  9.315e-02 0.002194
## [45,]  5  9.317e-02 0.001999
## [46,]  5  9.319e-02 0.001822
## [47,]  5  9.321e-02 0.001660
## [48,]  5  9.322e-02 0.001512
## [49,]  5  9.323e-02 0.001378
## [50,]  5  9.324e-02 0.001256
## [1] "lambda min is at location 15"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 157 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 214 cells and 6 genes..."
## [1] "evaluation accuracy ElasticNet 0.607476635514019"
## [1] "evaluation accuracy LDA 0.635514018691589"
## [1] "done training for bootstrap 3, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 2.67379679144385"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 187 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 20.855614973262"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 4.28571428571429"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 140 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 3.57142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 0.75187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 133 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 34.5864661654135"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 12.5"
## [1] "Number of genes in the target data, but not in model genes is 4994"
## [1] "Number of genes in the model present in the target data is 6"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 6 genes and 40 cells. The trained model has 6 genes"
## [1] "first 10 genes in model "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "first 10 genes in target "
## [1] "GATA4"  "VIM"    "SNAI2"  "GJA1"   "MAP2"   "TMEM88"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 20"</code></pre>
</div>
<div id="summarise-results" class="section level2">
<h2 class="hasAnchor">
<a href="#summarise-results" class="anchor"></a>2.3 Summarise results</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># display the list of result information in the LASOLDA_dat object </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(LSOLDA_dat)</code></pre></div>
<pre><code>## [1] "Accuracy"          "ElasticNetGenes"   "Deviance"         
## [4] "ElasticNetFit"     "LDAFit"            "predictor_S1"     
## [7] "ElasticNetPredict" "LDAPredict"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>ElasticNetPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[1]][[2]]
## [1] 17.1123
## 
## [[1]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[1]][[4]]
## [1] 32.14286
## 
## [[1]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[1]][[6]]
## [1] 89.47368
## 
## [[1]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 72.5
## 
## 
## [[2]]
## [[2]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[2]][[2]]
## [1] 4.278075
## 
## [[2]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[2]][[4]]
## [1] 22.85714
## 
## [[2]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 7.518797
## 
## [[2]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 17.5
## 
## 
## [[3]]
## [[3]][[1]]
## [1] "ElasticNet for subpop1 in target mixedpop2"
## 
## [[3]][[2]]
## [1] 2.673797
## 
## [[3]][[3]]
## [1] "ElasticNet for subpop2 in target mixedpop2"
## 
## [[3]][[4]]
## [1] 4.285714
## 
## [[3]][[5]]
## [1] "ElasticNet for subpop3 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 0.7518797
## 
## [[3]][[7]]
## [1] "ElasticNet for subpop4 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 12.5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LSOLDA_dat<span class="op">$</span>LDAPredict</code></pre></div>
<pre><code>## [[1]]
## [[1]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[1]][[2]]
## [1] 15.50802
## 
## [[1]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[1]][[4]]
## [1] 10.71429
## 
## [[1]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[1]][[6]]
## [1] 27.06767
## 
## [[1]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[1]][[8]]
## [1] 20
## 
## 
## [[2]]
## [[2]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[2]][[2]]
## [1] 8.02139
## 
## [[2]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[2]][[4]]
## [1] 22.85714
## 
## [[2]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[2]][[6]]
## [1] 17.29323
## 
## [[2]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[2]][[8]]
## [1] 10
## 
## 
## [[3]]
## [[3]][[1]]
## [1] "LDA for subpop 1 in target mixedpop2"
## 
## [[3]][[2]]
## [1] 20.85561
## 
## [[3]][[3]]
## [1] "LDA for subpop 2 in target mixedpop2"
## 
## [[3]][[4]]
## [1] 3.571429
## 
## [[3]][[5]]
## [1] "LDA for subpop 3 in target mixedpop2"
## 
## [[3]][[6]]
## [1] 34.58647
## 
## [[3]][[7]]
## [1] "LDA for subpop 4 in target mixedpop2"
## 
## [[3]][[8]]
## [1] 20</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary results LDA</span>
sum_pred_lda &lt;-<span class="st"> </span><span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># summary results Lasso to show the percent of cells classified as cells belonging </span>
sum_pred_lasso &lt;-<span class="st"> </span><span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co"># plot summary results </span>
plot_sum &lt;-<span class="cf">function</span>(sum_dat){
  sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(sum_dat)
  sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/na.fail">na.omit</a></span>(sum_dat_tf)
  sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(sum_dat[, <span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(sum_dat)],<span class="dv">1</span>, <span class="cf">function</span>(x){<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(x))})
  sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"for subpop "</span>,<span class="st">"sp"</span>,  sum_dat<span class="op">$</span>names )
  sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"in target mixedpop"</span>,<span class="st">"in p"</span>,  sum_dat<span class="op">$</span>names) 
  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sum_dat_tf) &lt;-<span class="st"> </span>sum_dat<span class="op">$</span>names
  <span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(sum_dat_tf, <span class="dt">las=</span><span class="dv">2</span>)
}
<span class="kw">plot_sum</span>(sum_pred_lasso)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_sum</span>(sum_pred_lda)</code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## [1] 66.50943 63.25581 63.55140</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary maximum deviance explained by the model </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## $allDeviance
## [1] "0.06926" "0.08183" "0.0773" 
## 
## $DeviMax
##          dat_DE$Dfd          Deviance           DEgenes
## 1                 0           0.08183    genes_cluster1
## 2                 1           0.08183    genes_cluster1
## 3                 2           0.08183    genes_cluster1
## 4 remaining DEgenes remaining DEgenes remaining DEgenes
## 
## $LassoGenesMax
## NULL</code></pre>
</div>
</div>
<div id="a-complete-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-complete-workflow-of-the-scgps" class="anchor"></a>3. A complete workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given an unknown mixed population, find clusters and estimate relationship between clusters</em></p>
<div id="identify-clusters-in-a-dataset-using-core" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-core" class="anchor"></a>3.1 Identify clusters in a dataset using CORE</h2>
<p><em>(skip this step if clusters are known)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find clustering information in an expresion data using CORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames ) 

CORE_cluster &lt;-<span class="st"> </span><span class="kw">CORE_scGPS</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>)</code></pre></div>
<pre><code>## [1] "Performing 1 round of filtering"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "No more outliers detected in filtering round 1"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "writing clustering result for run 1"
## [1] "writing clustering result for run 2"
## [1] "writing clustering result for run 3"
## [1] "writing clustering result for run 4"
## [1] "writing clustering result for run 5"
## [1] "writing clustering result for run 6"
## [1] "writing clustering result for run 7"
## [1] "writing clustering result for run 8"
## [1] "writing clustering result for run 9"
## [1] "writing clustering result for run 10"
## [1] "writing clustering result for run 11"
## [1] "writing clustering result for run 12"
## [1] "writing clustering result for run 13"
## [1] "writing clustering result for run 14"
## [1] "writing clustering result for run 15"
## [1] "writing clustering result for run 16"
## [1] "writing clustering result for run 17"
## [1] "writing clustering result for run 18"
## [1] "writing clustering result for run 19"
## [1] "writing clustering result for run 20"
## [1] "writing clustering result for run 21"
## [1] "writing clustering result for run 22"
## [1] "writing clustering result for run 23"
## [1] "writing clustering result for run 24"
## [1] "writing clustering result for run 25"
## [1] "writing clustering result for run 26"
## [1] "writing clustering result for run 27"
## [1] "writing clustering result for run 28"
## [1] "writing clustering result for run 29"
## [1] "writing clustering result for run 30"
## [1] "writing clustering result for run 31"
## [1] "writing clustering result for run 32"
## [1] "writing clustering result for run 33"
## [1] "writing clustering result for run 34"
## [1] "writing clustering result for run 35"
## [1] "writing clustering result for run 36"
## [1] "writing clustering result for run 37"
## [1] "writing clustering result for run 38"
## [1] "writing clustering result for run 39"
## [1] "writing clustering result for run 40"
## [1] "Done clustering, moving to stability calculation..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done finding optimal clustering..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># to update the clustering information, users can ...</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)

clustering_after_outlier_removal &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unname">unname</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(CORE_cluster<span class="op">$</span>Cluster[[optimal_index]]))
corresponding_cells_after_outlier_removal &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>cellsForClustering
original_cells_before_removal &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">2</span>]
corresponding_index &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(corresponding_cells_after_outlier_removal, original_cells_before_removal )
<span class="co"># check the matching </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/identical">identical</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/character">as.character</a></span>(original_cells_before_removal[corresponding_index]),corresponding_cells_after_outlier_removal)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># create new object with the new clustering after removing outliers </span>
mixedpop2_post_clustering &lt;-<span class="st"> </span>mixedpop2[,corresponding_index]
<span class="kw">colData</span>(mixedpop2_post_clustering)[,<span class="dv">1</span>] &lt;-<span class="st"> </span>clustering_after_outlier_removal</code></pre></div>
</div>
<div id="identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution" class="anchor"></a>3.2 Identify clusters in a dataset using SCORE (Stable Clustering at Optimal REsolution)</h2>
<p><em>(skip this step if clusters are known)</em> <em>(SCORE aims to get stable subpopulation results, by introducing bagging aggregation and bootstrapping to the CORE algorithm)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find clustering information in an expresion data using SCORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames ) 

SCORE_test &lt;-<span class="st"> </span><span class="kw">CORE_scGPS_bagging</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>,
                          <span class="dt">bagging_run =</span> <span class="dv">20</span>, <span class="dt">subsample_proportion =</span> .<span class="dv">8</span>)</code></pre></div>
<pre><code>## [1] "Performing 1 round of filtering"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "No more outliers detected in filtering round 1"
## [1] "Identifying top variable genes"
## [1] "Calculating distance matrix"
## [1] "Performing hierarchical clustering"
## [1] "Finding clustering information"
## [1] "500 cells left after filtering"
## [1] "Running 20 bagging runs, with 0.8 subsampling..."
## [1] "Done clustering, moving to stability calculation..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."
## [1] "Done calculating stability..."
## [1] "Start finding optimal clustering..."</code></pre>
</div>
<div id="visualise-all-cluster-results-in-all-iterations" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-all-cluster-results-in-all-iterations" class="anchor"></a>3.3 Visualise all cluster results in all iterations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/dev">dev.off</a></span>()</code></pre></div>
<pre><code>## null device 
##           1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##3.2.1 plot CORE clustering 
p1 &lt;-<span class="st"> </span><span class="kw">plot_CORE</span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster, <span class="dt">color_branch =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#208eb7"</span>, <span class="st">"#6ce9d3"</span>, <span class="st">"#1c5e39"</span>, <span class="st">"#8fca40"</span>, <span class="st">"#154975"</span>, <span class="st">"#b1c8eb"</span>))
p1 </code></pre></div>
<pre><code>## $mar
## [1] 1 5 0 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#extract optimal index identified by CORE_scGPS</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)
<span class="co">#plot one optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> CORE_cluster<span class="op">$</span>tree,
                   <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(CORE_cluster<span class="op">$</span>Cluster[optimal_index]), <span class="dt">shift =</span> <span class="op">-</span><span class="dv">2000</span>)</code></pre></div>
<pre><code>## [1] "Ordering and assigning labels..."
## [1] 2
## [1]  24 112  NA  NA  NA  NA  NA  NA
## [1] 3
## [1]  24 112 224  NA  NA  NA  NA  NA
## [1] 4
## [1]  24 112 224 299  NA  NA  NA  NA
## [1] 5
## [1]  24 112 224 299 335  NA  NA  NA
## [1] 6
## [1]  24 112 224 299 335 367  NA  NA
## [1] 7
## [1]  24 112 224 299 335 367 414  NA
## [1] 8
## [1]  24 112 224 299 335 367 414 470
## [1] "Plotting the colored dendrogram now...."</code></pre>
<pre><code>## [1] "Plotting the bar underneath now...."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##3.2.2 plot SCORE clustering 
<span class="kw">plot_CORE</span>(SCORE_test<span class="op">$</span>tree, <span class="dt">list_clusters =</span> SCORE_test<span class="op">$</span>Cluster)<span class="co">#plot all clustering bars </span>
<span class="co">#plot one stable optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> SCORE_test<span class="op">$</span>tree,
                   <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(SCORE_test<span class="op">$</span>Cluster[SCORE_test<span class="op">$</span>optimal_index]), <span class="dt">shift =</span> <span class="op">-</span><span class="dv">100</span>)</code></pre></div>
<pre><code>## [1] "Ordering and assigning labels..."
## [1] 2
## [1] 162 335  NA
## [1] 3
## [1] 162 335 423
## [1] "Plotting the colored dendrogram now...."</code></pre>
<pre><code>## [1] "Plotting the bar underneath now...."</code></pre>
</div>
<div id="compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="anchor"></a>3.4 Compare clustering results with other dimensional reduction methods (e.g., CIDR)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(cidr)
t &lt;-<span class="st"> </span><span class="kw">CIDR_scGPS</span>(<span class="dt">expression.matrix=</span><span class="kw">assay</span>(mixedpop2))</code></pre></div>
<pre><code>## [1] "building cidr object..."
## [1] "determine dropout candidates..."
## [1] "determine the imputation weighting threshold..."
## [1] "computes the _CIDR_ dissimilarity matrix..."
## [1] "PCA plot with proportion of variance explained..."</code></pre>
<p><img src="vignette_files/figure-html/compare_clustering-1.png" width="700"></p>
<pre><code>## [1] "find the number of PC..."
## [1] "perform clustering..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="kw">plotReduced_scGPS</span>(t, <span class="dt">color_fac =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]),<span class="dt">palletes =</span><span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])))
p2</code></pre></div>
<p><img src="vignette_files/figure-html/compare_clustering-2.png" width="700"></p>
</div>
<div id="find-gene-markers-and-annotate-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#find-gene-markers-and-annotate-clusters" class="anchor"></a>3.5 Find gene markers and annotate clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#load gene list (this can be any lists of user-selected genes)</span>
genes &lt;-GeneList
genes &lt;-genes<span class="op">$</span>Merged_unique

<span class="co">#the gene list can also be objectively identified by differential expression analysis</span>
<span class="co">#cluster information is requied for findMarkers_scGPS. Here, we use CORE results. </span>

<span class="co">#colData(mixedpop2)[,1] &lt;- unlist(SCORE_test$Cluster[SCORE_test$optimal_index])</span>

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(locfit))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DESeq))

DEgenes &lt;-<span class="st"> </span><span class="kw">findMarkers_scGPS</span>(<span class="dt">expression_matrix=</span><span class="kw">assay</span>(mixedpop2), <span class="dt">cluster =</span> <span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>],
                             <span class="dt">selected_cluster=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])) </code></pre></div>
<pre><code>## [1] "Start estimate dispersions for cluster 1..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 1..."
## [1] "Done nbinom test for cluster 1 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 2..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 2..."
## [1] "Done nbinom test for cluster 2 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 3..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 3..."
## [1] "Done nbinom test for cluster 3 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."
## [1] "Start estimate dispersions for cluster 4..."
## [1] "Done estimate dispersions. Start nbinom test for cluster 4..."
## [1] "Done nbinom test for cluster 4 ..."
## [1] "Adjust foldchange by subtracting basemean to 1..."</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#the output contains dataframes for each cluster.</span>
<span class="co">#the data frame contains all genes, sorted by p-values </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(DEgenes)</code></pre></div>
<pre><code>## [1] "DE_Subpop1vsRemaining" "DE_Subpop2vsRemaining" "DE_Subpop3vsRemaining"
## [4] "DE_Subpop4vsRemaining"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#you can annotate the identified clusters </span>
DEgeneList_1vsOthers &lt;-<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id

<span class="co">#users need to check the format of the gene input to make sure they are consistent to </span>
<span class="co">#the gene names in the expression matrix </span>

<span class="co">#the following command saves the file "PathwayEnrichment.xlsx" to the working dir</span>
<span class="co">#use 500 top DE genes </span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DOSE))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ReactomePA))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(clusterProfiler))
enrichment_test &lt;-<span class="st"> </span><span class="kw">annotate_scGPS</span>(DEgeneList_1vsOthers[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>], <span class="dt">pvalueCutoff=</span><span class="fl">0.05</span>, <span class="dt">gene_symbol=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] "Original gene number in geneList"
## [1] 500
## [1] "Number of genes successfully converted"
## [1] 486</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#the enrichment outputs can be displayed by running</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/ReactomePA/topics/reexports">dotplot</a></span>(enrichment_test, <span class="dt">showCategory=</span><span class="dv">10</span>, <span class="dt">font.size =</span> <span class="dv">6</span>)</code></pre></div>
<p><img src="vignette_files/figure-html/find_markers-1.png" width="700"></p>
</div>
</div>
<div id="relationship-between-clusters-within-one-sample-or-between-two-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples" class="anchor"></a>4. Relationship between clusters within one sample or between two samples</h1>
<p><em>The purpose of this workflow is to solve the following task: given one or two unknown mixed population(s) and clusters in each mixed population, estimate and visualise relationship between clusters</em></p>
<div id="start-the-scgps-prediction-to-find-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#start-the-scgps-prediction-to-find-relationship-between-clusters" class="anchor"></a>4.1 Start the scGPS prediction to find relationship between clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#select a subpopulation, and input gene list </span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#note make sure the format for genes input here is the same to the format for genes in the mixedpop1 and mixedpop2  </span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>]

<span class="co">#run the test bootstrap with nboots = 2 runs</span>

cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sink">sink</a></span>(<span class="st">"temp"</span>)
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID, 
     <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),
     <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 153 genes for training model"
## [1] "use 153 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 2 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 2.548e-01
##  [2,]  1  3.189e-02 2.321e-01
##  [3,]  1  5.879e-02 2.115e-01
##  [4,]  1  8.176e-02 1.927e-01
##  [5,]  2  1.066e-01 1.756e-01
##  [6,]  3  1.341e-01 1.600e-01
##  [7,]  3  1.644e-01 1.458e-01
##  [8,]  3  1.908e-01 1.328e-01
##  [9,]  5  2.184e-01 1.210e-01
## [10,]  5  2.467e-01 1.103e-01
## [11,]  8  2.747e-01 1.005e-01
## [12,]  9  3.018e-01 9.156e-02
## [13,]  9  3.268e-01 8.342e-02
## [14,] 13  3.521e-01 7.601e-02
## [15,] 16  3.794e-01 6.926e-02
## [16,] 18  4.076e-01 6.311e-02
## [17,] 20  4.358e-01 5.750e-02
## [18,] 24  4.628e-01 5.239e-02
## [19,] 25  4.883e-01 4.774e-02
## [20,] 30  5.130e-01 4.350e-02
## [21,] 30  5.374e-01 3.963e-02
## [22,] 32  5.613e-01 3.611e-02
## [23,] 37  5.855e-01 3.290e-02
## [24,] 43  6.096e-01 2.998e-02
## [25,] 44  6.328e-01 2.732e-02
## [26,] 44  6.544e-01 2.489e-02
## [27,] 44  6.745e-01 2.268e-02
## [28,] 48  6.936e-01 2.066e-02
## [29,] 50  7.121e-01 1.883e-02
## [30,] 50  7.295e-01 1.716e-02
## [31,] 52  7.465e-01 1.563e-02
## [32,] 53  7.630e-01 1.424e-02
## [33,] 58  7.792e-01 1.298e-02
## [34,] 61  7.953e-01 1.182e-02
## [35,] 65  8.112e-01 1.077e-02
## [36,] 65  8.265e-01 9.817e-03
## [37,] 66  8.406e-01 8.945e-03
## [38,] 67  8.537e-01 8.150e-03
## [39,] 69  8.659e-01 7.426e-03
## [40,] 70  8.774e-01 6.767e-03
## [41,] 70  8.880e-01 6.166e-03
## [42,] 69  8.977e-01 5.618e-03
## [43,] 70  9.066e-01 5.119e-03
## [44,] 70  9.146e-01 4.664e-03
## [45,] 72  9.220e-01 4.250e-03
## [46,] 72  9.289e-01 3.872e-03
## [47,] 73  9.352e-01 3.528e-03
## [48,] 74  9.409e-01 3.215e-03
## [49,] 74  9.462e-01 2.929e-03
## [50,] 74  9.509e-01 2.669e-03
## [51,] 74  9.553e-01 2.432e-03
## [52,] 75  9.593e-01 2.216e-03
## [53,] 73  9.629e-01 2.019e-03
## [54,] 75  9.661e-01 1.840e-03
## [55,] 76  9.691e-01 1.676e-03
## [56,] 77  9.719e-01 1.527e-03
## [57,] 77  9.744e-01 1.392e-03
## [58,] 78  9.766e-01 1.268e-03
## [59,] 79  9.787e-01 1.155e-03
## [60,] 79  9.806e-01 1.053e-03
## [61,] 79  9.823e-01 9.592e-04
## [62,] 79  9.839e-01 8.739e-04
## [63,] 80  9.853e-01 7.963e-04
## [64,] 81  9.866e-01 7.256e-04
## [65,] 81  9.878e-01 6.611e-04
## [66,] 82  9.889e-01 6.024e-04
## [67,] 83  9.898e-01 5.489e-04
## [68,] 82  9.907e-01 5.001e-04
## [69,] 81  9.916e-01 4.557e-04
## [70,] 81  9.923e-01 4.152e-04
## [71,] 81  9.930e-01 3.783e-04
## [72,] 80  9.936e-01 3.447e-04
## [73,] 81  9.942e-01 3.141e-04
## [74,] 81  9.947e-01 2.862e-04
## [75,] 81  9.952e-01 2.608e-04
## [76,] 80  9.956e-01 2.376e-04
## [77,] 81  9.960e-01 2.165e-04
## [78,] 81  9.963e-01 1.973e-04
## [79,] 81  9.967e-01 1.797e-04
## [80,] 80  9.969e-01 1.638e-04
## [81,] 80  9.972e-01 1.492e-04
## [82,] 80  9.975e-01 1.360e-04
## [83,] 80  9.977e-01 1.239e-04
## [84,] 80  9.979e-01 1.129e-04
## [85,] 80  9.981e-01 1.028e-04
## [86,] 80  9.982e-01 9.371e-05
## [87,] 80  9.984e-01 8.539e-05
## [88,] 81  9.985e-01 7.780e-05
## [89,] 81  9.987e-01 7.089e-05
## [90,] 81  9.988e-01 6.459e-05
## [91,] 82  9.989e-01 5.885e-05
## [92,] 82  9.990e-01 5.362e-05
## [93,] 82  9.991e-01 4.886e-05
## [1] "lambda min is at location 22"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 151 genes..."
## [1] "evaluation accuracy ElasticNet 0.84375"
## [1] "evaluation accuracy LDA 0.683035714285714"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 187 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 40.6417112299465"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 187 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 73.7967914438503"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 140 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 90"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 140 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 91.4285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 133 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 19.5488721804511"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 133 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 51.1278195488722"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 40 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 50"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 40 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 80"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 153 genes for training model"
## [1] "use 153 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 2 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df       %Dev    Lambda
##  [1,]  0 -2.563e-15 2.112e-01
##  [2,]  1  2.189e-02 1.924e-01
##  [3,]  4  4.822e-02 1.753e-01
##  [4,]  4  9.277e-02 1.598e-01
##  [5,]  4  1.310e-01 1.456e-01
##  [6,]  5  1.661e-01 1.326e-01
##  [7,]  6  1.990e-01 1.208e-01
##  [8,]  8  2.339e-01 1.101e-01
##  [9,]  8  2.653e-01 1.003e-01
## [10,]  9  2.934e-01 9.142e-02
## [11,] 12  3.231e-01 8.330e-02
## [12,] 15  3.556e-01 7.590e-02
## [13,] 17  3.862e-01 6.915e-02
## [14,] 21  4.170e-01 6.301e-02
## [15,] 22  4.459e-01 5.741e-02
## [16,] 25  4.740e-01 5.231e-02
## [17,] 30  5.023e-01 4.767e-02
## [18,] 30  5.296e-01 4.343e-02
## [19,] 30  5.545e-01 3.957e-02
## [20,] 34  5.782e-01 3.606e-02
## [21,] 35  6.009e-01 3.285e-02
## [22,] 35  6.219e-01 2.994e-02
## [23,] 36  6.413e-01 2.728e-02
## [24,] 38  6.595e-01 2.485e-02
## [25,] 42  6.777e-01 2.264e-02
## [26,] 45  6.956e-01 2.063e-02
## [27,] 47  7.132e-01 1.880e-02
## [28,] 48  7.307e-01 1.713e-02
## [29,] 53  7.483e-01 1.561e-02
## [30,] 54  7.649e-01 1.422e-02
## [31,] 61  7.812e-01 1.296e-02
## [32,] 63  7.974e-01 1.181e-02
## [33,] 62  8.130e-01 1.076e-02
## [34,] 64  8.275e-01 9.802e-03
## [35,] 65  8.412e-01 8.932e-03
## [36,] 68  8.545e-01 8.138e-03
## [37,] 70  8.671e-01 7.415e-03
## [38,] 71  8.786e-01 6.756e-03
## [39,] 72  8.893e-01 6.156e-03
## [40,] 73  8.991e-01 5.609e-03
## [41,] 72  9.080e-01 5.111e-03
## [42,] 74  9.161e-01 4.657e-03
## [43,] 75  9.236e-01 4.243e-03
## [44,] 75  9.303e-01 3.866e-03
## [45,] 76  9.366e-01 3.523e-03
## [46,] 77  9.422e-01 3.210e-03
## [47,] 78  9.474e-01 2.925e-03
## [48,] 78  9.520e-01 2.665e-03
## [49,] 78  9.563e-01 2.428e-03
## [50,] 79  9.602e-01 2.212e-03
## [51,] 78  9.637e-01 2.016e-03
## [52,] 78  9.670e-01 1.837e-03
## [53,] 79  9.699e-01 1.674e-03
## [54,] 79  9.726e-01 1.525e-03
## [55,] 79  9.750e-01 1.389e-03
## [56,] 79  9.773e-01 1.266e-03
## [57,] 79  9.793e-01 1.154e-03
## [58,] 79  9.811e-01 1.051e-03
## [59,] 79  9.828e-01 9.577e-04
## [60,] 79  9.843e-01 8.726e-04
## [61,] 79  9.857e-01 7.951e-04
## [62,] 79  9.870e-01 7.245e-04
## [63,] 79  9.881e-01 6.601e-04
## [64,] 79  9.892e-01 6.015e-04
## [65,] 80  9.901e-01 5.480e-04
## [66,] 80  9.910e-01 4.993e-04
## [67,] 80  9.918e-01 4.550e-04
## [68,] 80  9.925e-01 4.146e-04
## [69,] 80  9.932e-01 3.777e-04
## [70,] 79  9.938e-01 3.442e-04
## [71,] 79  9.943e-01 3.136e-04
## [72,] 79  9.948e-01 2.857e-04
## [73,] 79  9.953e-01 2.604e-04
## [74,] 79  9.957e-01 2.372e-04
## [75,] 79  9.961e-01 2.162e-04
## [76,] 79  9.964e-01 1.970e-04
## [77,] 80  9.967e-01 1.795e-04
## [78,] 80  9.970e-01 1.635e-04
## [79,] 80  9.973e-01 1.490e-04
## [80,] 80  9.975e-01 1.358e-04
## [81,] 80  9.978e-01 1.237e-04
## [82,] 80  9.979e-01 1.127e-04
## [83,] 80  9.981e-01 1.027e-04
## [84,] 81  9.983e-01 9.357e-05
## [85,] 81  9.984e-01 8.526e-05
## [86,] 81  9.986e-01 7.768e-05
## [87,] 81  9.987e-01 7.078e-05
## [88,] 81  9.988e-01 6.449e-05
## [89,] 82  9.989e-01 5.876e-05
## [90,] 81  9.990e-01 5.354e-05
## [1] "lambda min is at location 77"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 151 genes..."
## [1] "evaluation accuracy ElasticNet 0.776785714285714"
## [1] "evaluation accuracy LDA 0.65625"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 187 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 43.3155080213904"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 187 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 9.09090909090909"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 140 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 98.5714285714286"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 140 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 45.7142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 133 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 62.406015037594"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 133 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 12.0300751879699"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4849"
## [1] "Number of genes in the model present in the target data is 151"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 151 genes and 40 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 80"
## [1] "Number of genes in the target data, but not in model genes is 4852"
## [1] "Number of genes in the model present in the target data is 148"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are HHIP-AS1"   
## [2] "genes to be added to target subpop are THUMPD3-AS1"
## [3] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 151 genes and 40 cells. The trained model has 151 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "CCT4"    "ACTR10"  "TMEM223" "SH3BGRL" "TMEM98"  "GATA4"  
##  [8] "LETMD1"  "PSMD11"  "SUMF2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 17.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sink">sink</a></span>()</code></pre></div>
</div>
<div id="display-summary-results-for-the-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#display-summary-results-for-the-prediction" class="anchor"></a>4.2 Display summary results for the prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#get the number of rows for the summary matrix </span>
row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

<span class="co">#summary results LDA to to show the percent of cells classified as cells belonging by LDA classifier </span>
<span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster )</code></pre></div>
<pre><code>##                 V1               V2                                names
## 1 73.7967914438503 9.09090909090909 LDA for subpop 1 in target mixedpop2
## 2 91.4285714285714 45.7142857142857 LDA for subpop 2 in target mixedpop2
## 3 51.1278195488722 12.0300751879699 LDA for subpop 3 in target mixedpop2
## 4               80             17.5 LDA for subpop 4 in target mixedpop2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#summary results Lasso to show the percent of cells classified as cells belonging by Lasso classifier</span>
<span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster)</code></pre></div>
<pre><code>##                 V1               V2
## 1 40.6417112299465 43.3155080213904
## 2               90 98.5714285714286
## 3 19.5488721804511  62.406015037594
## 4               50               80
##                                        names
## 1 ElasticNet for subpop1 in target mixedpop2
## 2 ElasticNet for subpop2 in target mixedpop2
## 3 ElasticNet for subpop3 in target mixedpop2
## 4 ElasticNet for subpop4 in target mixedpop2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary maximum deviance explained by the model during the model training </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## $allDeviance
## [1] "0.5613" "0.9967"
## 
## $DeviMax
##           dat_DE$Dfd          Deviance           DEgenes
## 1                  0            0.9967    genes_cluster1
## 2                  1            0.9967    genes_cluster1
## 3                  4            0.9967    genes_cluster1
## 4                  5            0.9967    genes_cluster1
## 5                  6            0.9967    genes_cluster1
## 6                  8            0.9967    genes_cluster1
## 7                  9            0.9967    genes_cluster1
## 8                 12            0.9967    genes_cluster1
## 9                 15            0.9967    genes_cluster1
## 10                17            0.9967    genes_cluster1
## 11                21            0.9967    genes_cluster1
## 12                22            0.9967    genes_cluster1
## 13                25            0.9967    genes_cluster1
## 14                30            0.9967    genes_cluster1
## 15                34            0.9967    genes_cluster1
## 16                35            0.9967    genes_cluster1
## 17                36            0.9967    genes_cluster1
## 18                38            0.9967    genes_cluster1
## 19                42            0.9967    genes_cluster1
## 20                45            0.9967    genes_cluster1
## 21                47            0.9967    genes_cluster1
## 22                48            0.9967    genes_cluster1
## 23                53            0.9967    genes_cluster1
## 24                54            0.9967    genes_cluster1
## 25                61            0.9967    genes_cluster1
## 26                62            0.9967    genes_cluster1
## 27                63            0.9967    genes_cluster1
## 28                64            0.9967    genes_cluster1
## 29                65            0.9967    genes_cluster1
## 30                68            0.9967    genes_cluster1
## 31                70            0.9967    genes_cluster1
## 32                71            0.9967    genes_cluster1
## 33                72            0.9967    genes_cluster1
## 34                73            0.9967    genes_cluster1
## 35                74            0.9967    genes_cluster1
## 36                75            0.9967    genes_cluster1
## 37                76            0.9967    genes_cluster1
## 38                77            0.9967    genes_cluster1
## 39                78            0.9967    genes_cluster1
## 40                79            0.9967    genes_cluster1
## 41                80            0.9967    genes_cluster1
## 42 remaining DEgenes remaining DEgenes remaining DEgenes
## 
## $LassoGenesMax
## NULL</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)</code></pre></div>
<pre><code>## [1] 68.30357 65.62500</code></pre>
</div>
<div id="plot-the-relationship-between-clusters-in-one-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-one-sample" class="anchor"></a>4.3 Plot the relationship between clusters in one sample</h2>
<p><em>Here we look at one example use case to find relationship between clusters within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#cluster_mixedpop2 &lt;- as.numeric(as.vector(colData(mixedpop2)[,1]))</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] <span class="co">#top 200 gene markers distinguishing cluster 1 </span>

LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 187 cells as source subpop"
## [1] "Total 313 cells in remaining subpops"
## [1] "subsampling 94 cells for training source subpop"
## [1] "subsampling 94 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 188 cells for testing model"
## [1] "rename remaining subpops to 2_3_4"
## [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 3.364e-15 0.304500
##   [2,]  1 2.378e-02 0.290600
##   [3,]  1 4.550e-02 0.277400
##   [4,]  1 6.540e-02 0.264800
##   [5,]  1 8.368e-02 0.252800
##   [6,]  1 1.005e-01 0.241300
##   [7,]  1 1.160e-01 0.230300
##   [8,]  1 1.303e-01 0.219900
##   [9,]  1 1.435e-01 0.209900
##  [10,]  1 1.557e-01 0.200300
##  [11,]  1 1.670e-01 0.191200
##  [12,]  1 1.775e-01 0.182500
##  [13,]  2 1.879e-01 0.174200
##  [14,]  3 1.999e-01 0.166300
##  [15,]  4 2.152e-01 0.158800
##  [16,]  4 2.305e-01 0.151500
##  [17,]  4 2.448e-01 0.144700
##  [18,]  6 2.593e-01 0.138100
##  [19,]  8 2.755e-01 0.131800
##  [20,]  8 2.910e-01 0.125800
##  [21,] 10 3.071e-01 0.120100
##  [22,] 10 3.243e-01 0.114600
##  [23,] 13 3.423e-01 0.109400
##  [24,] 13 3.600e-01 0.104500
##  [25,] 14 3.772e-01 0.099700
##  [26,] 15 3.940e-01 0.095170
##  [27,] 15 4.102e-01 0.090850
##  [28,] 16 4.255e-01 0.086720
##  [29,] 16 4.404e-01 0.082770
##  [30,] 17 4.547e-01 0.079010
##  [31,] 18 4.688e-01 0.075420
##  [32,] 18 4.832e-01 0.071990
##  [33,] 19 4.970e-01 0.068720
##  [34,] 19 5.103e-01 0.065600
##  [35,] 20 5.235e-01 0.062620
##  [36,] 20 5.366e-01 0.059770
##  [37,] 21 5.494e-01 0.057050
##  [38,] 22 5.618e-01 0.054460
##  [39,] 23 5.737e-01 0.051990
##  [40,] 24 5.864e-01 0.049620
##  [41,] 25 5.991e-01 0.047370
##  [42,] 25 6.115e-01 0.045210
##  [43,] 26 6.235e-01 0.043160
##  [44,] 27 6.354e-01 0.041200
##  [45,] 29 6.471e-01 0.039320
##  [46,] 29 6.587e-01 0.037540
##  [47,] 30 6.699e-01 0.035830
##  [48,] 32 6.807e-01 0.034200
##  [49,] 32 6.912e-01 0.032650
##  [50,] 36 7.018e-01 0.031160
##  [51,] 37 7.126e-01 0.029750
##  [52,] 38 7.234e-01 0.028400
##  [53,] 40 7.340e-01 0.027110
##  [54,] 41 7.445e-01 0.025870
##  [55,] 42 7.546e-01 0.024700
##  [56,] 45 7.647e-01 0.023570
##  [57,] 46 7.745e-01 0.022500
##  [58,] 47 7.839e-01 0.021480
##  [59,] 48 7.929e-01 0.020500
##  [60,] 51 8.016e-01 0.019570
##  [61,] 52 8.101e-01 0.018680
##  [62,] 53 8.183e-01 0.017830
##  [63,] 54 8.261e-01 0.017020
##  [64,] 54 8.337e-01 0.016250
##  [65,] 54 8.409e-01 0.015510
##  [66,] 55 8.479e-01 0.014810
##  [67,] 55 8.545e-01 0.014130
##  [68,] 57 8.609e-01 0.013490
##  [69,] 57 8.670e-01 0.012880
##  [70,] 58 8.728e-01 0.012290
##  [71,] 60 8.785e-01 0.011730
##  [72,] 60 8.838e-01 0.011200
##  [73,] 61 8.890e-01 0.010690
##  [74,] 62 8.940e-01 0.010200
##  [75,] 62 8.987e-01 0.009741
##  [76,] 62 9.032e-01 0.009298
##  [77,] 62 9.075e-01 0.008876
##  [78,] 62 9.116e-01 0.008472
##  [79,] 63 9.156e-01 0.008087
##  [80,] 63 9.193e-01 0.007720
##  [81,] 63 9.229e-01 0.007369
##  [82,] 64 9.264e-01 0.007034
##  [83,] 64 9.297e-01 0.006714
##  [84,] 64 9.328e-01 0.006409
##  [85,] 66 9.358e-01 0.006118
##  [86,] 67 9.387e-01 0.005840
##  [87,] 67 9.414e-01 0.005574
##  [88,] 67 9.441e-01 0.005321
##  [89,] 67 9.466e-01 0.005079
##  [90,] 68 9.490e-01 0.004848
##  [91,] 69 9.513e-01 0.004628
##  [92,] 70 9.535e-01 0.004417
##  [93,] 70 9.556e-01 0.004217
##  [94,] 70 9.576e-01 0.004025
##  [95,] 70 9.596e-01 0.003842
##  [96,] 70 9.614e-01 0.003667
##  [97,] 70 9.632e-01 0.003501
##  [98,] 70 9.648e-01 0.003342
##  [99,] 70 9.664e-01 0.003190
## [100,] 70 9.680e-01 0.003045
## [1] "lambda min is at location 47"
## [1] "the leave-out cells in the source subpop is 93"
## [1] "use 94 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 187 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.844919786096257"
## [1] "evaluation accuracy LDA 0.689839572192513"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 75.4010695187166"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 43.3155080213904"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 19.2857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 7.5187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 15.7894736842105"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 20"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 20"
## [1] "Total 187 cells as source subpop"
## [1] "Total 313 cells in remaining subpops"
## [1] "subsampling 94 cells for training source subpop"
## [1] "subsampling 94 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 188 cells for testing model"
## [1] "rename remaining subpops to 2_3_4"
## [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 3.364e-15 0.287300
##   [2,]  1 2.116e-02 0.274200
##   [3,]  2 4.381e-02 0.261700
##   [4,]  2 6.554e-02 0.249800
##   [5,]  2 8.550e-02 0.238500
##   [6,]  2 1.039e-01 0.227600
##   [7,]  2 1.208e-01 0.217300
##   [8,]  2 1.364e-01 0.207400
##   [9,]  2 1.509e-01 0.198000
##  [10,]  3 1.655e-01 0.189000
##  [11,]  3 1.833e-01 0.180400
##  [12,]  3 2.000e-01 0.172200
##  [13,]  6 2.191e-01 0.164400
##  [14,]  6 2.392e-01 0.156900
##  [15,]  6 2.581e-01 0.149800
##  [16,]  6 2.758e-01 0.143000
##  [17,]  8 2.930e-01 0.136500
##  [18,]  8 3.098e-01 0.130300
##  [19,]  8 3.258e-01 0.124300
##  [20,]  9 3.412e-01 0.118700
##  [21,] 10 3.575e-01 0.113300
##  [22,] 13 3.749e-01 0.108200
##  [23,] 16 3.949e-01 0.103200
##  [24,] 18 4.150e-01 0.098540
##  [25,] 18 4.343e-01 0.094060
##  [26,] 18 4.526e-01 0.089790
##  [27,] 18 4.700e-01 0.085710
##  [28,] 18 4.865e-01 0.081810
##  [29,] 20 5.023e-01 0.078090
##  [30,] 22 5.180e-01 0.074540
##  [31,] 23 5.333e-01 0.071160
##  [32,] 24 5.480e-01 0.067920
##  [33,] 24 5.621e-01 0.064830
##  [34,] 25 5.755e-01 0.061890
##  [35,] 27 5.884e-01 0.059070
##  [36,] 28 6.011e-01 0.056390
##  [37,] 28 6.133e-01 0.053830
##  [38,] 31 6.252e-01 0.051380
##  [39,] 31 6.370e-01 0.049050
##  [40,] 32 6.484e-01 0.046820
##  [41,] 33 6.594e-01 0.044690
##  [42,] 34 6.699e-01 0.042660
##  [43,] 36 6.800e-01 0.040720
##  [44,] 36 6.899e-01 0.038870
##  [45,] 37 6.993e-01 0.037100
##  [46,] 37 7.083e-01 0.035410
##  [47,] 37 7.170e-01 0.033800
##  [48,] 38 7.254e-01 0.032270
##  [49,] 41 7.340e-01 0.030800
##  [50,] 42 7.426e-01 0.029400
##  [51,] 42 7.509e-01 0.028070
##  [52,] 42 7.589e-01 0.026790
##  [53,] 44 7.667e-01 0.025570
##  [54,] 44 7.744e-01 0.024410
##  [55,] 44 7.817e-01 0.023300
##  [56,] 45 7.894e-01 0.022240
##  [57,] 45 7.968e-01 0.021230
##  [58,] 46 8.040e-01 0.020270
##  [59,] 48 8.110e-01 0.019340
##  [60,] 52 8.179e-01 0.018470
##  [61,] 54 8.249e-01 0.017630
##  [62,] 54 8.319e-01 0.016820
##  [63,] 53 8.386e-01 0.016060
##  [64,] 54 8.450e-01 0.015330
##  [65,] 54 8.513e-01 0.014630
##  [66,] 53 8.574e-01 0.013970
##  [67,] 56 8.633e-01 0.013330
##  [68,] 55 8.690e-01 0.012730
##  [69,] 56 8.746e-01 0.012150
##  [70,] 58 8.799e-01 0.011600
##  [71,] 58 8.851e-01 0.011070
##  [72,] 57 8.900e-01 0.010570
##  [73,] 57 8.947e-01 0.010090
##  [74,] 57 8.992e-01 0.009628
##  [75,] 58 9.035e-01 0.009190
##  [76,] 58 9.077e-01 0.008772
##  [77,] 58 9.117e-01 0.008374
##  [78,] 60 9.156e-01 0.007993
##  [79,] 60 9.193e-01 0.007630
##  [80,] 60 9.228e-01 0.007283
##  [81,] 64 9.262e-01 0.006952
##  [82,] 64 9.295e-01 0.006636
##  [83,] 64 9.327e-01 0.006334
##  [84,] 67 9.357e-01 0.006047
##  [85,] 67 9.386e-01 0.005772
##  [86,] 68 9.413e-01 0.005509
##  [87,] 67 9.440e-01 0.005259
##  [88,] 67 9.465e-01 0.005020
##  [89,] 67 9.489e-01 0.004792
##  [90,] 67 9.512e-01 0.004574
##  [91,] 67 9.534e-01 0.004366
##  [92,] 68 9.555e-01 0.004168
##  [93,] 68 9.575e-01 0.003978
##  [94,] 68 9.594e-01 0.003797
##  [95,] 68 9.613e-01 0.003625
##  [96,] 68 9.630e-01 0.003460
##  [97,] 67 9.647e-01 0.003303
##  [98,] 67 9.663e-01 0.003153
##  [99,] 67 9.678e-01 0.003009
## [100,] 67 9.693e-01 0.002873
## [1] "lambda min is at location 40"
## [1] "the leave-out cells in the source subpop is 93"
## [1] "use 94 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 187 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.877005347593583"
## [1] "evaluation accuracy LDA 0.614973262032086"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 78.6096256684492"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 44.3850267379679"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 20.7142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 5.26315789473684"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 15.7894736842105"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 25"
## [1] "Number of genes in the target data, but not in model genes is 4803"
## [1] "Number of genes in the model present in the target data is 197"
## [1] "There are 3 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 3 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-281P23.2"
## [2] "genes to be added to target subpop are RP11-395G23.3"
## [3] "genes to be added to target subpop are RP11-11N9.4"  
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "SRI"      "KPNA2"    "PALLD"    "FAM210B"  "TMEM54"  
##  [7] "CCDC85B"  "SERPINE2" "CCNA2"    "AAMP"    
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 10"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]

LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 140 cells as source subpop"
## [1] "Total 360 cells in remaining subpops"
## [1] "subsampling 70 cells for training source subpop"
## [1] "subsampling 70 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 140 cells for testing model"
## [1] "rename remaining subpops to 1_3_4"
## [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 2.723e-15 0.390500
##   [2,]  1 3.913e-02 0.372700
##   [3,]  1 7.502e-02 0.355800
##   [4,]  1 1.081e-01 0.339600
##   [5,]  1 1.389e-01 0.324200
##   [6,]  1 1.675e-01 0.309400
##   [7,]  1 1.943e-01 0.295400
##   [8,]  1 2.194e-01 0.281900
##   [9,]  2 2.435e-01 0.269100
##  [10,]  3 2.732e-01 0.256900
##  [11,]  4 3.026e-01 0.245200
##  [12,]  5 3.317e-01 0.234100
##  [13,]  5 3.590e-01 0.223400
##  [14,]  5 3.847e-01 0.213300
##  [15,]  5 4.088e-01 0.203600
##  [16,]  6 4.324e-01 0.194300
##  [17,]  6 4.549e-01 0.185500
##  [18,]  7 4.762e-01 0.177100
##  [19,]  7 4.964e-01 0.169000
##  [20,]  7 5.155e-01 0.161300
##  [21,]  7 5.335e-01 0.154000
##  [22,]  7 5.506e-01 0.147000
##  [23,]  7 5.668e-01 0.140300
##  [24,]  7 5.821e-01 0.133900
##  [25,]  7 5.967e-01 0.127900
##  [26,]  7 6.105e-01 0.122000
##  [27,]  7 6.237e-01 0.116500
##  [28,]  7 6.362e-01 0.111200
##  [29,]  7 6.481e-01 0.106200
##  [30,]  8 6.595e-01 0.101300
##  [31,]  8 6.709e-01 0.096720
##  [32,]  8 6.818e-01 0.092320
##  [33,]  9 6.923e-01 0.088130
##  [34,]  9 7.026e-01 0.084120
##  [35,]  9 7.125e-01 0.080300
##  [36,] 12 7.222e-01 0.076650
##  [37,] 12 7.314e-01 0.073170
##  [38,] 12 7.403e-01 0.069840
##  [39,] 13 7.488e-01 0.066670
##  [40,] 13 7.571e-01 0.063640
##  [41,] 13 7.651e-01 0.060740
##  [42,] 14 7.728e-01 0.057980
##  [43,] 15 7.805e-01 0.055350
##  [44,] 15 7.878e-01 0.052830
##  [45,] 15 7.949e-01 0.050430
##  [46,] 16 8.019e-01 0.048140
##  [47,] 16 8.090e-01 0.045950
##  [48,] 17 8.158e-01 0.043860
##  [49,] 17 8.224e-01 0.041870
##  [50,] 21 8.292e-01 0.039970
##  [51,] 22 8.363e-01 0.038150
##  [52,] 23 8.432e-01 0.036410
##  [53,] 23 8.498e-01 0.034760
##  [54,] 24 8.561e-01 0.033180
##  [55,] 24 8.623e-01 0.031670
##  [56,] 24 8.682e-01 0.030230
##  [57,] 24 8.739e-01 0.028860
##  [58,] 26 8.793e-01 0.027550
##  [59,] 27 8.845e-01 0.026290
##  [60,] 27 8.895e-01 0.025100
##  [61,] 27 8.943e-01 0.023960
##  [62,] 27 8.989e-01 0.022870
##  [63,] 27 9.033e-01 0.021830
##  [64,] 27 9.075e-01 0.020840
##  [65,] 28 9.116e-01 0.019890
##  [66,] 28 9.155e-01 0.018990
##  [67,] 28 9.192e-01 0.018120
##  [68,] 28 9.227e-01 0.017300
##  [69,] 29 9.261e-01 0.016510
##  [70,] 29 9.294e-01 0.015760
##  [71,] 29 9.326e-01 0.015050
##  [72,] 30 9.356e-01 0.014360
##  [73,] 30 9.385e-01 0.013710
##  [74,] 30 9.412e-01 0.013090
##  [75,] 30 9.438e-01 0.012490
##  [76,] 31 9.463e-01 0.011920
##  [77,] 31 9.487e-01 0.011380
##  [78,] 31 9.510e-01 0.010860
##  [79,] 31 9.532e-01 0.010370
##  [80,] 31 9.553e-01 0.009900
##  [81,] 31 9.572e-01 0.009450
##  [82,] 31 9.591e-01 0.009020
##  [83,] 31 9.609e-01 0.008610
##  [84,] 31 9.627e-01 0.008219
##  [85,] 31 9.643e-01 0.007845
##  [86,] 31 9.659e-01 0.007489
##  [87,] 31 9.674e-01 0.007148
##  [88,] 32 9.688e-01 0.006823
##  [89,] 32 9.702e-01 0.006513
##  [90,] 32 9.716e-01 0.006217
##  [91,] 32 9.728e-01 0.005935
##  [92,] 32 9.740e-01 0.005665
##  [93,] 32 9.752e-01 0.005407
##  [94,] 32 9.763e-01 0.005162
##  [95,] 32 9.773e-01 0.004927
##  [96,] 32 9.783e-01 0.004703
##  [97,] 32 9.793e-01 0.004489
##  [98,] 32 9.802e-01 0.004285
##  [99,] 32 9.811e-01 0.004091
## [100,] 32 9.819e-01 0.003905
## [1] "lambda min is at location 26"
## [1] "the leave-out cells in the source subpop is 70"
## [1] "use 70 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 140 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.935714285714286"
## [1] "evaluation accuracy LDA 0.942857142857143"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 8.55614973262032"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 5.88235294117647"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 97.1428571428571"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 97.8571428571428"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 0.75187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 6.01503759398496"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 30"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 42.5"
## [1] "Total 140 cells as source subpop"
## [1] "Total 360 cells in remaining subpops"
## [1] "subsampling 70 cells for training source subpop"
## [1] "subsampling 70 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 140 cells for testing model"
## [1] "rename remaining subpops to 1_3_4"
## [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev   Lambda
##   [1,]  0 2.723e-15 0.407500
##   [2,]  1 4.260e-02 0.389000
##   [3,]  1 8.161e-02 0.371300
##   [4,]  1 1.175e-01 0.354400
##   [5,]  1 1.506e-01 0.338300
##   [6,]  1 1.813e-01 0.322900
##   [7,]  1 2.098e-01 0.308300
##   [8,]  1 2.364e-01 0.294300
##   [9,]  1 2.613e-01 0.280900
##  [10,]  2 2.848e-01 0.268100
##  [11,]  2 3.110e-01 0.255900
##  [12,]  2 3.355e-01 0.244300
##  [13,]  2 3.586e-01 0.233200
##  [14,]  2 3.803e-01 0.222600
##  [15,]  2 4.007e-01 0.212500
##  [16,]  2 4.200e-01 0.202800
##  [17,]  3 4.388e-01 0.193600
##  [18,]  3 4.585e-01 0.184800
##  [19,]  3 4.771e-01 0.176400
##  [20,]  3 4.948e-01 0.168400
##  [21,]  3 5.116e-01 0.160700
##  [22,]  3 5.276e-01 0.153400
##  [23,]  4 5.435e-01 0.146500
##  [24,]  4 5.586e-01 0.139800
##  [25,]  4 5.730e-01 0.133400
##  [26,]  6 5.870e-01 0.127400
##  [27,]  6 6.015e-01 0.121600
##  [28,]  7 6.156e-01 0.116100
##  [29,]  7 6.292e-01 0.110800
##  [30,]  7 6.421e-01 0.105800
##  [31,] 10 6.551e-01 0.100900
##  [32,] 10 6.681e-01 0.096360
##  [33,] 10 6.805e-01 0.091980
##  [34,] 10 6.923e-01 0.087800
##  [35,] 11 7.036e-01 0.083810
##  [36,] 13 7.146e-01 0.080000
##  [37,] 14 7.254e-01 0.076360
##  [38,] 14 7.358e-01 0.072890
##  [39,] 14 7.457e-01 0.069580
##  [40,] 14 7.552e-01 0.066410
##  [41,] 15 7.643e-01 0.063400
##  [42,] 16 7.732e-01 0.060510
##  [43,] 16 7.822e-01 0.057760
##  [44,] 16 7.908e-01 0.055140
##  [45,] 16 7.991e-01 0.052630
##  [46,] 17 8.070e-01 0.050240
##  [47,] 17 8.146e-01 0.047960
##  [48,] 18 8.220e-01 0.045780
##  [49,] 18 8.292e-01 0.043700
##  [50,] 18 8.360e-01 0.041710
##  [51,] 18 8.426e-01 0.039810
##  [52,] 18 8.488e-01 0.038000
##  [53,] 18 8.549e-01 0.036280
##  [54,] 19 8.606e-01 0.034630
##  [55,] 19 8.662e-01 0.033050
##  [56,] 20 8.716e-01 0.031550
##  [57,] 21 8.767e-01 0.030120
##  [58,] 21 8.818e-01 0.028750
##  [59,] 21 8.866e-01 0.027440
##  [60,] 21 8.913e-01 0.026200
##  [61,] 21 8.957e-01 0.025000
##  [62,] 21 9.000e-01 0.023870
##  [63,] 21 9.041e-01 0.022780
##  [64,] 22 9.080e-01 0.021750
##  [65,] 23 9.118e-01 0.020760
##  [66,] 24 9.155e-01 0.019820
##  [67,] 25 9.191e-01 0.018920
##  [68,] 25 9.225e-01 0.018060
##  [69,] 25 9.259e-01 0.017230
##  [70,] 26 9.291e-01 0.016450
##  [71,] 27 9.321e-01 0.015700
##  [72,] 27 9.351e-01 0.014990
##  [73,] 27 9.379e-01 0.014310
##  [74,] 28 9.406e-01 0.013660
##  [75,] 28 9.432e-01 0.013040
##  [76,] 28 9.456e-01 0.012440
##  [77,] 28 9.480e-01 0.011880
##  [78,] 28 9.503e-01 0.011340
##  [79,] 29 9.524e-01 0.010820
##  [80,] 30 9.546e-01 0.010330
##  [81,] 30 9.566e-01 0.009862
##  [82,] 30 9.585e-01 0.009414
##  [83,] 30 9.604e-01 0.008986
##  [84,] 30 9.621e-01 0.008578
##  [85,] 29 9.638e-01 0.008188
##  [86,] 29 9.654e-01 0.007816
##  [87,] 29 9.670e-01 0.007460
##  [88,] 29 9.684e-01 0.007121
##  [89,] 29 9.699e-01 0.006798
##  [90,] 30 9.712e-01 0.006489
##  [91,] 30 9.725e-01 0.006194
##  [92,] 30 9.737e-01 0.005912
##  [93,] 30 9.749e-01 0.005644
##  [94,] 30 9.760e-01 0.005387
##  [95,] 30 9.771e-01 0.005142
##  [96,] 30 9.781e-01 0.004908
##  [97,] 30 9.791e-01 0.004685
##  [98,] 30 9.800e-01 0.004472
##  [99,] 30 9.809e-01 0.004269
## [100,] 30 9.818e-01 0.004075
## [1] "lambda min is at location 53"
## [1] "the leave-out cells in the source subpop is 70"
## [1] "use 70 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 140 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.942857142857143"
## [1] "evaluation accuracy LDA 0.9"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 12.8342245989305"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 1.06951871657754"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 98.5714285714286"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 84.2857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 12.0300751879699"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 0"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4801"
## [1] "Number of genes in the model present in the target data is 199"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-395G23.3"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "C1QBP"    "PLA2G16"  "RAB31"    "CYR61"    "TMEM54"  
##  [7] "SERPINE2" "DBI"      "CCKBR"    "TMOD1"   
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 20"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 133 cells as source subpop"
## [1] "Total 367 cells in remaining subpops"
## [1] "subsampling 66 cells for training source subpop"
## [1] "subsampling 66 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 132 cells for testing model"
## [1] "rename remaining subpops to 1_2_4"
## [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev   Lambda
##   [1,]  0 -2.403e-15 0.259800
##   [2,]  1  1.731e-02 0.248000
##   [3,]  1  3.313e-02 0.236800
##   [4,]  1  4.760e-02 0.226000
##   [5,]  2  6.658e-02 0.215700
##   [6,]  2  8.515e-02 0.205900
##   [7,]  2  1.022e-01 0.196600
##   [8,]  3  1.214e-01 0.187600
##   [9,]  3  1.413e-01 0.179100
##  [10,]  3  1.598e-01 0.171000
##  [11,]  5  1.809e-01 0.163200
##  [12,]  7  2.026e-01 0.155800
##  [13,]  8  2.274e-01 0.148700
##  [14,]  9  2.510e-01 0.141900
##  [15,] 10  2.737e-01 0.135500
##  [16,] 10  2.956e-01 0.129300
##  [17,] 11  3.165e-01 0.123400
##  [18,] 13  3.378e-01 0.117800
##  [19,] 13  3.589e-01 0.112500
##  [20,] 15  3.798e-01 0.107400
##  [21,] 15  4.000e-01 0.102500
##  [22,] 16  4.193e-01 0.097830
##  [23,] 19  4.386e-01 0.093380
##  [24,] 21  4.585e-01 0.089140
##  [25,] 22  4.777e-01 0.085080
##  [26,] 22  4.959e-01 0.081220
##  [27,] 24  5.134e-01 0.077530
##  [28,] 24  5.305e-01 0.074000
##  [29,] 25  5.467e-01 0.070640
##  [30,] 26  5.624e-01 0.067430
##  [31,] 28  5.783e-01 0.064360
##  [32,] 30  5.942e-01 0.061440
##  [33,] 31  6.096e-01 0.058650
##  [34,] 31  6.243e-01 0.055980
##  [35,] 31  6.384e-01 0.053440
##  [36,] 32  6.521e-01 0.051010
##  [37,] 32  6.654e-01 0.048690
##  [38,] 32  6.781e-01 0.046480
##  [39,] 35  6.905e-01 0.044360
##  [40,] 35  7.028e-01 0.042350
##  [41,] 35  7.147e-01 0.040420
##  [42,] 37  7.262e-01 0.038580
##  [43,] 37  7.372e-01 0.036830
##  [44,] 37  7.478e-01 0.035160
##  [45,] 38  7.580e-01 0.033560
##  [46,] 38  7.678e-01 0.032030
##  [47,] 39  7.773e-01 0.030580
##  [48,] 39  7.866e-01 0.029190
##  [49,] 39  7.955e-01 0.027860
##  [50,] 39  8.039e-01 0.026600
##  [51,] 40  8.121e-01 0.025390
##  [52,] 40  8.199e-01 0.024230
##  [53,] 40  8.274e-01 0.023130
##  [54,] 40  8.346e-01 0.022080
##  [55,] 41  8.415e-01 0.021080
##  [56,] 42  8.483e-01 0.020120
##  [57,] 42  8.547e-01 0.019200
##  [58,] 43  8.609e-01 0.018330
##  [59,] 43  8.669e-01 0.017500
##  [60,] 43  8.726e-01 0.016700
##  [61,] 43  8.781e-01 0.015940
##  [62,] 43  8.833e-01 0.015220
##  [63,] 43  8.883e-01 0.014530
##  [64,] 45  8.931e-01 0.013870
##  [65,] 46  8.978e-01 0.013240
##  [66,] 46  9.022e-01 0.012630
##  [67,] 47  9.064e-01 0.012060
##  [68,] 48  9.105e-01 0.011510
##  [69,] 49  9.144e-01 0.010990
##  [70,] 50  9.182e-01 0.010490
##  [71,] 50  9.218e-01 0.010010
##  [72,] 51  9.253e-01 0.009558
##  [73,] 51  9.286e-01 0.009123
##  [74,] 50  9.318e-01 0.008709
##  [75,] 50  9.348e-01 0.008313
##  [76,] 51  9.377e-01 0.007935
##  [77,] 51  9.405e-01 0.007574
##  [78,] 51  9.432e-01 0.007230
##  [79,] 51  9.457e-01 0.006901
##  [80,] 49  9.481e-01 0.006588
##  [81,] 49  9.505e-01 0.006288
##  [82,] 49  9.527e-01 0.006003
##  [83,] 49  9.548e-01 0.005730
##  [84,] 50  9.568e-01 0.005469
##  [85,] 50  9.588e-01 0.005221
##  [86,] 50  9.606e-01 0.004983
##  [87,] 50  9.624e-01 0.004757
##  [88,] 50  9.641e-01 0.004541
##  [89,] 52  9.657e-01 0.004334
##  [90,] 52  9.672e-01 0.004137
##  [91,] 52  9.687e-01 0.003949
##  [92,] 52  9.701e-01 0.003770
##  [93,] 52  9.715e-01 0.003598
##  [94,] 52  9.727e-01 0.003435
##  [95,] 52  9.740e-01 0.003279
##  [96,] 52  9.751e-01 0.003130
##  [97,] 52  9.763e-01 0.002987
##  [98,] 52  9.773e-01 0.002852
##  [99,] 53  9.783e-01 0.002722
## [100,] 53  9.793e-01 0.002598
## [1] "lambda min is at location 53"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 66 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.857142857142857"
## [1] "evaluation accuracy LDA 0.721804511278195"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 56.6844919786096"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 17.1122994652406"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 19.2857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 9.77443609022556"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 84.2105263157895"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 7.5"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 27.5"
## [1] "Total 133 cells as source subpop"
## [1] "Total 367 cells in remaining subpops"
## [1] "subsampling 66 cells for training source subpop"
## [1] "subsampling 66 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 132 cells for testing model"
## [1] "rename remaining subpops to 1_2_4"
## [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev   Lambda
##   [1,]  0 -2.403e-15 0.233500
##   [2,]  1  1.398e-02 0.222900
##   [3,]  1  2.675e-02 0.212800
##   [4,]  2  4.113e-02 0.203100
##   [5,]  2  5.940e-02 0.193900
##   [6,]  4  7.876e-02 0.185100
##   [7,]  4  9.937e-02 0.176700
##   [8,]  6  1.189e-01 0.168600
##   [9,]  6  1.405e-01 0.161000
##  [10,]  6  1.606e-01 0.153700
##  [11,]  7  1.800e-01 0.146700
##  [12,]  7  1.982e-01 0.140000
##  [13,]  8  2.156e-01 0.133600
##  [14,] 11  2.367e-01 0.127600
##  [15,] 11  2.578e-01 0.121800
##  [16,] 12  2.782e-01 0.116200
##  [17,] 12  2.976e-01 0.110900
##  [18,] 18  3.186e-01 0.105900
##  [19,] 18  3.418e-01 0.101100
##  [20,] 18  3.636e-01 0.096500
##  [21,] 18  3.841e-01 0.092110
##  [22,] 20  4.037e-01 0.087920
##  [23,] 21  4.229e-01 0.083930
##  [24,] 21  4.412e-01 0.080110
##  [25,] 21  4.586e-01 0.076470
##  [26,] 23  4.751e-01 0.073000
##  [27,] 25  4.915e-01 0.069680
##  [28,] 25  5.071e-01 0.066510
##  [29,] 26  5.220e-01 0.063490
##  [30,] 27  5.364e-01 0.060600
##  [31,] 28  5.503e-01 0.057850
##  [32,] 31  5.644e-01 0.055220
##  [33,] 31  5.783e-01 0.052710
##  [34,] 33  5.919e-01 0.050310
##  [35,] 35  6.058e-01 0.048030
##  [36,] 35  6.193e-01 0.045840
##  [37,] 37  6.322e-01 0.043760
##  [38,] 38  6.452e-01 0.041770
##  [39,] 38  6.578e-01 0.039870
##  [40,] 38  6.698e-01 0.038060
##  [41,] 40  6.819e-01 0.036330
##  [42,] 41  6.940e-01 0.034680
##  [43,] 42  7.056e-01 0.033100
##  [44,] 42  7.167e-01 0.031600
##  [45,] 43  7.276e-01 0.030160
##  [46,] 43  7.380e-01 0.028790
##  [47,] 42  7.480e-01 0.027480
##  [48,] 41  7.576e-01 0.026230
##  [49,] 41  7.668e-01 0.025040
##  [50,] 42  7.756e-01 0.023900
##  [51,] 43  7.842e-01 0.022820
##  [52,] 44  7.925e-01 0.021780
##  [53,] 45  8.006e-01 0.020790
##  [54,] 45  8.082e-01 0.019840
##  [55,] 47  8.156e-01 0.018940
##  [56,] 47  8.228e-01 0.018080
##  [57,] 47  8.297e-01 0.017260
##  [58,] 47  8.363e-01 0.016480
##  [59,] 47  8.428e-01 0.015730
##  [60,] 49  8.491e-01 0.015010
##  [61,] 51  8.554e-01 0.014330
##  [62,] 51  8.616e-01 0.013680
##  [63,] 51  8.676e-01 0.013060
##  [64,] 53  8.733e-01 0.012460
##  [65,] 55  8.789e-01 0.011900
##  [66,] 55  8.843e-01 0.011360
##  [67,] 55  8.893e-01 0.010840
##  [68,] 55  8.942e-01 0.010350
##  [69,] 55  8.989e-01 0.009877
##  [70,] 56  9.033e-01 0.009428
##  [71,] 56  9.076e-01 0.008999
##  [72,] 56  9.117e-01 0.008590
##  [73,] 55  9.156e-01 0.008200
##  [74,] 55  9.193e-01 0.007827
##  [75,] 57  9.229e-01 0.007471
##  [76,] 57  9.264e-01 0.007132
##  [77,] 56  9.297e-01 0.006808
##  [78,] 56  9.329e-01 0.006498
##  [79,] 56  9.359e-01 0.006203
##  [80,] 56  9.388e-01 0.005921
##  [81,] 56  9.415e-01 0.005652
##  [82,] 56  9.442e-01 0.005395
##  [83,] 57  9.467e-01 0.005150
##  [84,] 59  9.491e-01 0.004916
##  [85,] 60  9.514e-01 0.004692
##  [86,] 60  9.536e-01 0.004479
##  [87,] 61  9.557e-01 0.004275
##  [88,] 61  9.577e-01 0.004081
##  [89,] 61  9.596e-01 0.003896
##  [90,] 61  9.614e-01 0.003719
##  [91,] 61  9.631e-01 0.003550
##  [92,] 61  9.648e-01 0.003388
##  [93,] 62  9.664e-01 0.003234
##  [94,] 63  9.679e-01 0.003087
##  [95,] 64  9.694e-01 0.002947
##  [96,] 63  9.708e-01 0.002813
##  [97,] 62  9.721e-01 0.002685
##  [98,] 61  9.734e-01 0.002563
##  [99,] 61  9.746e-01 0.002447
## [100,] 61  9.758e-01 0.002335
## [1] "lambda min is at location 46"
## [1] "the leave-out cells in the source subpop is 67"
## [1] "use 66 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.796992481203007"
## [1] "evaluation accuracy LDA 0.706766917293233"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 35.2941176470588"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 7.48663101604278"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 2.14285714285714"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 49.6240601503759"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 36.0902255639098"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 7.5"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "first 10 genes in target "
##  [1] "IGFBP6"   "KPNA2"    "FAM210B"  "TAGLN2"   "CYR61"    "SGO1"    
##  [7] "PTRHD1"   "CCDC85B"  "SERPINE2" "HNRNPAB" 
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 7.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">4</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop4vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat4 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 40 cells as source subpop"
## [1] "Total 460 cells in remaining subpops"
## [1] "subsampling 20 cells for training source subpop"
## [1] "subsampling 20 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 40 cells for testing model"
## [1] "rename remaining subpops to 1_2_3"
## [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df    %Dev   Lambda
##   [1,]  0 0.00000 0.229300
##   [2,]  3 0.01649 0.218900
##   [3,]  3 0.04008 0.208900
##   [4,]  5 0.06630 0.199400
##   [5,]  7 0.10220 0.190300
##   [6,]  7 0.13940 0.181700
##   [7,]  9 0.17720 0.173400
##   [8,]  9 0.21450 0.165600
##   [9,]  9 0.24910 0.158000
##  [10,]  9 0.28140 0.150800
##  [11,]  9 0.31140 0.144000
##  [12,] 10 0.34030 0.137400
##  [13,] 10 0.36870 0.131200
##  [14,] 11 0.39580 0.125200
##  [15,] 11 0.42160 0.119500
##  [16,] 12 0.44590 0.114100
##  [17,] 12 0.46930 0.108900
##  [18,] 13 0.49180 0.104000
##  [19,] 14 0.51360 0.099250
##  [20,] 14 0.53460 0.094740
##  [21,] 16 0.55470 0.090430
##  [22,] 16 0.57450 0.086320
##  [23,] 16 0.59330 0.082400
##  [24,] 17 0.61110 0.078650
##  [25,] 20 0.62830 0.075080
##  [26,] 20 0.64530 0.071670
##  [27,] 20 0.66140 0.068410
##  [28,] 20 0.67670 0.065300
##  [29,] 20 0.69130 0.062330
##  [30,] 20 0.70510 0.059500
##  [31,] 20 0.71830 0.056790
##  [32,] 20 0.73080 0.054210
##  [33,] 20 0.74270 0.051750
##  [34,] 20 0.75400 0.049400
##  [35,] 23 0.76500 0.047150
##  [36,] 23 0.77580 0.045010
##  [37,] 24 0.78600 0.042960
##  [38,] 25 0.79590 0.041010
##  [39,] 26 0.80540 0.039150
##  [40,] 26 0.81450 0.037370
##  [41,] 26 0.82310 0.035670
##  [42,] 26 0.83140 0.034050
##  [43,] 26 0.83920 0.032500
##  [44,] 26 0.84670 0.031020
##  [45,] 27 0.85380 0.029610
##  [46,] 28 0.86060 0.028270
##  [47,] 28 0.86720 0.026980
##  [48,] 28 0.87340 0.025760
##  [49,] 28 0.87930 0.024580
##  [50,] 28 0.88490 0.023470
##  [51,] 28 0.89020 0.022400
##  [52,] 28 0.89530 0.021380
##  [53,] 27 0.90020 0.020410
##  [54,] 27 0.90480 0.019480
##  [55,] 26 0.90920 0.018600
##  [56,] 26 0.91330 0.017750
##  [57,] 26 0.91730 0.016950
##  [58,] 26 0.92110 0.016170
##  [59,] 27 0.92470 0.015440
##  [60,] 25 0.92820 0.014740
##  [61,] 25 0.93150 0.014070
##  [62,] 25 0.93460 0.013430
##  [63,] 25 0.93760 0.012820
##  [64,] 25 0.94040 0.012240
##  [65,] 25 0.94310 0.011680
##  [66,] 25 0.94570 0.011150
##  [67,] 25 0.94810 0.010640
##  [68,] 26 0.95050 0.010160
##  [69,] 26 0.95280 0.009697
##  [70,] 28 0.95490 0.009256
##  [71,] 28 0.95700 0.008835
##  [72,] 28 0.95900 0.008434
##  [73,] 28 0.96080 0.008050
##  [74,] 28 0.96260 0.007684
##  [75,] 28 0.96430 0.007335
##  [76,] 28 0.96600 0.007002
##  [77,] 29 0.96750 0.006684
##  [78,] 30 0.96900 0.006380
##  [79,] 31 0.97040 0.006090
##  [80,] 31 0.97180 0.005813
##  [81,] 31 0.97310 0.005549
##  [82,] 30 0.97430 0.005297
##  [83,] 30 0.97540 0.005056
##  [84,] 30 0.97660 0.004826
##  [85,] 30 0.97760 0.004607
##  [86,] 30 0.97870 0.004397
##  [87,] 30 0.97960 0.004197
##  [88,] 30 0.98050 0.004007
##  [89,] 31 0.98140 0.003825
##  [90,] 31 0.98230 0.003651
##  [91,] 31 0.98310 0.003485
##  [92,] 32 0.98380 0.003326
##  [93,] 32 0.98460 0.003175
##  [94,] 32 0.98530 0.003031
##  [95,] 32 0.98600 0.002893
##  [96,] 33 0.98660 0.002762
##  [97,] 33 0.98720 0.002636
##  [98,] 32 0.98780 0.002516
##  [99,] 32 0.98830 0.002402
## [100,] 32 0.98890 0.002293
## [1] "lambda min is at location 23"
## [1] "the leave-out cells in the source subpop is 20"
## [1] "use 20 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 40 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.675"
## [1] "evaluation accuracy LDA 0.7"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 67.9144385026738"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 36.3636363636364"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 55"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 35.7142857142857"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 37.593984962406"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 15.0375939849624"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 92.5"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 72.5"
## [1] "Total 40 cells as source subpop"
## [1] "Total 460 cells in remaining subpops"
## [1] "subsampling 20 cells for training source subpop"
## [1] "subsampling 20 cells in remaining subpops for training"
## [1] "use 200 genes for training model"
## [1] "use 200 genes 40 cells for testing model"
## [1] "rename remaining subpops to 1_2_3"
## [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df    %Dev   Lambda
##   [1,]  0 0.00000 0.261900
##   [2,]  2 0.02512 0.250000
##   [3,]  2 0.05223 0.238600
##   [4,]  2 0.07720 0.227800
##   [5,]  3 0.10350 0.217400
##   [6,]  4 0.13100 0.207500
##   [7,]  4 0.15700 0.198100
##   [8,]  4 0.18130 0.189100
##   [9,]  5 0.20490 0.180500
##  [10,]  5 0.22890 0.172300
##  [11,]  6 0.25220 0.164500
##  [12,]  6 0.27430 0.157000
##  [13,]  6 0.29520 0.149900
##  [14,]  7 0.31820 0.143000
##  [15,]  7 0.34000 0.136500
##  [16,] 10 0.36370 0.130300
##  [17,] 12 0.39040 0.124400
##  [18,] 12 0.41570 0.118800
##  [19,] 14 0.44130 0.113400
##  [20,] 14 0.46670 0.108200
##  [21,] 15 0.49080 0.103300
##  [22,] 15 0.51380 0.098590
##  [23,] 15 0.53540 0.094110
##  [24,] 16 0.55600 0.089830
##  [25,] 16 0.57550 0.085750
##  [26,] 16 0.59400 0.081850
##  [27,] 16 0.61160 0.078130
##  [28,] 17 0.62840 0.074580
##  [29,] 17 0.64450 0.071190
##  [30,] 18 0.65990 0.067960
##  [31,] 19 0.67510 0.064870
##  [32,] 19 0.68960 0.061920
##  [33,] 19 0.70360 0.059100
##  [34,] 19 0.71700 0.056420
##  [35,] 19 0.72970 0.053850
##  [36,] 22 0.74190 0.051410
##  [37,] 23 0.75430 0.049070
##  [38,] 23 0.76610 0.046840
##  [39,] 23 0.77730 0.044710
##  [40,] 23 0.78790 0.042680
##  [41,] 23 0.79790 0.040740
##  [42,] 23 0.80750 0.038890
##  [43,] 23 0.81660 0.037120
##  [44,] 23 0.82520 0.035430
##  [45,] 24 0.83340 0.033820
##  [46,] 24 0.84120 0.032280
##  [47,] 24 0.84860 0.030820
##  [48,] 24 0.85570 0.029420
##  [49,] 26 0.86250 0.028080
##  [50,] 26 0.86900 0.026800
##  [51,] 26 0.87520 0.025580
##  [52,] 26 0.88100 0.024420
##  [53,] 26 0.88660 0.023310
##  [54,] 27 0.89190 0.022250
##  [55,] 27 0.89700 0.021240
##  [56,] 27 0.90180 0.020280
##  [57,] 27 0.90640 0.019350
##  [58,] 27 0.91070 0.018470
##  [59,] 27 0.91490 0.017630
##  [60,] 27 0.91880 0.016830
##  [61,] 27 0.92260 0.016070
##  [62,] 28 0.92620 0.015340
##  [63,] 29 0.92960 0.014640
##  [64,] 29 0.93290 0.013980
##  [65,] 29 0.93600 0.013340
##  [66,] 30 0.93890 0.012730
##  [67,] 30 0.94180 0.012150
##  [68,] 30 0.94450 0.011600
##  [69,] 30 0.94700 0.011080
##  [70,] 30 0.94950 0.010570
##  [71,] 30 0.95180 0.010090
##  [72,] 29 0.95400 0.009632
##  [73,] 29 0.95620 0.009195
##  [74,] 29 0.95820 0.008777
##  [75,] 29 0.96010 0.008378
##  [76,] 29 0.96190 0.007997
##  [77,] 29 0.96370 0.007634
##  [78,] 28 0.96540 0.007287
##  [79,] 28 0.96690 0.006955
##  [80,] 28 0.96850 0.006639
##  [81,] 28 0.96990 0.006338
##  [82,] 28 0.97130 0.006049
##  [83,] 28 0.97260 0.005775
##  [84,] 28 0.97390 0.005512
##  [85,] 28 0.97500 0.005262
##  [86,] 28 0.97620 0.005022
##  [87,] 28 0.97730 0.004794
##  [88,] 28 0.97830 0.004576
##  [89,] 27 0.97930 0.004368
##  [90,] 27 0.98020 0.004170
##  [91,] 27 0.98120 0.003980
##  [92,] 27 0.98200 0.003799
##  [93,] 27 0.98280 0.003627
##  [94,] 27 0.98360 0.003462
##  [95,] 27 0.98440 0.003304
##  [96,] 27 0.98510 0.003154
##  [97,] 27 0.98570 0.003011
##  [98,] 27 0.98640 0.002874
##  [99,] 27 0.98700 0.002743
## [100,] 27 0.98760 0.002619
## [1] "lambda min is at location 48"
## [1] "the leave-out cells in the source subpop is 20"
## [1] "use 20 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 40 cells and 200 genes..."
## [1] "evaluation accuracy ElasticNet 0.75"
## [1] "evaluation accuracy LDA 0.75"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 42.7807486631016"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 13.3689839572193"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 9.28571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 1.50375939849624"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 20.3007518796992"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4800"
## [1] "Number of genes in the model present in the target data is 200"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 60"
## [1] "Number of genes in the target data, but not in model genes is 4802"
## [1] "Number of genes in the model present in the target data is 198"
## [1] "There are 2 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are OTUD6B-AS1"  
## [2] "genes to be added to target subpop are RP11-160E2.6"
## [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"
## [1] "first 10 genes in model "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "first 10 genes in target "
##  [1] "HDDC2"   "C1QBP"   "PNKP"    "SRI"     "RAB10"   "C7orf73" "PARL"   
##  [8] "NDUFB9"  "CYR61"   "PSMB1"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 67.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#prepare table input for sankey plot </span>

LASSO_C1S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, 
                          <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#7570b3"</span>) 

LASSO_C2S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, 
                          <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#1b9e77"</span>)

LASSO_C3S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, 
                          <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#e7298a"</span>)
                          
LASSO_C4S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">4</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat4, 
                          <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#00FFFF"</span>)

combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2, LASSO_C4S2 )
combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)], <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))]) 

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))
Color &lt;-<span class="st"> </span>combined <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/tally.html">count</a></span>(Node, <span class="dt">color=</span>NodeGroup) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="dv">2</span>)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color<span class="op">$</span>color

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))
p1&lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/networkD3/topics/sankeyNetwork">sankeyNetwork</a></span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,  <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, 
                  <span class="dt">fontSize =</span> <span class="dv">22</span> )
p1</code></pre></div>
<div id="htmlwidget-446b85fc2cde45e4d13d" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-446b85fc2cde45e4d13d">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2,3,3,3,3],"target":[0,1,2,3,0,1,2,3,0,1,2,3,0,1,2,3],"value":[77.0053475935829,0,6.39097744360902,22.5,10.6951871657754,97.8571428571428,6.39097744360901,35,45.9893048128342,0,29.6992481203007,7.5,55.3475935828877,27.5,19.5488721804511,76.25],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a","#00FFFF","#00FFFF","#00FFFF","#00FFFF"]},"nodes":{"name":["C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#7570b3","#1b9e77","#e7298a","#00FFFF"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters-in-two-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-two-samples" class="anchor"></a>4.3 Plot the relationship between clusters in two samples</h2>
<p><em>Here we look at one example use case to find relationship between clusters within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] <span class="co">#top 200 gene markers distinguishing cluster 1 </span>
LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1, <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 63 genes for training model"
## [1] "use 63 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev    Lambda
##   [1,]  0 -2.563e-15 2.329e-01
##   [2,]  1  2.665e-02 2.122e-01
##   [3,]  2  6.048e-02 1.934e-01
##   [4,]  2  9.105e-02 1.762e-01
##   [5,]  5  1.318e-01 1.606e-01
##   [6,]  5  1.767e-01 1.463e-01
##   [7,]  6  2.163e-01 1.333e-01
##   [8,]  6  2.524e-01 1.215e-01
##   [9,]  6  2.846e-01 1.107e-01
##  [10,]  6  3.134e-01 1.008e-01
##  [11,]  6  3.394e-01 9.188e-02
##  [12,]  6  3.629e-01 8.371e-02
##  [13,]  8  3.859e-01 7.628e-02
##  [14,]  9  4.082e-01 6.950e-02
##  [15,] 13  4.316e-01 6.333e-02
##  [16,] 15  4.541e-01 5.770e-02
##  [17,] 15  4.760e-01 5.258e-02
##  [18,] 16  4.966e-01 4.790e-02
##  [19,] 18  5.173e-01 4.365e-02
##  [20,] 20  5.366e-01 3.977e-02
##  [21,] 20  5.561e-01 3.624e-02
##  [22,] 23  5.745e-01 3.302e-02
##  [23,] 23  5.921e-01 3.009e-02
##  [24,] 24  6.084e-01 2.741e-02
##  [25,] 25  6.234e-01 2.498e-02
##  [26,] 27  6.382e-01 2.276e-02
##  [27,] 27  6.520e-01 2.074e-02
##  [28,] 29  6.648e-01 1.889e-02
##  [29,] 30  6.775e-01 1.722e-02
##  [30,] 31  6.898e-01 1.569e-02
##  [31,] 33  7.015e-01 1.429e-02
##  [32,] 34  7.126e-01 1.302e-02
##  [33,] 35  7.235e-01 1.187e-02
##  [34,] 35  7.336e-01 1.081e-02
##  [35,] 36  7.430e-01 9.852e-03
##  [36,] 37  7.521e-01 8.976e-03
##  [37,] 40  7.608e-01 8.179e-03
##  [38,] 42  7.692e-01 7.452e-03
##  [39,] 44  7.776e-01 6.790e-03
##  [40,] 44  7.864e-01 6.187e-03
##  [41,] 44  7.951e-01 5.637e-03
##  [42,] 46  8.040e-01 5.137e-03
##  [43,] 46  8.125e-01 4.680e-03
##  [44,] 47  8.209e-01 4.265e-03
##  [45,] 48  8.290e-01 3.886e-03
##  [46,] 48  8.370e-01 3.541e-03
##  [47,] 49  8.450e-01 3.226e-03
##  [48,] 49  8.528e-01 2.939e-03
##  [49,] 50  8.606e-01 2.678e-03
##  [50,] 52  8.685e-01 2.440e-03
##  [51,] 53  8.763e-01 2.224e-03
##  [52,] 55  8.844e-01 2.026e-03
##  [53,] 57  8.921e-01 1.846e-03
##  [54,] 57  8.998e-01 1.682e-03
##  [55,] 56  9.070e-01 1.533e-03
##  [56,] 56  9.137e-01 1.396e-03
##  [57,] 56  9.202e-01 1.272e-03
##  [58,] 56  9.263e-01 1.159e-03
##  [59,] 56  9.320e-01 1.056e-03
##  [60,] 56  9.373e-01 9.625e-04
##  [61,] 56  9.422e-01 8.770e-04
##  [62,] 56  9.470e-01 7.991e-04
##  [63,] 57  9.514e-01 7.281e-04
##  [64,] 57  9.555e-01 6.634e-04
##  [65,] 57  9.594e-01 6.045e-04
##  [66,] 57  9.629e-01 5.508e-04
##  [67,] 56  9.661e-01 5.019e-04
##  [68,] 56  9.691e-01 4.573e-04
##  [69,] 57  9.718e-01 4.167e-04
##  [70,] 57  9.742e-01 3.796e-04
##  [71,] 57  9.765e-01 3.459e-04
##  [72,] 57  9.786e-01 3.152e-04
##  [73,] 57  9.805e-01 2.872e-04
##  [74,] 57  9.822e-01 2.617e-04
##  [75,] 57  9.837e-01 2.384e-04
##  [76,] 57  9.852e-01 2.172e-04
##  [77,] 57  9.865e-01 1.979e-04
##  [78,] 57  9.877e-01 1.804e-04
##  [79,] 57  9.888e-01 1.643e-04
##  [80,] 57  9.898e-01 1.497e-04
##  [81,] 57  9.907e-01 1.364e-04
##  [82,] 57  9.915e-01 1.243e-04
##  [83,] 57  9.923e-01 1.133e-04
##  [84,] 57  9.929e-01 1.032e-04
##  [85,] 57  9.936e-01 9.404e-05
##  [86,] 57  9.941e-01 8.568e-05
##  [87,] 58  9.946e-01 7.807e-05
##  [88,] 58  9.951e-01 7.114e-05
##  [89,] 58  9.955e-01 6.482e-05
##  [90,] 58  9.959e-01 5.906e-05
##  [91,] 58  9.963e-01 5.381e-05
##  [92,] 58  9.966e-01 4.903e-05
##  [93,] 58  9.969e-01 4.468e-05
##  [94,] 58  9.972e-01 4.071e-05
##  [95,] 58  9.974e-01 3.709e-05
##  [96,] 58  9.976e-01 3.380e-05
##  [97,] 58  9.978e-01 3.079e-05
##  [98,] 58  9.980e-01 2.806e-05
##  [99,] 58  9.982e-01 2.557e-05
## [100,] 58  9.984e-01 2.329e-05
## [1] "lambda min is at location 31"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 62 genes..."
## [1] "evaluation accuracy ElasticNet 0.825892857142857"
## [1] "evaluation accuracy LDA 0.799107142857143"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 66.8449197860963"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 5.3475935828877"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 85.7142857142857"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 62.1428571428571"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 31.5789473684211"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 5.26315789473684"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 60"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 17.5"
## [1] "Total 224 cells as source subpop"
## [1] "Total 366 cells in remaining subpops"
## [1] "subsampling 112 cells for training source subpop"
## [1] "subsampling 112 cells in remaining subpops for training"
## [1] "use 63 genes for training model"
## [1] "use 63 genes 224 cells for testing model"
## [1] "rename remaining subpops to 2_3"
## [1] "there are 112 cells in class 2_3 and 112 cells in class 1"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df       %Dev    Lambda
##   [1,]  0 -2.563e-15 2.694e-01
##   [2,]  1  3.566e-02 2.455e-01
##   [3,]  1  6.579e-02 2.237e-01
##   [4,]  1  9.159e-02 2.038e-01
##   [5,]  1  1.139e-01 1.857e-01
##   [6,]  1  1.333e-01 1.692e-01
##   [7,]  3  1.593e-01 1.542e-01
##   [8,]  5  1.964e-01 1.405e-01
##   [9,]  5  2.315e-01 1.280e-01
##  [10,]  6  2.631e-01 1.166e-01
##  [11,]  6  2.920e-01 1.063e-01
##  [12,]  6  3.177e-01 9.682e-02
##  [13,]  7  3.420e-01 8.822e-02
##  [14,]  8  3.649e-01 8.038e-02
##  [15,]  8  3.867e-01 7.324e-02
##  [16,] 11  4.098e-01 6.673e-02
##  [17,] 12  4.308e-01 6.081e-02
##  [18,] 13  4.499e-01 5.540e-02
##  [19,] 15  4.700e-01 5.048e-02
##  [20,] 16  4.892e-01 4.600e-02
##  [21,] 17  5.067e-01 4.191e-02
##  [22,] 17  5.232e-01 3.819e-02
##  [23,] 18  5.385e-01 3.479e-02
##  [24,] 19  5.535e-01 3.170e-02
##  [25,] 19  5.673e-01 2.889e-02
##  [26,] 21  5.801e-01 2.632e-02
##  [27,] 21  5.922e-01 2.398e-02
##  [28,] 23  6.046e-01 2.185e-02
##  [29,] 25  6.172e-01 1.991e-02
##  [30,] 26  6.304e-01 1.814e-02
##  [31,] 26  6.425e-01 1.653e-02
##  [32,] 27  6.541e-01 1.506e-02
##  [33,] 27  6.648e-01 1.372e-02
##  [34,] 30  6.753e-01 1.250e-02
##  [35,] 31  6.856e-01 1.139e-02
##  [36,] 32  6.954e-01 1.038e-02
##  [37,] 33  7.046e-01 9.459e-03
##  [38,] 37  7.143e-01 8.619e-03
##  [39,] 37  7.241e-01 7.853e-03
##  [40,] 37  7.333e-01 7.156e-03
##  [41,] 37  7.418e-01 6.520e-03
##  [42,] 39  7.498e-01 5.941e-03
##  [43,] 41  7.577e-01 5.413e-03
##  [44,] 43  7.657e-01 4.932e-03
##  [45,] 45  7.737e-01 4.494e-03
##  [46,] 45  7.815e-01 4.095e-03
##  [47,] 48  7.893e-01 3.731e-03
##  [48,] 51  7.976e-01 3.400e-03
##  [49,] 49  8.056e-01 3.097e-03
##  [50,] 51  8.135e-01 2.822e-03
##  [51,] 51  8.216e-01 2.572e-03
##  [52,] 51  8.295e-01 2.343e-03
##  [53,] 51  8.371e-01 2.135e-03
##  [54,] 52  8.445e-01 1.945e-03
##  [55,] 53  8.523e-01 1.773e-03
##  [56,] 53  8.603e-01 1.615e-03
##  [57,] 54  8.682e-01 1.472e-03
##  [58,] 55  8.762e-01 1.341e-03
##  [59,] 55  8.843e-01 1.222e-03
##  [60,] 56  8.921e-01 1.113e-03
##  [61,] 57  8.997e-01 1.014e-03
##  [62,] 57  9.073e-01 9.242e-04
##  [63,] 57  9.144e-01 8.421e-04
##  [64,] 58  9.212e-01 7.673e-04
##  [65,] 58  9.277e-01 6.991e-04
##  [66,] 58  9.338e-01 6.370e-04
##  [67,] 58  9.394e-01 5.804e-04
##  [68,] 59  9.448e-01 5.289e-04
##  [69,] 59  9.497e-01 4.819e-04
##  [70,] 59  9.542e-01 4.391e-04
##  [71,] 59  9.584e-01 4.001e-04
##  [72,] 59  9.621e-01 3.645e-04
##  [73,] 59  9.655e-01 3.321e-04
##  [74,] 59  9.687e-01 3.026e-04
##  [75,] 59  9.715e-01 2.757e-04
##  [76,] 59  9.740e-01 2.512e-04
##  [77,] 59  9.764e-01 2.289e-04
##  [78,] 59  9.785e-01 2.086e-04
##  [79,] 59  9.804e-01 1.901e-04
##  [80,] 59  9.822e-01 1.732e-04
##  [81,] 59  9.837e-01 1.578e-04
##  [82,] 58  9.852e-01 1.438e-04
##  [83,] 58  9.864e-01 1.310e-04
##  [84,] 59  9.877e-01 1.194e-04
##  [85,] 59  9.888e-01 1.088e-04
##  [86,] 59  9.897e-01 9.910e-05
##  [87,] 59  9.906e-01 9.029e-05
##  [88,] 59  9.915e-01 8.227e-05
##  [89,] 60  9.922e-01 7.496e-05
##  [90,] 60  9.928e-01 6.830e-05
##  [91,] 60  9.935e-01 6.224e-05
##  [92,] 60  9.940e-01 5.671e-05
##  [93,] 60  9.945e-01 5.167e-05
##  [94,] 60  9.950e-01 4.708e-05
##  [95,] 60  9.954e-01 4.290e-05
##  [96,] 60  9.958e-01 3.909e-05
##  [97,] 60  9.962e-01 3.561e-05
##  [98,] 60  9.965e-01 3.245e-05
##  [99,] 61  9.968e-01 2.957e-05
## [100,] 61  9.971e-01 2.694e-05
## [1] "lambda min is at location 24"
## [1] "the leave-out cells in the source subpop is 112"
## [1] "use 112 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 224 cells and 62 genes..."
## [1] "evaluation accuracy ElasticNet 0.816964285714286"
## [1] "evaluation accuracy LDA 0.776785714285714"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 36.8983957219251"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 187 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 39.572192513369"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 71.4285714285714"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 140 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 54.2857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 39.8496240601504"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 133 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 22.5563909774436"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4938"
## [1] "Number of genes in the model present in the target data is 62"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 57.5"
## [1] "Number of genes in the target data, but not in model genes is 4939"
## [1] "Number of genes in the model present in the target data is 61"
## [1] "There are 1 genes that are in the model, but not in target subpopulations"
## [1] "for dim conformation, add 1 random indexes for genes in model but not in target subpop, later to be replaced by 0"
## [1] "genes to be added to target subpop are RP11-11N9.4"
## [1] "the prediction (target) subop has 62 genes and 40 cells. The trained model has 62 genes"
## [1] "first 10 genes in model "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "first 10 genes in target "
##  [1] "SRI"     "SH3BGRL" "TMEM98"  "GATA4"   "LETMD1"  "IFITM2"  "VIM"    
##  [8] "SAE1"    "TPM4"    "HMGN2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 45"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 203 cells as source subpop"
## [1] "Total 387 cells in remaining subpops"
## [1] "subsampling 102 cells for training source subpop"
## [1] "subsampling 102 cells in remaining subpops for training"
## [1] "use 60 genes for training model"
## [1] "use 60 genes 204 cells for testing model"
## [1] "rename remaining subpops to 1_3"
## [1] "there are 102 cells in class 1_3 and 102 cells in class 2"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 1.281e-15 2.618e-01
##   [2,]  1 3.372e-02 2.385e-01
##   [3,]  1 6.244e-02 2.173e-01
##   [4,]  1 8.739e-02 1.980e-01
##   [5,]  2 1.113e-01 1.804e-01
##   [6,]  2 1.373e-01 1.644e-01
##   [7,]  4 1.669e-01 1.498e-01
##   [8,]  6 2.004e-01 1.365e-01
##   [9,]  6 2.317e-01 1.244e-01
##  [10,]  8 2.625e-01 1.133e-01
##  [11,]  9 2.922e-01 1.032e-01
##  [12,] 10 3.214e-01 9.407e-02
##  [13,] 10 3.482e-01 8.572e-02
##  [14,] 11 3.729e-01 7.810e-02
##  [15,] 12 3.954e-01 7.116e-02
##  [16,] 14 4.170e-01 6.484e-02
##  [17,] 15 4.385e-01 5.908e-02
##  [18,] 15 4.580e-01 5.383e-02
##  [19,] 18 4.771e-01 4.905e-02
##  [20,] 18 4.967e-01 4.469e-02
##  [21,] 19 5.149e-01 4.072e-02
##  [22,] 19 5.320e-01 3.711e-02
##  [23,] 19 5.477e-01 3.381e-02
##  [24,] 21 5.622e-01 3.081e-02
##  [25,] 21 5.761e-01 2.807e-02
##  [26,] 21 5.889e-01 2.558e-02
##  [27,] 23 6.009e-01 2.330e-02
##  [28,] 24 6.133e-01 2.123e-02
##  [29,] 25 6.257e-01 1.935e-02
##  [30,] 25 6.375e-01 1.763e-02
##  [31,] 26 6.490e-01 1.606e-02
##  [32,] 27 6.599e-01 1.464e-02
##  [33,] 27 6.701e-01 1.333e-02
##  [34,] 28 6.808e-01 1.215e-02
##  [35,] 28 6.907e-01 1.107e-02
##  [36,] 29 6.999e-01 1.009e-02
##  [37,] 29 7.084e-01 9.191e-03
##  [38,] 32 7.163e-01 8.375e-03
##  [39,] 34 7.249e-01 7.631e-03
##  [40,] 38 7.340e-01 6.953e-03
##  [41,] 37 7.438e-01 6.335e-03
##  [42,] 38 7.533e-01 5.772e-03
##  [43,] 40 7.633e-01 5.260e-03
##  [44,] 41 7.731e-01 4.792e-03
##  [45,] 43 7.833e-01 4.367e-03
##  [46,] 43 7.933e-01 3.979e-03
##  [47,] 44 8.030e-01 3.625e-03
##  [48,] 46 8.128e-01 3.303e-03
##  [49,] 46 8.227e-01 3.010e-03
##  [50,] 46 8.324e-01 2.742e-03
##  [51,] 46 8.421e-01 2.499e-03
##  [52,] 47 8.518e-01 2.277e-03
##  [53,] 47 8.615e-01 2.074e-03
##  [54,] 46 8.707e-01 1.890e-03
##  [55,] 47 8.798e-01 1.722e-03
##  [56,] 48 8.887e-01 1.569e-03
##  [57,] 49 8.973e-01 1.430e-03
##  [58,] 49 9.055e-01 1.303e-03
##  [59,] 49 9.133e-01 1.187e-03
##  [60,] 49 9.205e-01 1.082e-03
##  [61,] 52 9.273e-01 9.855e-04
##  [62,] 52 9.337e-01 8.980e-04
##  [63,] 51 9.395e-01 8.182e-04
##  [64,] 51 9.448e-01 7.455e-04
##  [65,] 51 9.498e-01 6.793e-04
##  [66,] 51 9.542e-01 6.189e-04
##  [67,] 51 9.583e-01 5.640e-04
##  [68,] 51 9.621e-01 5.139e-04
##  [69,] 51 9.655e-01 4.682e-04
##  [70,] 51 9.686e-01 4.266e-04
##  [71,] 51 9.714e-01 3.887e-04
##  [72,] 51 9.739e-01 3.542e-04
##  [73,] 50 9.763e-01 3.227e-04
##  [74,] 50 9.784e-01 2.941e-04
##  [75,] 50 9.803e-01 2.679e-04
##  [76,] 50 9.820e-01 2.441e-04
##  [77,] 51 9.836e-01 2.224e-04
##  [78,] 51 9.851e-01 2.027e-04
##  [79,] 51 9.864e-01 1.847e-04
##  [80,] 51 9.876e-01 1.683e-04
##  [81,] 51 9.887e-01 1.533e-04
##  [82,] 51 9.897e-01 1.397e-04
##  [83,] 51 9.906e-01 1.273e-04
##  [84,] 51 9.914e-01 1.160e-04
##  [85,] 51 9.922e-01 1.057e-04
##  [86,] 51 9.929e-01 9.629e-05
##  [87,] 51 9.935e-01 8.773e-05
##  [88,] 51 9.941e-01 7.994e-05
##  [89,] 51 9.945e-01 7.284e-05
##  [90,] 51 9.950e-01 6.637e-05
##  [91,] 51 9.954e-01 6.047e-05
##  [92,] 51 9.958e-01 5.510e-05
##  [93,] 52 9.962e-01 5.020e-05
##  [94,] 54 9.965e-01 4.574e-05
##  [95,] 54 9.968e-01 4.168e-05
##  [96,] 54 9.971e-01 3.798e-05
##  [97,] 54 9.973e-01 3.460e-05
##  [98,] 54 9.976e-01 3.153e-05
##  [99,] 54 9.978e-01 2.873e-05
## [100,] 55 9.980e-01 2.618e-05
## [1] "lambda min is at location 22"
## [1] "the leave-out cells in the source subpop is 101"
## [1] "use 102 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 203 cells and 59 genes..."
## [1] "evaluation accuracy ElasticNet 0.857142857142857"
## [1] "evaluation accuracy LDA 0.847290640394089"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 25.668449197861"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 28.3422459893048"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 13.5714285714286"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 47.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 66.1654135338346"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 30.8270676691729"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 47.5"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 45"
## [1] "Total 203 cells as source subpop"
## [1] "Total 387 cells in remaining subpops"
## [1] "subsampling 102 cells for training source subpop"
## [1] "subsampling 102 cells in remaining subpops for training"
## [1] "use 60 genes for training model"
## [1] "use 60 genes 204 cells for testing model"
## [1] "rename remaining subpops to 1_3"
## [1] "there are 102 cells in class 1_3 and 102 cells in class 2"
## [1] "removing 1 genes with no variance"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 1.281e-15 2.496e-01
##   [2,]  1 3.064e-02 2.274e-01
##   [3,]  1 5.662e-02 2.072e-01
##   [4,]  1 7.899e-02 1.888e-01
##   [5,]  1 9.848e-02 1.721e-01
##   [6,]  4 1.294e-01 1.568e-01
##   [7,]  4 1.587e-01 1.428e-01
##   [8,]  6 1.845e-01 1.302e-01
##   [9,]  8 2.149e-01 1.186e-01
##  [10,] 10 2.470e-01 1.081e-01
##  [11,] 11 2.764e-01 9.846e-02
##  [12,] 12 3.058e-01 8.971e-02
##  [13,] 12 3.326e-01 8.174e-02
##  [14,] 13 3.566e-01 7.448e-02
##  [15,] 13 3.808e-01 6.786e-02
##  [16,] 15 4.025e-01 6.183e-02
##  [17,] 16 4.234e-01 5.634e-02
##  [18,] 17 4.427e-01 5.134e-02
##  [19,] 17 4.603e-01 4.678e-02
##  [20,] 18 4.763e-01 4.262e-02
##  [21,] 19 4.914e-01 3.883e-02
##  [22,] 20 5.059e-01 3.538e-02
##  [23,] 21 5.191e-01 3.224e-02
##  [24,] 21 5.312e-01 2.938e-02
##  [25,] 23 5.425e-01 2.677e-02
##  [26,] 25 5.545e-01 2.439e-02
##  [27,] 25 5.656e-01 2.222e-02
##  [28,] 27 5.763e-01 2.025e-02
##  [29,] 27 5.870e-01 1.845e-02
##  [30,] 28 5.982e-01 1.681e-02
##  [31,] 29 6.087e-01 1.532e-02
##  [32,] 32 6.186e-01 1.396e-02
##  [33,] 32 6.284e-01 1.272e-02
##  [34,] 32 6.373e-01 1.159e-02
##  [35,] 33 6.454e-01 1.056e-02
##  [36,] 34 6.528e-01 9.619e-03
##  [37,] 35 6.597e-01 8.765e-03
##  [38,] 39 6.672e-01 7.986e-03
##  [39,] 39 6.754e-01 7.277e-03
##  [40,] 41 6.835e-01 6.630e-03
##  [41,] 41 6.911e-01 6.041e-03
##  [42,] 42 6.984e-01 5.505e-03
##  [43,] 44 7.055e-01 5.016e-03
##  [44,] 44 7.125e-01 4.570e-03
##  [45,] 45 7.190e-01 4.164e-03
##  [46,] 45 7.251e-01 3.794e-03
##  [47,] 45 7.309e-01 3.457e-03
##  [48,] 46 7.365e-01 3.150e-03
##  [49,] 46 7.420e-01 2.870e-03
##  [50,] 46 7.472e-01 2.615e-03
##  [51,] 46 7.523e-01 2.383e-03
##  [52,] 46 7.571e-01 2.171e-03
##  [53,] 46 7.616e-01 1.978e-03
##  [54,] 46 7.658e-01 1.802e-03
##  [55,] 47 7.697e-01 1.642e-03
##  [56,] 48 7.735e-01 1.496e-03
##  [57,] 50 7.775e-01 1.364e-03
##  [58,] 50 7.818e-01 1.242e-03
##  [59,] 50 7.862e-01 1.132e-03
##  [60,] 51 7.905e-01 1.031e-03
##  [61,] 51 7.948e-01 9.398e-04
##  [62,] 51 7.989e-01 8.563e-04
##  [63,] 51 8.031e-01 7.803e-04
##  [64,] 50 8.072e-01 7.109e-04
##  [65,] 49 8.112e-01 6.478e-04
##  [66,] 50 8.152e-01 5.902e-04
##  [67,] 50 8.189e-01 5.378e-04
##  [68,] 52 8.225e-01 4.900e-04
##  [69,] 53 8.273e-01 4.465e-04
##  [70,] 53 8.326e-01 4.068e-04
##  [71,] 53 8.380e-01 3.707e-04
##  [72,] 54 8.434e-01 3.378e-04
##  [73,] 54 8.489e-01 3.077e-04
##  [74,] 54 8.543e-01 2.804e-04
##  [75,] 54 8.598e-01 2.555e-04
##  [76,] 56 8.655e-01 2.328e-04
##  [77,] 56 8.723e-01 2.121e-04
##  [78,] 56 8.797e-01 1.933e-04
##  [79,] 55 8.866e-01 1.761e-04
##  [80,] 55 8.940e-01 1.605e-04
##  [81,] 53 9.016e-01 1.462e-04
##  [82,] 54 9.092e-01 1.332e-04
##  [83,] 55 9.174e-01 1.214e-04
##  [84,] 56 9.249e-01 1.106e-04
##  [85,] 56 9.322e-01 1.008e-04
##  [86,] 56 9.389e-01 9.182e-05
##  [87,] 56 9.447e-01 8.366e-05
##  [88,] 56 9.497e-01 7.623e-05
##  [89,] 57 9.542e-01 6.946e-05
##  [90,] 57 9.584e-01 6.329e-05
##  [91,] 57 9.619e-01 5.767e-05
##  [92,] 57 9.653e-01 5.254e-05
##  [93,] 57 9.684e-01 4.788e-05
##  [94,] 57 9.712e-01 4.362e-05
##  [95,] 57 9.737e-01 3.975e-05
##  [96,] 57 9.760e-01 3.622e-05
##  [97,] 57 9.780e-01 3.300e-05
##  [98,] 57 9.799e-01 3.007e-05
##  [99,] 57 9.816e-01 2.740e-05
## [100,] 57 9.831e-01 2.496e-05
## [1] "lambda min is at location 29"
## [1] "the leave-out cells in the source subpop is 101"
## [1] "use 102 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 203 cells and 59 genes..."
## [1] "evaluation accuracy ElasticNet 0.822660098522167"
## [1] "evaluation accuracy LDA 0.817733990147783"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 21.9251336898396"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 187 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 75.4010695187166"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 0.714285714285714"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 140 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 12.8571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 21.0526315789474"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 133 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 93.2330827067669"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 7.5"
## [1] "Number of genes in the target data, but not in model genes is 4941"
## [1] "Number of genes in the model present in the target data is 59"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 59 genes and 40 cells. The trained model has 59 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "first 10 genes in target "
##  [1] "CTGF"    "TMEM98"  "GATA4"   "IFITM2"  "NHP2"    "VIM"     "TPM4"   
##  [8] "IGFBP7"  "C1GALT1" "SNAI2"  
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 67.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</code></pre></div>
<pre><code>## [1] "Total 163 cells as source subpop"
## [1] "Total 427 cells in remaining subpops"
## [1] "subsampling 82 cells for training source subpop"
## [1] "subsampling 82 cells in remaining subpops for training"
## [1] "use 65 genes for training model"
## [1] "use 65 genes 164 cells for testing model"
## [1] "rename remaining subpops to 2_1"
## [1] "there are 82 cells in class 2_1 and 82 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##        Df      %Dev    Lambda
##   [1,]  0 2.403e-15 1.606e-01
##   [2,]  2 1.696e-02 1.463e-01
##   [3,]  3 3.708e-02 1.333e-01
##   [4,]  3 5.772e-02 1.215e-01
##   [5,]  3 7.541e-02 1.107e-01
##   [6,]  3 9.067e-02 1.008e-01
##   [7,]  3 1.039e-01 9.189e-02
##   [8,]  5 1.169e-01 8.372e-02
##   [9,]  7 1.357e-01 7.629e-02
##  [10,] 10 1.579e-01 6.951e-02
##  [11,] 11 1.797e-01 6.333e-02
##  [12,] 13 1.993e-01 5.771e-02
##  [13,] 15 2.171e-01 5.258e-02
##  [14,] 16 2.343e-01 4.791e-02
##  [15,] 17 2.506e-01 4.365e-02
##  [16,] 20 2.680e-01 3.978e-02
##  [17,] 25 2.910e-01 3.624e-02
##  [18,] 26 3.136e-01 3.302e-02
##  [19,] 28 3.352e-01 3.009e-02
##  [20,] 28 3.551e-01 2.742e-02
##  [21,] 29 3.738e-01 2.498e-02
##  [22,] 32 3.925e-01 2.276e-02
##  [23,] 32 4.109e-01 2.074e-02
##  [24,] 34 4.278e-01 1.890e-02
##  [25,] 38 4.439e-01 1.722e-02
##  [26,] 40 4.607e-01 1.569e-02
##  [27,] 42 4.772e-01 1.429e-02
##  [28,] 46 4.938e-01 1.302e-02
##  [29,] 47 5.107e-01 1.187e-02
##  [30,] 48 5.268e-01 1.081e-02
##  [31,] 47 5.417e-01 9.853e-03
##  [32,] 49 5.556e-01 8.977e-03
##  [33,] 49 5.687e-01 8.180e-03
##  [34,] 49 5.811e-01 7.453e-03
##  [35,] 49 5.928e-01 6.791e-03
##  [36,] 52 6.043e-01 6.188e-03
##  [37,] 52 6.150e-01 5.638e-03
##  [38,] 52 6.251e-01 5.137e-03
##  [39,] 52 6.345e-01 4.681e-03
##  [40,] 53 6.436e-01 4.265e-03
##  [41,] 55 6.527e-01 3.886e-03
##  [42,] 55 6.621e-01 3.541e-03
##  [43,] 54 6.711e-01 3.226e-03
##  [44,] 55 6.797e-01 2.940e-03
##  [45,] 56 6.882e-01 2.679e-03
##  [46,] 56 6.966e-01 2.441e-03
##  [47,] 56 7.048e-01 2.224e-03
##  [48,] 55 7.127e-01 2.026e-03
##  [49,] 57 7.207e-01 1.846e-03
##  [50,] 57 7.291e-01 1.682e-03
##  [51,] 57 7.374e-01 1.533e-03
##  [52,] 58 7.457e-01 1.397e-03
##  [53,] 59 7.542e-01 1.273e-03
##  [54,] 60 7.631e-01 1.159e-03
##  [55,] 60 7.731e-01 1.056e-03
##  [56,] 59 7.835e-01 9.626e-04
##  [57,] 61 7.953e-01 8.771e-04
##  [58,] 60 8.086e-01 7.992e-04
##  [59,] 61 8.212e-01 7.282e-04
##  [60,] 60 8.334e-01 6.635e-04
##  [61,] 62 8.465e-01 6.046e-04
##  [62,] 61 8.593e-01 5.508e-04
##  [63,] 63 8.715e-01 5.019e-04
##  [64,] 63 8.823e-01 4.573e-04
##  [65,] 62 8.935e-01 4.167e-04
##  [66,] 63 9.027e-01 3.797e-04
##  [67,] 61 9.114e-01 3.459e-04
##  [68,] 60 9.196e-01 3.152e-04
##  [69,] 61 9.264e-01 2.872e-04
##  [70,] 62 9.329e-01 2.617e-04
##  [71,] 62 9.392e-01 2.384e-04
##  [72,] 61 9.448e-01 2.173e-04
##  [73,] 61 9.498e-01 1.980e-04
##  [74,] 62 9.545e-01 1.804e-04
##  [75,] 62 9.586e-01 1.644e-04
##  [76,] 62 9.624e-01 1.498e-04
##  [77,] 62 9.658e-01 1.364e-04
##  [78,] 62 9.689e-01 1.243e-04
##  [79,] 62 9.717e-01 1.133e-04
##  [80,] 62 9.742e-01 1.032e-04
##  [81,] 62 9.765e-01 9.405e-05
##  [82,] 62 9.786e-01 8.569e-05
##  [83,] 62 9.805e-01 7.808e-05
##  [84,] 62 9.822e-01 7.114e-05
##  [85,] 62 9.837e-01 6.482e-05
##  [86,] 62 9.852e-01 5.907e-05
##  [87,] 62 9.865e-01 5.382e-05
##  [88,] 62 9.876e-01 4.904e-05
##  [89,] 62 9.887e-01 4.468e-05
##  [90,] 61 9.897e-01 4.071e-05
##  [91,] 60 9.906e-01 3.709e-05
##  [92,] 60 9.914e-01 3.380e-05
##  [93,] 60 9.921e-01 3.080e-05
##  [94,] 60 9.928e-01 2.806e-05
##  [95,] 60 9.934e-01 2.557e-05
##  [96,] 59 9.939e-01 2.330e-05
##  [97,] 59 9.945e-01 2.123e-05
##  [98,] 59 9.949e-01 1.934e-05
##  [99,] 59 9.953e-01 1.762e-05
## [100,] 59 9.957e-01 1.606e-05
## [1] "lambda min is at location 35"
## [1] "the leave-out cells in the source subpop is 81"
## [1] "use 82 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 163 cells and 65 genes..."
## [1] "evaluation accuracy ElasticNet 0.680981595092024"
## [1] "evaluation accuracy LDA 0.668711656441718"
## [1] "done training for bootstrap 1, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 44.9197860962567"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 80.2139037433155"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 7.14285714285714"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 1.42857142857143"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 84.2105263157895"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 84.2105263157895"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 35"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 52.5"
## [1] "Total 163 cells as source subpop"
## [1] "Total 427 cells in remaining subpops"
## [1] "subsampling 82 cells for training source subpop"
## [1] "subsampling 82 cells in remaining subpops for training"
## [1] "use 65 genes for training model"
## [1] "use 65 genes 164 cells for testing model"
## [1] "rename remaining subpops to 2_1"
## [1] "there are 82 cells in class 2_1 and 82 cells in class 3"
## [1] "standardizing prediction/target dataset"
## [1] "performning elasticnet model training..."
## [1] "performning LDA model training..."
## [1] "extracting deviance and best gene features..."
## 
## Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = as.vector(dataset$Cluster_class), family = "binomial") 
## 
##       Df      %Dev    Lambda
##  [1,]  0 2.403e-15 0.1574000
##  [2,]  1 1.215e-02 0.1434000
##  [3,]  1 2.230e-02 0.1307000
##  [4,]  2 3.627e-02 0.1191000
##  [5,]  4 5.208e-02 0.1085000
##  [6,]  6 7.402e-02 0.0988500
##  [7,]  7 9.445e-02 0.0900700
##  [8,] 10 1.155e-01 0.0820600
##  [9,] 11 1.432e-01 0.0747700
## [10,] 13 1.680e-01 0.0681300
## [11,] 14 1.933e-01 0.0620800
## [12,] 14 2.180e-01 0.0565600
## [13,] 14 2.396e-01 0.0515400
## [14,] 14 2.584e-01 0.0469600
## [15,] 16 2.763e-01 0.0427900
## [16,] 16 2.936e-01 0.0389900
## [17,] 17 3.092e-01 0.0355200
## [18,] 18 3.244e-01 0.0323700
## [19,] 20 3.396e-01 0.0294900
## [20,] 20 3.547e-01 0.0268700
## [21,] 21 3.686e-01 0.0244900
## [22,] 23 3.817e-01 0.0223100
## [23,] 28 3.946e-01 0.0203300
## [24,] 29 4.075e-01 0.0185200
## [25,] 29 4.190e-01 0.0168800
## [26,] 33 4.315e-01 0.0153800
## [27,] 35 4.438e-01 0.0140100
## [28,] 38 4.553e-01 0.0127700
## [29,] 41 4.670e-01 0.0116300
## [30,] 42 4.789e-01 0.0106000
## [31,] 45 4.901e-01 0.0096570
## [32,] 45 5.012e-01 0.0088000
## [33,] 46 5.112e-01 0.0080180
## [34,] 47 5.207e-01 0.0073060
## [35,] 46 5.296e-01 0.0066570
## [36,] 46 5.373e-01 0.0060650
## [37,] 48 5.443e-01 0.0055260
## [38,] 49 5.512e-01 0.0050350
## [39,] 51 5.574e-01 0.0045880
## [40,] 51 5.631e-01 0.0041800
## [41,] 52 5.683e-01 0.0038090
## [42,] 52 5.731e-01 0.0034710
## [43,] 53 5.775e-01 0.0031620
## [44,] 56 5.817e-01 0.0028810
## [45,] 56 5.858e-01 0.0026250
## [46,] 57 5.895e-01 0.0023920
## [47,] 58 5.930e-01 0.0021800
## [48,] 59 5.962e-01 0.0019860
## [49,] 61 5.992e-01 0.0018100
## [50,] 61 6.020e-01 0.0016490
## [51,] 62 6.048e-01 0.0015020
## [52,] 62 6.073e-01 0.0013690
## [53,] 63 6.096e-01 0.0012470
## [54,] 63 6.118e-01 0.0011370
## [55,] 64 6.137e-01 0.0010360
## [56,] 63 6.154e-01 0.0009435
## [57,] 61 6.169e-01 0.0008597
## [58,] 61 6.183e-01 0.0007833
## [59,] 61 6.194e-01 0.0007138
## [60,] 61 6.205e-01 0.0006503
## [61,] 61 6.214e-01 0.0005926
## [62,] 62 6.222e-01 0.0005399
## [63,] 63 6.229e-01 0.0004920
## [64,] 63 6.236e-01 0.0004483
## [65,] 63 6.242e-01 0.0004084
## [66,] 63 6.247e-01 0.0003722
## [67,] 63 6.252e-01 0.0003391
## [68,] 63 6.256e-01 0.0003090
## [69,] 63 6.259e-01 0.0002815
## [70,] 64 6.262e-01 0.0002565
## [71,] 64 6.265e-01 0.0002337
## [72,] 64 6.267e-01 0.0002130
## [73,] 64 6.269e-01 0.0001940
## [74,] 64 6.269e-01 0.0001768
## [1] "lambda min is at location 36"
## [1] "the leave-out cells in the source subpop is 81"
## [1] "use 82 target subpops cells for leave-out test set"
## [1] "standardizing the leave-out target and source subpops..."
## [1] "start ElasticNet prediction for estimating accuracy..."
## [1] "start LDA prediction for estimating accuracy for 163 cells and 65 genes..."
## [1] "evaluation accuracy ElasticNet 0.631901840490798"
## [1] "evaluation accuracy LDA 0.638036809815951"
## [1] "done training for bootstrap 2, moving to prediction..."
## [1] "standardizing target subpops before prediction..."
## [1] "predicting from source to target subpop 1..."
## [1] "number of cells in the target subpop 1 is 187"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 1 is 59.3582887700535"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 187 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 1 is 8.55614973262032"
## [1] "predicting from source to target subpop 2..."
## [1] "number of cells in the target subpop 2 is 140"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 2 is 2.14285714285714"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 140 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 2 is 9.28571428571429"
## [1] "predicting from source to target subpop 3..."
## [1] "number of cells in the target subpop 3 is 133"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 3 is 7.5187969924812"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 133 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 3 is 15.7894736842105"
## [1] "predicting from source to target subpop 4..."
## [1] "number of cells in the target subpop 4 is 40"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running elasticNet classification..."
## [1] "class probability prediction ElasticNet for target subpop 4 is 2.5"
## [1] "Number of genes in the target data, but not in model genes is 4935"
## [1] "Number of genes in the model present in the target data is 65"
## [1] "There are 0 genes that are in the model, but not in target subpopulations"
## [1] "the prediction (target) subop has 65 genes and 40 cells. The trained model has 65 genes"
## [1] "first 10 genes in model "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "first 10 genes in target "
##  [1] "CTGF"   "LETMD1" "SUMF2"  "IFITM2" "SAE1"   "PPP1CA" "TPM4"  
##  [8] "HMGN2"  "FKBP4"  "IGFBP7"
## [1] "running LDA classification..."
## [1] "class probability prediction LDA for target subpop 4 is 12.5"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#prepare table input for sankey plot </span>

LASSO_C1S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#7570b3"</span>)

LASSO_C2S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#1b9e77"</span>)

LASSO_C3S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#e7298a"</span>)


combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S1,LASSO_C2S1,LASSO_C3S1)

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)], <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))]) 
combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]


<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))
n &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(node_df<span class="op">$</span>Node))
getPalette =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/colorRamp">colorRampPalette</a></span>(RColorBrewer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">9</span>, <span class="st">"Set1"</span>))
Color =<span class="st"> </span><span class="kw">getPalette</span>(n)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))
p1&lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/networkD3/topics/sankeyNetwork">sankeyNetwork</a></span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,  <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, 
                  <span class="dt">fontSize =</span> <span class="dv">22</span> )
p1</code></pre></div>
<div id="htmlwidget-61e9779eea78eca92e33" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-61e9779eea78eca92e33">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2],"target":[3,4,5,6,3,4,5,6,3,4,5,6],"value":[51.8716577540107,78.5714285714286,35.7142857142858,58.75,23.7967914438503,7.14285714285716,43.609022556391,27.5,52.1390374331551,4.64285714285714,45.8646616541353,18.75],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a"]},"nodes":{"name":["C1_MP1","C2_MP1","C3_MP1","C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#E41A1C","#3E8E93","#7E6E85","#FF7F00","#E1C62F","#DB728C","#999999"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></code></pre></div>
</div>
<div id="annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with-a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with-a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="anchor"></a>4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with a reference dataset to see which cluster is most similar to the reference</h2>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation-instruction">1. Installation instruction</a></li>
      <li>
<a href="#a-simple-workflow-of-the-scgps">2. A simple workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#create-scgps-objects">2.1 Create scGPS objects</a></li>
      <li><a href="#run-prediction">2.2 Run prediction</a></li>
      <li><a href="#summarise-results">2.3 Summarise results</a></li>
      </ul>
</li>
      <li>
<a href="#a-complete-workflow-of-the-scgps">3. A complete workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#identify-clusters-in-a-dataset-using-core">3.1 Identify clusters in a dataset using CORE</a></li>
      <li><a href="#identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution">3.2 Identify clusters in a dataset using SCORE (Stable Clustering at Optimal REsolution)</a></li>
      <li><a href="#visualise-all-cluster-results-in-all-iterations">3.3 Visualise all cluster results in all iterations</a></li>
      <li><a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr">3.4 Compare clustering results with other dimensional reduction methods (e.g., CIDR)</a></li>
      <li><a href="#find-gene-markers-and-annotate-clusters">3.5 Find gene markers and annotate clusters</a></li>
      </ul>
</li>
      <li>
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples">4. Relationship between clusters within one sample or between two samples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#start-the-scgps-prediction-to-find-relationship-between-clusters">4.1 Start the scGPS prediction to find relationship between clusters</a></li>
      <li><a href="#display-summary-results-for-the-prediction">4.2 Display summary results for the prediction</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-one-sample">4.3 Plot the relationship between clusters in one sample</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-two-samples">4.3 Plot the relationship between clusters in two samples</a></li>
      <li><a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with-a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference">4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with a reference dataset to see which cluster is most similar to the reference</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Asenbouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
