<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGPS introduction â€¢ scGPS</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<script src="../extra.js"></script><meta property="og:title" content="scGPS introduction">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.9.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="vignette_files/htmlwidgets-1.3/htmlwidgets.js"></script><script src="vignette_files/d3-4.5.0/d3.min.js"></script><script src="vignette_files/sankey-1/sankey.js"></script><script src="vignette_files/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scGPS introduction</h1>
                        <h4 class="author">Quan Nguyen and Michael Thompson</h4>
            
            <h4 class="date">2019-02-07</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Prior to installing scGPS you need to install the SummarizedExperiment</span>
<span class="co"># bioconductor package as the following</span>
<span class="co"># source('https://bioconductor.org/biocLite.R') biocLite('SummarizedExperiment')</span>

<span class="co"># To install scGPS from github (Depending on the configuration of the local</span>
<span class="co"># computer or HPC, possible custom C++ compilation may be required - see</span>
<span class="co"># installation trouble-shootings below)</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"IMB-Computational-Genomics-Lab/scGPS"</span>)

<span class="co"># for C++ compilation trouble-shooting, manual download and installation can be</span>
<span class="co"># done from github</span>

git clone https<span class="op">:</span><span class="er">//</span>github.com<span class="op">/</span>IMB<span class="op">-</span>Computational<span class="op">-</span>Genomics<span class="op">-</span>Lab<span class="op">/</span>scGPS

<span class="co"># then check in scGPS/src if any of the precompiled (e.g.  those with *.so and</span>
<span class="co"># *.o) files exist and delete them before recompiling</span>

<span class="co"># create a Makevars file in the scGPS/src with one line: PKG_LIBS =</span>
<span class="co"># $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)</span>

<span class="co"># then with the scGPS as the R working directory, manually recompile scGPS in R</span>
<span class="co"># using devtools to load and install functions</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/document">document</a></span>()
<span class="co">#load the package to the workspace </span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>()</code></pre></div>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given a mixed population with known subpopulations, estimate transition scores between these subpopulation</em></p>
<div id="create-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#create-scgps-objects" class="anchor"></a>2.1 Create scGPS objects</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># load mixed population 1 (loaded from sample1 dataset, named it as day2)</span>

devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/'</span>) 

day2 &lt;-<span class="st"> </span>sample1
mixedpop1 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day2<span class="op">$</span>dat2_counts, <span class="dt">GeneMetadata =</span> day2<span class="op">$</span>dat2geneInfo, 
    <span class="dt">CellMetadata =</span> day2<span class="op">$</span>dat2_clusters)

<span class="co"># load mixed population 2 (loaded from sample2 dataset, named it as day5)</span>
day5 &lt;-<span class="st"> </span>sample2
mixedpop2 &lt;-<span class="st"> </span><span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo,
    <span class="dt">CellMetadata =</span> day5<span class="op">$</span>dat5_clusters)</code></pre></div>
</div>
<div id="run-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#run-prediction" class="anchor"></a>2.2 Run prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># select a subpopulation</span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co"># load gene list (this can be any lists of user selected genes)</span>
genes &lt;-<span class="st"> </span>GeneList
genes &lt;-<span class="st"> </span>genes<span class="op">$</span>Merged_unique
<span class="co"># load cluster information </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#run training </span>
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID, <span class="dt">listData =</span> <span class="kw">list</span>(),
     <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 224 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 366 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 112 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 112 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 107 genes for training model"</span>
<span class="co">#&gt; [1] "use 107 genes 224 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3"</span>
<span class="co">#&gt; [1] "there are 112 cells in class 2_3 and 112 cells in class 1"</span>
<span class="co">#&gt; [1] "removing 11 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev    Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 3.338e-01</span>
<span class="co">#&gt;   [2,]  1  5.472e-02 3.042e-01</span>
<span class="co">#&gt;   [3,]  2  1.036e-01 2.771e-01</span>
<span class="co">#&gt;   [4,]  2  1.492e-01 2.525e-01</span>
<span class="co">#&gt;   [5,]  2  1.881e-01 2.301e-01</span>
<span class="co">#&gt;   [6,]  2  2.216e-01 2.097e-01</span>
<span class="co">#&gt;   [7,]  2  2.505e-01 1.910e-01</span>
<span class="co">#&gt;   [8,]  2  2.756e-01 1.741e-01</span>
<span class="co">#&gt;   [9,]  2  2.974e-01 1.586e-01</span>
<span class="co">#&gt;  [10,]  3  3.203e-01 1.445e-01</span>
<span class="co">#&gt;  [11,]  3  3.418e-01 1.317e-01</span>
<span class="co">#&gt;  [12,]  3  3.605e-01 1.200e-01</span>
<span class="co">#&gt;  [13,]  3  3.769e-01 1.093e-01</span>
<span class="co">#&gt;  [14,]  3  3.912e-01 9.960e-02</span>
<span class="co">#&gt;  [15,]  3  4.038e-01 9.075e-02</span>
<span class="co">#&gt;  [16,]  5  4.165e-01 8.269e-02</span>
<span class="co">#&gt;  [17,]  7  4.320e-01 7.535e-02</span>
<span class="co">#&gt;  [18,]  8  4.480e-01 6.865e-02</span>
<span class="co">#&gt;  [19,]  9  4.629e-01 6.255e-02</span>
<span class="co">#&gt;  [20,] 10  4.766e-01 5.700e-02</span>
<span class="co">#&gt;  [21,] 11  4.911e-01 5.193e-02</span>
<span class="co">#&gt;  [22,] 12  5.057e-01 4.732e-02</span>
<span class="co">#&gt;  [23,] 13  5.191e-01 4.312e-02</span>
<span class="co">#&gt;  [24,] 13  5.315e-01 3.929e-02</span>
<span class="co">#&gt;  [25,] 16  5.443e-01 3.580e-02</span>
<span class="co">#&gt;  [26,] 16  5.564e-01 3.262e-02</span>
<span class="co">#&gt;  [27,] 18  5.683e-01 2.972e-02</span>
<span class="co">#&gt;  [28,] 20  5.797e-01 2.708e-02</span>
<span class="co">#&gt;  [29,] 24  5.937e-01 2.467e-02</span>
<span class="co">#&gt;  [30,] 25  6.075e-01 2.248e-02</span>
<span class="co">#&gt;  [31,] 30  6.220e-01 2.048e-02</span>
<span class="co">#&gt;  [32,] 33  6.387e-01 1.866e-02</span>
<span class="co">#&gt;  [33,] 37  6.551e-01 1.701e-02</span>
<span class="co">#&gt;  [34,] 37  6.717e-01 1.549e-02</span>
<span class="co">#&gt;  [35,] 39  6.880e-01 1.412e-02</span>
<span class="co">#&gt;  [36,] 40  7.040e-01 1.286e-02</span>
<span class="co">#&gt;  [37,] 42  7.194e-01 1.172e-02</span>
<span class="co">#&gt;  [38,] 42  7.338e-01 1.068e-02</span>
<span class="co">#&gt;  [39,] 43  7.475e-01 9.731e-03</span>
<span class="co">#&gt;  [40,] 45  7.609e-01 8.867e-03</span>
<span class="co">#&gt;  [41,] 47  7.742e-01 8.079e-03</span>
<span class="co">#&gt;  [42,] 47  7.878e-01 7.361e-03</span>
<span class="co">#&gt;  [43,] 51  7.998e-01 6.707e-03</span>
<span class="co">#&gt;  [44,] 54  8.120e-01 6.111e-03</span>
<span class="co">#&gt;  [45,] 53  8.237e-01 5.569e-03</span>
<span class="co">#&gt;  [46,] 53  8.347e-01 5.074e-03</span>
<span class="co">#&gt;  [47,] 55  8.451e-01 4.623e-03</span>
<span class="co">#&gt;  [48,] 55  8.551e-01 4.212e-03</span>
<span class="co">#&gt;  [49,] 57  8.644e-01 3.838e-03</span>
<span class="co">#&gt;  [50,] 58  8.731e-01 3.497e-03</span>
<span class="co">#&gt;  [51,] 58  8.812e-01 3.187e-03</span>
<span class="co">#&gt;  [52,] 59  8.888e-01 2.903e-03</span>
<span class="co">#&gt;  [53,] 62  8.961e-01 2.646e-03</span>
<span class="co">#&gt;  [54,] 63  9.032e-01 2.410e-03</span>
<span class="co">#&gt;  [55,] 63  9.099e-01 2.196e-03</span>
<span class="co">#&gt;  [56,] 64  9.162e-01 2.001e-03</span>
<span class="co">#&gt;  [57,] 64  9.223e-01 1.823e-03</span>
<span class="co">#&gt;  [58,] 65  9.279e-01 1.661e-03</span>
<span class="co">#&gt;  [59,] 65  9.333e-01 1.514e-03</span>
<span class="co">#&gt;  [60,] 66  9.383e-01 1.379e-03</span>
<span class="co">#&gt;  [61,] 67  9.431e-01 1.257e-03</span>
<span class="co">#&gt;  [62,] 68  9.475e-01 1.145e-03</span>
<span class="co">#&gt;  [63,] 68  9.517e-01 1.043e-03</span>
<span class="co">#&gt;  [64,] 69  9.556e-01 9.507e-04</span>
<span class="co">#&gt;  [65,] 70  9.593e-01 8.663e-04</span>
<span class="co">#&gt;  [66,] 69  9.627e-01 7.893e-04</span>
<span class="co">#&gt;  [67,] 69  9.660e-01 7.192e-04</span>
<span class="co">#&gt;  [68,] 69  9.690e-01 6.553e-04</span>
<span class="co">#&gt;  [69,] 69  9.717e-01 5.971e-04</span>
<span class="co">#&gt;  [70,] 69  9.742e-01 5.441e-04</span>
<span class="co">#&gt;  [71,] 70  9.766e-01 4.957e-04</span>
<span class="co">#&gt;  [72,] 69  9.786e-01 4.517e-04</span>
<span class="co">#&gt;  [73,] 69  9.805e-01 4.116e-04</span>
<span class="co">#&gt;  [74,] 69  9.823e-01 3.750e-04</span>
<span class="co">#&gt;  [75,] 68  9.839e-01 3.417e-04</span>
<span class="co">#&gt;  [76,] 67  9.853e-01 3.113e-04</span>
<span class="co">#&gt;  [77,] 68  9.866e-01 2.837e-04</span>
<span class="co">#&gt;  [78,] 68  9.878e-01 2.585e-04</span>
<span class="co">#&gt;  [79,] 67  9.889e-01 2.355e-04</span>
<span class="co">#&gt;  [80,] 67  9.899e-01 2.146e-04</span>
<span class="co">#&gt;  [81,] 66  9.908e-01 1.955e-04</span>
<span class="co">#&gt;  [82,] 65  9.916e-01 1.782e-04</span>
<span class="co">#&gt;  [83,] 65  9.924e-01 1.623e-04</span>
<span class="co">#&gt;  [84,] 65  9.930e-01 1.479e-04</span>
<span class="co">#&gt;  [85,] 65  9.936e-01 1.348e-04</span>
<span class="co">#&gt;  [86,] 66  9.942e-01 1.228e-04</span>
<span class="co">#&gt;  [87,] 66  9.947e-01 1.119e-04</span>
<span class="co">#&gt;  [88,] 66  9.952e-01 1.019e-04</span>
<span class="co">#&gt;  [89,] 66  9.956e-01 9.289e-05</span>
<span class="co">#&gt;  [90,] 66  9.960e-01 8.464e-05</span>
<span class="co">#&gt;  [91,] 66  9.963e-01 7.712e-05</span>
<span class="co">#&gt;  [92,] 66  9.967e-01 7.027e-05</span>
<span class="co">#&gt;  [93,] 66  9.970e-01 6.402e-05</span>
<span class="co">#&gt;  [94,] 67  9.972e-01 5.834e-05</span>
<span class="co">#&gt;  [95,] 67  9.975e-01 5.315e-05</span>
<span class="co">#&gt;  [96,] 67  9.977e-01 4.843e-05</span>
<span class="co">#&gt;  [97,] 66  9.979e-01 4.413e-05</span>
<span class="co">#&gt;  [98,] 66  9.981e-01 4.021e-05</span>
<span class="co">#&gt;  [99,] 66  9.982e-01 3.664e-05</span>
<span class="co">#&gt; [100,] 66  9.984e-01 3.338e-05</span>
<span class="co">#&gt; [1] "lambda min is at location 23"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 112"</span>
<span class="co">#&gt; [1] "use 112 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 224 cells and 96 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.839285714285714"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.732142857142857"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 187 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 6.95187165775401"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 187 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 67.379679144385"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 140 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 96.4285714285714"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 140 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 33.5714285714286"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 133 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 87.9699248120301"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 133 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 68.4210526315789"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 40 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 95"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 40 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "ABCA4"  "TPM3"   "RGS4"  </span>
<span class="co">#&gt;  [8] "RXRG"   "NR5A2"  "LEFTY1"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 55"</span>
<span class="co">#&gt; [1] "Total 224 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 366 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 112 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 112 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 107 genes for training model"</span>
<span class="co">#&gt; [1] "use 107 genes 224 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3"</span>
<span class="co">#&gt; [1] "there are 112 cells in class 2_3 and 112 cells in class 1"</span>
<span class="co">#&gt; [1] "removing 11 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev    Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 3.652e-01</span>
<span class="co">#&gt;   [2,]  1  6.549e-02 3.327e-01</span>
<span class="co">#&gt;   [3,]  1  1.206e-01 3.032e-01</span>
<span class="co">#&gt;   [4,]  2  1.738e-01 2.763e-01</span>
<span class="co">#&gt;   [5,]  2  2.216e-01 2.517e-01</span>
<span class="co">#&gt;   [6,]  2  2.630e-01 2.294e-01</span>
<span class="co">#&gt;   [7,]  2  2.988e-01 2.090e-01</span>
<span class="co">#&gt;   [8,]  2  3.301e-01 1.904e-01</span>
<span class="co">#&gt;   [9,]  2  3.575e-01 1.735e-01</span>
<span class="co">#&gt;  [10,]  2  3.815e-01 1.581e-01</span>
<span class="co">#&gt;  [11,]  2  4.027e-01 1.440e-01</span>
<span class="co">#&gt;  [12,]  3  4.214e-01 1.312e-01</span>
<span class="co">#&gt;  [13,]  4  4.412e-01 1.196e-01</span>
<span class="co">#&gt;  [14,]  4  4.603e-01 1.090e-01</span>
<span class="co">#&gt;  [15,]  4  4.773e-01 9.928e-02</span>
<span class="co">#&gt;  [16,]  4  4.922e-01 9.046e-02</span>
<span class="co">#&gt;  [17,]  4  5.055e-01 8.242e-02</span>
<span class="co">#&gt;  [18,]  4  5.172e-01 7.510e-02</span>
<span class="co">#&gt;  [19,]  4  5.277e-01 6.843e-02</span>
<span class="co">#&gt;  [20,]  4  5.371e-01 6.235e-02</span>
<span class="co">#&gt;  [21,]  4  5.454e-01 5.681e-02</span>
<span class="co">#&gt;  [22,]  4  5.528e-01 5.176e-02</span>
<span class="co">#&gt;  [23,]  5  5.596e-01 4.717e-02</span>
<span class="co">#&gt;  [24,]  8  5.675e-01 4.298e-02</span>
<span class="co">#&gt;  [25,]  8  5.766e-01 3.916e-02</span>
<span class="co">#&gt;  [26,]  9  5.852e-01 3.568e-02</span>
<span class="co">#&gt;  [27,]  9  5.935e-01 3.251e-02</span>
<span class="co">#&gt;  [28,] 11  6.028e-01 2.962e-02</span>
<span class="co">#&gt;  [29,] 12  6.125e-01 2.699e-02</span>
<span class="co">#&gt;  [30,] 15  6.215e-01 2.459e-02</span>
<span class="co">#&gt;  [31,] 16  6.307e-01 2.241e-02</span>
<span class="co">#&gt;  [32,] 19  6.395e-01 2.042e-02</span>
<span class="co">#&gt;  [33,] 24  6.489e-01 1.860e-02</span>
<span class="co">#&gt;  [34,] 25  6.595e-01 1.695e-02</span>
<span class="co">#&gt;  [35,] 27  6.696e-01 1.544e-02</span>
<span class="co">#&gt;  [36,] 27  6.788e-01 1.407e-02</span>
<span class="co">#&gt;  [37,] 32  6.885e-01 1.282e-02</span>
<span class="co">#&gt;  [38,] 36  6.981e-01 1.168e-02</span>
<span class="co">#&gt;  [39,] 39  7.081e-01 1.065e-02</span>
<span class="co">#&gt;  [40,] 46  7.192e-01 9.700e-03</span>
<span class="co">#&gt;  [41,] 52  7.328e-01 8.838e-03</span>
<span class="co">#&gt;  [42,] 52  7.465e-01 8.053e-03</span>
<span class="co">#&gt;  [43,] 54  7.593e-01 7.338e-03</span>
<span class="co">#&gt;  [44,] 55  7.713e-01 6.686e-03</span>
<span class="co">#&gt;  [45,] 57  7.826e-01 6.092e-03</span>
<span class="co">#&gt;  [46,] 57  7.932e-01 5.551e-03</span>
<span class="co">#&gt;  [47,] 56  8.031e-01 5.057e-03</span>
<span class="co">#&gt;  [48,] 56  8.124e-01 4.608e-03</span>
<span class="co">#&gt;  [49,] 56  8.210e-01 4.199e-03</span>
<span class="co">#&gt;  [50,] 57  8.292e-01 3.826e-03</span>
<span class="co">#&gt;  [51,] 57  8.370e-01 3.486e-03</span>
<span class="co">#&gt;  [52,] 57  8.443e-01 3.176e-03</span>
<span class="co">#&gt;  [53,] 58  8.514e-01 2.894e-03</span>
<span class="co">#&gt;  [54,] 61  8.586e-01 2.637e-03</span>
<span class="co">#&gt;  [55,] 62  8.658e-01 2.403e-03</span>
<span class="co">#&gt;  [56,] 63  8.731e-01 2.189e-03</span>
<span class="co">#&gt;  [57,] 63  8.805e-01 1.995e-03</span>
<span class="co">#&gt;  [58,] 62  8.875e-01 1.818e-03</span>
<span class="co">#&gt;  [59,] 63  8.944e-01 1.656e-03</span>
<span class="co">#&gt;  [60,] 64  9.011e-01 1.509e-03</span>
<span class="co">#&gt;  [61,] 67  9.076e-01 1.375e-03</span>
<span class="co">#&gt;  [62,] 68  9.145e-01 1.253e-03</span>
<span class="co">#&gt;  [63,] 68  9.214e-01 1.141e-03</span>
<span class="co">#&gt;  [64,] 68  9.281e-01 1.040e-03</span>
<span class="co">#&gt;  [65,] 68  9.343e-01 9.477e-04</span>
<span class="co">#&gt;  [66,] 68  9.401e-01 8.635e-04</span>
<span class="co">#&gt;  [67,] 70  9.457e-01 7.868e-04</span>
<span class="co">#&gt;  [68,] 70  9.506e-01 7.169e-04</span>
<span class="co">#&gt;  [69,] 70  9.552e-01 6.532e-04</span>
<span class="co">#&gt;  [70,] 70  9.591e-01 5.952e-04</span>
<span class="co">#&gt;  [71,] 70  9.628e-01 5.423e-04</span>
<span class="co">#&gt;  [72,] 69  9.663e-01 4.941e-04</span>
<span class="co">#&gt;  [73,] 68  9.693e-01 4.502e-04</span>
<span class="co">#&gt;  [74,] 68  9.721e-01 4.102e-04</span>
<span class="co">#&gt;  [75,] 68  9.746e-01 3.738e-04</span>
<span class="co">#&gt;  [76,] 67  9.769e-01 3.406e-04</span>
<span class="co">#&gt;  [77,] 67  9.789e-01 3.103e-04</span>
<span class="co">#&gt;  [78,] 67  9.808e-01 2.828e-04</span>
<span class="co">#&gt;  [79,] 67  9.825e-01 2.576e-04</span>
<span class="co">#&gt;  [80,] 67  9.841e-01 2.347e-04</span>
<span class="co">#&gt;  [81,] 67  9.855e-01 2.139e-04</span>
<span class="co">#&gt;  [82,] 67  9.868e-01 1.949e-04</span>
<span class="co">#&gt;  [83,] 67  9.880e-01 1.776e-04</span>
<span class="co">#&gt;  [84,] 67  9.890e-01 1.618e-04</span>
<span class="co">#&gt;  [85,] 67  9.900e-01 1.474e-04</span>
<span class="co">#&gt;  [86,] 67  9.909e-01 1.343e-04</span>
<span class="co">#&gt;  [87,] 67  9.917e-01 1.224e-04</span>
<span class="co">#&gt;  [88,] 67  9.924e-01 1.115e-04</span>
<span class="co">#&gt;  [89,] 67  9.931e-01 1.016e-04</span>
<span class="co">#&gt;  [90,] 66  9.937e-01 9.259e-05</span>
<span class="co">#&gt;  [91,] 66  9.943e-01 8.436e-05</span>
<span class="co">#&gt;  [92,] 66  9.948e-01 7.687e-05</span>
<span class="co">#&gt;  [93,] 66  9.952e-01 7.004e-05</span>
<span class="co">#&gt;  [94,] 66  9.956e-01 6.382e-05</span>
<span class="co">#&gt;  [95,] 66  9.960e-01 5.815e-05</span>
<span class="co">#&gt;  [96,] 66  9.964e-01 5.298e-05</span>
<span class="co">#&gt;  [97,] 66  9.967e-01 4.828e-05</span>
<span class="co">#&gt;  [98,] 66  9.970e-01 4.399e-05</span>
<span class="co">#&gt;  [99,] 66  9.972e-01 4.008e-05</span>
<span class="co">#&gt; [100,] 66  9.975e-01 3.652e-05</span>
<span class="co">#&gt; [1] "lambda min is at location 16"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 112"</span>
<span class="co">#&gt; [1] "use 112 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 224 cells and 96 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.825892857142857"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.78125"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 187 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 0"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 187 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 60.427807486631"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 140 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 67.8571428571429"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 140 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 90.7142857142857"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 133 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 15.7894736842105"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 133 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 46.6165413533835"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 40 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 55"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17306"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 96"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 96 genes and 40 cells. The trained model has 96 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "FABP3"  "LCK"    "DMBX1"  "TPM3"   "RGS4"   "RXRG"  </span>
<span class="co">#&gt;  [8] "NR5A2"  "LEFTY1" "LEFTY2"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 47.5"</span></code></pre></div>
</div>
<div id="summarise-results" class="section level2">
<h2 class="hasAnchor">
<a href="#summarise-results" class="anchor"></a>2.3 Summarise results</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># display the list of result information in the LASOLDA_dat object </span>
<span class="kw">names</span>(LSOLDA_dat)
<span class="co">#&gt; [1] "Accuracy"          "ElasticNetGenes"   "Deviance"         </span>
<span class="co">#&gt; [4] "ElasticNetFit"     "LDAFit"            "predictor_S1"     </span>
<span class="co">#&gt; [7] "ElasticNetPredict" "LDAPredict"</span>
LSOLDA_dat<span class="op">$</span>ElasticNetPredict
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; [[1]][[1]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[2]]</span>
<span class="co">#&gt; [1] 6.951872</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[3]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[4]]</span>
<span class="co">#&gt; [1] 96.42857</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[5]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[6]]</span>
<span class="co">#&gt; [1] 87.96992</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[7]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[8]]</span>
<span class="co">#&gt; [1] 95</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; [[2]][[1]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[2]]</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[3]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[4]]</span>
<span class="co">#&gt; [1] 67.85714</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[5]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[6]]</span>
<span class="co">#&gt; [1] 15.78947</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[7]]</span>
<span class="co">#&gt; [1] "ElasticNet for subpop4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[8]]</span>
<span class="co">#&gt; [1] 55</span>
LSOLDA_dat<span class="op">$</span>LDAPredict
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; [[1]][[1]]</span>
<span class="co">#&gt; [1] "LDA for subpop 1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[2]]</span>
<span class="co">#&gt; [1] 67.37968</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[3]]</span>
<span class="co">#&gt; [1] "LDA for subpop 2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[4]]</span>
<span class="co">#&gt; [1] 33.57143</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[5]]</span>
<span class="co">#&gt; [1] "LDA for subpop 3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[6]]</span>
<span class="co">#&gt; [1] 68.42105</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[7]]</span>
<span class="co">#&gt; [1] "LDA for subpop 4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[8]]</span>
<span class="co">#&gt; [1] 55</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; [[2]][[1]]</span>
<span class="co">#&gt; [1] "LDA for subpop 1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[2]]</span>
<span class="co">#&gt; [1] 60.42781</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[3]]</span>
<span class="co">#&gt; [1] "LDA for subpop 2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[4]]</span>
<span class="co">#&gt; [1] 90.71429</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[5]]</span>
<span class="co">#&gt; [1] "LDA for subpop 3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[6]]</span>
<span class="co">#&gt; [1] 46.61654</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[7]]</span>
<span class="co">#&gt; [1] "LDA for subpop 4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[8]]</span>
<span class="co">#&gt; [1] 47.5</span>

<span class="co"># summary results LDA</span>
<span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co">#&gt;                 V1               V2                                names</span>
<span class="co">#&gt; 1  67.379679144385  60.427807486631 LDA for subpop 1 in target mixedpop2</span>
<span class="co">#&gt; 2 33.5714285714286 90.7142857142857 LDA for subpop 2 in target mixedpop2</span>
<span class="co">#&gt; 3 68.4210526315789 46.6165413533835 LDA for subpop 3 in target mixedpop2</span>
<span class="co">#&gt; 4               55             47.5 LDA for subpop 4 in target mixedpop2</span>

<span class="co"># summary results Lasso to show the percent of cells classified as cells belonging </span>
<span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co">#&gt;                 V1               V2</span>
<span class="co">#&gt; 1 6.95187165775401                0</span>
<span class="co">#&gt; 2 96.4285714285714 67.8571428571429</span>
<span class="co">#&gt; 3 87.9699248120301 15.7894736842105</span>
<span class="co">#&gt; 4               95               55</span>
<span class="co">#&gt;                                        names</span>
<span class="co">#&gt; 1 ElasticNet for subpop1 in target mixedpop2</span>
<span class="co">#&gt; 2 ElasticNet for subpop2 in target mixedpop2</span>
<span class="co">#&gt; 3 ElasticNet for subpop3 in target mixedpop2</span>
<span class="co">#&gt; 4 ElasticNet for subpop4 in target mixedpop2</span>

<span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)
<span class="co">#&gt; [1] 73.21429 78.12500</span>

<span class="co"># summary maximum deviance explained by the model </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)
<span class="co">#&gt; $allDeviance</span>
<span class="co">#&gt; [1] "0.5191" "0.4922"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DeviMax</span>
<span class="co">#&gt;           dat_DE$Dfd          Deviance           DEgenes</span>
<span class="co">#&gt; 1                  0            0.5191    genes_cluster1</span>
<span class="co">#&gt; 2                  1            0.5191    genes_cluster1</span>
<span class="co">#&gt; 3                  2            0.5191    genes_cluster1</span>
<span class="co">#&gt; 4                  3            0.5191    genes_cluster1</span>
<span class="co">#&gt; 5                  5            0.5191    genes_cluster1</span>
<span class="co">#&gt; 6                  7            0.5191    genes_cluster1</span>
<span class="co">#&gt; 7                  8            0.5191    genes_cluster1</span>
<span class="co">#&gt; 8                  9            0.5191    genes_cluster1</span>
<span class="co">#&gt; 9                 10            0.5191    genes_cluster1</span>
<span class="co">#&gt; 10                11            0.5191    genes_cluster1</span>
<span class="co">#&gt; 11                12            0.5191    genes_cluster1</span>
<span class="co">#&gt; 12                13            0.5191    genes_cluster1</span>
<span class="co">#&gt; 13 remaining DEgenes remaining DEgenes remaining DEgenes</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LassoGenesMax</span>
<span class="co">#&gt; NULL</span></code></pre></div>
</div>
</div>
<div id="a-complete-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-complete-workflow-of-the-scgps" class="anchor"></a>3. A complete workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given an unknown mixed population, find clusters and estimate relationship between clusters</em></p>
<div id="identify-clusters-in-a-dataset-using-core" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-core" class="anchor"></a>3.1 Identify clusters in a dataset using CORE</h2>
<p><em>(skip this step if clusters are known)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># find clustering information in an expresion data using CORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw">colnames</span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw">data.frame</span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames ) 

CORE_cluster &lt;-<span class="st"> </span><span class="kw">CORE_scGPS</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw">c</span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; [1] "Performing 1 round of filtering"</span>
<span class="co">#&gt; [1] "Identifying top variable genes"</span>
<span class="co">#&gt; [1] "Calculating distance matrix"</span>
<span class="co">#&gt; [1] "Performing hierarchical clustering"</span>
<span class="co">#&gt; [1] "Finding clustering information"</span>
<span class="co">#&gt; [1] "No more outliers detected in filtering round 1"</span>
<span class="co">#&gt; [1] "Identifying top variable genes"</span>
<span class="co">#&gt; [1] "Calculating distance matrix"</span>
<span class="co">#&gt; [1] "Performing hierarchical clustering"</span>
<span class="co">#&gt; [1] "Finding clustering information"</span>
<span class="co">#&gt; [1] "writing clustering result for run 1"</span>
<span class="co">#&gt; [1] "writing clustering result for run 2"</span>
<span class="co">#&gt; [1] "writing clustering result for run 3"</span>
<span class="co">#&gt; [1] "writing clustering result for run 4"</span>
<span class="co">#&gt; [1] "writing clustering result for run 5"</span>
<span class="co">#&gt; [1] "writing clustering result for run 6"</span>
<span class="co">#&gt; [1] "writing clustering result for run 7"</span>
<span class="co">#&gt; [1] "writing clustering result for run 8"</span>
<span class="co">#&gt; [1] "writing clustering result for run 9"</span>
<span class="co">#&gt; [1] "writing clustering result for run 10"</span>
<span class="co">#&gt; [1] "writing clustering result for run 11"</span>
<span class="co">#&gt; [1] "writing clustering result for run 12"</span>
<span class="co">#&gt; [1] "writing clustering result for run 13"</span>
<span class="co">#&gt; [1] "writing clustering result for run 14"</span>
<span class="co">#&gt; [1] "writing clustering result for run 15"</span>
<span class="co">#&gt; [1] "writing clustering result for run 16"</span>
<span class="co">#&gt; [1] "writing clustering result for run 17"</span>
<span class="co">#&gt; [1] "writing clustering result for run 18"</span>
<span class="co">#&gt; [1] "writing clustering result for run 19"</span>
<span class="co">#&gt; [1] "writing clustering result for run 20"</span>
<span class="co">#&gt; [1] "writing clustering result for run 21"</span>
<span class="co">#&gt; [1] "writing clustering result for run 22"</span>
<span class="co">#&gt; [1] "writing clustering result for run 23"</span>
<span class="co">#&gt; [1] "writing clustering result for run 24"</span>
<span class="co">#&gt; [1] "writing clustering result for run 25"</span>
<span class="co">#&gt; [1] "writing clustering result for run 26"</span>
<span class="co">#&gt; [1] "writing clustering result for run 27"</span>
<span class="co">#&gt; [1] "writing clustering result for run 28"</span>
<span class="co">#&gt; [1] "writing clustering result for run 29"</span>
<span class="co">#&gt; [1] "writing clustering result for run 30"</span>
<span class="co">#&gt; [1] "writing clustering result for run 31"</span>
<span class="co">#&gt; [1] "writing clustering result for run 32"</span>
<span class="co">#&gt; [1] "writing clustering result for run 33"</span>
<span class="co">#&gt; [1] "writing clustering result for run 34"</span>
<span class="co">#&gt; [1] "writing clustering result for run 35"</span>
<span class="co">#&gt; [1] "writing clustering result for run 36"</span>
<span class="co">#&gt; [1] "writing clustering result for run 37"</span>
<span class="co">#&gt; [1] "writing clustering result for run 38"</span>
<span class="co">#&gt; [1] "writing clustering result for run 39"</span>
<span class="co">#&gt; [1] "writing clustering result for run 40"</span>
<span class="co">#&gt; [1] "Done clustering, moving to stability calculation..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done finding optimal clustering..."</span>

<span class="co"># to update the clustering information, users can ...</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw">which</span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)

clustering_after_outlier_removal &lt;-<span class="st"> </span><span class="kw">unname</span>(<span class="kw">unlist</span>(CORE_cluster<span class="op">$</span>Cluster[[optimal_index]]))
corresponding_cells_after_outlier_removal &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>cellsForClustering
original_cells_before_removal &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">2</span>]
corresponding_index &lt;-<span class="st"> </span><span class="kw">match</span>(corresponding_cells_after_outlier_removal, original_cells_before_removal )
<span class="co"># check the matching </span>
<span class="kw">identical</span>(<span class="kw">as.character</span>(original_cells_before_removal[corresponding_index]),corresponding_cells_after_outlier_removal)
<span class="co">#&gt; [1] TRUE</span>
<span class="co"># create new object with the new clustering after removing outliers </span>
mixedpop2_post_clustering &lt;-<span class="st"> </span>mixedpop2[,corresponding_index]
<span class="kw">colData</span>(mixedpop2_post_clustering)[,<span class="dv">1</span>] &lt;-<span class="st"> </span>clustering_after_outlier_removal</code></pre></div>
</div>
<div id="identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution" class="anchor"></a>3.2 Identify clusters in a dataset using SCORE (Stable Clustering at Optimal REsolution)</h2>
<p><em>(skip this step if clusters are known)</em> <em>(SCORE aims to get stable subpopulation results, by introducing bagging aggregation and bootstrapping to the CORE algorithm)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># find clustering information in an expresion data using SCORE</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw">colnames</span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw">data.frame</span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw">NewscGPS</span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames ) 

SCORE_test &lt;-<span class="st"> </span><span class="kw">CORE_scGPS_bagging</span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw">c</span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>,
                          <span class="dt">bagging_run =</span> <span class="dv">20</span>, <span class="dt">subsample_proportion =</span> .<span class="dv">8</span>)
<span class="co">#&gt; [1] "Performing 1 round of filtering"</span>
<span class="co">#&gt; [1] "Identifying top variable genes"</span>
<span class="co">#&gt; [1] "Calculating distance matrix"</span>
<span class="co">#&gt; [1] "Performing hierarchical clustering"</span>
<span class="co">#&gt; [1] "Finding clustering information"</span>
<span class="co">#&gt; [1] "No more outliers detected in filtering round 1"</span>
<span class="co">#&gt; [1] "Identifying top variable genes"</span>
<span class="co">#&gt; [1] "Calculating distance matrix"</span>
<span class="co">#&gt; [1] "Performing hierarchical clustering"</span>
<span class="co">#&gt; [1] "Finding clustering information"</span>
<span class="co">#&gt; [1] "500 cells left after filtering"</span>
<span class="co">#&gt; [1] "Running 20 bagging runs, with 0.8 subsampling..."</span>
<span class="co">#&gt; [1] "Done clustering, moving to stability calculation..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span></code></pre></div>
</div>
<div id="visualise-all-cluster-results-in-all-iterations" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-all-cluster-results-in-all-iterations" class="anchor"></a>3.3 Visualise all cluster results in all iterations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##3.2.1 plot CORE clustering 
<span class="kw">plot_CORE</span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster) <span class="co">#plot all clustering bars </span></code></pre></div>
<p><img src="man/figures/visualisation-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#extract optimal index identified by CORE_scGPS</span>
key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height
optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes
optimal_index =<span class="st"> </span><span class="kw">which</span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)
<span class="co">#plot one optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> CORE_cluster<span class="op">$</span>tree,
                   <span class="dt">optimal_cluster =</span> <span class="kw">unlist</span>(CORE_cluster<span class="op">$</span>Cluster[optimal_index]), <span class="dt">shift =</span> <span class="op">-</span><span class="dv">2000</span>)
<span class="co">#&gt; [1] "Ordering and assigning labels..."</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 128 270  NA</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] 128 270 393</span>
<span class="co">#&gt; [1] "Plotting the colored dendrogram now...."</span></code></pre></div>
<p><img src="man/figures/visualisation-2.png" width="700"></p>
<pre><code>#&gt; [1] "Plotting the bar underneath now...."
# you can customise the cluster color bars (provide color_branch values)
plot_CORE(CORE_cluster$tree, CORE_cluster$Cluster, color_branch = c("#208eb7", "#6ce9d3", "#1c5e39", "#8fca40", "#154975", "#b1c8eb"))</code></pre>
<p><img src="man/figures/visualisation-3.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
##3.2.2 plot SCORE clustering 
<span class="kw">plot_CORE</span>(SCORE_test<span class="op">$</span>tree, <span class="dt">list_clusters =</span> SCORE_test<span class="op">$</span>Cluster)<span class="co">#plot all clustering bars </span></code></pre></div>
<p><img src="man/figures/visualisation-4.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#plot one stable optimal clustering bar</span>
<span class="kw">plot_optimal_CORE</span>(<span class="dt">original_tree=</span> SCORE_test<span class="op">$</span>tree,
                   <span class="dt">optimal_cluster =</span> <span class="kw">unlist</span>(SCORE_test<span class="op">$</span>Cluster[SCORE_test<span class="op">$</span>optimal_index]), <span class="dt">shift =</span> <span class="op">-</span><span class="dv">100</span>)
<span class="co">#&gt; [1] "Ordering and assigning labels..."</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 128 270  NA  NA</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] 128 270 333  NA</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; [1] 128 270 333 440</span>
<span class="co">#&gt; [1] "Plotting the colored dendrogram now...."</span></code></pre></div>
<p><img src="man/figures/visualisation-5.png" width="700"></p>
<pre><code>#&gt; [1] "Plotting the bar underneath now...."</code></pre>
</div>
<div id="compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="anchor"></a>3.4 Compare clustering results with other dimensional reduction methods (e.g., CIDR)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(cidr)
t &lt;-<span class="st"> </span><span class="kw">CIDR_scGPS</span>(<span class="dt">expression.matrix=</span><span class="kw">assay</span>(mixedpop2))
<span class="co">#&gt; [1] "building cidr object..."</span>
<span class="co">#&gt; [1] "determine dropout candidates..."</span>
<span class="co">#&gt; [1] "determine the imputation weighting threshold..."</span>
<span class="co">#&gt; [1] "computes the _CIDR_ dissimilarity matrix..."</span>
<span class="co">#&gt; [1] "PCA plot with proportion of variance explained..."</span></code></pre></div>
<p><img src="man/figures/compare_clustering-1.png" width="700"></p>
<pre><code>#&gt; [1] "find the number of PC..."
#&gt; [1] "perform clustering..."
p2 &lt;-plotReduced_scGPS(t, color_fac = factor(colData(mixedpop2)[,1]),palletes =1:length(unique(colData(mixedpop2)[,1])))
p2</code></pre>
<p><img src="man/figures/compare_clustering-2.png" width="700"></p>
</div>
<div id="find-gene-markers-and-annotate-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#find-gene-markers-and-annotate-clusters" class="anchor"></a>3.5 Find gene markers and annotate clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#load gene list (this can be any lists of user-selected genes)</span>
genes &lt;-GeneList
genes &lt;-genes<span class="op">$</span>Merged_unique

<span class="co">#the gene list can also be objectively identified by differential expression analysis</span>
<span class="co">#cluster information is requied for findMarkers_scGPS. Here, we use CORE results. </span>

<span class="co">#colData(mixedpop2)[,1] &lt;- unlist(SCORE_test$Cluster[SCORE_test$optimal_index])</span>

<span class="kw">suppressMessages</span>(<span class="kw">library</span>(locfit))
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(DESeq))

DEgenes &lt;-<span class="st"> </span><span class="kw">findMarkers_scGPS</span>(<span class="dt">expression_matrix=</span><span class="kw">assay</span>(mixedpop2), <span class="dt">cluster =</span> <span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>],
                             <span class="dt">selected_cluster=</span><span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])) 
<span class="co">#&gt; [1] "Start estimate dispersions for cluster 1..."</span>
<span class="co">#&gt; [1] "Done estimate dispersions. Start nbinom test for cluster 1..."</span>
<span class="co">#&gt; [1] "Done nbinom test for cluster 1 ..."</span>
<span class="co">#&gt; [1] "Adjust foldchange by subtracting basemean to 1..."</span>
<span class="co">#&gt; [1] "Start estimate dispersions for cluster 2..."</span>
<span class="co">#&gt; [1] "Done estimate dispersions. Start nbinom test for cluster 2..."</span>
<span class="co">#&gt; [1] "Done nbinom test for cluster 2 ..."</span>
<span class="co">#&gt; [1] "Adjust foldchange by subtracting basemean to 1..."</span>
<span class="co">#&gt; [1] "Start estimate dispersions for cluster 3..."</span>
<span class="co">#&gt; [1] "Done estimate dispersions. Start nbinom test for cluster 3..."</span>
<span class="co">#&gt; [1] "Done nbinom test for cluster 3 ..."</span>
<span class="co">#&gt; [1] "Adjust foldchange by subtracting basemean to 1..."</span>
<span class="co">#&gt; [1] "Start estimate dispersions for cluster 4..."</span>
<span class="co">#&gt; [1] "Done estimate dispersions. Start nbinom test for cluster 4..."</span>
<span class="co">#&gt; [1] "Done nbinom test for cluster 4 ..."</span>
<span class="co">#&gt; [1] "Adjust foldchange by subtracting basemean to 1..."</span>

<span class="co">#the output contains dataframes for each cluster.</span>
<span class="co">#the data frame contains all genes, sorted by p-values </span>
<span class="kw">names</span>(DEgenes)
<span class="co">#&gt; [1] "DE_Subpop1vsRemaining" "DE_Subpop2vsRemaining" "DE_Subpop3vsRemaining"</span>
<span class="co">#&gt; [4] "DE_Subpop4vsRemaining"</span>

<span class="co">#you can annotate the identified clusters </span>
DEgeneList_1vsOthers &lt;-<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id

<span class="co">#users need to check the format of the gene input to make sure they are consistent to </span>
<span class="co">#the gene names in the expression matrix </span>

<span class="co">#the following command saves the file "PathwayEnrichment.xlsx" to the working dir</span>
<span class="co">#use 500 top DE genes </span>
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(DOSE))
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(ReactomePA))
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(clusterProfiler))
enrichment_test &lt;-<span class="st"> </span><span class="kw">annotate_scGPS</span>(DEgeneList_1vsOthers[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>], <span class="dt">pvalueCutoff=</span><span class="fl">0.05</span>, <span class="dt">gene_symbol=</span><span class="ot">TRUE</span>)
<span class="co">#&gt; [1] "Original gene number in geneList"</span>
<span class="co">#&gt; [1] 500</span>
<span class="co">#&gt; [1] "Number of genes successfully converted"</span>
<span class="co">#&gt; [1] 487</span>

<span class="co">#the enrichment outputs can be displayed by running</span>
<span class="kw"><a href="http://www.rdocumentation.org/packages/ReactomePA/topics/reexports">dotplot</a></span>(enrichment_test, <span class="dt">showCategory=</span><span class="dv">15</span>)</code></pre></div>
<p><img src="man/figures/find_markers-1.png" width="700"></p>
</div>
</div>
<div id="relationship-between-clusters-within-one-sample-or-between-two-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples" class="anchor"></a>4. Relationship between clusters within one sample or between two samples</h1>
<p><em>The purpose of this workflow is to solve the following task: given one or two unknown mixed population(s) and clusters in each mixed population, estimate and visualise relationship between clusters</em></p>
<div id="start-the-scgps-prediction-to-find-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#start-the-scgps-prediction-to-find-relationship-between-clusters" class="anchor"></a>4.1 Start the scGPS prediction to find relationship between clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#select a subpopulation, and input gene list </span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="co">#note make sure the format for genes input here is the same to the format for genes in the mixedpop1 and mixedpop2  </span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>]

<span class="co">#run the test bootstrap with nboots = 2 runs</span>

cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]

<span class="kw">sink</span>(<span class="st">"temp"</span>)
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID, 
     <span class="dt">listData =</span> <span class="kw">list</span>(),
     <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 224 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 366 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 112 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 112 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 482 genes for training model"</span>
<span class="co">#&gt; [1] "use 482 genes 224 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3"</span>
<span class="co">#&gt; [1] "there are 112 cells in class 2_3 and 112 cells in class 1"</span>
<span class="co">#&gt; [1] "removing 4 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 0.298300</span>
<span class="co">#&gt;   [2,]  2  2.551e-02 0.284700</span>
<span class="co">#&gt;   [3,]  3  5.542e-02 0.271800</span>
<span class="co">#&gt;   [4,]  3  8.453e-02 0.259400</span>
<span class="co">#&gt;   [5,]  4  1.128e-01 0.247600</span>
<span class="co">#&gt;   [6,]  4  1.397e-01 0.236400</span>
<span class="co">#&gt;   [7,]  4  1.648e-01 0.225600</span>
<span class="co">#&gt;   [8,]  5  1.885e-01 0.215400</span>
<span class="co">#&gt;   [9,]  5  2.116e-01 0.205600</span>
<span class="co">#&gt;  [10,]  5  2.332e-01 0.196200</span>
<span class="co">#&gt;  [11,]  5  2.535e-01 0.187300</span>
<span class="co">#&gt;  [12,]  5  2.726e-01 0.178800</span>
<span class="co">#&gt;  [13,]  5  2.905e-01 0.170700</span>
<span class="co">#&gt;  [14,]  5  3.075e-01 0.162900</span>
<span class="co">#&gt;  [15,]  6  3.237e-01 0.155500</span>
<span class="co">#&gt;  [16,]  6  3.404e-01 0.148400</span>
<span class="co">#&gt;  [17,]  7  3.569e-01 0.141700</span>
<span class="co">#&gt;  [18,]  7  3.729e-01 0.135300</span>
<span class="co">#&gt;  [19,]  8  3.883e-01 0.129100</span>
<span class="co">#&gt;  [20,]  8  4.030e-01 0.123200</span>
<span class="co">#&gt;  [21,]  9  4.170e-01 0.117600</span>
<span class="co">#&gt;  [22,] 10  4.306e-01 0.112300</span>
<span class="co">#&gt;  [23,] 11  4.437e-01 0.107200</span>
<span class="co">#&gt;  [24,] 12  4.571e-01 0.102300</span>
<span class="co">#&gt;  [25,] 13  4.714e-01 0.097670</span>
<span class="co">#&gt;  [26,] 14  4.859e-01 0.093230</span>
<span class="co">#&gt;  [27,] 14  5.002e-01 0.088990</span>
<span class="co">#&gt;  [28,] 15  5.139e-01 0.084950</span>
<span class="co">#&gt;  [29,] 16  5.270e-01 0.081090</span>
<span class="co">#&gt;  [30,] 18  5.401e-01 0.077400</span>
<span class="co">#&gt;  [31,] 18  5.527e-01 0.073880</span>
<span class="co">#&gt;  [32,] 19  5.647e-01 0.070530</span>
<span class="co">#&gt;  [33,] 20  5.766e-01 0.067320</span>
<span class="co">#&gt;  [34,] 22  5.881e-01 0.064260</span>
<span class="co">#&gt;  [35,] 22  5.993e-01 0.061340</span>
<span class="co">#&gt;  [36,] 23  6.101e-01 0.058550</span>
<span class="co">#&gt;  [37,] 23  6.207e-01 0.055890</span>
<span class="co">#&gt;  [38,] 23  6.308e-01 0.053350</span>
<span class="co">#&gt;  [39,] 23  6.405e-01 0.050920</span>
<span class="co">#&gt;  [40,] 25  6.500e-01 0.048610</span>
<span class="co">#&gt;  [41,] 25  6.592e-01 0.046400</span>
<span class="co">#&gt;  [42,] 25  6.680e-01 0.044290</span>
<span class="co">#&gt;  [43,] 25  6.764e-01 0.042280</span>
<span class="co">#&gt;  [44,] 25  6.845e-01 0.040360</span>
<span class="co">#&gt;  [45,] 25  6.923e-01 0.038520</span>
<span class="co">#&gt;  [46,] 26  6.998e-01 0.036770</span>
<span class="co">#&gt;  [47,] 28  7.072e-01 0.035100</span>
<span class="co">#&gt;  [48,] 31  7.146e-01 0.033510</span>
<span class="co">#&gt;  [49,] 31  7.219e-01 0.031980</span>
<span class="co">#&gt;  [50,] 34  7.290e-01 0.030530</span>
<span class="co">#&gt;  [51,] 36  7.363e-01 0.029140</span>
<span class="co">#&gt;  [52,] 40  7.437e-01 0.027820</span>
<span class="co">#&gt;  [53,] 44  7.512e-01 0.026550</span>
<span class="co">#&gt;  [54,] 46  7.589e-01 0.025350</span>
<span class="co">#&gt;  [55,] 48  7.667e-01 0.024190</span>
<span class="co">#&gt;  [56,] 48  7.746e-01 0.023090</span>
<span class="co">#&gt;  [57,] 50  7.821e-01 0.022040</span>
<span class="co">#&gt;  [58,] 50  7.895e-01 0.021040</span>
<span class="co">#&gt;  [59,] 52  7.968e-01 0.020090</span>
<span class="co">#&gt;  [60,] 53  8.042e-01 0.019170</span>
<span class="co">#&gt;  [61,] 55  8.114e-01 0.018300</span>
<span class="co">#&gt;  [62,] 55  8.184e-01 0.017470</span>
<span class="co">#&gt;  [63,] 54  8.251e-01 0.016680</span>
<span class="co">#&gt;  [64,] 55  8.316e-01 0.015920</span>
<span class="co">#&gt;  [65,] 58  8.380e-01 0.015190</span>
<span class="co">#&gt;  [66,] 59  8.442e-01 0.014500</span>
<span class="co">#&gt;  [67,] 62  8.503e-01 0.013840</span>
<span class="co">#&gt;  [68,] 62  8.565e-01 0.013220</span>
<span class="co">#&gt;  [69,] 63  8.626e-01 0.012610</span>
<span class="co">#&gt;  [70,] 66  8.684e-01 0.012040</span>
<span class="co">#&gt;  [71,] 66  8.742e-01 0.011490</span>
<span class="co">#&gt;  [72,] 71  8.798e-01 0.010970</span>
<span class="co">#&gt;  [73,] 71  8.853e-01 0.010470</span>
<span class="co">#&gt;  [74,] 74  8.905e-01 0.009997</span>
<span class="co">#&gt;  [75,] 75  8.955e-01 0.009542</span>
<span class="co">#&gt;  [76,] 75  9.003e-01 0.009109</span>
<span class="co">#&gt;  [77,] 76  9.050e-01 0.008695</span>
<span class="co">#&gt;  [78,] 77  9.094e-01 0.008299</span>
<span class="co">#&gt;  [79,] 78  9.137e-01 0.007922</span>
<span class="co">#&gt;  [80,] 77  9.177e-01 0.007562</span>
<span class="co">#&gt;  [81,] 76  9.215e-01 0.007218</span>
<span class="co">#&gt;  [82,] 77  9.251e-01 0.006890</span>
<span class="co">#&gt;  [83,] 79  9.286e-01 0.006577</span>
<span class="co">#&gt;  [84,] 79  9.319e-01 0.006278</span>
<span class="co">#&gt;  [85,] 79  9.350e-01 0.005993</span>
<span class="co">#&gt;  [86,] 80  9.380e-01 0.005721</span>
<span class="co">#&gt;  [87,] 80  9.409e-01 0.005461</span>
<span class="co">#&gt;  [88,] 81  9.436e-01 0.005212</span>
<span class="co">#&gt;  [89,] 81  9.462e-01 0.004975</span>
<span class="co">#&gt;  [90,] 78  9.486e-01 0.004749</span>
<span class="co">#&gt;  [91,] 79  9.510e-01 0.004533</span>
<span class="co">#&gt;  [92,] 79  9.532e-01 0.004327</span>
<span class="co">#&gt;  [93,] 80  9.554e-01 0.004131</span>
<span class="co">#&gt;  [94,] 81  9.574e-01 0.003943</span>
<span class="co">#&gt;  [95,] 81  9.594e-01 0.003764</span>
<span class="co">#&gt;  [96,] 81  9.612e-01 0.003593</span>
<span class="co">#&gt;  [97,] 81  9.630e-01 0.003429</span>
<span class="co">#&gt;  [98,] 82  9.647e-01 0.003273</span>
<span class="co">#&gt;  [99,] 82  9.663e-01 0.003125</span>
<span class="co">#&gt; [100,] 83  9.679e-01 0.002983</span>
<span class="co">#&gt; [1] "lambda min is at location 46"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 112"</span>
<span class="co">#&gt; [1] "use 112 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 224 cells and 478 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.870535714285714"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.848214285714286"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16924"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 478"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 187 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 58.8235294117647"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 187 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 82.3529411764706"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16924"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 478"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 140 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 30.7142857142857"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 140 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 68.5714285714286"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16924"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 478"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 133 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 17.2932330827068"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 133 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 71.4285714285714"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16924"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 478"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 40 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 15"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 478 genes and 40 cells. The trained model has 478 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "NPPB"     "CAPZB"   </span>
<span class="co">#&gt;  [7] "EPHB2"    "HMGN2"    "MARCKSL1" "BSDC1"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 62.5"</span>
<span class="co">#&gt; [1] "Total 224 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 366 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 112 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 112 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 482 genes for training model"</span>
<span class="co">#&gt; [1] "use 482 genes 224 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3"</span>
<span class="co">#&gt; [1] "there are 112 cells in class 2_3 and 112 cells in class 1"</span>
<span class="co">#&gt; [1] "removing 10 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 0.283300</span>
<span class="co">#&gt;   [2,]  3  2.340e-02 0.270500</span>
<span class="co">#&gt;   [3,]  4  5.700e-02 0.258200</span>
<span class="co">#&gt;   [4,]  5  8.831e-02 0.246400</span>
<span class="co">#&gt;   [5,]  6  1.187e-01 0.235200</span>
<span class="co">#&gt;   [6,]  6  1.473e-01 0.224500</span>
<span class="co">#&gt;   [7,]  7  1.744e-01 0.214300</span>
<span class="co">#&gt;   [8,]  7  1.997e-01 0.204600</span>
<span class="co">#&gt;   [9,]  8  2.240e-01 0.195300</span>
<span class="co">#&gt;  [10,]  8  2.468e-01 0.186400</span>
<span class="co">#&gt;  [11,]  8  2.682e-01 0.177900</span>
<span class="co">#&gt;  [12,]  9  2.891e-01 0.169900</span>
<span class="co">#&gt;  [13,]  9  3.089e-01 0.162100</span>
<span class="co">#&gt;  [14,]  9  3.275e-01 0.154800</span>
<span class="co">#&gt;  [15,]  9  3.451e-01 0.147700</span>
<span class="co">#&gt;  [16,] 10  3.618e-01 0.141000</span>
<span class="co">#&gt;  [17,] 11  3.780e-01 0.134600</span>
<span class="co">#&gt;  [18,] 11  3.934e-01 0.128500</span>
<span class="co">#&gt;  [19,] 11  4.080e-01 0.122600</span>
<span class="co">#&gt;  [20,] 11  4.219e-01 0.117100</span>
<span class="co">#&gt;  [21,] 11  4.350e-01 0.111800</span>
<span class="co">#&gt;  [22,] 11  4.475e-01 0.106700</span>
<span class="co">#&gt;  [23,] 13  4.599e-01 0.101800</span>
<span class="co">#&gt;  [24,] 14  4.720e-01 0.097200</span>
<span class="co">#&gt;  [25,] 15  4.842e-01 0.092780</span>
<span class="co">#&gt;  [26,] 17  4.972e-01 0.088560</span>
<span class="co">#&gt;  [27,] 19  5.106e-01 0.084540</span>
<span class="co">#&gt;  [28,] 19  5.249e-01 0.080690</span>
<span class="co">#&gt;  [29,] 19  5.384e-01 0.077030</span>
<span class="co">#&gt;  [30,] 20  5.512e-01 0.073520</span>
<span class="co">#&gt;  [31,] 21  5.637e-01 0.070180</span>
<span class="co">#&gt;  [32,] 22  5.758e-01 0.066990</span>
<span class="co">#&gt;  [33,] 22  5.875e-01 0.063950</span>
<span class="co">#&gt;  [34,] 22  5.986e-01 0.061040</span>
<span class="co">#&gt;  [35,] 24  6.095e-01 0.058270</span>
<span class="co">#&gt;  [36,] 25  6.204e-01 0.055620</span>
<span class="co">#&gt;  [37,] 27  6.313e-01 0.053090</span>
<span class="co">#&gt;  [38,] 30  6.427e-01 0.050680</span>
<span class="co">#&gt;  [39,] 30  6.542e-01 0.048370</span>
<span class="co">#&gt;  [40,] 30  6.652e-01 0.046180</span>
<span class="co">#&gt;  [41,] 30  6.757e-01 0.044080</span>
<span class="co">#&gt;  [42,] 31  6.859e-01 0.042070</span>
<span class="co">#&gt;  [43,] 33  6.958e-01 0.040160</span>
<span class="co">#&gt;  [44,] 36  7.057e-01 0.038340</span>
<span class="co">#&gt;  [45,] 37  7.154e-01 0.036590</span>
<span class="co">#&gt;  [46,] 37  7.248e-01 0.034930</span>
<span class="co">#&gt;  [47,] 40  7.340e-01 0.033340</span>
<span class="co">#&gt;  [48,] 41  7.433e-01 0.031830</span>
<span class="co">#&gt;  [49,] 41  7.522e-01 0.030380</span>
<span class="co">#&gt;  [50,] 43  7.610e-01 0.029000</span>
<span class="co">#&gt;  [51,] 43  7.695e-01 0.027680</span>
<span class="co">#&gt;  [52,] 43  7.776e-01 0.026420</span>
<span class="co">#&gt;  [53,] 45  7.856e-01 0.025220</span>
<span class="co">#&gt;  [54,] 47  7.935e-01 0.024080</span>
<span class="co">#&gt;  [55,] 48  8.014e-01 0.022980</span>
<span class="co">#&gt;  [56,] 48  8.090e-01 0.021940</span>
<span class="co">#&gt;  [57,] 49  8.164e-01 0.020940</span>
<span class="co">#&gt;  [58,] 49  8.236e-01 0.019990</span>
<span class="co">#&gt;  [59,] 49  8.305e-01 0.019080</span>
<span class="co">#&gt;  [60,] 49  8.371e-01 0.018210</span>
<span class="co">#&gt;  [61,] 49  8.436e-01 0.017380</span>
<span class="co">#&gt;  [62,] 50  8.498e-01 0.016590</span>
<span class="co">#&gt;  [63,] 50  8.559e-01 0.015840</span>
<span class="co">#&gt;  [64,] 52  8.617e-01 0.015120</span>
<span class="co">#&gt;  [65,] 52  8.674e-01 0.014430</span>
<span class="co">#&gt;  [66,] 54  8.729e-01 0.013780</span>
<span class="co">#&gt;  [67,] 54  8.782e-01 0.013150</span>
<span class="co">#&gt;  [68,] 55  8.832e-01 0.012550</span>
<span class="co">#&gt;  [69,] 56  8.881e-01 0.011980</span>
<span class="co">#&gt;  [70,] 56  8.928e-01 0.011440</span>
<span class="co">#&gt;  [71,] 58  8.974e-01 0.010920</span>
<span class="co">#&gt;  [72,] 59  9.018e-01 0.010420</span>
<span class="co">#&gt;  [73,] 59  9.061e-01 0.009948</span>
<span class="co">#&gt;  [74,] 60  9.101e-01 0.009496</span>
<span class="co">#&gt;  [75,] 62  9.140e-01 0.009064</span>
<span class="co">#&gt;  [76,] 62  9.178e-01 0.008652</span>
<span class="co">#&gt;  [77,] 62  9.214e-01 0.008259</span>
<span class="co">#&gt;  [78,] 62  9.249e-01 0.007884</span>
<span class="co">#&gt;  [79,] 62  9.282e-01 0.007525</span>
<span class="co">#&gt;  [80,] 62  9.314e-01 0.007183</span>
<span class="co">#&gt;  [81,] 62  9.345e-01 0.006857</span>
<span class="co">#&gt;  [82,] 62  9.374e-01 0.006545</span>
<span class="co">#&gt;  [83,] 62  9.402e-01 0.006248</span>
<span class="co">#&gt;  [84,] 63  9.429e-01 0.005964</span>
<span class="co">#&gt;  [85,] 64  9.454e-01 0.005693</span>
<span class="co">#&gt;  [86,] 66  9.479e-01 0.005434</span>
<span class="co">#&gt;  [87,] 67  9.502e-01 0.005187</span>
<span class="co">#&gt;  [88,] 67  9.525e-01 0.004951</span>
<span class="co">#&gt;  [89,] 68  9.546e-01 0.004726</span>
<span class="co">#&gt;  [90,] 68  9.567e-01 0.004511</span>
<span class="co">#&gt;  [91,] 68  9.587e-01 0.004306</span>
<span class="co">#&gt;  [92,] 68  9.605e-01 0.004111</span>
<span class="co">#&gt;  [93,] 68  9.623e-01 0.003924</span>
<span class="co">#&gt;  [94,] 69  9.640e-01 0.003745</span>
<span class="co">#&gt;  [95,] 69  9.657e-01 0.003575</span>
<span class="co">#&gt;  [96,] 69  9.672e-01 0.003413</span>
<span class="co">#&gt;  [97,] 70  9.687e-01 0.003258</span>
<span class="co">#&gt;  [98,] 70  9.701e-01 0.003110</span>
<span class="co">#&gt;  [99,] 70  9.715e-01 0.002968</span>
<span class="co">#&gt; [100,] 70  9.728e-01 0.002833</span>
<span class="co">#&gt; [1] "lambda min is at location 81"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 112"</span>
<span class="co">#&gt; [1] "use 112 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 224 cells and 472 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.883928571428571"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.84375"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 187 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 41.7112299465241"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16936"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 466"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 187 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 33.6898395721925"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 140 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 80"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16936"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 466"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 140 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 59.2857142857143"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 133 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 67.6691729323308"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16936"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 466"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 133 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 6.76691729323308"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16930"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 472"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 40 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 82.5"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 16936"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 466"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 472 genes and 40 cells. The trained model has 472 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"    "MRPL20"   "ERRFI1"   "MAD2L2"   "CAPZB"    "EPHB2"   </span>
<span class="co">#&gt;  [7] "HMGN2"    "MARCKSL1" "BSDC1"    "CDC20"   </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 37.5"</span>

<span class="kw">sink</span>()</code></pre></div>
</div>
<div id="display-summary-results-for-the-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#display-summary-results-for-the-prediction" class="anchor"></a>4.2 Display summary results for the prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#get the number of rows for the summary matrix </span>
row_cluster &lt;-<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

<span class="co">#summary results LDA to to show the percent of cells classified as cells belonging by LDA classifier </span>
<span class="kw">summary_prediction_lda</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster )
<span class="co">#&gt;                 V1               V2                                names</span>
<span class="co">#&gt; 1 82.3529411764706 33.6898395721925 LDA for subpop 1 in target mixedpop2</span>
<span class="co">#&gt; 2 68.5714285714286 59.2857142857143 LDA for subpop 2 in target mixedpop2</span>
<span class="co">#&gt; 3 71.4285714285714 6.76691729323308 LDA for subpop 3 in target mixedpop2</span>
<span class="co">#&gt; 4             62.5             37.5 LDA for subpop 4 in target mixedpop2</span>

<span class="co">#summary results Lasso to show the percent of cells classified as cells belonging by Lasso classifier</span>
<span class="kw">summary_prediction_lasso</span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster)
<span class="co">#&gt;                 V1               V2</span>
<span class="co">#&gt; 1 58.8235294117647 41.7112299465241</span>
<span class="co">#&gt; 2 30.7142857142857               80</span>
<span class="co">#&gt; 3 17.2932330827068 67.6691729323308</span>
<span class="co">#&gt; 4               15             82.5</span>
<span class="co">#&gt;                                        names</span>
<span class="co">#&gt; 1 ElasticNet for subpop1 in target mixedpop2</span>
<span class="co">#&gt; 2 ElasticNet for subpop2 in target mixedpop2</span>
<span class="co">#&gt; 3 ElasticNet for subpop3 in target mixedpop2</span>
<span class="co">#&gt; 4 ElasticNet for subpop4 in target mixedpop2</span>

<span class="co"># summary maximum deviance explained by the model during the model training </span>
<span class="kw">summary_deviance</span>(<span class="dt">object =</span> LSOLDA_dat)
<span class="co">#&gt; $allDeviance</span>
<span class="co">#&gt; [1] "0.6998" "0.9345"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DeviMax</span>
<span class="co">#&gt;           dat_DE$Dfd          Deviance           DEgenes</span>
<span class="co">#&gt; 1                  0            0.9345    genes_cluster1</span>
<span class="co">#&gt; 2                  3            0.9345    genes_cluster1</span>
<span class="co">#&gt; 3                  4            0.9345    genes_cluster1</span>
<span class="co">#&gt; 4                  5            0.9345    genes_cluster1</span>
<span class="co">#&gt; 5                  6            0.9345    genes_cluster1</span>
<span class="co">#&gt; 6                  7            0.9345    genes_cluster1</span>
<span class="co">#&gt; 7                  8            0.9345    genes_cluster1</span>
<span class="co">#&gt; 8                  9            0.9345    genes_cluster1</span>
<span class="co">#&gt; 9                 10            0.9345    genes_cluster1</span>
<span class="co">#&gt; 10                11            0.9345    genes_cluster1</span>
<span class="co">#&gt; 11                13            0.9345    genes_cluster1</span>
<span class="co">#&gt; 12                14            0.9345    genes_cluster1</span>
<span class="co">#&gt; 13                15            0.9345    genes_cluster1</span>
<span class="co">#&gt; 14                17            0.9345    genes_cluster1</span>
<span class="co">#&gt; 15                19            0.9345    genes_cluster1</span>
<span class="co">#&gt; 16                20            0.9345    genes_cluster1</span>
<span class="co">#&gt; 17                21            0.9345    genes_cluster1</span>
<span class="co">#&gt; 18                22            0.9345    genes_cluster1</span>
<span class="co">#&gt; 19                24            0.9345    genes_cluster1</span>
<span class="co">#&gt; 20                25            0.9345    genes_cluster1</span>
<span class="co">#&gt; 21                27            0.9345    genes_cluster1</span>
<span class="co">#&gt; 22                30            0.9345    genes_cluster1</span>
<span class="co">#&gt; 23                31            0.9345    genes_cluster1</span>
<span class="co">#&gt; 24                33            0.9345    genes_cluster1</span>
<span class="co">#&gt; 25                36            0.9345    genes_cluster1</span>
<span class="co">#&gt; 26                37            0.9345    genes_cluster1</span>
<span class="co">#&gt; 27                40            0.9345    genes_cluster1</span>
<span class="co">#&gt; 28                41            0.9345    genes_cluster1</span>
<span class="co">#&gt; 29                43            0.9345    genes_cluster1</span>
<span class="co">#&gt; 30                45            0.9345    genes_cluster1</span>
<span class="co">#&gt; 31                47            0.9345    genes_cluster1</span>
<span class="co">#&gt; 32                48            0.9345    genes_cluster1</span>
<span class="co">#&gt; 33                49            0.9345    genes_cluster1</span>
<span class="co">#&gt; 34                50            0.9345    genes_cluster1</span>
<span class="co">#&gt; 35                52            0.9345    genes_cluster1</span>
<span class="co">#&gt; 36                54            0.9345    genes_cluster1</span>
<span class="co">#&gt; 37                55            0.9345    genes_cluster1</span>
<span class="co">#&gt; 38                56            0.9345    genes_cluster1</span>
<span class="co">#&gt; 39                58            0.9345    genes_cluster1</span>
<span class="co">#&gt; 40                59            0.9345    genes_cluster1</span>
<span class="co">#&gt; 41                60            0.9345    genes_cluster1</span>
<span class="co">#&gt; 42                62            0.9345    genes_cluster1</span>
<span class="co">#&gt; 43 remaining DEgenes remaining DEgenes remaining DEgenes</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LassoGenesMax</span>
<span class="co">#&gt; NULL</span>

<span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span>
<span class="kw">summary_accuracy</span>(<span class="dt">object =</span> LSOLDA_dat)
<span class="co">#&gt; [1] 84.82143 84.37500</span></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters-in-one-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-one-sample" class="anchor"></a>4.3 Plot the relationship between clusters in one sample</h2>
<p><em>Here we look at one example use case to find relationship between clusters within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
<span class="co">#cluster_mixedpop2 &lt;- as.numeric(as.vector(colData(mixedpop2)[,1]))</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] <span class="co">#top 200 gene markers distinguishing cluster 1 </span>

LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2, <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 187 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 313 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 94 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 94 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 201 genes for training model"</span>
<span class="co">#&gt; [1] "use 201 genes 188 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3_4"</span>
<span class="co">#&gt; [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df      %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 3.364e-15 0.267900</span>
<span class="co">#&gt;   [2,]  1 1.842e-02 0.255700</span>
<span class="co">#&gt;   [3,]  2 3.703e-02 0.244100</span>
<span class="co">#&gt;   [4,]  3 5.758e-02 0.233000</span>
<span class="co">#&gt;   [5,]  3 7.751e-02 0.222400</span>
<span class="co">#&gt;   [6,]  3 9.606e-02 0.212300</span>
<span class="co">#&gt;   [7,]  4 1.135e-01 0.202600</span>
<span class="co">#&gt;   [8,]  4 1.300e-01 0.193400</span>
<span class="co">#&gt;   [9,]  4 1.457e-01 0.184600</span>
<span class="co">#&gt;  [10,]  5 1.611e-01 0.176200</span>
<span class="co">#&gt;  [11,]  7 1.770e-01 0.168200</span>
<span class="co">#&gt;  [12,]  8 1.964e-01 0.160600</span>
<span class="co">#&gt;  [13,] 10 2.153e-01 0.153300</span>
<span class="co">#&gt;  [14,] 10 2.333e-01 0.146300</span>
<span class="co">#&gt;  [15,] 11 2.508e-01 0.139700</span>
<span class="co">#&gt;  [16,] 11 2.674e-01 0.133300</span>
<span class="co">#&gt;  [17,] 12 2.834e-01 0.127300</span>
<span class="co">#&gt;  [18,] 13 2.995e-01 0.121500</span>
<span class="co">#&gt;  [19,] 13 3.157e-01 0.116000</span>
<span class="co">#&gt;  [20,] 13 3.310e-01 0.110700</span>
<span class="co">#&gt;  [21,] 14 3.460e-01 0.105700</span>
<span class="co">#&gt;  [22,] 14 3.609e-01 0.100900</span>
<span class="co">#&gt;  [23,] 14 3.751e-01 0.096270</span>
<span class="co">#&gt;  [24,] 15 3.888e-01 0.091900</span>
<span class="co">#&gt;  [25,] 16 4.040e-01 0.087720</span>
<span class="co">#&gt;  [26,] 16 4.188e-01 0.083730</span>
<span class="co">#&gt;  [27,] 18 4.336e-01 0.079930</span>
<span class="co">#&gt;  [28,] 18 4.482e-01 0.076290</span>
<span class="co">#&gt;  [29,] 18 4.622e-01 0.072830</span>
<span class="co">#&gt;  [30,] 18 4.757e-01 0.069520</span>
<span class="co">#&gt;  [31,] 19 4.889e-01 0.066360</span>
<span class="co">#&gt;  [32,] 19 5.021e-01 0.063340</span>
<span class="co">#&gt;  [33,] 19 5.148e-01 0.060460</span>
<span class="co">#&gt;  [34,] 20 5.271e-01 0.057710</span>
<span class="co">#&gt;  [35,] 20 5.394e-01 0.055090</span>
<span class="co">#&gt;  [36,] 20 5.512e-01 0.052590</span>
<span class="co">#&gt;  [37,] 20 5.627e-01 0.050200</span>
<span class="co">#&gt;  [38,] 24 5.745e-01 0.047910</span>
<span class="co">#&gt;  [39,] 24 5.863e-01 0.045740</span>
<span class="co">#&gt;  [40,] 24 5.977e-01 0.043660</span>
<span class="co">#&gt;  [41,] 25 6.087e-01 0.041670</span>
<span class="co">#&gt;  [42,] 26 6.195e-01 0.039780</span>
<span class="co">#&gt;  [43,] 27 6.302e-01 0.037970</span>
<span class="co">#&gt;  [44,] 27 6.405e-01 0.036250</span>
<span class="co">#&gt;  [45,] 28 6.506e-01 0.034600</span>
<span class="co">#&gt;  [46,] 28 6.603e-01 0.033030</span>
<span class="co">#&gt;  [47,] 30 6.698e-01 0.031520</span>
<span class="co">#&gt;  [48,] 32 6.796e-01 0.030090</span>
<span class="co">#&gt;  [49,] 32 6.892e-01 0.028720</span>
<span class="co">#&gt;  [50,] 34 6.987e-01 0.027420</span>
<span class="co">#&gt;  [51,] 34 7.080e-01 0.026170</span>
<span class="co">#&gt;  [52,] 34 7.170e-01 0.024980</span>
<span class="co">#&gt;  [53,] 34 7.256e-01 0.023850</span>
<span class="co">#&gt;  [54,] 34 7.340e-01 0.022760</span>
<span class="co">#&gt;  [55,] 33 7.421e-01 0.021730</span>
<span class="co">#&gt;  [56,] 33 7.499e-01 0.020740</span>
<span class="co">#&gt;  [57,] 33 7.573e-01 0.019800</span>
<span class="co">#&gt;  [58,] 36 7.648e-01 0.018900</span>
<span class="co">#&gt;  [59,] 37 7.723e-01 0.018040</span>
<span class="co">#&gt;  [60,] 37 7.794e-01 0.017220</span>
<span class="co">#&gt;  [61,] 37 7.864e-01 0.016440</span>
<span class="co">#&gt;  [62,] 35 7.930e-01 0.015690</span>
<span class="co">#&gt;  [63,] 36 7.995e-01 0.014980</span>
<span class="co">#&gt;  [64,] 36 8.058e-01 0.014300</span>
<span class="co">#&gt;  [65,] 40 8.120e-01 0.013650</span>
<span class="co">#&gt;  [66,] 39 8.181e-01 0.013030</span>
<span class="co">#&gt;  [67,] 39 8.239e-01 0.012430</span>
<span class="co">#&gt;  [68,] 42 8.297e-01 0.011870</span>
<span class="co">#&gt;  [69,] 43 8.356e-01 0.011330</span>
<span class="co">#&gt;  [70,] 44 8.413e-01 0.010810</span>
<span class="co">#&gt;  [71,] 45 8.469e-01 0.010320</span>
<span class="co">#&gt;  [72,] 46 8.524e-01 0.009854</span>
<span class="co">#&gt;  [73,] 48 8.576e-01 0.009406</span>
<span class="co">#&gt;  [74,] 49 8.629e-01 0.008978</span>
<span class="co">#&gt;  [75,] 49 8.681e-01 0.008570</span>
<span class="co">#&gt;  [76,] 50 8.731e-01 0.008181</span>
<span class="co">#&gt;  [77,] 50 8.781e-01 0.007809</span>
<span class="co">#&gt;  [78,] 50 8.829e-01 0.007454</span>
<span class="co">#&gt;  [79,] 51 8.875e-01 0.007115</span>
<span class="co">#&gt;  [80,] 52 8.921e-01 0.006792</span>
<span class="co">#&gt;  [81,] 52 8.966e-01 0.006483</span>
<span class="co">#&gt;  [82,] 52 9.010e-01 0.006188</span>
<span class="co">#&gt;  [83,] 52 9.051e-01 0.005907</span>
<span class="co">#&gt;  [84,] 52 9.091e-01 0.005639</span>
<span class="co">#&gt;  [85,] 54 9.130e-01 0.005382</span>
<span class="co">#&gt;  [86,] 55 9.168e-01 0.005138</span>
<span class="co">#&gt;  [87,] 54 9.203e-01 0.004904</span>
<span class="co">#&gt;  [88,] 55 9.238e-01 0.004681</span>
<span class="co">#&gt;  [89,] 55 9.272e-01 0.004469</span>
<span class="co">#&gt;  [90,] 57 9.304e-01 0.004265</span>
<span class="co">#&gt;  [91,] 57 9.335e-01 0.004072</span>
<span class="co">#&gt;  [92,] 57 9.364e-01 0.003887</span>
<span class="co">#&gt;  [93,] 56 9.393e-01 0.003710</span>
<span class="co">#&gt;  [94,] 56 9.420e-01 0.003541</span>
<span class="co">#&gt;  [95,] 57 9.445e-01 0.003380</span>
<span class="co">#&gt;  [96,] 56 9.470e-01 0.003227</span>
<span class="co">#&gt;  [97,] 57 9.494e-01 0.003080</span>
<span class="co">#&gt;  [98,] 57 9.517e-01 0.002940</span>
<span class="co">#&gt;  [99,] 58 9.538e-01 0.002806</span>
<span class="co">#&gt; [100,] 58 9.559e-01 0.002679</span>
<span class="co">#&gt; [1] "lambda min is at location 64"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 93"</span>
<span class="co">#&gt; [1] "use 94 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 187 cells and 201 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.909090909090909"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.641711229946524"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 95.1871657754011"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 81.8181818181818"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 25.7142857142857"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 25"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 79.6992481203008"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 23.3082706766917"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 72.5"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 30"</span>
<span class="co">#&gt; [1] "Total 187 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 313 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 94 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 94 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 201 genes for training model"</span>
<span class="co">#&gt; [1] "use 201 genes 188 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3_4"</span>
<span class="co">#&gt; [1] "there are 94 cells in class 2_3_4 and 94 cells in class 1"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df      %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 3.364e-15 0.281200</span>
<span class="co">#&gt;   [2,]  2 2.601e-02 0.268400</span>
<span class="co">#&gt;   [3,]  2 5.045e-02 0.256200</span>
<span class="co">#&gt;   [4,]  2 7.292e-02 0.244600</span>
<span class="co">#&gt;   [5,]  3 9.507e-02 0.233500</span>
<span class="co">#&gt;   [6,]  4 1.184e-01 0.222800</span>
<span class="co">#&gt;   [7,]  4 1.409e-01 0.212700</span>
<span class="co">#&gt;   [8,]  5 1.625e-01 0.203000</span>
<span class="co">#&gt;   [9,]  5 1.831e-01 0.193800</span>
<span class="co">#&gt;  [10,]  5 2.023e-01 0.185000</span>
<span class="co">#&gt;  [11,]  6 2.210e-01 0.176600</span>
<span class="co">#&gt;  [12,]  6 2.386e-01 0.168600</span>
<span class="co">#&gt;  [13,]  6 2.554e-01 0.160900</span>
<span class="co">#&gt;  [14,]  6 2.712e-01 0.153600</span>
<span class="co">#&gt;  [15,]  7 2.864e-01 0.146600</span>
<span class="co">#&gt;  [16,]  7 3.011e-01 0.140000</span>
<span class="co">#&gt;  [17,]  7 3.151e-01 0.133600</span>
<span class="co">#&gt;  [18,]  7 3.285e-01 0.127500</span>
<span class="co">#&gt;  [19,]  9 3.423e-01 0.121700</span>
<span class="co">#&gt;  [20,] 10 3.578e-01 0.116200</span>
<span class="co">#&gt;  [21,] 11 3.732e-01 0.110900</span>
<span class="co">#&gt;  [22,] 12 3.883e-01 0.105900</span>
<span class="co">#&gt;  [23,] 12 4.028e-01 0.101100</span>
<span class="co">#&gt;  [24,] 12 4.168e-01 0.096460</span>
<span class="co">#&gt;  [25,] 12 4.301e-01 0.092080</span>
<span class="co">#&gt;  [26,] 13 4.433e-01 0.087890</span>
<span class="co">#&gt;  [27,] 13 4.570e-01 0.083900</span>
<span class="co">#&gt;  [28,] 13 4.701e-01 0.080080</span>
<span class="co">#&gt;  [29,] 13 4.828e-01 0.076440</span>
<span class="co">#&gt;  [30,] 13 4.951e-01 0.072970</span>
<span class="co">#&gt;  [31,] 14 5.069e-01 0.069650</span>
<span class="co">#&gt;  [32,] 15 5.187e-01 0.066490</span>
<span class="co">#&gt;  [33,] 15 5.301e-01 0.063470</span>
<span class="co">#&gt;  [34,] 16 5.414e-01 0.060580</span>
<span class="co">#&gt;  [35,] 17 5.523e-01 0.057830</span>
<span class="co">#&gt;  [36,] 18 5.633e-01 0.055200</span>
<span class="co">#&gt;  [37,] 20 5.741e-01 0.052690</span>
<span class="co">#&gt;  [38,] 20 5.849e-01 0.050300</span>
<span class="co">#&gt;  [39,] 20 5.952e-01 0.048010</span>
<span class="co">#&gt;  [40,] 22 6.053e-01 0.045830</span>
<span class="co">#&gt;  [41,] 23 6.152e-01 0.043740</span>
<span class="co">#&gt;  [42,] 25 6.248e-01 0.041760</span>
<span class="co">#&gt;  [43,] 25 6.346e-01 0.039860</span>
<span class="co">#&gt;  [44,] 27 6.443e-01 0.038050</span>
<span class="co">#&gt;  [45,] 29 6.543e-01 0.036320</span>
<span class="co">#&gt;  [46,] 30 6.642e-01 0.034670</span>
<span class="co">#&gt;  [47,] 31 6.738e-01 0.033090</span>
<span class="co">#&gt;  [48,] 32 6.832e-01 0.031590</span>
<span class="co">#&gt;  [49,] 33 6.925e-01 0.030150</span>
<span class="co">#&gt;  [50,] 36 7.019e-01 0.028780</span>
<span class="co">#&gt;  [51,] 37 7.113e-01 0.027470</span>
<span class="co">#&gt;  [52,] 37 7.205e-01 0.026220</span>
<span class="co">#&gt;  [53,] 37 7.292e-01 0.025030</span>
<span class="co">#&gt;  [54,] 38 7.379e-01 0.023890</span>
<span class="co">#&gt;  [55,] 38 7.462e-01 0.022810</span>
<span class="co">#&gt;  [56,] 38 7.543e-01 0.021770</span>
<span class="co">#&gt;  [57,] 39 7.620e-01 0.020780</span>
<span class="co">#&gt;  [58,] 39 7.697e-01 0.019840</span>
<span class="co">#&gt;  [59,] 40 7.772e-01 0.018940</span>
<span class="co">#&gt;  [60,] 40 7.844e-01 0.018080</span>
<span class="co">#&gt;  [61,] 39 7.914e-01 0.017250</span>
<span class="co">#&gt;  [62,] 39 7.981e-01 0.016470</span>
<span class="co">#&gt;  [63,] 40 8.045e-01 0.015720</span>
<span class="co">#&gt;  [64,] 40 8.108e-01 0.015010</span>
<span class="co">#&gt;  [65,] 40 8.169e-01 0.014320</span>
<span class="co">#&gt;  [66,] 40 8.228e-01 0.013670</span>
<span class="co">#&gt;  [67,] 39 8.286e-01 0.013050</span>
<span class="co">#&gt;  [68,] 42 8.341e-01 0.012460</span>
<span class="co">#&gt;  [69,] 44 8.396e-01 0.011890</span>
<span class="co">#&gt;  [70,] 47 8.453e-01 0.011350</span>
<span class="co">#&gt;  [71,] 47 8.510e-01 0.010840</span>
<span class="co">#&gt;  [72,] 46 8.564e-01 0.010340</span>
<span class="co">#&gt;  [73,] 46 8.616e-01 0.009873</span>
<span class="co">#&gt;  [74,] 49 8.668e-01 0.009424</span>
<span class="co">#&gt;  [75,] 50 8.722e-01 0.008996</span>
<span class="co">#&gt;  [76,] 49 8.774e-01 0.008587</span>
<span class="co">#&gt;  [77,] 50 8.822e-01 0.008197</span>
<span class="co">#&gt;  [78,] 50 8.869e-01 0.007824</span>
<span class="co">#&gt;  [79,] 50 8.915e-01 0.007469</span>
<span class="co">#&gt;  [80,] 50 8.958e-01 0.007129</span>
<span class="co">#&gt;  [81,] 51 9.000e-01 0.006805</span>
<span class="co">#&gt;  [82,] 51 9.040e-01 0.006496</span>
<span class="co">#&gt;  [83,] 52 9.080e-01 0.006201</span>
<span class="co">#&gt;  [84,] 52 9.117e-01 0.005919</span>
<span class="co">#&gt;  [85,] 51 9.154e-01 0.005650</span>
<span class="co">#&gt;  [86,] 51 9.188e-01 0.005393</span>
<span class="co">#&gt;  [87,] 52 9.222e-01 0.005148</span>
<span class="co">#&gt;  [88,] 52 9.254e-01 0.004914</span>
<span class="co">#&gt;  [89,] 53 9.285e-01 0.004691</span>
<span class="co">#&gt;  [90,] 53 9.316e-01 0.004477</span>
<span class="co">#&gt;  [91,] 53 9.345e-01 0.004274</span>
<span class="co">#&gt;  [92,] 53 9.374e-01 0.004080</span>
<span class="co">#&gt;  [93,] 53 9.401e-01 0.003894</span>
<span class="co">#&gt;  [94,] 55 9.427e-01 0.003717</span>
<span class="co">#&gt;  [95,] 55 9.452e-01 0.003548</span>
<span class="co">#&gt;  [96,] 55 9.476e-01 0.003387</span>
<span class="co">#&gt;  [97,] 56 9.499e-01 0.003233</span>
<span class="co">#&gt;  [98,] 57 9.521e-01 0.003086</span>
<span class="co">#&gt;  [99,] 57 9.543e-01 0.002946</span>
<span class="co">#&gt; [100,] 57 9.563e-01 0.002812</span>
<span class="co">#&gt; [1] "lambda min is at location 67"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 93"</span>
<span class="co">#&gt; [1] "use 94 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 187 cells and 201 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.86096256684492"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.668449197860963"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 95.7219251336899"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 74.8663101604278"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 22.1428571428571"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 27.8571428571429"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 78.1954887218045"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 30.0751879699248"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 70"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 7 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 7 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are NKX2-5"       </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [7] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 22.5"</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]

LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 140 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 360 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 70 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 70 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 201 genes for training model"</span>
<span class="co">#&gt; [1] "use 201 genes 140 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_3_4"</span>
<span class="co">#&gt; [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df      %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 2.723e-15 0.417400</span>
<span class="co">#&gt;   [2,]  2 4.491e-02 0.398400</span>
<span class="co">#&gt;   [3,]  3 9.256e-02 0.380300</span>
<span class="co">#&gt;   [4,]  3 1.373e-01 0.363000</span>
<span class="co">#&gt;   [5,]  4 1.795e-01 0.346500</span>
<span class="co">#&gt;   [6,]  4 2.193e-01 0.330800</span>
<span class="co">#&gt;   [7,]  4 2.562e-01 0.315800</span>
<span class="co">#&gt;   [8,]  4 2.906e-01 0.301400</span>
<span class="co">#&gt;   [9,]  4 3.227e-01 0.287700</span>
<span class="co">#&gt;  [10,]  5 3.529e-01 0.274600</span>
<span class="co">#&gt;  [11,]  5 3.814e-01 0.262100</span>
<span class="co">#&gt;  [12,]  6 4.082e-01 0.250200</span>
<span class="co">#&gt;  [13,]  7 4.346e-01 0.238900</span>
<span class="co">#&gt;  [14,]  7 4.595e-01 0.228000</span>
<span class="co">#&gt;  [15,]  7 4.830e-01 0.217600</span>
<span class="co">#&gt;  [16,]  7 5.051e-01 0.207700</span>
<span class="co">#&gt;  [17,]  7 5.260e-01 0.198300</span>
<span class="co">#&gt;  [18,]  7 5.458e-01 0.189300</span>
<span class="co">#&gt;  [19,]  7 5.645e-01 0.180700</span>
<span class="co">#&gt;  [20,]  7 5.823e-01 0.172500</span>
<span class="co">#&gt;  [21,]  8 5.991e-01 0.164600</span>
<span class="co">#&gt;  [22,]  9 6.153e-01 0.157200</span>
<span class="co">#&gt;  [23,]  9 6.310e-01 0.150000</span>
<span class="co">#&gt;  [24,]  9 6.459e-01 0.143200</span>
<span class="co">#&gt;  [25,]  9 6.601e-01 0.136700</span>
<span class="co">#&gt;  [26,]  9 6.736e-01 0.130500</span>
<span class="co">#&gt;  [27,]  9 6.864e-01 0.124500</span>
<span class="co">#&gt;  [28,]  9 6.987e-01 0.118900</span>
<span class="co">#&gt;  [29,] 10 7.103e-01 0.113500</span>
<span class="co">#&gt;  [30,] 10 7.214e-01 0.108300</span>
<span class="co">#&gt;  [31,] 10 7.320e-01 0.103400</span>
<span class="co">#&gt;  [32,] 10 7.422e-01 0.098700</span>
<span class="co">#&gt;  [33,] 10 7.518e-01 0.094210</span>
<span class="co">#&gt;  [34,] 10 7.611e-01 0.089930</span>
<span class="co">#&gt;  [35,] 11 7.700e-01 0.085840</span>
<span class="co">#&gt;  [36,] 11 7.786e-01 0.081940</span>
<span class="co">#&gt;  [37,] 12 7.867e-01 0.078210</span>
<span class="co">#&gt;  [38,] 14 7.947e-01 0.074660</span>
<span class="co">#&gt;  [39,] 15 8.023e-01 0.071270</span>
<span class="co">#&gt;  [40,] 15 8.097e-01 0.068030</span>
<span class="co">#&gt;  [41,] 16 8.167e-01 0.064940</span>
<span class="co">#&gt;  [42,] 17 8.238e-01 0.061980</span>
<span class="co">#&gt;  [43,] 17 8.305e-01 0.059170</span>
<span class="co">#&gt;  [44,] 17 8.371e-01 0.056480</span>
<span class="co">#&gt;  [45,] 17 8.433e-01 0.053910</span>
<span class="co">#&gt;  [46,] 17 8.494e-01 0.051460</span>
<span class="co">#&gt;  [47,] 17 8.552e-01 0.049120</span>
<span class="co">#&gt;  [48,] 16 8.608e-01 0.046890</span>
<span class="co">#&gt;  [49,] 16 8.662e-01 0.044760</span>
<span class="co">#&gt;  [50,] 16 8.714e-01 0.042720</span>
<span class="co">#&gt;  [51,] 16 8.764e-01 0.040780</span>
<span class="co">#&gt;  [52,] 17 8.813e-01 0.038930</span>
<span class="co">#&gt;  [53,] 18 8.861e-01 0.037160</span>
<span class="co">#&gt;  [54,] 19 8.907e-01 0.035470</span>
<span class="co">#&gt;  [55,] 19 8.953e-01 0.033860</span>
<span class="co">#&gt;  [56,] 19 8.996e-01 0.032320</span>
<span class="co">#&gt;  [57,] 19 9.038e-01 0.030850</span>
<span class="co">#&gt;  [58,] 20 9.078e-01 0.029450</span>
<span class="co">#&gt;  [59,] 20 9.118e-01 0.028110</span>
<span class="co">#&gt;  [60,] 20 9.156e-01 0.026830</span>
<span class="co">#&gt;  [61,] 20 9.192e-01 0.025610</span>
<span class="co">#&gt;  [62,] 20 9.227e-01 0.024450</span>
<span class="co">#&gt;  [63,] 20 9.260e-01 0.023340</span>
<span class="co">#&gt;  [64,] 21 9.292e-01 0.022280</span>
<span class="co">#&gt;  [65,] 22 9.323e-01 0.021260</span>
<span class="co">#&gt;  [66,] 22 9.352e-01 0.020300</span>
<span class="co">#&gt;  [67,] 22 9.381e-01 0.019370</span>
<span class="co">#&gt;  [68,] 22 9.408e-01 0.018490</span>
<span class="co">#&gt;  [69,] 22 9.434e-01 0.017650</span>
<span class="co">#&gt;  [70,] 22 9.458e-01 0.016850</span>
<span class="co">#&gt;  [71,] 23 9.482e-01 0.016080</span>
<span class="co">#&gt;  [72,] 23 9.505e-01 0.015350</span>
<span class="co">#&gt;  [73,] 23 9.527e-01 0.014660</span>
<span class="co">#&gt;  [74,] 23 9.547e-01 0.013990</span>
<span class="co">#&gt;  [75,] 22 9.567e-01 0.013350</span>
<span class="co">#&gt;  [76,] 22 9.586e-01 0.012750</span>
<span class="co">#&gt;  [77,] 22 9.604e-01 0.012170</span>
<span class="co">#&gt;  [78,] 22 9.621e-01 0.011610</span>
<span class="co">#&gt;  [79,] 22 9.638e-01 0.011090</span>
<span class="co">#&gt;  [80,] 22 9.654e-01 0.010580</span>
<span class="co">#&gt;  [81,] 22 9.669e-01 0.010100</span>
<span class="co">#&gt;  [82,] 21 9.683e-01 0.009643</span>
<span class="co">#&gt;  [83,] 21 9.697e-01 0.009204</span>
<span class="co">#&gt;  [84,] 21 9.711e-01 0.008786</span>
<span class="co">#&gt;  [85,] 21 9.723e-01 0.008387</span>
<span class="co">#&gt;  [86,] 21 9.735e-01 0.008006</span>
<span class="co">#&gt;  [87,] 21 9.747e-01 0.007642</span>
<span class="co">#&gt;  [88,] 22 9.758e-01 0.007294</span>
<span class="co">#&gt;  [89,] 23 9.769e-01 0.006963</span>
<span class="co">#&gt;  [90,] 23 9.779e-01 0.006646</span>
<span class="co">#&gt;  [91,] 23 9.789e-01 0.006344</span>
<span class="co">#&gt;  [92,] 23 9.799e-01 0.006056</span>
<span class="co">#&gt;  [93,] 23 9.808e-01 0.005781</span>
<span class="co">#&gt;  [94,] 23 9.816e-01 0.005518</span>
<span class="co">#&gt;  [95,] 23 9.825e-01 0.005267</span>
<span class="co">#&gt;  [96,] 23 9.832e-01 0.005028</span>
<span class="co">#&gt;  [97,] 23 9.840e-01 0.004799</span>
<span class="co">#&gt;  [98,] 23 9.847e-01 0.004581</span>
<span class="co">#&gt;  [99,] 23 9.854e-01 0.004373</span>
<span class="co">#&gt; [100,] 23 9.860e-01 0.004174</span>
<span class="co">#&gt; [1] "lambda min is at location 19"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 70"</span>
<span class="co">#&gt; [1] "use 70 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 140 cells and 201 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.9"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.907142857142857"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 9.09090909090909"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 4.27807486631016"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 99.2857142857143"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 96.4285714285714"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 25.5639097744361"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 2.25563909774436"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 45"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 32.5"</span>
<span class="co">#&gt; [1] "Total 140 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 360 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 70 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 70 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 201 genes for training model"</span>
<span class="co">#&gt; [1] "use 201 genes 140 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_3_4"</span>
<span class="co">#&gt; [1] "there are 70 cells in class 1_3_4 and 70 cells in class 2"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df      %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 2.723e-15 0.404300</span>
<span class="co">#&gt;   [2,]  1 4.193e-02 0.385900</span>
<span class="co">#&gt;   [3,]  1 8.031e-02 0.368300</span>
<span class="co">#&gt;   [4,]  1 1.156e-01 0.351600</span>
<span class="co">#&gt;   [5,]  3 1.546e-01 0.335600</span>
<span class="co">#&gt;   [6,]  4 1.931e-01 0.320400</span>
<span class="co">#&gt;   [7,]  4 2.293e-01 0.305800</span>
<span class="co">#&gt;   [8,]  4 2.631e-01 0.291900</span>
<span class="co">#&gt;   [9,]  4 2.947e-01 0.278600</span>
<span class="co">#&gt;  [10,]  4 3.242e-01 0.266000</span>
<span class="co">#&gt;  [11,]  4 3.519e-01 0.253900</span>
<span class="co">#&gt;  [12,]  5 3.781e-01 0.242300</span>
<span class="co">#&gt;  [13,]  5 4.029e-01 0.231300</span>
<span class="co">#&gt;  [14,]  5 4.263e-01 0.220800</span>
<span class="co">#&gt;  [15,]  5 4.483e-01 0.210800</span>
<span class="co">#&gt;  [16,]  6 4.692e-01 0.201200</span>
<span class="co">#&gt;  [17,]  6 4.890e-01 0.192100</span>
<span class="co">#&gt;  [18,]  6 5.078e-01 0.183300</span>
<span class="co">#&gt;  [19,]  8 5.267e-01 0.175000</span>
<span class="co">#&gt;  [20,]  9 5.453e-01 0.167000</span>
<span class="co">#&gt;  [21,] 10 5.630e-01 0.159400</span>
<span class="co">#&gt;  [22,] 10 5.799e-01 0.152200</span>
<span class="co">#&gt;  [23,] 10 5.961e-01 0.145300</span>
<span class="co">#&gt;  [24,] 10 6.114e-01 0.138700</span>
<span class="co">#&gt;  [25,] 10 6.260e-01 0.132400</span>
<span class="co">#&gt;  [26,] 10 6.399e-01 0.126400</span>
<span class="co">#&gt;  [27,] 10 6.533e-01 0.120600</span>
<span class="co">#&gt;  [28,] 10 6.661e-01 0.115100</span>
<span class="co">#&gt;  [29,] 10 6.783e-01 0.109900</span>
<span class="co">#&gt;  [30,] 10 6.899e-01 0.104900</span>
<span class="co">#&gt;  [31,] 10 7.010e-01 0.100100</span>
<span class="co">#&gt;  [32,] 11 7.117e-01 0.095590</span>
<span class="co">#&gt;  [33,] 11 7.219e-01 0.091240</span>
<span class="co">#&gt;  [34,] 11 7.316e-01 0.087090</span>
<span class="co">#&gt;  [35,] 11 7.409e-01 0.083140</span>
<span class="co">#&gt;  [36,] 12 7.501e-01 0.079360</span>
<span class="co">#&gt;  [37,] 13 7.589e-01 0.075750</span>
<span class="co">#&gt;  [38,] 13 7.675e-01 0.072310</span>
<span class="co">#&gt;  [39,] 13 7.757e-01 0.069020</span>
<span class="co">#&gt;  [40,] 13 7.836e-01 0.065880</span>
<span class="co">#&gt;  [41,] 13 7.911e-01 0.062890</span>
<span class="co">#&gt;  [42,] 13 7.983e-01 0.060030</span>
<span class="co">#&gt;  [43,] 13 8.053e-01 0.057300</span>
<span class="co">#&gt;  [44,] 13 8.119e-01 0.054700</span>
<span class="co">#&gt;  [45,] 13 8.183e-01 0.052210</span>
<span class="co">#&gt;  [46,] 13 8.244e-01 0.049840</span>
<span class="co">#&gt;  [47,] 13 8.303e-01 0.047570</span>
<span class="co">#&gt;  [48,] 13 8.360e-01 0.045410</span>
<span class="co">#&gt;  [49,] 13 8.414e-01 0.043350</span>
<span class="co">#&gt;  [50,] 13 8.467e-01 0.041380</span>
<span class="co">#&gt;  [51,] 13 8.517e-01 0.039500</span>
<span class="co">#&gt;  [52,] 14 8.565e-01 0.037700</span>
<span class="co">#&gt;  [53,] 14 8.614e-01 0.035990</span>
<span class="co">#&gt;  [54,] 14 8.663e-01 0.034350</span>
<span class="co">#&gt;  [55,] 15 8.711e-01 0.032790</span>
<span class="co">#&gt;  [56,] 15 8.757e-01 0.031300</span>
<span class="co">#&gt;  [57,] 15 8.801e-01 0.029880</span>
<span class="co">#&gt;  [58,] 16 8.844e-01 0.028520</span>
<span class="co">#&gt;  [59,] 16 8.887e-01 0.027220</span>
<span class="co">#&gt;  [60,] 18 8.929e-01 0.025990</span>
<span class="co">#&gt;  [61,] 19 8.969e-01 0.024800</span>
<span class="co">#&gt;  [62,] 21 9.009e-01 0.023680</span>
<span class="co">#&gt;  [63,] 20 9.049e-01 0.022600</span>
<span class="co">#&gt;  [64,] 20 9.086e-01 0.021570</span>
<span class="co">#&gt;  [65,] 20 9.122e-01 0.020590</span>
<span class="co">#&gt;  [66,] 20 9.157e-01 0.019660</span>
<span class="co">#&gt;  [67,] 21 9.191e-01 0.018760</span>
<span class="co">#&gt;  [68,] 21 9.223e-01 0.017910</span>
<span class="co">#&gt;  [69,] 21 9.254e-01 0.017100</span>
<span class="co">#&gt;  [70,] 20 9.284e-01 0.016320</span>
<span class="co">#&gt;  [71,] 20 9.312e-01 0.015580</span>
<span class="co">#&gt;  [72,] 20 9.339e-01 0.014870</span>
<span class="co">#&gt;  [73,] 19 9.366e-01 0.014190</span>
<span class="co">#&gt;  [74,] 19 9.391e-01 0.013550</span>
<span class="co">#&gt;  [75,] 19 9.415e-01 0.012930</span>
<span class="co">#&gt;  [76,] 20 9.438e-01 0.012350</span>
<span class="co">#&gt;  [77,] 20 9.460e-01 0.011780</span>
<span class="co">#&gt;  [78,] 21 9.482e-01 0.011250</span>
<span class="co">#&gt;  [79,] 21 9.503e-01 0.010740</span>
<span class="co">#&gt;  [80,] 22 9.523e-01 0.010250</span>
<span class="co">#&gt;  [81,] 22 9.542e-01 0.009784</span>
<span class="co">#&gt;  [82,] 22 9.560e-01 0.009339</span>
<span class="co">#&gt;  [83,] 22 9.578e-01 0.008914</span>
<span class="co">#&gt;  [84,] 22 9.595e-01 0.008509</span>
<span class="co">#&gt;  [85,] 21 9.612e-01 0.008122</span>
<span class="co">#&gt;  [86,] 21 9.628e-01 0.007753</span>
<span class="co">#&gt;  [87,] 22 9.643e-01 0.007401</span>
<span class="co">#&gt;  [88,] 22 9.658e-01 0.007064</span>
<span class="co">#&gt;  [89,] 22 9.673e-01 0.006743</span>
<span class="co">#&gt;  [90,] 22 9.687e-01 0.006437</span>
<span class="co">#&gt;  [91,] 22 9.701e-01 0.006144</span>
<span class="co">#&gt;  [92,] 22 9.714e-01 0.005865</span>
<span class="co">#&gt;  [93,] 22 9.726e-01 0.005598</span>
<span class="co">#&gt;  [94,] 22 9.738e-01 0.005344</span>
<span class="co">#&gt;  [95,] 22 9.749e-01 0.005101</span>
<span class="co">#&gt;  [96,] 22 9.760e-01 0.004869</span>
<span class="co">#&gt;  [97,] 22 9.771e-01 0.004648</span>
<span class="co">#&gt;  [98,] 22 9.780e-01 0.004437</span>
<span class="co">#&gt;  [99,] 23 9.790e-01 0.004235</span>
<span class="co">#&gt; [100,] 23 9.799e-01 0.004043</span>
<span class="co">#&gt; [1] "lambda min is at location 25"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 70"</span>
<span class="co">#&gt; [1] "use 70 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 140 cells and 201 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.942857142857143"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.821428571428571"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 1.60427807486631"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 0.53475935828877"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 100"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 2.85714285714286"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 75.187969924812"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 0.75187969924812"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 77.5"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17206"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 196"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 22.5"</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 133 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 367 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 66 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 66 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 200 genes for training model"</span>
<span class="co">#&gt; [1] "use 200 genes 132 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_2_4"</span>
<span class="co">#&gt; [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.403e-15 0.275200</span>
<span class="co">#&gt;   [2,]  1  1.942e-02 0.262700</span>
<span class="co">#&gt;   [3,]  3  4.646e-02 0.250700</span>
<span class="co">#&gt;   [4,]  4  7.869e-02 0.239300</span>
<span class="co">#&gt;   [5,]  4  1.098e-01 0.228400</span>
<span class="co">#&gt;   [6,]  4  1.385e-01 0.218100</span>
<span class="co">#&gt;   [7,]  5  1.653e-01 0.208200</span>
<span class="co">#&gt;   [8,]  7  1.934e-01 0.198700</span>
<span class="co">#&gt;   [9,]  7  2.205e-01 0.189700</span>
<span class="co">#&gt;  [10,]  7  2.458e-01 0.181000</span>
<span class="co">#&gt;  [11,]  8  2.695e-01 0.172800</span>
<span class="co">#&gt;  [12,]  9  2.925e-01 0.165000</span>
<span class="co">#&gt;  [13,] 10  3.141e-01 0.157500</span>
<span class="co">#&gt;  [14,] 11  3.348e-01 0.150300</span>
<span class="co">#&gt;  [15,] 12  3.548e-01 0.143500</span>
<span class="co">#&gt;  [16,] 13  3.737e-01 0.137000</span>
<span class="co">#&gt;  [17,] 14  3.918e-01 0.130700</span>
<span class="co">#&gt;  [18,] 15  4.091e-01 0.124800</span>
<span class="co">#&gt;  [19,] 15  4.256e-01 0.119100</span>
<span class="co">#&gt;  [20,] 15  4.413e-01 0.113700</span>
<span class="co">#&gt;  [21,] 15  4.563e-01 0.108500</span>
<span class="co">#&gt;  [22,] 15  4.705e-01 0.103600</span>
<span class="co">#&gt;  [23,] 17  4.843e-01 0.098890</span>
<span class="co">#&gt;  [24,] 16  4.985e-01 0.094400</span>
<span class="co">#&gt;  [25,] 17  5.123e-01 0.090100</span>
<span class="co">#&gt;  [26,] 17  5.254e-01 0.086010</span>
<span class="co">#&gt;  [27,] 20  5.383e-01 0.082100</span>
<span class="co">#&gt;  [28,] 22  5.522e-01 0.078370</span>
<span class="co">#&gt;  [29,] 23  5.657e-01 0.074810</span>
<span class="co">#&gt;  [30,] 24  5.786e-01 0.071410</span>
<span class="co">#&gt;  [31,] 25  5.913e-01 0.068160</span>
<span class="co">#&gt;  [32,] 26  6.041e-01 0.065060</span>
<span class="co">#&gt;  [33,] 27  6.165e-01 0.062110</span>
<span class="co">#&gt;  [34,] 30  6.286e-01 0.059280</span>
<span class="co">#&gt;  [35,] 33  6.412e-01 0.056590</span>
<span class="co">#&gt;  [36,] 34  6.537e-01 0.054020</span>
<span class="co">#&gt;  [37,] 35  6.663e-01 0.051560</span>
<span class="co">#&gt;  [38,] 35  6.786e-01 0.049220</span>
<span class="co">#&gt;  [39,] 35  6.904e-01 0.046980</span>
<span class="co">#&gt;  [40,] 35  7.017e-01 0.044850</span>
<span class="co">#&gt;  [41,] 35  7.127e-01 0.042810</span>
<span class="co">#&gt;  [42,] 35  7.232e-01 0.040860</span>
<span class="co">#&gt;  [43,] 35  7.335e-01 0.039000</span>
<span class="co">#&gt;  [44,] 36  7.435e-01 0.037230</span>
<span class="co">#&gt;  [45,] 36  7.530e-01 0.035540</span>
<span class="co">#&gt;  [46,] 36  7.625e-01 0.033920</span>
<span class="co">#&gt;  [47,] 38  7.718e-01 0.032380</span>
<span class="co">#&gt;  [48,] 38  7.808e-01 0.030910</span>
<span class="co">#&gt;  [49,] 40  7.897e-01 0.029510</span>
<span class="co">#&gt;  [50,] 39  7.983e-01 0.028160</span>
<span class="co">#&gt;  [51,] 39  8.065e-01 0.026880</span>
<span class="co">#&gt;  [52,] 40  8.144e-01 0.025660</span>
<span class="co">#&gt;  [53,] 40  8.220e-01 0.024500</span>
<span class="co">#&gt;  [54,] 40  8.293e-01 0.023380</span>
<span class="co">#&gt;  [55,] 40  8.363e-01 0.022320</span>
<span class="co">#&gt;  [56,] 40  8.430e-01 0.021310</span>
<span class="co">#&gt;  [57,] 41  8.495e-01 0.020340</span>
<span class="co">#&gt;  [58,] 40  8.558e-01 0.019410</span>
<span class="co">#&gt;  [59,] 40  8.619e-01 0.018530</span>
<span class="co">#&gt;  [60,] 39  8.677e-01 0.017690</span>
<span class="co">#&gt;  [61,] 38  8.732e-01 0.016880</span>
<span class="co">#&gt;  [62,] 39  8.785e-01 0.016120</span>
<span class="co">#&gt;  [63,] 39  8.836e-01 0.015380</span>
<span class="co">#&gt;  [64,] 39  8.885e-01 0.014680</span>
<span class="co">#&gt;  [65,] 39  8.932e-01 0.014020</span>
<span class="co">#&gt;  [66,] 39  8.977e-01 0.013380</span>
<span class="co">#&gt;  [67,] 39  9.021e-01 0.012770</span>
<span class="co">#&gt;  [68,] 41  9.063e-01 0.012190</span>
<span class="co">#&gt;  [69,] 41  9.104e-01 0.011640</span>
<span class="co">#&gt;  [70,] 42  9.143e-01 0.011110</span>
<span class="co">#&gt;  [71,] 42  9.181e-01 0.010600</span>
<span class="co">#&gt;  [72,] 42  9.217e-01 0.010120</span>
<span class="co">#&gt;  [73,] 42  9.251e-01 0.009662</span>
<span class="co">#&gt;  [74,] 42  9.284e-01 0.009222</span>
<span class="co">#&gt;  [75,] 42  9.316e-01 0.008803</span>
<span class="co">#&gt;  [76,] 41  9.347e-01 0.008403</span>
<span class="co">#&gt;  [77,] 41  9.376e-01 0.008021</span>
<span class="co">#&gt;  [78,] 41  9.403e-01 0.007657</span>
<span class="co">#&gt;  [79,] 41  9.430e-01 0.007309</span>
<span class="co">#&gt;  [80,] 41  9.455e-01 0.006976</span>
<span class="co">#&gt;  [81,] 41  9.479e-01 0.006659</span>
<span class="co">#&gt;  [82,] 41  9.502e-01 0.006357</span>
<span class="co">#&gt;  [83,] 41  9.525e-01 0.006068</span>
<span class="co">#&gt;  [84,] 42  9.546e-01 0.005792</span>
<span class="co">#&gt;  [85,] 43  9.566e-01 0.005529</span>
<span class="co">#&gt;  [86,] 43  9.585e-01 0.005277</span>
<span class="co">#&gt;  [87,] 42  9.604e-01 0.005038</span>
<span class="co">#&gt;  [88,] 42  9.621e-01 0.004809</span>
<span class="co">#&gt;  [89,] 42  9.638e-01 0.004590</span>
<span class="co">#&gt;  [90,] 42  9.654e-01 0.004381</span>
<span class="co">#&gt;  [91,] 42  9.670e-01 0.004182</span>
<span class="co">#&gt;  [92,] 42  9.685e-01 0.003992</span>
<span class="co">#&gt;  [93,] 41  9.699e-01 0.003811</span>
<span class="co">#&gt;  [94,] 41  9.712e-01 0.003638</span>
<span class="co">#&gt;  [95,] 41  9.725e-01 0.003472</span>
<span class="co">#&gt;  [96,] 41  9.737e-01 0.003314</span>
<span class="co">#&gt;  [97,] 41  9.749e-01 0.003164</span>
<span class="co">#&gt;  [98,] 41  9.760e-01 0.003020</span>
<span class="co">#&gt;  [99,] 41  9.771e-01 0.002883</span>
<span class="co">#&gt; [100,] 41  9.781e-01 0.002752</span>
<span class="co">#&gt; [1] "lambda min is at location 90"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 67"</span>
<span class="co">#&gt; [1] "use 66 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.857142857142857"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.804511278195489"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 29.4117647058824"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 1.60427807486631"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 0"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 0.714285714285714"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 65.4135338345865"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 38.3458646616541"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 5"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 7.5"</span>
<span class="co">#&gt; [1] "Total 133 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 367 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 66 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 66 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 200 genes for training model"</span>
<span class="co">#&gt; [1] "use 200 genes 132 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_2_4"</span>
<span class="co">#&gt; [1] "there are 66 cells in class 1_2_4 and 66 cells in class 3"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.403e-15 0.280000</span>
<span class="co">#&gt;   [2,]  1  2.012e-02 0.267300</span>
<span class="co">#&gt;   [3,]  2  4.298e-02 0.255100</span>
<span class="co">#&gt;   [4,]  3  6.714e-02 0.243500</span>
<span class="co">#&gt;   [5,]  4  9.551e-02 0.232500</span>
<span class="co">#&gt;   [6,]  6  1.229e-01 0.221900</span>
<span class="co">#&gt;   [7,] 11  1.524e-01 0.211800</span>
<span class="co">#&gt;   [8,] 14  1.824e-01 0.202200</span>
<span class="co">#&gt;   [9,] 15  2.130e-01 0.193000</span>
<span class="co">#&gt;  [10,] 16  2.418e-01 0.184200</span>
<span class="co">#&gt;  [11,] 17  2.697e-01 0.175900</span>
<span class="co">#&gt;  [12,] 18  2.963e-01 0.167900</span>
<span class="co">#&gt;  [13,] 19  3.220e-01 0.160200</span>
<span class="co">#&gt;  [14,] 19  3.465e-01 0.152900</span>
<span class="co">#&gt;  [15,] 19  3.697e-01 0.146000</span>
<span class="co">#&gt;  [16,] 19  3.916e-01 0.139400</span>
<span class="co">#&gt;  [17,] 19  4.123e-01 0.133000</span>
<span class="co">#&gt;  [18,] 20  4.321e-01 0.127000</span>
<span class="co">#&gt;  [19,] 22  4.516e-01 0.121200</span>
<span class="co">#&gt;  [20,] 23  4.707e-01 0.115700</span>
<span class="co">#&gt;  [21,] 23  4.892e-01 0.110400</span>
<span class="co">#&gt;  [22,] 23  5.067e-01 0.105400</span>
<span class="co">#&gt;  [23,] 22  5.234e-01 0.100600</span>
<span class="co">#&gt;  [24,] 22  5.393e-01 0.096060</span>
<span class="co">#&gt;  [25,] 22  5.544e-01 0.091690</span>
<span class="co">#&gt;  [26,] 23  5.690e-01 0.087520</span>
<span class="co">#&gt;  [27,] 23  5.829e-01 0.083550</span>
<span class="co">#&gt;  [28,] 24  5.968e-01 0.079750</span>
<span class="co">#&gt;  [29,] 24  6.104e-01 0.076120</span>
<span class="co">#&gt;  [30,] 25  6.234e-01 0.072660</span>
<span class="co">#&gt;  [31,] 26  6.365e-01 0.069360</span>
<span class="co">#&gt;  [32,] 26  6.491e-01 0.066210</span>
<span class="co">#&gt;  [33,] 27  6.613e-01 0.063200</span>
<span class="co">#&gt;  [34,] 28  6.732e-01 0.060330</span>
<span class="co">#&gt;  [35,] 28  6.846e-01 0.057580</span>
<span class="co">#&gt;  [36,] 28  6.955e-01 0.054970</span>
<span class="co">#&gt;  [37,] 27  7.060e-01 0.052470</span>
<span class="co">#&gt;  [38,] 28  7.161e-01 0.050080</span>
<span class="co">#&gt;  [39,] 28  7.266e-01 0.047810</span>
<span class="co">#&gt;  [40,] 28  7.366e-01 0.045630</span>
<span class="co">#&gt;  [41,] 28  7.461e-01 0.043560</span>
<span class="co">#&gt;  [42,] 29  7.558e-01 0.041580</span>
<span class="co">#&gt;  [43,] 29  7.650e-01 0.039690</span>
<span class="co">#&gt;  [44,] 29  7.739e-01 0.037890</span>
<span class="co">#&gt;  [45,] 29  7.824e-01 0.036160</span>
<span class="co">#&gt;  [46,] 29  7.905e-01 0.034520</span>
<span class="co">#&gt;  [47,] 29  7.983e-01 0.032950</span>
<span class="co">#&gt;  [48,] 30  8.058e-01 0.031450</span>
<span class="co">#&gt;  [49,] 29  8.131e-01 0.030020</span>
<span class="co">#&gt;  [50,] 28  8.199e-01 0.028660</span>
<span class="co">#&gt;  [51,] 28  8.266e-01 0.027360</span>
<span class="co">#&gt;  [52,] 29  8.329e-01 0.026110</span>
<span class="co">#&gt;  [53,] 30  8.392e-01 0.024930</span>
<span class="co">#&gt;  [54,] 31  8.452e-01 0.023790</span>
<span class="co">#&gt;  [55,] 31  8.511e-01 0.022710</span>
<span class="co">#&gt;  [56,] 30  8.569e-01 0.021680</span>
<span class="co">#&gt;  [57,] 30  8.624e-01 0.020690</span>
<span class="co">#&gt;  [58,] 31  8.678e-01 0.019750</span>
<span class="co">#&gt;  [59,] 33  8.730e-01 0.018860</span>
<span class="co">#&gt;  [60,] 34  8.780e-01 0.018000</span>
<span class="co">#&gt;  [61,] 34  8.829e-01 0.017180</span>
<span class="co">#&gt;  [62,] 37  8.876e-01 0.016400</span>
<span class="co">#&gt;  [63,] 37  8.922e-01 0.015650</span>
<span class="co">#&gt;  [64,] 38  8.967e-01 0.014940</span>
<span class="co">#&gt;  [65,] 38  9.010e-01 0.014260</span>
<span class="co">#&gt;  [66,] 38  9.052e-01 0.013620</span>
<span class="co">#&gt;  [67,] 39  9.092e-01 0.013000</span>
<span class="co">#&gt;  [68,] 39  9.130e-01 0.012410</span>
<span class="co">#&gt;  [69,] 39  9.169e-01 0.011840</span>
<span class="co">#&gt;  [70,] 39  9.206e-01 0.011300</span>
<span class="co">#&gt;  [71,] 40  9.242e-01 0.010790</span>
<span class="co">#&gt;  [72,] 40  9.276e-01 0.010300</span>
<span class="co">#&gt;  [73,] 40  9.308e-01 0.009832</span>
<span class="co">#&gt;  [74,] 40  9.339e-01 0.009385</span>
<span class="co">#&gt;  [75,] 40  9.369e-01 0.008958</span>
<span class="co">#&gt;  [76,] 40  9.397e-01 0.008551</span>
<span class="co">#&gt;  [77,] 40  9.424e-01 0.008162</span>
<span class="co">#&gt;  [78,] 40  9.450e-01 0.007791</span>
<span class="co">#&gt;  [79,] 40  9.474e-01 0.007437</span>
<span class="co">#&gt;  [80,] 40  9.498e-01 0.007099</span>
<span class="co">#&gt;  [81,] 41  9.521e-01 0.006777</span>
<span class="co">#&gt;  [82,] 42  9.542e-01 0.006469</span>
<span class="co">#&gt;  [83,] 42  9.563e-01 0.006175</span>
<span class="co">#&gt;  [84,] 43  9.583e-01 0.005894</span>
<span class="co">#&gt;  [85,] 42  9.602e-01 0.005626</span>
<span class="co">#&gt;  [86,] 42  9.620e-01 0.005370</span>
<span class="co">#&gt;  [87,] 42  9.637e-01 0.005126</span>
<span class="co">#&gt;  [88,] 42  9.653e-01 0.004893</span>
<span class="co">#&gt;  [89,] 42  9.669e-01 0.004671</span>
<span class="co">#&gt;  [90,] 43  9.684e-01 0.004459</span>
<span class="co">#&gt;  [91,] 43  9.698e-01 0.004256</span>
<span class="co">#&gt;  [92,] 43  9.712e-01 0.004062</span>
<span class="co">#&gt;  [93,] 43  9.725e-01 0.003878</span>
<span class="co">#&gt;  [94,] 43  9.738e-01 0.003702</span>
<span class="co">#&gt;  [95,] 43  9.750e-01 0.003533</span>
<span class="co">#&gt;  [96,] 43  9.761e-01 0.003373</span>
<span class="co">#&gt;  [97,] 43  9.772e-01 0.003219</span>
<span class="co">#&gt;  [98,] 44  9.782e-01 0.003073</span>
<span class="co">#&gt;  [99,] 44  9.792e-01 0.002933</span>
<span class="co">#&gt; [100,] 44  9.801e-01 0.002800</span>
<span class="co">#&gt; [1] "lambda min is at location 60"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 67"</span>
<span class="co">#&gt; [1] "use 66 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 133 cells and 200 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.902255639097744"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.87218045112782"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 27.807486631016"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 187 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 2.13903743315508"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 2.14285714285714"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 140 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 2.14285714285714"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 97.7443609022556"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 133 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 65.4135338345865"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17202"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 200"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 20"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17204"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 198"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 200 genes and 40 cells. The trained model has 200 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 10"</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">4</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop4vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat4 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 40 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 460 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 20 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 20 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 201 genes for training model"</span>
<span class="co">#&gt; [1] "use 201 genes 40 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_2_3"</span>
<span class="co">#&gt; [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df    %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 0.00000 0.259500</span>
<span class="co">#&gt;   [2,]  1 0.01727 0.247700</span>
<span class="co">#&gt;   [3,]  1 0.03311 0.236400</span>
<span class="co">#&gt;   [4,]  2 0.05058 0.225700</span>
<span class="co">#&gt;   [5,]  3 0.06912 0.215400</span>
<span class="co">#&gt;   [6,]  3 0.08696 0.205600</span>
<span class="co">#&gt;   [7,]  4 0.10710 0.196300</span>
<span class="co">#&gt;   [8,]  4 0.12610 0.187300</span>
<span class="co">#&gt;   [9,]  4 0.14380 0.178800</span>
<span class="co">#&gt;  [10,]  4 0.16030 0.170700</span>
<span class="co">#&gt;  [11,]  4 0.17570 0.162900</span>
<span class="co">#&gt;  [12,]  5 0.19600 0.155500</span>
<span class="co">#&gt;  [13,]  5 0.21580 0.148500</span>
<span class="co">#&gt;  [14,]  5 0.23450 0.141700</span>
<span class="co">#&gt;  [15,]  6 0.25270 0.135300</span>
<span class="co">#&gt;  [16,]  6 0.27630 0.129100</span>
<span class="co">#&gt;  [17,]  6 0.29840 0.123300</span>
<span class="co">#&gt;  [18,]  6 0.31920 0.117700</span>
<span class="co">#&gt;  [19,]  6 0.33890 0.112300</span>
<span class="co">#&gt;  [20,]  6 0.35740 0.107200</span>
<span class="co">#&gt;  [21,]  6 0.37500 0.102300</span>
<span class="co">#&gt;  [22,]  7 0.39330 0.097680</span>
<span class="co">#&gt;  [23,]  7 0.41140 0.093240</span>
<span class="co">#&gt;  [24,]  7 0.42850 0.089010</span>
<span class="co">#&gt;  [25,]  8 0.44480 0.084960</span>
<span class="co">#&gt;  [26,] 10 0.46170 0.081100</span>
<span class="co">#&gt;  [27,] 10 0.48000 0.077410</span>
<span class="co">#&gt;  [28,] 11 0.49750 0.073890</span>
<span class="co">#&gt;  [29,] 12 0.51520 0.070540</span>
<span class="co">#&gt;  [30,] 12 0.53250 0.067330</span>
<span class="co">#&gt;  [31,] 15 0.55100 0.064270</span>
<span class="co">#&gt;  [32,] 15 0.57170 0.061350</span>
<span class="co">#&gt;  [33,] 16 0.59160 0.058560</span>
<span class="co">#&gt;  [34,] 16 0.61050 0.055900</span>
<span class="co">#&gt;  [35,] 18 0.62940 0.053360</span>
<span class="co">#&gt;  [36,] 19 0.64760 0.050930</span>
<span class="co">#&gt;  [37,] 19 0.66490 0.048620</span>
<span class="co">#&gt;  [38,] 19 0.68120 0.046410</span>
<span class="co">#&gt;  [39,] 20 0.69660 0.044300</span>
<span class="co">#&gt;  [40,] 20 0.71120 0.042280</span>
<span class="co">#&gt;  [41,] 20 0.72510 0.040360</span>
<span class="co">#&gt;  [42,] 20 0.73820 0.038530</span>
<span class="co">#&gt;  [43,] 20 0.75060 0.036780</span>
<span class="co">#&gt;  [44,] 20 0.76240 0.035110</span>
<span class="co">#&gt;  [45,] 20 0.77360 0.033510</span>
<span class="co">#&gt;  [46,] 20 0.78430 0.031990</span>
<span class="co">#&gt;  [47,] 20 0.79440 0.030530</span>
<span class="co">#&gt;  [48,] 20 0.80400 0.029150</span>
<span class="co">#&gt;  [49,] 20 0.81320 0.027820</span>
<span class="co">#&gt;  [50,] 20 0.82190 0.026560</span>
<span class="co">#&gt;  [51,] 19 0.83020 0.025350</span>
<span class="co">#&gt;  [52,] 19 0.83800 0.024200</span>
<span class="co">#&gt;  [53,] 19 0.84540 0.023100</span>
<span class="co">#&gt;  [54,] 19 0.85260 0.022050</span>
<span class="co">#&gt;  [55,] 19 0.85930 0.021050</span>
<span class="co">#&gt;  [56,] 19 0.86580 0.020090</span>
<span class="co">#&gt;  [57,] 20 0.87190 0.019180</span>
<span class="co">#&gt;  [58,] 21 0.87780 0.018300</span>
<span class="co">#&gt;  [59,] 21 0.88340 0.017470</span>
<span class="co">#&gt;  [60,] 21 0.88880 0.016680</span>
<span class="co">#&gt;  [61,] 21 0.89390 0.015920</span>
<span class="co">#&gt;  [62,] 21 0.89870 0.015200</span>
<span class="co">#&gt;  [63,] 21 0.90340 0.014510</span>
<span class="co">#&gt;  [64,] 21 0.90780 0.013850</span>
<span class="co">#&gt;  [65,] 21 0.91200 0.013220</span>
<span class="co">#&gt;  [66,] 21 0.91600 0.012620</span>
<span class="co">#&gt;  [67,] 21 0.91990 0.012040</span>
<span class="co">#&gt;  [68,] 21 0.92350 0.011500</span>
<span class="co">#&gt;  [69,] 21 0.92700 0.010970</span>
<span class="co">#&gt;  [70,] 21 0.93030 0.010470</span>
<span class="co">#&gt;  [71,] 22 0.93350 0.009998</span>
<span class="co">#&gt;  [72,] 22 0.93650 0.009544</span>
<span class="co">#&gt;  [73,] 22 0.93950 0.009110</span>
<span class="co">#&gt;  [74,] 22 0.94220 0.008696</span>
<span class="co">#&gt;  [75,] 23 0.94480 0.008301</span>
<span class="co">#&gt;  [76,] 23 0.94740 0.007923</span>
<span class="co">#&gt;  [77,] 22 0.94980 0.007563</span>
<span class="co">#&gt;  [78,] 22 0.95210 0.007220</span>
<span class="co">#&gt;  [79,] 22 0.95430 0.006891</span>
<span class="co">#&gt;  [80,] 22 0.95640 0.006578</span>
<span class="co">#&gt;  [81,] 22 0.95840 0.006279</span>
<span class="co">#&gt;  [82,] 22 0.96030 0.005994</span>
<span class="co">#&gt;  [83,] 22 0.96210 0.005721</span>
<span class="co">#&gt;  [84,] 22 0.96380 0.005461</span>
<span class="co">#&gt;  [85,] 22 0.96550 0.005213</span>
<span class="co">#&gt;  [86,] 21 0.96700 0.004976</span>
<span class="co">#&gt;  [87,] 21 0.96860 0.004750</span>
<span class="co">#&gt;  [88,] 21 0.97000 0.004534</span>
<span class="co">#&gt;  [89,] 21 0.97140 0.004328</span>
<span class="co">#&gt;  [90,] 21 0.97270 0.004131</span>
<span class="co">#&gt;  [91,] 21 0.97390 0.003943</span>
<span class="co">#&gt;  [92,] 21 0.97510 0.003764</span>
<span class="co">#&gt;  [93,] 21 0.97620 0.003593</span>
<span class="co">#&gt;  [94,] 21 0.97730 0.003430</span>
<span class="co">#&gt;  [95,] 21 0.97830 0.003274</span>
<span class="co">#&gt;  [96,] 21 0.97930 0.003125</span>
<span class="co">#&gt;  [97,] 21 0.98030 0.002983</span>
<span class="co">#&gt;  [98,] 21 0.98120 0.002848</span>
<span class="co">#&gt;  [99,] 21 0.98200 0.002718</span>
<span class="co">#&gt; [100,] 21 0.98280 0.002595</span>
<span class="co">#&gt; [1] "lambda min is at location 61"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 20"</span>
<span class="co">#&gt; [1] "use 20 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 40 cells and 201 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.825"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.725"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 44.9197860962567"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 34.2245989304813"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 42.8571428571429"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 85.7142857142857"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 5.26315789473684"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 54.8872180451128"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 72.5"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 60"</span>
<span class="co">#&gt; [1] "Total 40 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 460 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 20 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 20 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 201 genes for training model"</span>
<span class="co">#&gt; [1] "use 201 genes 40 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_2_3"</span>
<span class="co">#&gt; [1] "there are 20 cells in class 1_2_3 and 20 cells in class 4"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df    %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 0.00000 0.358100</span>
<span class="co">#&gt;   [2,]  1 0.03289 0.341800</span>
<span class="co">#&gt;   [3,]  1 0.06302 0.326200</span>
<span class="co">#&gt;   [4,]  1 0.09075 0.311400</span>
<span class="co">#&gt;   [5,]  1 0.11640 0.297300</span>
<span class="co">#&gt;   [6,]  1 0.14010 0.283800</span>
<span class="co">#&gt;   [7,]  1 0.16230 0.270900</span>
<span class="co">#&gt;   [8,]  1 0.18290 0.258500</span>
<span class="co">#&gt;   [9,]  1 0.20220 0.246800</span>
<span class="co">#&gt;  [10,]  1 0.22030 0.235600</span>
<span class="co">#&gt;  [11,]  1 0.23720 0.224900</span>
<span class="co">#&gt;  [12,]  2 0.25490 0.214600</span>
<span class="co">#&gt;  [13,]  2 0.27380 0.204900</span>
<span class="co">#&gt;  [14,]  2 0.29160 0.195600</span>
<span class="co">#&gt;  [15,]  2 0.30850 0.186700</span>
<span class="co">#&gt;  [16,]  2 0.32440 0.178200</span>
<span class="co">#&gt;  [17,]  3 0.33960 0.170100</span>
<span class="co">#&gt;  [18,]  3 0.35410 0.162400</span>
<span class="co">#&gt;  [19,]  3 0.36780 0.155000</span>
<span class="co">#&gt;  [20,]  4 0.38240 0.147900</span>
<span class="co">#&gt;  [21,]  4 0.39680 0.141200</span>
<span class="co">#&gt;  [22,]  6 0.41230 0.134800</span>
<span class="co">#&gt;  [23,]  8 0.42780 0.128700</span>
<span class="co">#&gt;  [24,]  8 0.44510 0.122800</span>
<span class="co">#&gt;  [25,]  8 0.46140 0.117200</span>
<span class="co">#&gt;  [26,]  7 0.47680 0.111900</span>
<span class="co">#&gt;  [27,]  7 0.49110 0.106800</span>
<span class="co">#&gt;  [28,]  7 0.50470 0.102000</span>
<span class="co">#&gt;  [29,]  8 0.51860 0.097340</span>
<span class="co">#&gt;  [30,]  8 0.53260 0.092920</span>
<span class="co">#&gt;  [31,]  9 0.54670 0.088690</span>
<span class="co">#&gt;  [32,]  9 0.56070 0.084660</span>
<span class="co">#&gt;  [33,] 11 0.57490 0.080810</span>
<span class="co">#&gt;  [34,] 11 0.59010 0.077140</span>
<span class="co">#&gt;  [35,] 11 0.60460 0.073630</span>
<span class="co">#&gt;  [36,] 12 0.61890 0.070290</span>
<span class="co">#&gt;  [37,] 12 0.63260 0.067090</span>
<span class="co">#&gt;  [38,] 12 0.64580 0.064040</span>
<span class="co">#&gt;  [39,] 12 0.65830 0.061130</span>
<span class="co">#&gt;  [40,] 12 0.67020 0.058350</span>
<span class="co">#&gt;  [41,] 12 0.68170 0.055700</span>
<span class="co">#&gt;  [42,] 13 0.69280 0.053170</span>
<span class="co">#&gt;  [43,] 13 0.70360 0.050750</span>
<span class="co">#&gt;  [44,] 14 0.71510 0.048450</span>
<span class="co">#&gt;  [45,] 15 0.72630 0.046240</span>
<span class="co">#&gt;  [46,] 16 0.73800 0.044140</span>
<span class="co">#&gt;  [47,] 17 0.74960 0.042140</span>
<span class="co">#&gt;  [48,] 17 0.76100 0.040220</span>
<span class="co">#&gt;  [49,] 16 0.77190 0.038390</span>
<span class="co">#&gt;  [50,] 17 0.78220 0.036650</span>
<span class="co">#&gt;  [51,] 18 0.79210 0.034980</span>
<span class="co">#&gt;  [52,] 18 0.80160 0.033390</span>
<span class="co">#&gt;  [53,] 18 0.81070 0.031870</span>
<span class="co">#&gt;  [54,] 17 0.81910 0.030430</span>
<span class="co">#&gt;  [55,] 17 0.82710 0.029040</span>
<span class="co">#&gt;  [56,] 16 0.83470 0.027720</span>
<span class="co">#&gt;  [57,] 17 0.84230 0.026460</span>
<span class="co">#&gt;  [58,] 18 0.84960 0.025260</span>
<span class="co">#&gt;  [59,] 18 0.85660 0.024110</span>
<span class="co">#&gt;  [60,] 19 0.86330 0.023020</span>
<span class="co">#&gt;  [61,] 19 0.86970 0.021970</span>
<span class="co">#&gt;  [62,] 19 0.87580 0.020970</span>
<span class="co">#&gt;  [63,] 19 0.88160 0.020020</span>
<span class="co">#&gt;  [64,] 18 0.88700 0.019110</span>
<span class="co">#&gt;  [65,] 18 0.89210 0.018240</span>
<span class="co">#&gt;  [66,] 18 0.89690 0.017410</span>
<span class="co">#&gt;  [67,] 18 0.90150 0.016620</span>
<span class="co">#&gt;  [68,] 18 0.90600 0.015860</span>
<span class="co">#&gt;  [69,] 19 0.91020 0.015140</span>
<span class="co">#&gt;  [70,] 19 0.91420 0.014450</span>
<span class="co">#&gt;  [71,] 19 0.91800 0.013800</span>
<span class="co">#&gt;  [72,] 20 0.92180 0.013170</span>
<span class="co">#&gt;  [73,] 19 0.92540 0.012570</span>
<span class="co">#&gt;  [74,] 19 0.92880 0.012000</span>
<span class="co">#&gt;  [75,] 19 0.93200 0.011460</span>
<span class="co">#&gt;  [76,] 19 0.93510 0.010930</span>
<span class="co">#&gt;  [77,] 19 0.93810 0.010440</span>
<span class="co">#&gt;  [78,] 19 0.94090 0.009963</span>
<span class="co">#&gt;  [79,] 19 0.94360 0.009510</span>
<span class="co">#&gt;  [80,] 18 0.94610 0.009078</span>
<span class="co">#&gt;  [81,] 18 0.94850 0.008665</span>
<span class="co">#&gt;  [82,] 18 0.95080 0.008272</span>
<span class="co">#&gt;  [83,] 18 0.95300 0.007896</span>
<span class="co">#&gt;  [84,] 18 0.95510 0.007537</span>
<span class="co">#&gt;  [85,] 18 0.95710 0.007194</span>
<span class="co">#&gt;  [86,] 18 0.95900 0.006867</span>
<span class="co">#&gt;  [87,] 18 0.96080 0.006555</span>
<span class="co">#&gt;  [88,] 18 0.96260 0.006257</span>
<span class="co">#&gt;  [89,] 18 0.96420 0.005973</span>
<span class="co">#&gt;  [90,] 18 0.96580 0.005701</span>
<span class="co">#&gt;  [91,] 18 0.96740 0.005442</span>
<span class="co">#&gt;  [92,] 17 0.96880 0.005195</span>
<span class="co">#&gt;  [93,] 17 0.97020 0.004959</span>
<span class="co">#&gt;  [94,] 17 0.97150 0.004733</span>
<span class="co">#&gt;  [95,] 18 0.97280 0.004518</span>
<span class="co">#&gt;  [96,] 18 0.97400 0.004313</span>
<span class="co">#&gt;  [97,] 18 0.97520 0.004117</span>
<span class="co">#&gt;  [98,] 18 0.97630 0.003930</span>
<span class="co">#&gt;  [99,] 18 0.97740 0.003751</span>
<span class="co">#&gt; [100,] 19 0.97840 0.003581</span>
<span class="co">#&gt; [1] "lambda min is at location 11"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 20"</span>
<span class="co">#&gt; [1] "use 20 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 40 cells and 201 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.65"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.75"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 49.1978609625668"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 187 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 8.02139037433155"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 100"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 140 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 23.5714285714286"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 100"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 133 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 10.5263157894737"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 100"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17201"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 201"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 201 genes and 40 cells. The trained model has 201 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "RNF207"  "AKR7A2"  "CAPZB"   "NUDC"    "EIF3I"   "LSM10"   "CAP1"   </span>
<span class="co">#&gt;  [8] "PRDX1"   "AKR1A1"  "GADD45A"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 60"</span>


<span class="co">#prepare table input for sankey plot </span>

LASSO_C1S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, 
                          <span class="dt">nPredSubpop =</span> <span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#7570b3"</span>) 

LASSO_C2S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, 
                          <span class="dt">nPredSubpop =</span> <span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#1b9e77"</span>)

LASSO_C3S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, 
                          <span class="dt">nPredSubpop =</span> <span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#e7298a"</span>)
                          
LASSO_C4S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">4</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat4, 
                          <span class="dt">nPredSubpop =</span> <span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])), 
                          <span class="dt">Nodes_group =</span><span class="st">"#00FFFF"</span>)

combined &lt;-<span class="st"> </span><span class="kw">rbind</span>(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2, LASSO_C4S2 )
combined &lt;-<span class="st"> </span>combined[<span class="kw">is.na</span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw">list</span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)], <span class="dt">Links=</span>combined[,<span class="kw">c</span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw">ncol</span>(combined))]) 

<span class="kw">library</span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw">as.vector</span>(<span class="kw">sort</span>(<span class="kw">unique</span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw">as.vector</span>(<span class="kw">sort</span>(<span class="kw">unique</span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw">unique</span>(<span class="kw">c</span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw">data.frame</span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw">as.numeric</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw">suppressMessages</span>(<span class="kw">library</span>(dplyr))
Color &lt;-<span class="st"> </span>combined <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/tally.html">count</a></span>(Node, <span class="dt">color=</span>NodeGroup) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="dv">2</span>)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color<span class="op">$</span>color

<span class="kw">suppressMessages</span>(<span class="kw">library</span>(networkD3))
p1&lt;-<span class="kw"><a href="http://www.rdocumentation.org/packages/networkD3/topics/sankeyNetwork">sankeyNetwork</a></span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,  <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, 
                  <span class="dt">fontSize =</span> <span class="dv">22</span> )
p1</code></pre></div>
<div id="htmlwidget-254b81233138ffe4e59f" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-254b81233138ffe4e59f">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2,3,3,3,3],"target":[0,1,2,3,0,1,2,3,0,1,2,3,0,1,2,3],"value":[95.4545454545455,23.9285714285714,78.9473684210526,71.25,5.3475935828877,99.6428571428572,50.375939849624,61.25,28.6096256684492,1.07142857142857,81.578947368421,12.5,47.0588235294118,71.4285714285714,52.6315789473684,86.25],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a","#00FFFF","#00FFFF","#00FFFF","#00FFFF"]},"nodes":{"name":["C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#7570b3","#1b9e77","#e7298a","#00FFFF"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters-in-two-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-two-samples" class="anchor"></a>4.3 Plot the relationship between clusters in two samples</h2>
<p><em>Here we look at one example use case to find relationship between clusters within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>
cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]
cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]
row_cluster &lt;-<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] <span class="co">#top 200 gene markers distinguishing cluster 1 </span>
LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1, <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 224 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 366 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 112 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 112 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 191 genes for training model"</span>
<span class="co">#&gt; [1] "use 191 genes 224 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3"</span>
<span class="co">#&gt; [1] "there are 112 cells in class 2_3 and 112 cells in class 1"</span>
<span class="co">#&gt; [1] "removing 2 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;       Df       %Dev    Lambda</span>
<span class="co">#&gt;  [1,]  0 -2.563e-15 2.944e-01</span>
<span class="co">#&gt;  [2,]  2  5.140e-02 2.683e-01</span>
<span class="co">#&gt;  [3,]  2  9.935e-02 2.445e-01</span>
<span class="co">#&gt;  [4,]  4  1.480e-01 2.227e-01</span>
<span class="co">#&gt;  [5,]  6  1.942e-01 2.030e-01</span>
<span class="co">#&gt;  [6,]  6  2.354e-01 1.849e-01</span>
<span class="co">#&gt;  [7,]  6  2.718e-01 1.685e-01</span>
<span class="co">#&gt;  [8,]  7  3.049e-01 1.535e-01</span>
<span class="co">#&gt;  [9,]  7  3.355e-01 1.399e-01</span>
<span class="co">#&gt; [10,]  7  3.630e-01 1.275e-01</span>
<span class="co">#&gt; [11,]  9  3.900e-01 1.161e-01</span>
<span class="co">#&gt; [12,] 10  4.156e-01 1.058e-01</span>
<span class="co">#&gt; [13,] 11  4.401e-01 9.642e-02</span>
<span class="co">#&gt; [14,] 12  4.639e-01 8.785e-02</span>
<span class="co">#&gt; [15,] 12  4.873e-01 8.005e-02</span>
<span class="co">#&gt; [16,] 12  5.086e-01 7.294e-02</span>
<span class="co">#&gt; [17,] 13  5.282e-01 6.646e-02</span>
<span class="co">#&gt; [18,] 15  5.470e-01 6.055e-02</span>
<span class="co">#&gt; [19,] 14  5.649e-01 5.517e-02</span>
<span class="co">#&gt; [20,] 18  5.834e-01 5.027e-02</span>
<span class="co">#&gt; [21,] 24  6.021e-01 4.581e-02</span>
<span class="co">#&gt; [22,] 26  6.224e-01 4.174e-02</span>
<span class="co">#&gt; [23,] 28  6.416e-01 3.803e-02</span>
<span class="co">#&gt; [24,] 29  6.596e-01 3.465e-02</span>
<span class="co">#&gt; [25,] 30  6.770e-01 3.157e-02</span>
<span class="co">#&gt; [26,] 34  6.941e-01 2.877e-02</span>
<span class="co">#&gt; [27,] 34  7.105e-01 2.621e-02</span>
<span class="co">#&gt; [28,] 38  7.264e-01 2.388e-02</span>
<span class="co">#&gt; [29,] 38  7.416e-01 2.176e-02</span>
<span class="co">#&gt; [30,] 40  7.560e-01 1.983e-02</span>
<span class="co">#&gt; [31,] 41  7.696e-01 1.807e-02</span>
<span class="co">#&gt; [32,] 42  7.827e-01 1.646e-02</span>
<span class="co">#&gt; [33,] 42  7.950e-01 1.500e-02</span>
<span class="co">#&gt; [34,] 42  8.066e-01 1.367e-02</span>
<span class="co">#&gt; [35,] 42  8.178e-01 1.245e-02</span>
<span class="co">#&gt; [36,] 42  8.290e-01 1.135e-02</span>
<span class="co">#&gt; [37,] 43  8.395e-01 1.034e-02</span>
<span class="co">#&gt; [38,] 46  8.499e-01 9.420e-03</span>
<span class="co">#&gt; [39,] 49  8.606e-01 8.583e-03</span>
<span class="co">#&gt; [40,] 51  8.712e-01 7.821e-03</span>
<span class="co">#&gt; [41,] 53  8.812e-01 7.126e-03</span>
<span class="co">#&gt; [42,] 56  8.909e-01 6.493e-03</span>
<span class="co">#&gt; [43,] 57  9.000e-01 5.916e-03</span>
<span class="co">#&gt; [44,] 57  9.085e-01 5.391e-03</span>
<span class="co">#&gt; [45,] 61  9.165e-01 4.912e-03</span>
<span class="co">#&gt; [46,] 62  9.239e-01 4.475e-03</span>
<span class="co">#&gt; [47,] 62  9.307e-01 4.078e-03</span>
<span class="co">#&gt; [48,] 62  9.369e-01 3.715e-03</span>
<span class="co">#&gt; [49,] 61  9.425e-01 3.385e-03</span>
<span class="co">#&gt; [50,] 62  9.476e-01 3.085e-03</span>
<span class="co">#&gt; [51,] 62  9.523e-01 2.811e-03</span>
<span class="co">#&gt; [52,] 64  9.566e-01 2.561e-03</span>
<span class="co">#&gt; [53,] 64  9.605e-01 2.333e-03</span>
<span class="co">#&gt; [54,] 64  9.640e-01 2.126e-03</span>
<span class="co">#&gt; [55,] 64  9.672e-01 1.937e-03</span>
<span class="co">#&gt; [56,] 64  9.701e-01 1.765e-03</span>
<span class="co">#&gt; [57,] 66  9.728e-01 1.608e-03</span>
<span class="co">#&gt; [58,] 66  9.752e-01 1.465e-03</span>
<span class="co">#&gt; [59,] 66  9.774e-01 1.335e-03</span>
<span class="co">#&gt; [60,] 66  9.795e-01 1.217e-03</span>
<span class="co">#&gt; [61,] 68  9.813e-01 1.109e-03</span>
<span class="co">#&gt; [62,] 68  9.830e-01 1.010e-03</span>
<span class="co">#&gt; [63,] 68  9.845e-01 9.204e-04</span>
<span class="co">#&gt; [64,] 68  9.859e-01 8.386e-04</span>
<span class="co">#&gt; [65,] 68  9.871e-01 7.641e-04</span>
<span class="co">#&gt; [66,] 68  9.883e-01 6.962e-04</span>
<span class="co">#&gt; [67,] 69  9.893e-01 6.344e-04</span>
<span class="co">#&gt; [68,] 69  9.902e-01 5.780e-04</span>
<span class="co">#&gt; [69,] 70  9.911e-01 5.267e-04</span>
<span class="co">#&gt; [70,] 71  9.919e-01 4.799e-04</span>
<span class="co">#&gt; [71,] 71  9.926e-01 4.372e-04</span>
<span class="co">#&gt; [72,] 72  9.933e-01 3.984e-04</span>
<span class="co">#&gt; [73,] 72  9.939e-01 3.630e-04</span>
<span class="co">#&gt; [74,] 73  9.944e-01 3.308e-04</span>
<span class="co">#&gt; [75,] 73  9.949e-01 3.014e-04</span>
<span class="co">#&gt; [76,] 73  9.954e-01 2.746e-04</span>
<span class="co">#&gt; [77,] 73  9.958e-01 2.502e-04</span>
<span class="co">#&gt; [78,] 73  9.961e-01 2.280e-04</span>
<span class="co">#&gt; [79,] 74  9.965e-01 2.077e-04</span>
<span class="co">#&gt; [80,] 75  9.968e-01 1.893e-04</span>
<span class="co">#&gt; [81,] 75  9.971e-01 1.725e-04</span>
<span class="co">#&gt; [82,] 75  9.973e-01 1.571e-04</span>
<span class="co">#&gt; [83,] 73  9.976e-01 1.432e-04</span>
<span class="co">#&gt; [84,] 73  9.978e-01 1.305e-04</span>
<span class="co">#&gt; [85,] 73  9.980e-01 1.189e-04</span>
<span class="co">#&gt; [86,] 73  9.982e-01 1.083e-04</span>
<span class="co">#&gt; [87,] 74  9.983e-01 9.869e-05</span>
<span class="co">#&gt; [88,] 75  9.985e-01 8.992e-05</span>
<span class="co">#&gt; [89,] 75  9.986e-01 8.193e-05</span>
<span class="co">#&gt; [90,] 75  9.987e-01 7.465e-05</span>
<span class="co">#&gt; [91,] 75  9.988e-01 6.802e-05</span>
<span class="co">#&gt; [92,] 75  9.989e-01 6.198e-05</span>
<span class="co">#&gt; [93,] 75  9.990e-01 5.647e-05</span>
<span class="co">#&gt; [1] "lambda min is at location 30"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 112"</span>
<span class="co">#&gt; [1] "use 112 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 224 cells and 189 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.870535714285714"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.584821428571429"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 187 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 65.7754010695187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 187 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 22.4598930481283"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 140 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 70"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 140 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 82.8571428571429"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 133 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 78.1954887218045"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 133 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 36.0902255639098"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 40 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 80"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 40 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 67.5"</span>
<span class="co">#&gt; [1] "Total 224 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 366 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 112 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 112 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 191 genes for training model"</span>
<span class="co">#&gt; [1] "use 191 genes 224 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_3"</span>
<span class="co">#&gt; [1] "there are 112 cells in class 2_3 and 112 cells in class 1"</span>
<span class="co">#&gt; [1] "removing 2 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;       Df       %Dev    Lambda</span>
<span class="co">#&gt;  [1,]  0 -2.563e-15 2.859e-01</span>
<span class="co">#&gt;  [2,]  4  6.198e-02 2.605e-01</span>
<span class="co">#&gt;  [3,]  5  1.211e-01 2.374e-01</span>
<span class="co">#&gt;  [4,]  5  1.729e-01 2.163e-01</span>
<span class="co">#&gt;  [5,]  5  2.180e-01 1.971e-01</span>
<span class="co">#&gt;  [6,]  5  2.578e-01 1.796e-01</span>
<span class="co">#&gt;  [7,]  6  2.936e-01 1.636e-01</span>
<span class="co">#&gt;  [8,]  8  3.272e-01 1.491e-01</span>
<span class="co">#&gt;  [9,]  8  3.591e-01 1.358e-01</span>
<span class="co">#&gt; [10,]  9  3.880e-01 1.238e-01</span>
<span class="co">#&gt; [11,]  9  4.149e-01 1.128e-01</span>
<span class="co">#&gt; [12,] 10  4.406e-01 1.028e-01</span>
<span class="co">#&gt; [13,] 11  4.672e-01 9.364e-02</span>
<span class="co">#&gt; [14,] 12  4.933e-01 8.532e-02</span>
<span class="co">#&gt; [15,] 12  5.177e-01 7.774e-02</span>
<span class="co">#&gt; [16,] 12  5.399e-01 7.083e-02</span>
<span class="co">#&gt; [17,] 14  5.607e-01 6.454e-02</span>
<span class="co">#&gt; [18,] 15  5.844e-01 5.881e-02</span>
<span class="co">#&gt; [19,] 16  6.062e-01 5.358e-02</span>
<span class="co">#&gt; [20,] 20  6.276e-01 4.882e-02</span>
<span class="co">#&gt; [21,] 20  6.483e-01 4.448e-02</span>
<span class="co">#&gt; [22,] 21  6.673e-01 4.053e-02</span>
<span class="co">#&gt; [23,] 21  6.853e-01 3.693e-02</span>
<span class="co">#&gt; [24,] 23  7.026e-01 3.365e-02</span>
<span class="co">#&gt; [25,] 26  7.194e-01 3.066e-02</span>
<span class="co">#&gt; [26,] 28  7.355e-01 2.794e-02</span>
<span class="co">#&gt; [27,] 32  7.507e-01 2.546e-02</span>
<span class="co">#&gt; [28,] 32  7.655e-01 2.319e-02</span>
<span class="co">#&gt; [29,] 34  7.800e-01 2.113e-02</span>
<span class="co">#&gt; [30,] 35  7.937e-01 1.926e-02</span>
<span class="co">#&gt; [31,] 38  8.069e-01 1.755e-02</span>
<span class="co">#&gt; [32,] 38  8.197e-01 1.599e-02</span>
<span class="co">#&gt; [33,] 38  8.316e-01 1.457e-02</span>
<span class="co">#&gt; [34,] 39  8.429e-01 1.327e-02</span>
<span class="co">#&gt; [35,] 39  8.535e-01 1.209e-02</span>
<span class="co">#&gt; [36,] 39  8.635e-01 1.102e-02</span>
<span class="co">#&gt; [37,] 43  8.733e-01 1.004e-02</span>
<span class="co">#&gt; [38,] 44  8.827e-01 9.148e-03</span>
<span class="co">#&gt; [39,] 45  8.918e-01 8.336e-03</span>
<span class="co">#&gt; [40,] 46  9.005e-01 7.595e-03</span>
<span class="co">#&gt; [41,] 47  9.086e-01 6.920e-03</span>
<span class="co">#&gt; [42,] 51  9.162e-01 6.306e-03</span>
<span class="co">#&gt; [43,] 54  9.233e-01 5.745e-03</span>
<span class="co">#&gt; [44,] 54  9.299e-01 5.235e-03</span>
<span class="co">#&gt; [45,] 55  9.359e-01 4.770e-03</span>
<span class="co">#&gt; [46,] 55  9.414e-01 4.346e-03</span>
<span class="co">#&gt; [47,] 56  9.465e-01 3.960e-03</span>
<span class="co">#&gt; [48,] 56  9.512e-01 3.608e-03</span>
<span class="co">#&gt; [49,] 58  9.554e-01 3.288e-03</span>
<span class="co">#&gt; [50,] 58  9.593e-01 2.996e-03</span>
<span class="co">#&gt; [51,] 58  9.629e-01 2.730e-03</span>
<span class="co">#&gt; [52,] 58  9.661e-01 2.487e-03</span>
<span class="co">#&gt; [53,] 59  9.691e-01 2.266e-03</span>
<span class="co">#&gt; [54,] 59  9.718e-01 2.065e-03</span>
<span class="co">#&gt; [55,] 59  9.743e-01 1.881e-03</span>
<span class="co">#&gt; [56,] 60  9.765e-01 1.714e-03</span>
<span class="co">#&gt; [57,] 59  9.786e-01 1.562e-03</span>
<span class="co">#&gt; [58,] 61  9.805e-01 1.423e-03</span>
<span class="co">#&gt; [59,] 60  9.822e-01 1.297e-03</span>
<span class="co">#&gt; [60,] 61  9.838e-01 1.182e-03</span>
<span class="co">#&gt; [61,] 62  9.852e-01 1.077e-03</span>
<span class="co">#&gt; [62,] 63  9.865e-01 9.809e-04</span>
<span class="co">#&gt; [63,] 63  9.877e-01 8.938e-04</span>
<span class="co">#&gt; [64,] 63  9.888e-01 8.144e-04</span>
<span class="co">#&gt; [65,] 63  9.898e-01 7.420e-04</span>
<span class="co">#&gt; [66,] 62  9.907e-01 6.761e-04</span>
<span class="co">#&gt; [67,] 62  9.915e-01 6.161e-04</span>
<span class="co">#&gt; [68,] 62  9.923e-01 5.613e-04</span>
<span class="co">#&gt; [69,] 62  9.929e-01 5.115e-04</span>
<span class="co">#&gt; [70,] 62  9.936e-01 4.660e-04</span>
<span class="co">#&gt; [71,] 62  9.941e-01 4.246e-04</span>
<span class="co">#&gt; [72,] 62  9.946e-01 3.869e-04</span>
<span class="co">#&gt; [73,] 62  9.951e-01 3.525e-04</span>
<span class="co">#&gt; [74,] 62  9.956e-01 3.212e-04</span>
<span class="co">#&gt; [75,] 62  9.959e-01 2.927e-04</span>
<span class="co">#&gt; [76,] 63  9.963e-01 2.667e-04</span>
<span class="co">#&gt; [77,] 63  9.966e-01 2.430e-04</span>
<span class="co">#&gt; [78,] 64  9.969e-01 2.214e-04</span>
<span class="co">#&gt; [79,] 64  9.972e-01 2.017e-04</span>
<span class="co">#&gt; [80,] 65  9.974e-01 1.838e-04</span>
<span class="co">#&gt; [81,] 65  9.977e-01 1.675e-04</span>
<span class="co">#&gt; [82,] 65  9.979e-01 1.526e-04</span>
<span class="co">#&gt; [83,] 65  9.981e-01 1.390e-04</span>
<span class="co">#&gt; [84,] 65  9.982e-01 1.267e-04</span>
<span class="co">#&gt; [85,] 65  9.984e-01 1.154e-04</span>
<span class="co">#&gt; [86,] 63  9.985e-01 1.052e-04</span>
<span class="co">#&gt; [87,] 63  9.987e-01 9.584e-05</span>
<span class="co">#&gt; [88,] 65  9.988e-01 8.732e-05</span>
<span class="co">#&gt; [89,] 65  9.989e-01 7.957e-05</span>
<span class="co">#&gt; [90,] 65  9.990e-01 7.250e-05</span>
<span class="co">#&gt; [91,] 65  9.991e-01 6.606e-05</span>
<span class="co">#&gt; [1] "lambda min is at location 22"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 112"</span>
<span class="co">#&gt; [1] "use 112 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 224 cells and 189 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.848214285714286"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.660714285714286"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 187 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 31.5508021390374"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 187 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 20.855614973262"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 140 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 100"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 140 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 87.1428571428571"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 133 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 90.9774436090226"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 133 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 38.3458646616541"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17213"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 189"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 40 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 95"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17219"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 183"</span>
<span class="co">#&gt; [1] "There are 6 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 6 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1" </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [6] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 189 genes and 40 cells. The trained model has 189 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "MXRA8"     "HMGN2"     "PRDX1"     "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "CKS1B"     "FDPS"     </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 47.5"</span>


c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 203 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 387 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 102 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 102 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 192 genes for training model"</span>
<span class="co">#&gt; [1] "use 192 genes 204 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_3"</span>
<span class="co">#&gt; [1] "there are 102 cells in class 1_3 and 102 cells in class 2"</span>
<span class="co">#&gt; [1] "removing 6 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;       Df      %Dev    Lambda</span>
<span class="co">#&gt;  [1,]  0 1.281e-15 3.194e-01</span>
<span class="co">#&gt;  [2,]  1 5.014e-02 2.910e-01</span>
<span class="co">#&gt;  [3,]  1 9.259e-02 2.651e-01</span>
<span class="co">#&gt;  [4,]  3 1.363e-01 2.416e-01</span>
<span class="co">#&gt;  [5,]  3 1.815e-01 2.201e-01</span>
<span class="co">#&gt;  [6,]  3 2.208e-01 2.006e-01</span>
<span class="co">#&gt;  [7,]  3 2.554e-01 1.827e-01</span>
<span class="co">#&gt;  [8,]  4 2.884e-01 1.665e-01</span>
<span class="co">#&gt;  [9,]  4 3.180e-01 1.517e-01</span>
<span class="co">#&gt; [10,]  4 3.442e-01 1.382e-01</span>
<span class="co">#&gt; [11,]  4 3.676e-01 1.260e-01</span>
<span class="co">#&gt; [12,]  4 3.885e-01 1.148e-01</span>
<span class="co">#&gt; [13,]  5 4.091e-01 1.046e-01</span>
<span class="co">#&gt; [14,]  8 4.325e-01 9.528e-02</span>
<span class="co">#&gt; [15,]  9 4.567e-01 8.682e-02</span>
<span class="co">#&gt; [16,]  9 4.794e-01 7.911e-02</span>
<span class="co">#&gt; [17,] 11 5.001e-01 7.208e-02</span>
<span class="co">#&gt; [18,] 11 5.206e-01 6.568e-02</span>
<span class="co">#&gt; [19,] 13 5.395e-01 5.984e-02</span>
<span class="co">#&gt; [20,] 15 5.585e-01 5.453e-02</span>
<span class="co">#&gt; [21,] 16 5.765e-01 4.968e-02</span>
<span class="co">#&gt; [22,] 19 5.957e-01 4.527e-02</span>
<span class="co">#&gt; [23,] 20 6.144e-01 4.125e-02</span>
<span class="co">#&gt; [24,] 21 6.322e-01 3.758e-02</span>
<span class="co">#&gt; [25,] 23 6.505e-01 3.424e-02</span>
<span class="co">#&gt; [26,] 23 6.676e-01 3.120e-02</span>
<span class="co">#&gt; [27,] 25 6.843e-01 2.843e-02</span>
<span class="co">#&gt; [28,] 27 7.009e-01 2.590e-02</span>
<span class="co">#&gt; [29,] 29 7.167e-01 2.360e-02</span>
<span class="co">#&gt; [30,] 32 7.322e-01 2.151e-02</span>
<span class="co">#&gt; [31,] 33 7.483e-01 1.960e-02</span>
<span class="co">#&gt; [32,] 36 7.647e-01 1.785e-02</span>
<span class="co">#&gt; [33,] 37 7.806e-01 1.627e-02</span>
<span class="co">#&gt; [34,] 41 7.958e-01 1.482e-02</span>
<span class="co">#&gt; [35,] 43 8.100e-01 1.351e-02</span>
<span class="co">#&gt; [36,] 44 8.235e-01 1.231e-02</span>
<span class="co">#&gt; [37,] 44 8.362e-01 1.121e-02</span>
<span class="co">#&gt; [38,] 48 8.482e-01 1.022e-02</span>
<span class="co">#&gt; [39,] 48 8.599e-01 9.309e-03</span>
<span class="co">#&gt; [40,] 48 8.708e-01 8.482e-03</span>
<span class="co">#&gt; [41,] 50 8.809e-01 7.729e-03</span>
<span class="co">#&gt; [42,] 52 8.904e-01 7.042e-03</span>
<span class="co">#&gt; [43,] 54 8.994e-01 6.417e-03</span>
<span class="co">#&gt; [44,] 53 9.076e-01 5.847e-03</span>
<span class="co">#&gt; [45,] 55 9.152e-01 5.327e-03</span>
<span class="co">#&gt; [46,] 55 9.223e-01 4.854e-03</span>
<span class="co">#&gt; [47,] 56 9.288e-01 4.423e-03</span>
<span class="co">#&gt; [48,] 56 9.348e-01 4.030e-03</span>
<span class="co">#&gt; [49,] 57 9.404e-01 3.672e-03</span>
<span class="co">#&gt; [50,] 58 9.455e-01 3.346e-03</span>
<span class="co">#&gt; [51,] 57 9.502e-01 3.048e-03</span>
<span class="co">#&gt; [52,] 57 9.545e-01 2.778e-03</span>
<span class="co">#&gt; [53,] 58 9.585e-01 2.531e-03</span>
<span class="co">#&gt; [54,] 58 9.621e-01 2.306e-03</span>
<span class="co">#&gt; [55,] 59 9.654e-01 2.101e-03</span>
<span class="co">#&gt; [56,] 60 9.685e-01 1.914e-03</span>
<span class="co">#&gt; [57,] 59 9.713e-01 1.744e-03</span>
<span class="co">#&gt; [58,] 59 9.738e-01 1.589e-03</span>
<span class="co">#&gt; [59,] 62 9.761e-01 1.448e-03</span>
<span class="co">#&gt; [60,] 62 9.782e-01 1.320e-03</span>
<span class="co">#&gt; [61,] 62 9.802e-01 1.202e-03</span>
<span class="co">#&gt; [62,] 63 9.819e-01 1.096e-03</span>
<span class="co">#&gt; [63,] 62 9.835e-01 9.982e-04</span>
<span class="co">#&gt; [64,] 64 9.850e-01 9.095e-04</span>
<span class="co">#&gt; [65,] 64 9.863e-01 8.287e-04</span>
<span class="co">#&gt; [66,] 64 9.875e-01 7.551e-04</span>
<span class="co">#&gt; [67,] 64 9.886e-01 6.880e-04</span>
<span class="co">#&gt; [68,] 64 9.896e-01 6.269e-04</span>
<span class="co">#&gt; [69,] 65 9.905e-01 5.712e-04</span>
<span class="co">#&gt; [70,] 65 9.914e-01 5.205e-04</span>
<span class="co">#&gt; [71,] 65 9.921e-01 4.742e-04</span>
<span class="co">#&gt; [72,] 65 9.928e-01 4.321e-04</span>
<span class="co">#&gt; [73,] 66 9.935e-01 3.937e-04</span>
<span class="co">#&gt; [74,] 66 9.940e-01 3.587e-04</span>
<span class="co">#&gt; [75,] 65 9.946e-01 3.269e-04</span>
<span class="co">#&gt; [76,] 64 9.951e-01 2.978e-04</span>
<span class="co">#&gt; [77,] 64 9.955e-01 2.714e-04</span>
<span class="co">#&gt; [78,] 65 9.959e-01 2.473e-04</span>
<span class="co">#&gt; [79,] 65 9.962e-01 2.253e-04</span>
<span class="co">#&gt; [80,] 65 9.966e-01 2.053e-04</span>
<span class="co">#&gt; [81,] 65 9.969e-01 1.870e-04</span>
<span class="co">#&gt; [82,] 65 9.972e-01 1.704e-04</span>
<span class="co">#&gt; [83,] 65 9.974e-01 1.553e-04</span>
<span class="co">#&gt; [84,] 65 9.976e-01 1.415e-04</span>
<span class="co">#&gt; [85,] 65 9.978e-01 1.289e-04</span>
<span class="co">#&gt; [86,] 65 9.980e-01 1.175e-04</span>
<span class="co">#&gt; [87,] 66 9.982e-01 1.070e-04</span>
<span class="co">#&gt; [88,] 66 9.984e-01 9.753e-05</span>
<span class="co">#&gt; [89,] 66 9.985e-01 8.886e-05</span>
<span class="co">#&gt; [90,] 65 9.986e-01 8.097e-05</span>
<span class="co">#&gt; [91,] 66 9.988e-01 7.378e-05</span>
<span class="co">#&gt; [92,] 66 9.989e-01 6.722e-05</span>
<span class="co">#&gt; [93,] 66 9.990e-01 6.125e-05</span>
<span class="co">#&gt; [94,] 66 9.991e-01 5.581e-05</span>
<span class="co">#&gt; [1] "lambda min is at location 25"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 101"</span>
<span class="co">#&gt; [1] "use 102 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 203 cells and 186 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.896551724137931"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.625615763546798"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17216"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 186"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 187 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 80.7486631016043"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17221"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 181"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 187 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 97.8609625668449"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17216"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 186"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 140 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 11.4285714285714"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17221"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 181"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 140 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 91.4285714285714"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17216"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 186"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 133 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 69.9248120300752"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17221"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 181"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 133 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 88.7218045112782"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17216"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 186"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 40 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 62.5"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17221"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 181"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 186 genes and 40 cells. The trained model has 186 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 95"</span>
<span class="co">#&gt; [1] "Total 203 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 387 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 102 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 102 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 192 genes for training model"</span>
<span class="co">#&gt; [1] "use 192 genes 204 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 1_3"</span>
<span class="co">#&gt; [1] "there are 102 cells in class 1_3 and 102 cells in class 2"</span>
<span class="co">#&gt; [1] "removing 5 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;       Df      %Dev    Lambda</span>
<span class="co">#&gt;  [1,]  0 1.281e-15 3.176e-01</span>
<span class="co">#&gt;  [2,]  1 4.963e-02 2.894e-01</span>
<span class="co">#&gt;  [3,]  1 9.181e-02 2.637e-01</span>
<span class="co">#&gt;  [4,]  1 1.283e-01 2.403e-01</span>
<span class="co">#&gt;  [5,]  3 1.675e-01 2.189e-01</span>
<span class="co">#&gt;  [6,]  3 2.093e-01 1.995e-01</span>
<span class="co">#&gt;  [7,]  4 2.471e-01 1.818e-01</span>
<span class="co">#&gt;  [8,]  4 2.811e-01 1.656e-01</span>
<span class="co">#&gt;  [9,]  5 3.110e-01 1.509e-01</span>
<span class="co">#&gt; [10,]  5 3.383e-01 1.375e-01</span>
<span class="co">#&gt; [11,]  6 3.668e-01 1.253e-01</span>
<span class="co">#&gt; [12,]  6 3.938e-01 1.141e-01</span>
<span class="co">#&gt; [13,]  6 4.180e-01 1.040e-01</span>
<span class="co">#&gt; [14,]  7 4.399e-01 9.477e-02</span>
<span class="co">#&gt; [15,]  8 4.617e-01 8.635e-02</span>
<span class="co">#&gt; [16,]  8 4.816e-01 7.868e-02</span>
<span class="co">#&gt; [17,] 11 5.018e-01 7.169e-02</span>
<span class="co">#&gt; [18,] 12 5.207e-01 6.532e-02</span>
<span class="co">#&gt; [19,] 15 5.393e-01 5.952e-02</span>
<span class="co">#&gt; [20,] 16 5.584e-01 5.423e-02</span>
<span class="co">#&gt; [21,] 16 5.756e-01 4.941e-02</span>
<span class="co">#&gt; [22,] 16 5.911e-01 4.502e-02</span>
<span class="co">#&gt; [23,] 16 6.052e-01 4.102e-02</span>
<span class="co">#&gt; [24,] 17 6.186e-01 3.738e-02</span>
<span class="co">#&gt; [25,] 19 6.319e-01 3.406e-02</span>
<span class="co">#&gt; [26,] 19 6.448e-01 3.103e-02</span>
<span class="co">#&gt; [27,] 20 6.567e-01 2.827e-02</span>
<span class="co">#&gt; [28,] 21 6.676e-01 2.576e-02</span>
<span class="co">#&gt; [29,] 24 6.801e-01 2.347e-02</span>
<span class="co">#&gt; [30,] 27 6.935e-01 2.139e-02</span>
<span class="co">#&gt; [31,] 32 7.080e-01 1.949e-02</span>
<span class="co">#&gt; [32,] 35 7.235e-01 1.776e-02</span>
<span class="co">#&gt; [33,] 36 7.398e-01 1.618e-02</span>
<span class="co">#&gt; [34,] 37 7.551e-01 1.474e-02</span>
<span class="co">#&gt; [35,] 42 7.707e-01 1.343e-02</span>
<span class="co">#&gt; [36,] 46 7.870e-01 1.224e-02</span>
<span class="co">#&gt; [37,] 47 8.024e-01 1.115e-02</span>
<span class="co">#&gt; [38,] 50 8.171e-01 1.016e-02</span>
<span class="co">#&gt; [39,] 51 8.318e-01 9.259e-03</span>
<span class="co">#&gt; [40,] 53 8.454e-01 8.436e-03</span>
<span class="co">#&gt; [41,] 55 8.581e-01 7.687e-03</span>
<span class="co">#&gt; [42,] 55 8.700e-01 7.004e-03</span>
<span class="co">#&gt; [43,] 59 8.812e-01 6.382e-03</span>
<span class="co">#&gt; [44,] 60 8.917e-01 5.815e-03</span>
<span class="co">#&gt; [45,] 60 9.011e-01 5.298e-03</span>
<span class="co">#&gt; [46,] 62 9.099e-01 4.828e-03</span>
<span class="co">#&gt; [47,] 64 9.178e-01 4.399e-03</span>
<span class="co">#&gt; [48,] 67 9.251e-01 4.008e-03</span>
<span class="co">#&gt; [49,] 68 9.318e-01 3.652e-03</span>
<span class="co">#&gt; [50,] 69 9.378e-01 3.327e-03</span>
<span class="co">#&gt; [51,] 69 9.433e-01 3.032e-03</span>
<span class="co">#&gt; [52,] 70 9.484e-01 2.762e-03</span>
<span class="co">#&gt; [53,] 72 9.530e-01 2.517e-03</span>
<span class="co">#&gt; [54,] 72 9.573e-01 2.293e-03</span>
<span class="co">#&gt; [55,] 73 9.610e-01 2.090e-03</span>
<span class="co">#&gt; [56,] 73 9.645e-01 1.904e-03</span>
<span class="co">#&gt; [57,] 73 9.677e-01 1.735e-03</span>
<span class="co">#&gt; [58,] 73 9.707e-01 1.581e-03</span>
<span class="co">#&gt; [59,] 73 9.733e-01 1.440e-03</span>
<span class="co">#&gt; [60,] 73 9.756e-01 1.312e-03</span>
<span class="co">#&gt; [61,] 73 9.778e-01 1.196e-03</span>
<span class="co">#&gt; [62,] 74 9.798e-01 1.090e-03</span>
<span class="co">#&gt; [63,] 74 9.816e-01 9.928e-04</span>
<span class="co">#&gt; [64,] 75 9.832e-01 9.046e-04</span>
<span class="co">#&gt; [65,] 75 9.847e-01 8.242e-04</span>
<span class="co">#&gt; [66,] 77 9.861e-01 7.510e-04</span>
<span class="co">#&gt; [67,] 77 9.873e-01 6.843e-04</span>
<span class="co">#&gt; [68,] 77 9.884e-01 6.235e-04</span>
<span class="co">#&gt; [69,] 77 9.895e-01 5.681e-04</span>
<span class="co">#&gt; [70,] 77 9.904e-01 5.176e-04</span>
<span class="co">#&gt; [71,] 77 9.913e-01 4.717e-04</span>
<span class="co">#&gt; [72,] 77 9.920e-01 4.298e-04</span>
<span class="co">#&gt; [73,] 78 9.927e-01 3.916e-04</span>
<span class="co">#&gt; [74,] 78 9.934e-01 3.568e-04</span>
<span class="co">#&gt; [75,] 78 9.940e-01 3.251e-04</span>
<span class="co">#&gt; [76,] 78 9.945e-01 2.962e-04</span>
<span class="co">#&gt; [77,] 78 9.950e-01 2.699e-04</span>
<span class="co">#&gt; [78,] 78 9.954e-01 2.459e-04</span>
<span class="co">#&gt; [79,] 78 9.958e-01 2.241e-04</span>
<span class="co">#&gt; [80,] 79 9.962e-01 2.042e-04</span>
<span class="co">#&gt; [81,] 79 9.965e-01 1.860e-04</span>
<span class="co">#&gt; [82,] 79 9.968e-01 1.695e-04</span>
<span class="co">#&gt; [83,] 79 9.971e-01 1.544e-04</span>
<span class="co">#&gt; [84,] 79 9.974e-01 1.407e-04</span>
<span class="co">#&gt; [85,] 79 9.976e-01 1.282e-04</span>
<span class="co">#&gt; [86,] 80 9.978e-01 1.168e-04</span>
<span class="co">#&gt; [87,] 80 9.980e-01 1.065e-04</span>
<span class="co">#&gt; [88,] 81 9.982e-01 9.700e-05</span>
<span class="co">#&gt; [89,] 81 9.983e-01 8.838e-05</span>
<span class="co">#&gt; [90,] 82 9.985e-01 8.053e-05</span>
<span class="co">#&gt; [91,] 82 9.986e-01 7.337e-05</span>
<span class="co">#&gt; [92,] 82 9.987e-01 6.686e-05</span>
<span class="co">#&gt; [93,] 82 9.989e-01 6.092e-05</span>
<span class="co">#&gt; [94,] 82 9.990e-01 5.550e-05</span>
<span class="co">#&gt; [95,] 82 9.990e-01 5.057e-05</span>
<span class="co">#&gt; [1] "lambda min is at location 20"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 101"</span>
<span class="co">#&gt; [1] "use 102 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 203 cells and 187 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.896551724137931"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.556650246305419"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17215"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 187"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 187 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 55.0802139037433"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17220"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 182"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 187 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 71.1229946524064"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17215"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 187"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 140 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 41.4285714285714"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17220"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 182"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 140 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 85"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17215"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 187"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 133 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 61.6541353383459"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17220"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 182"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 133 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 76.6917293233083"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17215"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 187"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 40 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 55"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17220"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 182"</span>
<span class="co">#&gt; [1] "There are 5 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 5 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are HLA-C"        </span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"        </span>
<span class="co">#&gt; [3] "genes to be added to target subpop are RP11-419I17.1"</span>
<span class="co">#&gt; [4] "genes to be added to target subpop are RP11-834C11.4"</span>
<span class="co">#&gt; [5] "genes to be added to target subpop are GATA6-AS1"    </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 187 genes and 40 cells. The trained model has 187 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"      "MARCKSL1"  "GNG5"      "SH3GLB1"   "LINC01356"</span>
<span class="co">#&gt;  [6] "MLLT11"    "S100A16"   "TPM3"      "RAB25"     "APOA2"    </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 90"</span>

c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw">bootstrap_scGPS</span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,
     <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)
<span class="co">#&gt; [1] "Total 163 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 427 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 82 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 82 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 196 genes for training model"</span>
<span class="co">#&gt; [1] "use 196 genes 164 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_1"</span>
<span class="co">#&gt; [1] "there are 82 cells in class 2_1 and 82 cells in class 3"</span>
<span class="co">#&gt; [1] "removing 3 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df      %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 2.403e-15 0.166300</span>
<span class="co">#&gt;   [2,]  2 8.918e-03 0.158800</span>
<span class="co">#&gt;   [3,]  2 1.994e-02 0.151600</span>
<span class="co">#&gt;   [4,]  4 3.431e-02 0.144700</span>
<span class="co">#&gt;   [5,]  5 4.968e-02 0.138100</span>
<span class="co">#&gt;   [6,]  5 6.411e-02 0.131800</span>
<span class="co">#&gt;   [7,]  5 7.749e-02 0.125800</span>
<span class="co">#&gt;   [8,]  6 9.008e-02 0.120100</span>
<span class="co">#&gt;   [9,]  7 1.049e-01 0.114700</span>
<span class="co">#&gt;  [10,]  9 1.195e-01 0.109400</span>
<span class="co">#&gt;  [11,]  9 1.334e-01 0.104500</span>
<span class="co">#&gt;  [12,] 12 1.469e-01 0.099720</span>
<span class="co">#&gt;  [13,] 12 1.605e-01 0.095190</span>
<span class="co">#&gt;  [14,] 13 1.738e-01 0.090860</span>
<span class="co">#&gt;  [15,] 14 1.867e-01 0.086730</span>
<span class="co">#&gt;  [16,] 16 2.020e-01 0.082790</span>
<span class="co">#&gt;  [17,] 16 2.170e-01 0.079030</span>
<span class="co">#&gt;  [18,] 17 2.316e-01 0.075440</span>
<span class="co">#&gt;  [19,] 17 2.454e-01 0.072010</span>
<span class="co">#&gt;  [20,] 18 2.592e-01 0.068730</span>
<span class="co">#&gt;  [21,] 20 2.734e-01 0.065610</span>
<span class="co">#&gt;  [22,] 20 2.871e-01 0.062630</span>
<span class="co">#&gt;  [23,] 20 3.001e-01 0.059780</span>
<span class="co">#&gt;  [24,] 23 3.125e-01 0.057060</span>
<span class="co">#&gt;  [25,] 23 3.255e-01 0.054470</span>
<span class="co">#&gt;  [26,] 22 3.388e-01 0.052000</span>
<span class="co">#&gt;  [27,] 24 3.517e-01 0.049630</span>
<span class="co">#&gt;  [28,] 24 3.648e-01 0.047380</span>
<span class="co">#&gt;  [29,] 26 3.778e-01 0.045220</span>
<span class="co">#&gt;  [30,] 29 3.911e-01 0.043170</span>
<span class="co">#&gt;  [31,] 31 4.045e-01 0.041210</span>
<span class="co">#&gt;  [32,] 30 4.175e-01 0.039330</span>
<span class="co">#&gt;  [33,] 32 4.299e-01 0.037540</span>
<span class="co">#&gt;  [34,] 34 4.427e-01 0.035840</span>
<span class="co">#&gt;  [35,] 36 4.578e-01 0.034210</span>
<span class="co">#&gt;  [36,] 36 4.724e-01 0.032650</span>
<span class="co">#&gt;  [37,] 37 4.864e-01 0.031170</span>
<span class="co">#&gt;  [38,] 38 4.998e-01 0.029750</span>
<span class="co">#&gt;  [39,] 38 5.129e-01 0.028400</span>
<span class="co">#&gt;  [40,] 41 5.258e-01 0.027110</span>
<span class="co">#&gt;  [41,] 44 5.391e-01 0.025880</span>
<span class="co">#&gt;  [42,] 44 5.522e-01 0.024700</span>
<span class="co">#&gt;  [43,] 45 5.650e-01 0.023580</span>
<span class="co">#&gt;  [44,] 45 5.772e-01 0.022510</span>
<span class="co">#&gt;  [45,] 46 5.889e-01 0.021480</span>
<span class="co">#&gt;  [46,] 49 6.006e-01 0.020510</span>
<span class="co">#&gt;  [47,] 53 6.126e-01 0.019580</span>
<span class="co">#&gt;  [48,] 55 6.253e-01 0.018690</span>
<span class="co">#&gt;  [49,] 57 6.377e-01 0.017840</span>
<span class="co">#&gt;  [50,] 59 6.499e-01 0.017030</span>
<span class="co">#&gt;  [51,] 62 6.627e-01 0.016250</span>
<span class="co">#&gt;  [52,] 63 6.755e-01 0.015510</span>
<span class="co">#&gt;  [53,] 65 6.879e-01 0.014810</span>
<span class="co">#&gt;  [54,] 66 7.002e-01 0.014140</span>
<span class="co">#&gt;  [55,] 66 7.120e-01 0.013490</span>
<span class="co">#&gt;  [56,] 69 7.236e-01 0.012880</span>
<span class="co">#&gt;  [57,] 70 7.353e-01 0.012290</span>
<span class="co">#&gt;  [58,] 71 7.467e-01 0.011740</span>
<span class="co">#&gt;  [59,] 72 7.578e-01 0.011200</span>
<span class="co">#&gt;  [60,] 73 7.684e-01 0.010690</span>
<span class="co">#&gt;  [61,] 74 7.785e-01 0.010210</span>
<span class="co">#&gt;  [62,] 77 7.884e-01 0.009743</span>
<span class="co">#&gt;  [63,] 77 7.979e-01 0.009300</span>
<span class="co">#&gt;  [64,] 77 8.070e-01 0.008877</span>
<span class="co">#&gt;  [65,] 77 8.156e-01 0.008474</span>
<span class="co">#&gt;  [66,] 77 8.239e-01 0.008089</span>
<span class="co">#&gt;  [67,] 79 8.319e-01 0.007721</span>
<span class="co">#&gt;  [68,] 81 8.396e-01 0.007370</span>
<span class="co">#&gt;  [69,] 82 8.471e-01 0.007035</span>
<span class="co">#&gt;  [70,] 82 8.542e-01 0.006715</span>
<span class="co">#&gt;  [71,] 82 8.609e-01 0.006410</span>
<span class="co">#&gt;  [72,] 84 8.673e-01 0.006119</span>
<span class="co">#&gt;  [73,] 84 8.735e-01 0.005841</span>
<span class="co">#&gt;  [74,] 84 8.794e-01 0.005575</span>
<span class="co">#&gt;  [75,] 84 8.850e-01 0.005322</span>
<span class="co">#&gt;  [76,] 85 8.904e-01 0.005080</span>
<span class="co">#&gt;  [77,] 85 8.955e-01 0.004849</span>
<span class="co">#&gt;  [78,] 85 9.004e-01 0.004629</span>
<span class="co">#&gt;  [79,] 85 9.050e-01 0.004418</span>
<span class="co">#&gt;  [80,] 85 9.094e-01 0.004217</span>
<span class="co">#&gt;  [81,] 85 9.136e-01 0.004026</span>
<span class="co">#&gt;  [82,] 85 9.176e-01 0.003843</span>
<span class="co">#&gt;  [83,] 85 9.214e-01 0.003668</span>
<span class="co">#&gt;  [84,] 85 9.251e-01 0.003501</span>
<span class="co">#&gt;  [85,] 85 9.285e-01 0.003342</span>
<span class="co">#&gt;  [86,] 85 9.318e-01 0.003190</span>
<span class="co">#&gt;  [87,] 86 9.349e-01 0.003045</span>
<span class="co">#&gt;  [88,] 86 9.379e-01 0.002907</span>
<span class="co">#&gt;  [89,] 87 9.408e-01 0.002775</span>
<span class="co">#&gt;  [90,] 88 9.435e-01 0.002649</span>
<span class="co">#&gt;  [91,] 88 9.461e-01 0.002528</span>
<span class="co">#&gt;  [92,] 88 9.486e-01 0.002413</span>
<span class="co">#&gt;  [93,] 88 9.509e-01 0.002304</span>
<span class="co">#&gt;  [94,] 88 9.532e-01 0.002199</span>
<span class="co">#&gt;  [95,] 89 9.554e-01 0.002099</span>
<span class="co">#&gt;  [96,] 88 9.574e-01 0.002004</span>
<span class="co">#&gt;  [97,] 88 9.594e-01 0.001913</span>
<span class="co">#&gt;  [98,] 89 9.613e-01 0.001826</span>
<span class="co">#&gt;  [99,] 89 9.630e-01 0.001743</span>
<span class="co">#&gt; [100,] 89 9.647e-01 0.001663</span>
<span class="co">#&gt; [1] "lambda min is at location 28"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 81"</span>
<span class="co">#&gt; [1] "use 82 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 163 cells and 193 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.705521472392638"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.570552147239264"</span>
<span class="co">#&gt; [1] "done training for bootstrap 1, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17209"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 193"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 187 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 45.9893048128342"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17211"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 191"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 187 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 22.4598930481283"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17209"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 193"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 140 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 3.57142857142857"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17211"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 191"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 140 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 26.4285714285714"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17209"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 193"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 133 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 11.2781954887218"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17211"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 191"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 133 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 53.3834586466165"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17209"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 193"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 40 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 5"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17211"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 191"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 193 genes and 40 cells. The trained model has 193 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "DDOST"     "STMN1"     "HMGN2"     "CDC20"     "NASP"     </span>
<span class="co">#&gt;  [6] "CYR61"     "CNN3"      "ATP5F1"    "RHOC"      "LINC01356"</span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Total 163 cells as source subpop"</span>
<span class="co">#&gt; [1] "Total 427 cells in remaining subpops"</span>
<span class="co">#&gt; [1] "subsampling 82 cells for training source subpop"</span>
<span class="co">#&gt; [1] "subsampling 82 cells in remaining subpops for training"</span>
<span class="co">#&gt; [1] "use 196 genes for training model"</span>
<span class="co">#&gt; [1] "use 196 genes 164 cells for testing model"</span>
<span class="co">#&gt; [1] "rename remaining subpops to 2_1"</span>
<span class="co">#&gt; [1] "there are 82 cells in class 2_1 and 82 cells in class 3"</span>
<span class="co">#&gt; [1] "removing 2 genes with no variance"</span>
<span class="co">#&gt; [1] "standardizing prediction/target dataset"</span>
<span class="co">#&gt; [1] "performning elasticnet model training..."</span>
<span class="co">#&gt; [1] "performning LDA model training..."</span>
<span class="co">#&gt; [1] "extracting deviance and best gene features..."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = as.matrix(dataset[, -which(colnames(dataset) == "Cluster_class")]),      y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df      %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 2.403e-15 0.165100</span>
<span class="co">#&gt;   [2,]  1 6.993e-03 0.157600</span>
<span class="co">#&gt;   [3,]  1 1.338e-02 0.150500</span>
<span class="co">#&gt;   [4,]  1 1.922e-02 0.143600</span>
<span class="co">#&gt;   [5,]  3 2.574e-02 0.137100</span>
<span class="co">#&gt;   [6,]  3 3.648e-02 0.130900</span>
<span class="co">#&gt;   [7,]  3 4.634e-02 0.124900</span>
<span class="co">#&gt;   [8,]  4 5.553e-02 0.119200</span>
<span class="co">#&gt;   [9,]  5 6.486e-02 0.113800</span>
<span class="co">#&gt;  [10,]  8 7.587e-02 0.108700</span>
<span class="co">#&gt;  [11,] 10 9.081e-02 0.103700</span>
<span class="co">#&gt;  [12,] 13 1.088e-01 0.099000</span>
<span class="co">#&gt;  [13,] 16 1.289e-01 0.094500</span>
<span class="co">#&gt;  [14,] 19 1.508e-01 0.090200</span>
<span class="co">#&gt;  [15,] 19 1.717e-01 0.086100</span>
<span class="co">#&gt;  [16,] 20 1.916e-01 0.082190</span>
<span class="co">#&gt;  [17,] 22 2.113e-01 0.078450</span>
<span class="co">#&gt;  [18,] 22 2.308e-01 0.074890</span>
<span class="co">#&gt;  [19,] 22 2.492e-01 0.071490</span>
<span class="co">#&gt;  [20,] 25 2.678e-01 0.068240</span>
<span class="co">#&gt;  [21,] 26 2.867e-01 0.065130</span>
<span class="co">#&gt;  [22,] 26 3.049e-01 0.062170</span>
<span class="co">#&gt;  [23,] 27 3.225e-01 0.059350</span>
<span class="co">#&gt;  [24,] 30 3.402e-01 0.056650</span>
<span class="co">#&gt;  [25,] 30 3.575e-01 0.054080</span>
<span class="co">#&gt;  [26,] 32 3.748e-01 0.051620</span>
<span class="co">#&gt;  [27,] 33 3.915e-01 0.049270</span>
<span class="co">#&gt;  [28,] 35 4.080e-01 0.047030</span>
<span class="co">#&gt;  [29,] 37 4.246e-01 0.044890</span>
<span class="co">#&gt;  [30,] 40 4.410e-01 0.042850</span>
<span class="co">#&gt;  [31,] 40 4.576e-01 0.040910</span>
<span class="co">#&gt;  [32,] 42 4.735e-01 0.039050</span>
<span class="co">#&gt;  [33,] 44 4.887e-01 0.037270</span>
<span class="co">#&gt;  [34,] 44 5.037e-01 0.035580</span>
<span class="co">#&gt;  [35,] 43 5.180e-01 0.033960</span>
<span class="co">#&gt;  [36,] 43 5.316e-01 0.032420</span>
<span class="co">#&gt;  [37,] 43 5.446e-01 0.030940</span>
<span class="co">#&gt;  [38,] 45 5.572e-01 0.029540</span>
<span class="co">#&gt;  [39,] 46 5.694e-01 0.028200</span>
<span class="co">#&gt;  [40,] 47 5.812e-01 0.026910</span>
<span class="co">#&gt;  [41,] 48 5.930e-01 0.025690</span>
<span class="co">#&gt;  [42,] 51 6.050e-01 0.024520</span>
<span class="co">#&gt;  [43,] 52 6.170e-01 0.023410</span>
<span class="co">#&gt;  [44,] 53 6.289e-01 0.022340</span>
<span class="co">#&gt;  [45,] 54 6.410e-01 0.021330</span>
<span class="co">#&gt;  [46,] 56 6.529e-01 0.020360</span>
<span class="co">#&gt;  [47,] 59 6.655e-01 0.019430</span>
<span class="co">#&gt;  [48,] 60 6.777e-01 0.018550</span>
<span class="co">#&gt;  [49,] 60 6.896e-01 0.017710</span>
<span class="co">#&gt;  [50,] 62 7.014e-01 0.016900</span>
<span class="co">#&gt;  [51,] 62 7.128e-01 0.016130</span>
<span class="co">#&gt;  [52,] 63 7.236e-01 0.015400</span>
<span class="co">#&gt;  [53,] 64 7.344e-01 0.014700</span>
<span class="co">#&gt;  [54,] 65 7.449e-01 0.014030</span>
<span class="co">#&gt;  [55,] 67 7.551e-01 0.013400</span>
<span class="co">#&gt;  [56,] 67 7.650e-01 0.012790</span>
<span class="co">#&gt;  [57,] 67 7.745e-01 0.012210</span>
<span class="co">#&gt;  [58,] 67 7.835e-01 0.011650</span>
<span class="co">#&gt;  [59,] 67 7.923e-01 0.011120</span>
<span class="co">#&gt;  [60,] 68 8.008e-01 0.010620</span>
<span class="co">#&gt;  [61,] 69 8.091e-01 0.010130</span>
<span class="co">#&gt;  [62,] 69 8.171e-01 0.009672</span>
<span class="co">#&gt;  [63,] 69 8.247e-01 0.009233</span>
<span class="co">#&gt;  [64,] 70 8.321e-01 0.008813</span>
<span class="co">#&gt;  [65,] 71 8.394e-01 0.008412</span>
<span class="co">#&gt;  [66,] 73 8.464e-01 0.008030</span>
<span class="co">#&gt;  [67,] 74 8.533e-01 0.007665</span>
<span class="co">#&gt;  [68,] 75 8.599e-01 0.007317</span>
<span class="co">#&gt;  [69,] 75 8.662e-01 0.006984</span>
<span class="co">#&gt;  [70,] 75 8.722e-01 0.006667</span>
<span class="co">#&gt;  [71,] 75 8.780e-01 0.006364</span>
<span class="co">#&gt;  [72,] 77 8.835e-01 0.006074</span>
<span class="co">#&gt;  [73,] 78 8.888e-01 0.005798</span>
<span class="co">#&gt;  [74,] 77 8.938e-01 0.005535</span>
<span class="co">#&gt;  [75,] 77 8.986e-01 0.005283</span>
<span class="co">#&gt;  [76,] 77 9.031e-01 0.005043</span>
<span class="co">#&gt;  [77,] 77 9.075e-01 0.004814</span>
<span class="co">#&gt;  [78,] 76 9.117e-01 0.004595</span>
<span class="co">#&gt;  [79,] 78 9.157e-01 0.004386</span>
<span class="co">#&gt;  [80,] 78 9.195e-01 0.004187</span>
<span class="co">#&gt;  [81,] 78 9.231e-01 0.003997</span>
<span class="co">#&gt;  [82,] 78 9.266e-01 0.003815</span>
<span class="co">#&gt;  [83,] 77 9.299e-01 0.003642</span>
<span class="co">#&gt;  [84,] 77 9.331e-01 0.003476</span>
<span class="co">#&gt;  [85,] 77 9.361e-01 0.003318</span>
<span class="co">#&gt;  [86,] 77 9.390e-01 0.003167</span>
<span class="co">#&gt;  [87,] 77 9.417e-01 0.003023</span>
<span class="co">#&gt;  [88,] 77 9.444e-01 0.002886</span>
<span class="co">#&gt;  [89,] 79 9.469e-01 0.002755</span>
<span class="co">#&gt;  [90,] 79 9.493e-01 0.002629</span>
<span class="co">#&gt;  [91,] 79 9.516e-01 0.002510</span>
<span class="co">#&gt;  [92,] 79 9.538e-01 0.002396</span>
<span class="co">#&gt;  [93,] 79 9.559e-01 0.002287</span>
<span class="co">#&gt;  [94,] 79 9.579e-01 0.002183</span>
<span class="co">#&gt;  [95,] 80 9.598e-01 0.002084</span>
<span class="co">#&gt;  [96,] 81 9.617e-01 0.001989</span>
<span class="co">#&gt;  [97,] 81 9.634e-01 0.001899</span>
<span class="co">#&gt;  [98,] 81 9.651e-01 0.001812</span>
<span class="co">#&gt;  [99,] 81 9.667e-01 0.001730</span>
<span class="co">#&gt; [100,] 81 9.682e-01 0.001651</span>
<span class="co">#&gt; [1] "lambda min is at location 75"</span>
<span class="co">#&gt; [1] "the leave-out cells in the source subpop is 81"</span>
<span class="co">#&gt; [1] "use 82 target subpops cells for leave-out test set"</span>
<span class="co">#&gt; [1] "standardizing the leave-out target and source subpops..."</span>
<span class="co">#&gt; [1] "start ElasticNet prediction for estimating accuracy..."</span>
<span class="co">#&gt; [1] "start LDA prediction for estimating accuracy for 163 cells and 194 genes..."</span>
<span class="co">#&gt; [1] "evaluation accuracy ElasticNet 0.662576687116564"</span>
<span class="co">#&gt; [1] "evaluation accuracy LDA 0.558282208588957"</span>
<span class="co">#&gt; [1] "done training for bootstrap 2, moving to prediction..."</span>
<span class="co">#&gt; [1] "standardizing target subpops before prediction..."</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 1..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 1 is 187"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 187 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 1 is 22.9946524064171"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17210"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 192"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 187 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 1 is 66.8449197860963"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 2..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 2 is 140"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 140 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 2 is 49.2857142857143"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17210"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 192"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 140 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 2 is 70"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 3..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 3 is 133"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 133 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 3 is 72.1804511278195"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17210"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 192"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 133 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 3 is 39.0977443609023"</span>
<span class="co">#&gt; [1] "predicting from source to target subpop 4..."</span>
<span class="co">#&gt; [1] "number of cells in the target subpop 4 is 40"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17208"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 194"</span>
<span class="co">#&gt; [1] "There are 0 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 40 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running elasticNet classification..."</span>
<span class="co">#&gt; [1] "class probability prediction ElasticNet for target subpop 4 is 65"</span>
<span class="co">#&gt; [1] "Number of genes in the target data, but not in model genes is 17210"</span>
<span class="co">#&gt; [1] "Number of genes in the model present in the target data is 192"</span>
<span class="co">#&gt; [1] "There are 2 genes that are in the model, but not in target subpopulations"</span>
<span class="co">#&gt; [1] "for dim conformation, add 2 random indexes for genes in model but not in target subpop, later to be replaced by 0"</span>
<span class="co">#&gt; [1] "genes to be added to target subpop are EPB41L4A-AS1"</span>
<span class="co">#&gt; [2] "genes to be added to target subpop are HLA-B"       </span>
<span class="co">#&gt; [1] "the prediction (target) subop has 194 genes and 40 cells. The trained model has 194 genes"</span>
<span class="co">#&gt; [1] "first 10 genes in model "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "first 10 genes in target "</span>
<span class="co">#&gt;  [1] "NPPB"   "DDOST"  "STMN1"  "HMGN2"  "CDC20"  "NASP"   "CYR61" </span>
<span class="co">#&gt;  [8] "CNN3"   "ATP5F1" "RHOC"  </span>
<span class="co">#&gt; [1] "running LDA classification..."</span>
<span class="co">#&gt; [1] "class probability prediction LDA for target subpop 4 is 57.5"</span>

<span class="co">#prepare table input for sankey plot </span>

LASSO_C1S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#7570b3"</span>)

LASSO_C2S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#1b9e77"</span>)

LASSO_C3S1  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#e7298a"</span>)


combined &lt;-<span class="st"> </span><span class="kw">rbind</span>(LASSO_C1S1,LASSO_C2S1,LASSO_C3S1)

nboots =<span class="st"> </span><span class="dv">2</span>
<span class="co">#links: source, target, value</span>
<span class="co">#source: node, nodegroup</span>
combined_D3obj &lt;-<span class="kw">list</span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)], <span class="dt">Links=</span>combined[,<span class="kw">c</span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw">ncol</span>(combined))]) 
combined &lt;-<span class="st"> </span>combined[<span class="kw">is.na</span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]


<span class="kw">library</span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw">as.vector</span>(<span class="kw">sort</span>(<span class="kw">unique</span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw">as.vector</span>(<span class="kw">sort</span>(<span class="kw">unique</span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw">unique</span>(<span class="kw">c</span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw">data.frame</span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw">as.numeric</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw">suppressMessages</span>(<span class="kw">library</span>(dplyr))
n &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(node_df<span class="op">$</span>Node))
getPalette =<span class="st"> </span><span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">9</span>, <span class="st">"Set1"</span>))
Color =<span class="st"> </span><span class="kw">getPalette</span>(n)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(networkD3))
p1&lt;-<span class="kw"><a href="http://www.rdocumentation.org/packages/networkD3/topics/sankeyNetwork">sankeyNetwork</a></span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,  <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, 
                  <span class="dt">fontSize =</span> <span class="dv">22</span> )
p1</code></pre></div>
<div id="htmlwidget-8096071e9cae3d45e5d0" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-8096071e9cae3d45e5d0">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2],"target":[3,4,5,6,3,4,5,6,3,4,5,6],"value":[48.663101604278,85,84.5864661654136,87.5,67.9144385026738,26.4285714285714,65.7894736842105,58.75,34.4919786096256,26.4285714285714,41.7293233082707,35],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a"]},"nodes":{"name":["C1_MP1","C2_MP1","C3_MP1","C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#E41A1C","#3E8E93","#7E6E85","#FF7F00","#E1C62F","#DB728C","#999999"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span>
##R Setting Information
<span class="co">#sessionInfo()</span>
<span class="co">#rmarkdown::render("/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/vignettes/vignette.Rmd",html_document(toc = TRUE, toc_depth = 3))</span>
<span class="co">#rmarkdown::render("/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/vignettes/vignette.Rmd",pdf_document(toc = TRUE, toc_depth = 3))</span></code></pre></div>
</div>
<div id="annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with-a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with-a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference" class="anchor"></a>4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with a reference dataset to see which cluster is most similar to the reference</h2>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation-instruction">1. Installation instruction</a></li>
      <li>
<a href="#a-simple-workflow-of-the-scgps">2. A simple workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#create-scgps-objects">2.1 Create scGPS objects</a></li>
      <li><a href="#run-prediction">2.2 Run prediction</a></li>
      <li><a href="#summarise-results">2.3 Summarise results</a></li>
      </ul>
</li>
      <li>
<a href="#a-complete-workflow-of-the-scgps">3. A complete workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#identify-clusters-in-a-dataset-using-core">3.1 Identify clusters in a dataset using CORE</a></li>
      <li><a href="#identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution">3.2 Identify clusters in a dataset using SCORE (Stable Clustering at Optimal REsolution)</a></li>
      <li><a href="#visualise-all-cluster-results-in-all-iterations">3.3 Visualise all cluster results in all iterations</a></li>
      <li><a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr">3.4 Compare clustering results with other dimensional reduction methods (e.g., CIDR)</a></li>
      <li><a href="#find-gene-markers-and-annotate-clusters">3.5 Find gene markers and annotate clusters</a></li>
      </ul>
</li>
      <li>
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples">4. Relationship between clusters within one sample or between two samples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#start-the-scgps-prediction-to-find-relationship-between-clusters">4.1 Start the scGPS prediction to find relationship between clusters</a></li>
      <li><a href="#display-summary-results-for-the-prediction">4.2 Display summary results for the prediction</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-one-sample">4.3 Plot the relationship between clusters in one sample</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-two-samples">4.3 Plot the relationship between clusters in two samples</a></li>
      <li><a href="#annotation-scgps-prediction-can-be-used-to-compare-scgps-clusters-with-a-reference-dataset-to-see-which-cluster-is-most-similar-to-the-reference">4.4 Annotation: scGPS prediction can be used to compare scGPS clusters with a reference dataset to see which cluster is most similar to the reference</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Asenbouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
