<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGPS introduction â€¢ scGPS</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scGPS introduction">
<meta property="og:description" content="scGPS is a single cell RNA analysis framework to decompose a mixed population into clusters (SCORE), followed by analysing the relationship between clusters (scGPS)">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>

<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Benchmarking
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../benchmarking/benchMarking_baron.html" target="_blank">Baron</a>
      <a href="../benchmarking/benchMarking_camp.html" target="_blank">Camp</a>
      <a href="../benchmarking/benchMarking_klein.html" target="_blank">Klein</a>
      <a href="../benchmarking/benchMarking_koh.html" target="_blank">Koh</a>
      <a href="../benchmarking/benchMarking_kumar.html" target="_blank">Kumar</a>
      <a href="../benchmarking/benchMarking_yan.html" target="_blank">Yan</a>
    </li>
  </ul>
</li>


      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="vignette_files/htmlwidgets-1.3/htmlwidgets.js"></script><script src="vignette_files/d3-4.5.0/d3.min.js"></script><script src="vignette_files/sankey-1/sankey.js"></script><script src="vignette_files/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scGPS introduction</h1>
                        <h4 class="author">Quan Nguyen and Michael Thompson</h4>
            
            <h4 class="date">2019-06-25</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="co"># To install scGPS from github (Depending on the configuration of the local</span></a>
<a class="sourceLine" id="cb1-2" title="2"><span class="co"># computer or HPC, possible custom C++ compilation may be required - see</span></a>
<a class="sourceLine" id="cb1-3" title="3"><span class="co"># installation trouble-shootings below)</span></a>
<a class="sourceLine" id="cb1-4" title="4">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"IMB-Computational-Genomics-Lab/scGPS"</span>)</a>
<a class="sourceLine" id="cb1-5" title="5"></a>
<a class="sourceLine" id="cb1-6" title="6"><span class="co"># for C++ compilation trouble-shooting, manual download and installation can be</span></a>
<a class="sourceLine" id="cb1-7" title="7"><span class="co"># done from github</span></a>
<a class="sourceLine" id="cb1-8" title="8"></a>
<a class="sourceLine" id="cb1-9" title="9">git clone https<span class="op">:</span><span class="er">//</span>github.com<span class="op">/</span>IMB<span class="op">-</span>Computational<span class="op">-</span>Genomics<span class="op">-</span>Lab<span class="op">/</span>scGPS</a>
<a class="sourceLine" id="cb1-10" title="10"></a>
<a class="sourceLine" id="cb1-11" title="11"><span class="co"># then check in scGPS/src if any of the precompiled (e.g.  those with *.so and</span></a>
<a class="sourceLine" id="cb1-12" title="12"><span class="co"># *.o) files exist and delete them before recompiling</span></a>
<a class="sourceLine" id="cb1-13" title="13"></a>
<a class="sourceLine" id="cb1-14" title="14"><span class="co"># then with the scGPS as the R working directory, manually install and load</span></a>
<a class="sourceLine" id="cb1-15" title="15"><span class="co"># using devtools functionality</span></a>
<a class="sourceLine" id="cb1-16" title="16"><span class="co"># Install the package</span></a>
<a class="sourceLine" id="cb1-17" title="17">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/install">install</a></span>()</a>
<a class="sourceLine" id="cb1-18" title="18"><span class="co">#load the package to the workspace </span></a>
<a class="sourceLine" id="cb1-19" title="19"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scGPS)</a></code></pre></div>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p>The purpose of this workflow is to solve the following task:</p>
<ul>
<li>Given a mixed population with known subpopulations, estimate transition scores between these subpopulation.</li>
</ul>
<div id="create-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#create-scgps-objects" class="anchor"></a>2.1 Create scGPS objects</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"></a>
<a class="sourceLine" id="cb2-2" title="2"><span class="co"># load mixed population 1 (loaded from day_2_cardio_cell_sample dataset, named it as day2)</span></a>
<a class="sourceLine" id="cb2-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scGPS)</a>
<a class="sourceLine" id="cb2-4" title="4"></a>
<a class="sourceLine" id="cb2-5" title="5">day2 &lt;-<span class="st"> </span>day_<span class="dv">2</span>_cardio_cell_sample</a>
<a class="sourceLine" id="cb2-6" title="6">mixedpop1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/new_scGPS_object.html">new_scGPS_object</a></span>(<span class="dt">ExpressionMatrix =</span> day2<span class="op">$</span>dat2_counts,</a>
<a class="sourceLine" id="cb2-7" title="7">    <span class="dt">GeneMetadata =</span> day2<span class="op">$</span>dat2geneInfo, <span class="dt">CellMetadata =</span> day2<span class="op">$</span>dat2_clusters)</a>
<a class="sourceLine" id="cb2-8" title="8"></a>
<a class="sourceLine" id="cb2-9" title="9"><span class="co"># load mixed population 2 (loaded from day_5_cardio_cell_sample dataset, named it as day5)</span></a>
<a class="sourceLine" id="cb2-10" title="10">day5 &lt;-<span class="st"> </span>day_<span class="dv">5</span>_cardio_cell_sample</a>
<a class="sourceLine" id="cb2-11" title="11">mixedpop2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/new_scGPS_object.html">new_scGPS_object</a></span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts,</a>
<a class="sourceLine" id="cb2-12" title="12">    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> day5<span class="op">$</span>dat5_clusters)</a></code></pre></div>
</div>
<div id="run-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#run-prediction" class="anchor"></a>2.2 Run prediction</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"></a>
<a class="sourceLine" id="cb3-2" title="2"><span class="co"># select a subpopulation</span></a>
<a class="sourceLine" id="cb3-3" title="3">c_selectID &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb3-4" title="4"><span class="co"># load gene list (this can be any lists of user selected genes)</span></a>
<a class="sourceLine" id="cb3-5" title="5">genes &lt;-<span class="st"> </span>training_gene_sample</a>
<a class="sourceLine" id="cb3-6" title="6">genes &lt;-<span class="st"> </span>genes<span class="op">$</span>Merged_unique</a>
<a class="sourceLine" id="cb3-7" title="7"><span class="co"># load cluster information </span></a>
<a class="sourceLine" id="cb3-8" title="8">cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb3-9" title="9">cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb3-10" title="10"><span class="co">#run training (running nboots = 3 here, but recommend to use nboots = 50-100)</span></a>
<a class="sourceLine" id="cb3-11" title="11">LSOLDA_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">3</span>, <span class="dt">mixedpop1 =</span> mixedpop1, </a>
<a class="sourceLine" id="cb3-12" title="12">    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID,</a>
<a class="sourceLine" id="cb3-13" title="13">    <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(), <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1, </a>
<a class="sourceLine" id="cb3-14" title="14">    <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2, <span class="dt">trainset_ratio =</span> <span class="fl">0.7</span>)</a>
<a class="sourceLine" id="cb3-15" title="15"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(LSOLDA_dat)</a>
<a class="sourceLine" id="cb3-16" title="16"><span class="co">#&gt; [1] "Accuracy"          "ElasticNetGenes"   "Deviance"         </span></a>
<a class="sourceLine" id="cb3-17" title="17"><span class="co">#&gt; [4] "ElasticNetFit"     "LDAFit"            "predictor_S1"     </span></a>
<a class="sourceLine" id="cb3-18" title="18"><span class="co">#&gt; [7] "ElasticNetPredict" "LDAPredict"</span></a></code></pre></div>
</div>
<div id="summarise-results" class="section level2">
<h2 class="hasAnchor">
<a href="#summarise-results" class="anchor"></a>2.3 Summarise results</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="co"># summary results LDA</span></a>
<a class="sourceLine" id="cb4-2" title="2">sum_pred_lda &lt;-<span class="st"> </span><span class="kw"><a href="../reference/summary_prediction_lda.html">summary_prediction_lda</a></span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb4-3" title="3"><span class="co"># summary results Lasso to show the percent of cells</span></a>
<a class="sourceLine" id="cb4-4" title="4"><span class="co"># classified as cells belonging </span></a>
<a class="sourceLine" id="cb4-5" title="5">sum_pred_lasso &lt;-<span class="st"> </span><span class="kw"><a href="../reference/summary_prediction_lasso.html">summary_prediction_lasso</a></span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat,</a>
<a class="sourceLine" id="cb4-6" title="6">    <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb4-7" title="7"><span class="co"># plot summary results </span></a>
<a class="sourceLine" id="cb4-8" title="8">plot_sum &lt;-<span class="cf">function</span>(sum_dat){</a>
<a class="sourceLine" id="cb4-9" title="9">    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(sum_dat)</a>
<a class="sourceLine" id="cb4-10" title="10">    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/na.fail">na.omit</a></span>(sum_dat_tf)</a>
<a class="sourceLine" id="cb4-11" title="11">    sum_dat_tf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(sum_dat[, <span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(sum_dat)],<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb4-12" title="12">        <span class="cf">function</span>(x){<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(x))})</a>
<a class="sourceLine" id="cb4-13" title="13">    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"ElasticNet for subpop"</span>,<span class="st">"sp"</span>,  sum_dat<span class="op">$</span>names )</a>
<a class="sourceLine" id="cb4-14" title="14">    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"in target mixedpop"</span>,<span class="st">"in p"</span>,  sum_dat<span class="op">$</span>names) </a>
<a class="sourceLine" id="cb4-15" title="15">    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"LDA for subpop"</span>,<span class="st">"sp"</span>,  sum_dat<span class="op">$</span>names )</a>
<a class="sourceLine" id="cb4-16" title="16">    sum_dat<span class="op">$</span>names &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">gsub</a></span>(<span class="st">"in target mixedpop"</span>,<span class="st">"in p"</span>,  sum_dat<span class="op">$</span>names)</a>
<a class="sourceLine" id="cb4-17" title="17">    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sum_dat_tf) &lt;-<span class="st"> </span>sum_dat<span class="op">$</span>names</a>
<a class="sourceLine" id="cb4-18" title="18">    <span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(sum_dat_tf, <span class="dt">las=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb4-19" title="19">}</a>
<a class="sourceLine" id="cb4-20" title="20"><span class="kw">plot_sum</span>(sum_pred_lasso)</a></code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-1.png" width="700"></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw">plot_sum</span>(sum_pred_lda)</a></code></pre></div>
<p><img src="vignette_files/figure-html/summarise_results-2.png" width="700"></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set </span></a>
<a class="sourceLine" id="cb6-2" title="2"><span class="kw"><a href="../reference/summary_accuracy.html">summary_accuracy</a></span>(<span class="dt">object =</span> LSOLDA_dat)</a>
<a class="sourceLine" id="cb6-3" title="3"><span class="co">#&gt; [1] 67.28972 63.67925 66.82464</span></a>
<a class="sourceLine" id="cb6-4" title="4"><span class="co"># summary maximum deviance explained by the model </span></a>
<a class="sourceLine" id="cb6-5" title="5"><span class="kw"><a href="../reference/summary_deviance.html">summary_deviance</a></span>(<span class="dt">object =</span> LSOLDA_dat)</a>
<a class="sourceLine" id="cb6-6" title="6"><span class="co">#&gt; $allDeviance</span></a>
<a class="sourceLine" id="cb6-7" title="7"><span class="co">#&gt; [1] "0.0596"  "0.08978" "0.07554"</span></a>
<a class="sourceLine" id="cb6-8" title="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-9" title="9"><span class="co">#&gt; $DeviMax</span></a>
<a class="sourceLine" id="cb6-10" title="10"><span class="co">#&gt;          dat_DE$Dfd          Deviance           DEgenes</span></a>
<a class="sourceLine" id="cb6-11" title="11"><span class="co">#&gt; 1                 0           0.08978    genes_cluster1</span></a>
<a class="sourceLine" id="cb6-12" title="12"><span class="co">#&gt; 2                 1           0.08978    genes_cluster1</span></a>
<a class="sourceLine" id="cb6-13" title="13"><span class="co">#&gt; 3                 2           0.08978    genes_cluster1</span></a>
<a class="sourceLine" id="cb6-14" title="14"><span class="co">#&gt; 4                 3           0.08978    genes_cluster1</span></a>
<a class="sourceLine" id="cb6-15" title="15"><span class="co">#&gt; 5                 4           0.08978    genes_cluster1</span></a>
<a class="sourceLine" id="cb6-16" title="16"><span class="co">#&gt; 6 remaining DEgenes remaining DEgenes remaining DEgenes</span></a>
<a class="sourceLine" id="cb6-17" title="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-18" title="18"><span class="co">#&gt; $LassoGenesMax</span></a>
<a class="sourceLine" id="cb6-19" title="19"><span class="co">#&gt; NULL</span></a></code></pre></div>
</div>
</div>
<div id="a-complete-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-complete-workflow-of-the-scgps" class="anchor"></a>3. A complete workflow of the scGPS:</h1>
<p>The purpose of this workflow is to solve the following task:</p>
<ul>
<li>Given an unknown mixed population, find clusters and estimate relationship between clusters</li>
</ul>
<div id="identify-clusters-in-a-dataset-using-core" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-core" class="anchor"></a>3.1 Identify clusters in a dataset using CORE</h2>
<p>(skip this step if clusters are known)</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"></a>
<a class="sourceLine" id="cb7-2" title="2"><span class="co"># find clustering information in an expresion data using CORE</span></a>
<a class="sourceLine" id="cb7-3" title="3">day5 &lt;-<span class="st"> </span>day_<span class="dv">5</span>_cardio_cell_sample</a>
<a class="sourceLine" id="cb7-4" title="4">cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)</a>
<a class="sourceLine" id="cb7-5" title="5">cluster &lt;-day5<span class="op">$</span>dat5_clusters</a>
<a class="sourceLine" id="cb7-6" title="6">cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)</a>
<a class="sourceLine" id="cb7-7" title="7">mixedpop2 &lt;-<span class="kw"><a href="../reference/new_scGPS_object.html">new_scGPS_object</a></span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts,</a>
<a class="sourceLine" id="cb7-8" title="8">                    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames)</a>
<a class="sourceLine" id="cb7-9" title="9"></a>
<a class="sourceLine" id="cb7-10" title="10">CORE_cluster &lt;-<span class="st"> </span><span class="kw"><a href="../reference/CORE_clustering.html">CORE_clustering</a></span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb7-11" title="11"></a>
<a class="sourceLine" id="cb7-12" title="12"><span class="co"># to update the clustering information, users can ...</span></a>
<a class="sourceLine" id="cb7-13" title="13">key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height</a>
<a class="sourceLine" id="cb7-14" title="14">optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes</a>
<a class="sourceLine" id="cb7-15" title="15">optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)</a>
<a class="sourceLine" id="cb7-16" title="16"></a>
<a class="sourceLine" id="cb7-17" title="17">clustering_after_outlier_removal &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unname">unname</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(</a>
<a class="sourceLine" id="cb7-18" title="18"> CORE_cluster<span class="op">$</span>Cluster[[optimal_index]]))</a>
<a class="sourceLine" id="cb7-19" title="19">corresponding_cells_after_outlier_removal &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>cellsForClustering</a>
<a class="sourceLine" id="cb7-20" title="20">original_cells_before_removal &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">2</span>]</a>
<a class="sourceLine" id="cb7-21" title="21">corresponding_index &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(corresponding_cells_after_outlier_removal,</a>
<a class="sourceLine" id="cb7-22" title="22">                            original_cells_before_removal )</a>
<a class="sourceLine" id="cb7-23" title="23"><span class="co"># check the matching</span></a>
<a class="sourceLine" id="cb7-24" title="24"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/identical">identical</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/character">as.character</a></span>(original_cells_before_removal[corresponding_index]),</a>
<a class="sourceLine" id="cb7-25" title="25">         corresponding_cells_after_outlier_removal)</a>
<a class="sourceLine" id="cb7-26" title="26"><span class="co">#&gt; [1] TRUE</span></a>
<a class="sourceLine" id="cb7-27" title="27"><span class="co"># create new object with the new clustering after removing outliers</span></a>
<a class="sourceLine" id="cb7-28" title="28">mixedpop2_post_clustering &lt;-<span class="st"> </span>mixedpop2[,corresponding_index]</a>
<a class="sourceLine" id="cb7-29" title="29"><span class="kw">colData</span>(mixedpop2_post_clustering)[,<span class="dv">1</span>] &lt;-<span class="st"> </span>clustering_after_outlier_removal</a></code></pre></div>
</div>
<div id="identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution" class="anchor"></a>3.2 Identify clusters in a dataset using SCORE (Stable Clustering at Optimal REsolution)</h2>
<p>(skip this step if clusters are known)</p>
<p>(SCORE aims to get stable subpopulation results by introducing bagging aggregation and bootstrapping to the CORE algorithm)</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1"></a>
<a class="sourceLine" id="cb8-2" title="2"><span class="co"># find clustering information in an expresion data using SCORE</span></a>
<a class="sourceLine" id="cb8-3" title="3">day5 &lt;-<span class="st"> </span>day_<span class="dv">5</span>_cardio_cell_sample</a>
<a class="sourceLine" id="cb8-4" title="4">cellnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(day5<span class="op">$</span>dat5_counts)</a>
<a class="sourceLine" id="cb8-5" title="5">cluster &lt;-day5<span class="op">$</span>dat5_clusters</a>
<a class="sourceLine" id="cb8-6" title="6">cellnames &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)</a>
<a class="sourceLine" id="cb8-7" title="7">mixedpop2 &lt;-<span class="kw"><a href="../reference/new_scGPS_object.html">new_scGPS_object</a></span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts,</a>
<a class="sourceLine" id="cb8-8" title="8">                    <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames )</a>
<a class="sourceLine" id="cb8-9" title="9"></a>
<a class="sourceLine" id="cb8-10" title="10">SCORE_test &lt;-<span class="st"> </span><span class="kw"><a href="../reference/CORE_bagging.html">CORE_bagging</a></span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb8-11" title="11">                                <span class="dt">bagging_run =</span> <span class="dv">20</span>, <span class="dt">subsample_proportion =</span> <span class="fl">.8</span>)</a></code></pre></div>
</div>
<div id="visualise-all-cluster-results-in-all-iterations" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-all-cluster-results-in-all-iterations" class="anchor"></a>3.3 Visualise all cluster results in all iterations</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/dev">dev.off</a></span>()</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="co">#&gt; null device </span></a>
<a class="sourceLine" id="cb9-3" title="3"><span class="co">#&gt;           1</span></a>
<a class="sourceLine" id="cb9-4" title="4"><span class="co">##3.2.1 plot CORE clustering</span></a>
<a class="sourceLine" id="cb9-5" title="5">p1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_CORE.html">plot_CORE</a></span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster,</a>
<a class="sourceLine" id="cb9-6" title="6">               <span class="dt">color_branch =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#208eb7"</span>, <span class="st">"#6ce9d3"</span>, <span class="st">"#1c5e39"</span>, <span class="st">"#8fca40"</span>, <span class="st">"#154975"</span>,</a>
<a class="sourceLine" id="cb9-7" title="7">                                <span class="st">"#b1c8eb"</span>))</a>
<a class="sourceLine" id="cb9-8" title="8">p1</a>
<a class="sourceLine" id="cb9-9" title="9"><span class="co">#&gt; $mar</span></a>
<a class="sourceLine" id="cb9-10" title="10"><span class="co">#&gt; [1] 1 5 0 1</span></a>
<a class="sourceLine" id="cb9-11" title="11"><span class="co">#extract optimal index identified by CORE</span></a>
<a class="sourceLine" id="cb9-12" title="12">key_height &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height</a>
<a class="sourceLine" id="cb9-13" title="13">optimal_res &lt;-<span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes</a>
<a class="sourceLine" id="cb9-14" title="14">optimal_index =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(key_height <span class="op">==</span><span class="st"> </span>optimal_res)</a>
<a class="sourceLine" id="cb9-15" title="15"><span class="co">#plot one optimal clustering bar</span></a>
<a class="sourceLine" id="cb9-16" title="16"><span class="kw"><a href="../reference/plot_optimal_CORE.html">plot_optimal_CORE</a></span>(<span class="dt">original_tree=</span> CORE_cluster<span class="op">$</span>tree,</a>
<a class="sourceLine" id="cb9-17" title="17">                 <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(CORE_cluster<span class="op">$</span>Cluster[optimal_index]),</a>
<a class="sourceLine" id="cb9-18" title="18">                 <span class="dt">shift =</span> <span class="dv">-2000</span>)</a>
<a class="sourceLine" id="cb9-19" title="19"><span class="co">#&gt; Ordering and assigning labels...</span></a>
<a class="sourceLine" id="cb9-20" title="20"><span class="co">#&gt; 2</span></a>
<a class="sourceLine" id="cb9-21" title="21"><span class="co">#&gt; 162335NA</span></a>
<a class="sourceLine" id="cb9-22" title="22"><span class="co">#&gt; 3</span></a>
<a class="sourceLine" id="cb9-23" title="23"><span class="co">#&gt; 162335423</span></a>
<a class="sourceLine" id="cb9-24" title="24"><span class="co">#&gt; Plotting the colored dendrogram now....</span></a>
<a class="sourceLine" id="cb9-25" title="25"><span class="co">#&gt; Plotting the bar underneath now....</span></a>
<a class="sourceLine" id="cb9-26" title="26"></a>
<a class="sourceLine" id="cb9-27" title="27"><span class="co">##3.2.2 plot SCORE clustering</span></a>
<a class="sourceLine" id="cb9-28" title="28"><span class="co">#plot all clustering bars</span></a>
<a class="sourceLine" id="cb9-29" title="29"><span class="kw"><a href="../reference/plot_CORE.html">plot_CORE</a></span>(SCORE_test<span class="op">$</span>tree, <span class="dt">list_clusters =</span> SCORE_test<span class="op">$</span>Cluster)</a>
<a class="sourceLine" id="cb9-30" title="30"><span class="co">#plot one stable optimal clustering bar</span></a>
<a class="sourceLine" id="cb9-31" title="31"><span class="kw"><a href="../reference/plot_optimal_CORE.html">plot_optimal_CORE</a></span>(<span class="dt">original_tree=</span> SCORE_test<span class="op">$</span>tree,</a>
<a class="sourceLine" id="cb9-32" title="32">                 <span class="dt">optimal_cluster =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>(SCORE_test<span class="op">$</span>Cluster[SCORE_test<span class="op">$</span>optimal_index]),</a>
<a class="sourceLine" id="cb9-33" title="33">                 <span class="dt">shift =</span> <span class="dv">-100</span>)</a>
<a class="sourceLine" id="cb9-34" title="34"><span class="co">#&gt; Ordering and assigning labels...</span></a>
<a class="sourceLine" id="cb9-35" title="35"><span class="co">#&gt; 2</span></a>
<a class="sourceLine" id="cb9-36" title="36"><span class="co">#&gt; 162335NA</span></a>
<a class="sourceLine" id="cb9-37" title="37"><span class="co">#&gt; 3</span></a>
<a class="sourceLine" id="cb9-38" title="38"><span class="co">#&gt; 162335423</span></a>
<a class="sourceLine" id="cb9-39" title="39"><span class="co">#&gt; Plotting the colored dendrogram now....</span></a>
<a class="sourceLine" id="cb9-40" title="40"><span class="co">#&gt; Plotting the bar underneath now....</span></a></code></pre></div>
</div>
<div id="compare-clustering-results-with-other-dimensional-reduction-methods-e-g--tsne" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--tsne" class="anchor"></a>3.4 Compare clustering results with other dimensional reduction methods (e.g., tSNE)</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">t &lt;-<span class="st"> </span><span class="kw"><a href="../reference/tSNE.html">tSNE</a></span>(<span class="dt">expression.mat=</span><span class="kw">assay</span>(mixedpop2))</a>
<a class="sourceLine" id="cb10-2" title="2"><span class="co">#&gt; Preparing PCA inputs using the top 1500 genes ...</span></a>
<a class="sourceLine" id="cb10-3" title="3"><span class="co">#&gt; Computing PCA values...</span></a>
<a class="sourceLine" id="cb10-4" title="4"><span class="co">#&gt; Running tSNE ...</span></a>
<a class="sourceLine" id="cb10-5" title="5">p2 &lt;-<span class="kw"><a href="../reference/plot_reduced.html">plot_reduced</a></span>(t, <span class="dt">color_fac =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]),</a>
<a class="sourceLine" id="cb10-6" title="6">                      <span class="dt">palletes =</span><span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])))</a>
<a class="sourceLine" id="cb10-7" title="7">p2</a></code></pre></div>
<p><img src="vignette_files/figure-html/compare_clustering-1.png" width="480"></p>
</div>
<div id="find-gene-markers-and-annotate-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#find-gene-markers-and-annotate-clusters" class="anchor"></a>3.5 Find gene markers and annotate clusters</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1"><span class="co">#load gene list (this can be any lists of user-selected genes)</span></a>
<a class="sourceLine" id="cb11-2" title="2">genes &lt;-training_gene_sample</a>
<a class="sourceLine" id="cb11-3" title="3">genes &lt;-genes<span class="op">$</span>Merged_unique</a>
<a class="sourceLine" id="cb11-4" title="4"></a>
<a class="sourceLine" id="cb11-5" title="5"><span class="co">#the gene list can also be objectively identified by differential expression</span></a>
<a class="sourceLine" id="cb11-6" title="6"><span class="co">#analysis cluster information is requied for find_markers. Here, we use</span></a>
<a class="sourceLine" id="cb11-7" title="7"><span class="co">#CORE results.</span></a>
<a class="sourceLine" id="cb11-8" title="8"></a>
<a class="sourceLine" id="cb11-9" title="9"><span class="co">#colData(mixedpop2)[,1] &lt;- unlist(SCORE_test$Cluster[SCORE_test$optimal_index])</span></a>
<a class="sourceLine" id="cb11-10" title="10"></a>
<a class="sourceLine" id="cb11-11" title="11"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(locfit))</a>
<a class="sourceLine" id="cb11-12" title="12"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DESeq))</a>
<a class="sourceLine" id="cb11-13" title="13"></a>
<a class="sourceLine" id="cb11-14" title="14">DEgenes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find_markers.html">find_markers</a></span>(<span class="dt">expression_matrix=</span><span class="kw">assay</span>(mixedpop2),</a>
<a class="sourceLine" id="cb11-15" title="15">                            <span class="dt">cluster =</span> <span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>],</a>
<a class="sourceLine" id="cb11-16" title="16">                            <span class="dt">selected_cluster=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))</a>
<a class="sourceLine" id="cb11-17" title="17"></a>
<a class="sourceLine" id="cb11-18" title="18"><span class="co">#the output contains dataframes for each cluster.</span></a>
<a class="sourceLine" id="cb11-19" title="19"><span class="co">#the data frame contains all genes, sorted by p-values</span></a>
<a class="sourceLine" id="cb11-20" title="20"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(DEgenes)</a>
<a class="sourceLine" id="cb11-21" title="21"><span class="co">#&gt; [1] NA                      NA                      NA                     </span></a>
<a class="sourceLine" id="cb11-22" title="22"><span class="co">#&gt; [4] NA                      "DE_Subpop1vsRemaining" "DE_Subpop2vsRemaining"</span></a>
<a class="sourceLine" id="cb11-23" title="23"><span class="co">#&gt; [7] "DE_Subpop3vsRemaining" "DE_Subpop4vsRemaining"</span></a>
<a class="sourceLine" id="cb11-24" title="24"></a>
<a class="sourceLine" id="cb11-25" title="25"><span class="co">#you can annotate the identified clusters</span></a>
<a class="sourceLine" id="cb11-26" title="26">DEgeneList_1vsOthers &lt;-<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id</a>
<a class="sourceLine" id="cb11-27" title="27"></a>
<a class="sourceLine" id="cb11-28" title="28"><span class="co">#users need to check the format of the gene input to make sure they are</span></a>
<a class="sourceLine" id="cb11-29" title="29"><span class="co">#consistent to the gene names in the expression matrix</span></a>
<a class="sourceLine" id="cb11-30" title="30"></a>
<a class="sourceLine" id="cb11-31" title="31"><span class="co">#the following command saves the file "PathwayEnrichment.xlsx" to the</span></a>
<a class="sourceLine" id="cb11-32" title="32"><span class="co">#working dir</span></a>
<a class="sourceLine" id="cb11-33" title="33"><span class="co">#use 500 top DE genes</span></a>
<a class="sourceLine" id="cb11-34" title="34"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DOSE))</a>
<a class="sourceLine" id="cb11-35" title="35"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ReactomePA))</a>
<a class="sourceLine" id="cb11-36" title="36"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(clusterProfiler))</a>
<a class="sourceLine" id="cb11-37" title="37">genes500 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">as.factor</a></span>(DEgeneList_1vsOthers[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq_len</a></span>(<span class="dv">500</span>)])</a>
<a class="sourceLine" id="cb11-38" title="38">enrichment_test &lt;-<span class="st"> </span><span class="kw"><a href="../reference/annotate_clusters.html">annotate_clusters</a></span>(genes, <span class="dt">pvalueCutoff=</span><span class="fl">0.05</span>, <span class="dt">gene_symbol=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb11-39" title="39"></a>
<a class="sourceLine" id="cb11-40" title="40"><span class="co">#the enrichment outputs can be displayed by running</span></a>
<a class="sourceLine" id="cb11-41" title="41">clusterProfiler<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/clusterProfiler/topics/reexports">dotplot</a></span>(enrichment_test, <span class="dt">showCategory=</span><span class="dv">10</span>, <span class="dt">font.size =</span> <span class="dv">6</span>)</a></code></pre></div>
<p><img src="vignette_files/figure-html/find_markers-1.png" width="672"></p>
</div>
</div>
<div id="relationship-between-clusters-within-one-sample-or-between-two-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples" class="anchor"></a>4. Relationship between clusters within one sample or between two samples</h1>
<p>The purpose of this workflow is to solve the following task:</p>
<ul>
<li>Given one or two unknown mixed population(s) and clusters in each mixed population, estimate</li>
<li>Visualise relationship between clusters*</li>
</ul>
<div id="start-the-scgps-prediction-to-find-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#start-the-scgps-prediction-to-find-relationship-between-clusters" class="anchor"></a>4.1 Start the scGPS prediction to find relationship between clusters</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1"></a>
<a class="sourceLine" id="cb12-2" title="2"><span class="co">#select a subpopulation, and input gene list</span></a>
<a class="sourceLine" id="cb12-3" title="3">c_selectID &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb12-4" title="4"><span class="co">#note make sure the format for genes input here is the same to the format</span></a>
<a class="sourceLine" id="cb12-5" title="5"><span class="co">#for genes in the mixedpop1 and mixedpop2</span></a>
<a class="sourceLine" id="cb12-6" title="6">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>]</a>
<a class="sourceLine" id="cb12-7" title="7"></a>
<a class="sourceLine" id="cb12-8" title="8"><span class="co">#run the test bootstrap with nboots = 2 runs</span></a>
<a class="sourceLine" id="cb12-9" title="9"></a>
<a class="sourceLine" id="cb12-10" title="10">cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb12-11" title="11">cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb12-12" title="12"></a>
<a class="sourceLine" id="cb12-13" title="13">LSOLDA_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1,</a>
<a class="sourceLine" id="cb12-14" title="14">                             <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, <span class="dt">c_selectID  =</span> c_selectID,</a>
<a class="sourceLine" id="cb12-15" title="15">                             <span class="dt">listData =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb12-16" title="16">                             <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,</a>
<a class="sourceLine" id="cb12-17" title="17">                             <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a></code></pre></div>
</div>
<div id="display-summary-results-for-the-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#display-summary-results-for-the-prediction" class="anchor"></a>4.2 Display summary results for the prediction</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1"><span class="co">#get the number of rows for the summary matrix</span></a>
<a class="sourceLine" id="cb13-2" title="2">row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))</a>
<a class="sourceLine" id="cb13-3" title="3"></a>
<a class="sourceLine" id="cb13-4" title="4"><span class="co">#summary results LDA to to show the percent of cells classified as cells</span></a>
<a class="sourceLine" id="cb13-5" title="5"><span class="co">#belonging by LDA classifier</span></a>
<a class="sourceLine" id="cb13-6" title="6"><span class="kw"><a href="../reference/summary_prediction_lda.html">summary_prediction_lda</a></span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster )</a>
<a class="sourceLine" id="cb13-7" title="7"><span class="co">#&gt;                 V1               V2                                names</span></a>
<a class="sourceLine" id="cb13-8" title="8"><span class="co">#&gt; 1 66.8449197860963 2.67379679144385 LDA for subpop 1 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-9" title="9"><span class="co">#&gt; 2 92.1428571428571 40.7142857142857 LDA for subpop 2 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-10" title="10"><span class="co">#&gt; 3 60.9022556390977 6.01503759398496 LDA for subpop 3 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-11" title="11"><span class="co">#&gt; 4               70             17.5 LDA for subpop 4 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-12" title="12"></a>
<a class="sourceLine" id="cb13-13" title="13"><span class="co">#summary results Lasso to show the percent of cells classified as cells</span></a>
<a class="sourceLine" id="cb13-14" title="14"><span class="co">#belonging by Lasso classifier</span></a>
<a class="sourceLine" id="cb13-15" title="15"><span class="kw"><a href="../reference/summary_prediction_lasso.html">summary_prediction_lasso</a></span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster)</a>
<a class="sourceLine" id="cb13-16" title="16"><span class="co">#&gt;                 V1               V2</span></a>
<a class="sourceLine" id="cb13-17" title="17"><span class="co">#&gt; 1 26.2032085561497 24.5989304812834</span></a>
<a class="sourceLine" id="cb13-18" title="18"><span class="co">#&gt; 2 95.7142857142857 95.7142857142857</span></a>
<a class="sourceLine" id="cb13-19" title="19"><span class="co">#&gt; 3 68.4210526315789 78.1954887218045</span></a>
<a class="sourceLine" id="cb13-20" title="20"><span class="co">#&gt; 4             92.5             87.5</span></a>
<a class="sourceLine" id="cb13-21" title="21"><span class="co">#&gt;                                        names</span></a>
<a class="sourceLine" id="cb13-22" title="22"><span class="co">#&gt; 1 ElasticNet for subpop1 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-23" title="23"><span class="co">#&gt; 2 ElasticNet for subpop2 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-24" title="24"><span class="co">#&gt; 3 ElasticNet for subpop3 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-25" title="25"><span class="co">#&gt; 4 ElasticNet for subpop4 in target mixedpop2</span></a>
<a class="sourceLine" id="cb13-26" title="26"></a>
<a class="sourceLine" id="cb13-27" title="27"><span class="co"># summary maximum deviance explained by the model during the model training</span></a>
<a class="sourceLine" id="cb13-28" title="28"><span class="kw"><a href="../reference/summary_deviance.html">summary_deviance</a></span>(<span class="dt">object =</span> LSOLDA_dat)</a>
<a class="sourceLine" id="cb13-29" title="29"><span class="co">#&gt; $allDeviance</span></a>
<a class="sourceLine" id="cb13-30" title="30"><span class="co">#&gt; [1] "0.3624" "0.5885"</span></a>
<a class="sourceLine" id="cb13-31" title="31"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-32" title="32"><span class="co">#&gt; $DeviMax</span></a>
<a class="sourceLine" id="cb13-33" title="33"><span class="co">#&gt;           dat_DE$Dfd          Deviance           DEgenes</span></a>
<a class="sourceLine" id="cb13-34" title="34"><span class="co">#&gt; 1                  0            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-35" title="35"><span class="co">#&gt; 2                  1            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-36" title="36"><span class="co">#&gt; 3                  2            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-37" title="37"><span class="co">#&gt; 4                  5            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-38" title="38"><span class="co">#&gt; 5                  6            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-39" title="39"><span class="co">#&gt; 6                  8            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-40" title="40"><span class="co">#&gt; 7                  9            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-41" title="41"><span class="co">#&gt; 8                 11            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-42" title="42"><span class="co">#&gt; 9                 13            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-43" title="43"><span class="co">#&gt; 10                16            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-44" title="44"><span class="co">#&gt; 11                17            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-45" title="45"><span class="co">#&gt; 12                19            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-46" title="46"><span class="co">#&gt; 13                20            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-47" title="47"><span class="co">#&gt; 14                22            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-48" title="48"><span class="co">#&gt; 15                24            0.5885    genes_cluster1</span></a>
<a class="sourceLine" id="cb13-49" title="49"><span class="co">#&gt; 16 remaining DEgenes remaining DEgenes remaining DEgenes</span></a>
<a class="sourceLine" id="cb13-50" title="50"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-51" title="51"><span class="co">#&gt; $LassoGenesMax</span></a>
<a class="sourceLine" id="cb13-52" title="52"><span class="co">#&gt; NULL</span></a>
<a class="sourceLine" id="cb13-53" title="53"></a>
<a class="sourceLine" id="cb13-54" title="54"><span class="co"># summary accuracy to check the model accuracy in the leave-out test set</span></a>
<a class="sourceLine" id="cb13-55" title="55"><span class="kw"><a href="../reference/summary_accuracy.html">summary_accuracy</a></span>(<span class="dt">object =</span> LSOLDA_dat)</a>
<a class="sourceLine" id="cb13-56" title="56"><span class="co">#&gt; [1] 69.64286 70.53571</span></a></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters-in-one-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-one-sample" class="anchor"></a>4.3 Plot the relationship between clusters in one sample</h2>
<p>Here we look at one example use case to find relationship between clusters within one sample or between two sample</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="co">#run prediction for 3 clusters</span></a>
<a class="sourceLine" id="cb14-2" title="2">cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb14-3" title="3">cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb14-4" title="4"><span class="co">#cluster_mixedpop2 &lt;- as.numeric(as.vector(colData(mixedpop2)[,1]))</span></a>
<a class="sourceLine" id="cb14-5" title="5"></a>
<a class="sourceLine" id="cb14-6" title="6">c_selectID &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb14-7" title="7"><span class="co">#top 200 gene markers distinguishing cluster 1</span></a>
<a class="sourceLine" id="cb14-8" title="8">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb14-9" title="9"></a>
<a class="sourceLine" id="cb14-10" title="10">LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop2,</a>
<a class="sourceLine" id="cb14-11" title="11">                        <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb14-12" title="12">                        <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,</a>
<a class="sourceLine" id="cb14-13" title="13">                        <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a>
<a class="sourceLine" id="cb14-14" title="14"></a>
<a class="sourceLine" id="cb14-15" title="15">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb14-16" title="16">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb14-17" title="17"></a>
<a class="sourceLine" id="cb14-18" title="18">LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2,</a>
<a class="sourceLine" id="cb14-19" title="19">                        <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb14-20" title="20">                        <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,</a>
<a class="sourceLine" id="cb14-21" title="21">                        <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a>
<a class="sourceLine" id="cb14-22" title="22"></a>
<a class="sourceLine" id="cb14-23" title="23">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span></a>
<a class="sourceLine" id="cb14-24" title="24">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb14-25" title="25">LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2,</a>
<a class="sourceLine" id="cb14-26" title="26">                        <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb14-27" title="27">                        <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,</a>
<a class="sourceLine" id="cb14-28" title="28">                        <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a>
<a class="sourceLine" id="cb14-29" title="29"></a>
<a class="sourceLine" id="cb14-30" title="30">c_selectID &lt;-<span class="st"> </span><span class="dv">4</span></a>
<a class="sourceLine" id="cb14-31" title="31">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop4vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb14-32" title="32">LSOLDA_dat4 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2,</a>
<a class="sourceLine" id="cb14-33" title="33">                        <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb14-34" title="34">                        <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop2,</a>
<a class="sourceLine" id="cb14-35" title="35">                        <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a>
<a class="sourceLine" id="cb14-36" title="36"></a>
<a class="sourceLine" id="cb14-37" title="37"></a>
<a class="sourceLine" id="cb14-38" title="38"><span class="co">#prepare table input for sankey plot</span></a>
<a class="sourceLine" id="cb14-39" title="39"></a>
<a class="sourceLine" id="cb14-40" title="40">LASSO_C1S2  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/reformat_LASSO.html">reformat_LASSO</a></span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb14-41" title="41">                             <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1,</a>
<a class="sourceLine" id="cb14-42" title="42">                             <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])),</a>
<a class="sourceLine" id="cb14-43" title="43">                             <span class="dt">Nodes_group =</span><span class="st">"#7570b3"</span>)</a>
<a class="sourceLine" id="cb14-44" title="44"></a>
<a class="sourceLine" id="cb14-45" title="45">LASSO_C2S2  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/reformat_LASSO.html">reformat_LASSO</a></span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>,</a>
<a class="sourceLine" id="cb14-46" title="46">                             <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2,</a>
<a class="sourceLine" id="cb14-47" title="47">                             <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])),</a>
<a class="sourceLine" id="cb14-48" title="48">                             <span class="dt">Nodes_group =</span><span class="st">"#1b9e77"</span>)</a>
<a class="sourceLine" id="cb14-49" title="49"></a>
<a class="sourceLine" id="cb14-50" title="50">LASSO_C3S2  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/reformat_LASSO.html">reformat_LASSO</a></span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>,</a>
<a class="sourceLine" id="cb14-51" title="51">                             <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3,</a>
<a class="sourceLine" id="cb14-52" title="52">                             <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])),</a>
<a class="sourceLine" id="cb14-53" title="53">                             <span class="dt">Nodes_group =</span><span class="st">"#e7298a"</span>)</a>
<a class="sourceLine" id="cb14-54" title="54"></a>
<a class="sourceLine" id="cb14-55" title="55">LASSO_C4S2  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/reformat_LASSO.html">reformat_LASSO</a></span>(<span class="dt">c_selectID=</span><span class="dv">4</span>, <span class="dt">mp_selectID =</span><span class="dv">2</span>,</a>
<a class="sourceLine" id="cb14-56" title="56">                             <span class="dt">LSOLDA_dat=</span>LSOLDA_dat4,</a>
<a class="sourceLine" id="cb14-57" title="57">                             <span class="dt">nPredSubpop =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>])),</a>
<a class="sourceLine" id="cb14-58" title="58">                             <span class="dt">Nodes_group =</span><span class="st">"#00FFFF"</span>)</a>
<a class="sourceLine" id="cb14-59" title="59"></a>
<a class="sourceLine" id="cb14-60" title="60">combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2, LASSO_C4S2 )</a>
<a class="sourceLine" id="cb14-61" title="61">combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]</a>
<a class="sourceLine" id="cb14-62" title="62"></a>
<a class="sourceLine" id="cb14-63" title="63">nboots =<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb14-64" title="64"><span class="co">#links: source, target, value</span></a>
<a class="sourceLine" id="cb14-65" title="65"><span class="co">#source: node, nodegroup</span></a>
<a class="sourceLine" id="cb14-66" title="66">combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)],</a>
<a class="sourceLine" id="cb14-67" title="67">                     <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))])</a>
<a class="sourceLine" id="cb14-68" title="68"></a>
<a class="sourceLine" id="cb14-69" title="69"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)</a>
<a class="sourceLine" id="cb14-70" title="70"></a>
<a class="sourceLine" id="cb14-71" title="71">Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))</a>
<a class="sourceLine" id="cb14-72" title="72">Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))</a>
<a class="sourceLine" id="cb14-73" title="73">Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))</a>
<a class="sourceLine" id="cb14-74" title="74"></a>
<a class="sourceLine" id="cb14-75" title="75"><span class="co">#assign IDs for Source (start from 0)</span></a>
<a class="sourceLine" id="cb14-76" title="76">Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source</a>
<a class="sourceLine" id="cb14-77" title="77">Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target</a>
<a class="sourceLine" id="cb14-78" title="78"></a>
<a class="sourceLine" id="cb14-79" title="79"><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){</a>
<a class="sourceLine" id="cb14-80" title="80">   Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="dv">-1</span></a>
<a class="sourceLine" id="cb14-81" title="81">   Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="dv">-1</span></a>
<a class="sourceLine" id="cb14-82" title="82">}</a>
<a class="sourceLine" id="cb14-83" title="83"><span class="co"># </span></a>
<a class="sourceLine" id="cb14-84" title="84">combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)</a>
<a class="sourceLine" id="cb14-85" title="85">combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)</a>
<a class="sourceLine" id="cb14-86" title="86">combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup</a>
<a class="sourceLine" id="cb14-87" title="87"></a>
<a class="sourceLine" id="cb14-88" title="88"><span class="co">#prepare node info</span></a>
<a class="sourceLine" id="cb14-89" title="89">node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)</a>
<a class="sourceLine" id="cb14-90" title="90">node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))</a>
<a class="sourceLine" id="cb14-91" title="91"></a>
<a class="sourceLine" id="cb14-92" title="92"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))</a>
<a class="sourceLine" id="cb14-93" title="93">Color &lt;-<span class="st"> </span>combined <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(Node, <span class="dt">color=</span>NodeGroup) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb14-94" title="94">node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color<span class="op">$</span>color</a>
<a class="sourceLine" id="cb14-95" title="95"></a>
<a class="sourceLine" id="cb14-96" title="96"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))</a>
<a class="sourceLine" id="cb14-97" title="97">p1&lt;-<span class="kw">sankeyNetwork</span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,</a>
<a class="sourceLine" id="cb14-98" title="98">                 <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>,</a>
<a class="sourceLine" id="cb14-99" title="99">                 <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, <span class="dt">fontSize =</span> <span class="dv">22</span> )</a>
<a class="sourceLine" id="cb14-100" title="100">p1</a></code></pre></div>
<div id="htmlwidget-eb9d68251b896b134464" style="width:480px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-eb9d68251b896b134464">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2,3,3,3,3],"target":[0,1,2,3,0,1,2,3,0,1,2,3,0,1,2,3],"value":[84.75935828877,0,3.38345864661654,12.5,2.40641711229947,100,27.0676691729323,57.5,48.1283422459893,0,32.3308270676691,7.5,64.7058823529412,79.2857142857143,83.8345864661654,96.25],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a","#00FFFF","#00FFFF","#00FFFF","#00FFFF"]},"nodes":{"name":["C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#7570b3","#1b9e77","#e7298a","#00FFFF"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1"></a>
<a class="sourceLine" id="cb15-2" title="2"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></a></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters-in-two-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters-in-two-samples" class="anchor"></a>4.3 Plot the relationship between clusters in two samples</h2>
<p>Here we look at one example use case to find relationship between clusters within one sample or between two sample</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1"><span class="co">#run prediction for 3 clusters</span></a>
<a class="sourceLine" id="cb16-2" title="2">cluster_mixedpop1 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop1)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb16-3" title="3">cluster_mixedpop2 &lt;-<span class="st"> </span><span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb16-4" title="4">row_cluster &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))</a>
<a class="sourceLine" id="cb16-5" title="5"></a>
<a class="sourceLine" id="cb16-6" title="6">c_selectID &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb16-7" title="7"><span class="co">#top 200 gene markers distinguishing cluster 1</span></a>
<a class="sourceLine" id="cb16-8" title="8">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb16-9" title="9">LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1,</a>
<a class="sourceLine" id="cb16-10" title="10">                        <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb16-11" title="11">                        <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,</a>
<a class="sourceLine" id="cb16-12" title="12">                        <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a>
<a class="sourceLine" id="cb16-13" title="13"></a>
<a class="sourceLine" id="cb16-14" title="14"></a>
<a class="sourceLine" id="cb16-15" title="15">c_selectID &lt;-<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb16-16" title="16">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb16-17" title="17">LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1,</a>
<a class="sourceLine" id="cb16-18" title="18">                        <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb16-19" title="19">                        <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,</a>
<a class="sourceLine" id="cb16-20" title="20">                        <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a>
<a class="sourceLine" id="cb16-21" title="21"></a>
<a class="sourceLine" id="cb16-22" title="22">c_selectID &lt;-<span class="st"> </span><span class="dv">3</span></a>
<a class="sourceLine" id="cb16-23" title="23">genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb16-24" title="24">LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_prediction.html">bootstrap_prediction</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop1,</a>
<a class="sourceLine" id="cb16-25" title="25">                        <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(),</a>
<a class="sourceLine" id="cb16-26" title="26">                        <span class="dt">cluster_mixedpop1 =</span> cluster_mixedpop1,</a>
<a class="sourceLine" id="cb16-27" title="27">                        <span class="dt">cluster_mixedpop2 =</span> cluster_mixedpop2)</a>
<a class="sourceLine" id="cb16-28" title="28"></a>
<a class="sourceLine" id="cb16-29" title="29"><span class="co">#prepare table input for sankey plot</span></a>
<a class="sourceLine" id="cb16-30" title="30"></a>
<a class="sourceLine" id="cb16-31" title="31">LASSO_C1S1  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/reformat_LASSO.html">reformat_LASSO</a></span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb16-32" title="32">                             <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#7570b3"</span>)</a>
<a class="sourceLine" id="cb16-33" title="33"></a>
<a class="sourceLine" id="cb16-34" title="34">LASSO_C2S1  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/reformat_LASSO.html">reformat_LASSO</a></span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb16-35" title="35">                             <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#1b9e77"</span>)</a>
<a class="sourceLine" id="cb16-36" title="36"></a>
<a class="sourceLine" id="cb16-37" title="37">LASSO_C3S1  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/reformat_LASSO.html">reformat_LASSO</a></span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">mp_selectID =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb16-38" title="38">                             <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#e7298a"</span>)</a>
<a class="sourceLine" id="cb16-39" title="39"></a>
<a class="sourceLine" id="cb16-40" title="40"></a>
<a class="sourceLine" id="cb16-41" title="41">combined &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind</a></span>(LASSO_C1S1,LASSO_C2S1,LASSO_C3S1)</a>
<a class="sourceLine" id="cb16-42" title="42"></a>
<a class="sourceLine" id="cb16-43" title="43">nboots =<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb16-44" title="44"><span class="co">#links: source, target, value</span></a>
<a class="sourceLine" id="cb16-45" title="45"><span class="co">#source: node, nodegroup</span></a>
<a class="sourceLine" id="cb16-46" title="46">combined_D3obj &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">Nodes=</span>combined[,(nboots<span class="op">+</span><span class="dv">3</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">4</span>)],</a>
<a class="sourceLine" id="cb16-47" title="47">                     <span class="dt">Links=</span>combined[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>((nboots<span class="op">+</span><span class="dv">2</span>)<span class="op">:</span>(nboots<span class="op">+</span><span class="dv">1</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(combined))])</a>
<a class="sourceLine" id="cb16-48" title="48">combined &lt;-<span class="st"> </span>combined[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]</a>
<a class="sourceLine" id="cb16-49" title="49"></a>
<a class="sourceLine" id="cb16-50" title="50"></a>
<a class="sourceLine" id="cb16-51" title="51"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3)</a>
<a class="sourceLine" id="cb16-52" title="52"></a>
<a class="sourceLine" id="cb16-53" title="53">Node_source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))</a>
<a class="sourceLine" id="cb16-54" title="54">Node_target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))</a>
<a class="sourceLine" id="cb16-55" title="55">Node_all &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(Node_source, Node_target))</a>
<a class="sourceLine" id="cb16-56" title="56"></a>
<a class="sourceLine" id="cb16-57" title="57"><span class="co">#assign IDs for Source (start from 0)</span></a>
<a class="sourceLine" id="cb16-58" title="58">Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source</a>
<a class="sourceLine" id="cb16-59" title="59">Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target</a>
<a class="sourceLine" id="cb16-60" title="60"></a>
<a class="sourceLine" id="cb16-61" title="61"><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)){</a>
<a class="sourceLine" id="cb16-62" title="62">   Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="dv">-1</span></a>
<a class="sourceLine" id="cb16-63" title="63">   Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="dv">-1</span></a>
<a class="sourceLine" id="cb16-64" title="64">}</a>
<a class="sourceLine" id="cb16-65" title="65"></a>
<a class="sourceLine" id="cb16-66" title="66">combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Source)</a>
<a class="sourceLine" id="cb16-67" title="67">combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(Target)</a>
<a class="sourceLine" id="cb16-68" title="68">combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup</a>
<a class="sourceLine" id="cb16-69" title="69"></a>
<a class="sourceLine" id="cb16-70" title="70"><span class="co">#prepare node info</span></a>
<a class="sourceLine" id="cb16-71" title="71">node_df &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">Node=</span>Node_all)</a>
<a class="sourceLine" id="cb16-72" title="72">node_df<span class="op">$</span>id &lt;-<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))</a>
<a class="sourceLine" id="cb16-73" title="73"></a>
<a class="sourceLine" id="cb16-74" title="74"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr))</a>
<a class="sourceLine" id="cb16-75" title="75">n &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(node_df<span class="op">$</span>Node))</a>
<a class="sourceLine" id="cb16-76" title="76">getPalette =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/colorRamp">colorRampPalette</a></span>(RColorBrewer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">9</span>, <span class="st">"Set1"</span>))</a>
<a class="sourceLine" id="cb16-77" title="77">Color =<span class="st"> </span><span class="kw">getPalette</span>(n)</a>
<a class="sourceLine" id="cb16-78" title="78">node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color</a>
<a class="sourceLine" id="cb16-79" title="79"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(networkD3))</a>
<a class="sourceLine" id="cb16-80" title="80">p1&lt;-<span class="kw">sankeyNetwork</span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,</a>
<a class="sourceLine" id="cb16-81" title="81">                 <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>,</a>
<a class="sourceLine" id="cb16-82" title="82">                 <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, <span class="dt">fontSize =</span> <span class="dv">22</span>)</a>
<a class="sourceLine" id="cb16-83" title="83">p1</a></code></pre></div>
<div id="htmlwidget-57a7dbf72d90caa148a0" style="width:480px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-57a7dbf72d90caa148a0">{"x":{"links":{"source":[0,0,0,0,1,1,1,1,2,2,2,2],"target":[3,4,5,6,3,4,5,6,3,4,5,6],"value":[54.0106951871658,60,53.0075187969925,71.25,48.6631016042781,16.0714285714286,76.6917293233083,58.75,53.2085561497326,40.3571428571428,66.1654135338346,50],"group":["#7570b3","#7570b3","#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a","#e7298a","#e7298a"]},"nodes":{"name":["C1_MP1","C2_MP1","C3_MP1","C1_MP2","C2_MP2","C3_MP2","C4_MP2"],"group":["#E41A1C","#3E8E93","#7E6E85","#FF7F00","#E1C62F","#DB728C","#999999"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1"><span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span></a></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">session_info</a></span>()</a>
<a class="sourceLine" id="cb18-2" title="2"><span class="co">#&gt; - Session info ----------------------------------------------------------</span></a>
<a class="sourceLine" id="cb18-3" title="3"><span class="co">#&gt;  setting  value                       </span></a>
<a class="sourceLine" id="cb18-4" title="4"><span class="co">#&gt;  version  R version 3.5.1 (2018-07-02)</span></a>
<a class="sourceLine" id="cb18-5" title="5"><span class="co">#&gt;  os       Windows 10 x64              </span></a>
<a class="sourceLine" id="cb18-6" title="6"><span class="co">#&gt;  system   x86_64, mingw32             </span></a>
<a class="sourceLine" id="cb18-7" title="7"><span class="co">#&gt;  ui       RTerm                       </span></a>
<a class="sourceLine" id="cb18-8" title="8"><span class="co">#&gt;  language (EN)                        </span></a>
<a class="sourceLine" id="cb18-9" title="9"><span class="co">#&gt;  collate  English_Australia.1252      </span></a>
<a class="sourceLine" id="cb18-10" title="10"><span class="co">#&gt;  ctype    English_Australia.1252      </span></a>
<a class="sourceLine" id="cb18-11" title="11"><span class="co">#&gt;  tz       Australia/Brisbane          </span></a>
<a class="sourceLine" id="cb18-12" title="12"><span class="co">#&gt;  date     2019-06-25                  </span></a>
<a class="sourceLine" id="cb18-13" title="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-14" title="14"><span class="co">#&gt; - Packages --------------------------------------------------------------</span></a>
<a class="sourceLine" id="cb18-15" title="15"><span class="co">#&gt;  ! package              * version     date       lib</span></a>
<a class="sourceLine" id="cb18-16" title="16"><span class="co">#&gt;    annotate               1.60.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-17" title="17"><span class="co">#&gt;    AnnotationDbi        * 1.44.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-18" title="18"><span class="co">#&gt;    assertthat             0.2.0       2017-04-11 [1]</span></a>
<a class="sourceLine" id="cb18-19" title="19"><span class="co">#&gt;    backports              1.1.2       2017-12-13 [1]</span></a>
<a class="sourceLine" id="cb18-20" title="20"><span class="co">#&gt;    base64enc              0.1-3       2015-07-28 [1]</span></a>
<a class="sourceLine" id="cb18-21" title="21"><span class="co">#&gt;    bindr                  0.1.1       2018-03-13 [1]</span></a>
<a class="sourceLine" id="cb18-22" title="22"><span class="co">#&gt;    bindrcpp             * 0.2.2       2018-03-29 [1]</span></a>
<a class="sourceLine" id="cb18-23" title="23"><span class="co">#&gt;    Biobase              * 2.42.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-24" title="24"><span class="co">#&gt;    BiocGenerics         * 0.28.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-25" title="25"><span class="co">#&gt;    BiocParallel         * 1.16.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-26" title="26"><span class="co">#&gt;    bit                    1.1-14      2018-05-29 [1]</span></a>
<a class="sourceLine" id="cb18-27" title="27"><span class="co">#&gt;    bit64                  0.9-7       2017-05-08 [1]</span></a>
<a class="sourceLine" id="cb18-28" title="28"><span class="co">#&gt;    bitops                 1.0-6       2013-08-17 [1]</span></a>
<a class="sourceLine" id="cb18-29" title="29"><span class="co">#&gt;    blob                   1.1.1       2018-03-25 [1]</span></a>
<a class="sourceLine" id="cb18-30" title="30"><span class="co">#&gt;    callr                  3.0.0       2018-08-24 [1]</span></a>
<a class="sourceLine" id="cb18-31" title="31"><span class="co">#&gt;    caret                * 6.0-81      2018-11-20 [1]</span></a>
<a class="sourceLine" id="cb18-32" title="32"><span class="co">#&gt;    checkmate              1.8.5       2017-10-24 [1]</span></a>
<a class="sourceLine" id="cb18-33" title="33"><span class="co">#&gt;    class                  7.3-14      2015-08-30 [2]</span></a>
<a class="sourceLine" id="cb18-34" title="34"><span class="co">#&gt;    cli                    1.0.1       2018-09-25 [1]</span></a>
<a class="sourceLine" id="cb18-35" title="35"><span class="co">#&gt;    cluster                2.0.7-1     2018-04-13 [2]</span></a>
<a class="sourceLine" id="cb18-36" title="36"><span class="co">#&gt;    clusterProfiler      * 3.10.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-37" title="37"><span class="co">#&gt;    codetools              0.2-15      2016-10-05 [2]</span></a>
<a class="sourceLine" id="cb18-38" title="38"><span class="co">#&gt;    colorspace             1.3-2       2016-12-14 [1]</span></a>
<a class="sourceLine" id="cb18-39" title="39"><span class="co">#&gt;    commonmark             1.6         2018-09-30 [1]</span></a>
<a class="sourceLine" id="cb18-40" title="40"><span class="co">#&gt;    cowplot                0.9.3       2018-07-15 [1]</span></a>
<a class="sourceLine" id="cb18-41" title="41"><span class="co">#&gt;    crayon                 1.3.4       2017-09-16 [1]</span></a>
<a class="sourceLine" id="cb18-42" title="42"><span class="co">#&gt;    data.table             1.11.8      2018-09-30 [1]</span></a>
<a class="sourceLine" id="cb18-43" title="43"><span class="co">#&gt;    DBI                    1.0.0       2018-05-02 [1]</span></a>
<a class="sourceLine" id="cb18-44" title="44"><span class="co">#&gt;    DelayedArray         * 0.8.0       2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-45" title="45"><span class="co">#&gt;    dendextend             1.9.0       2018-10-19 [1]</span></a>
<a class="sourceLine" id="cb18-46" title="46"><span class="co">#&gt;    DEoptimR               1.0-8       2016-11-19 [1]</span></a>
<a class="sourceLine" id="cb18-47" title="47"><span class="co">#&gt;    desc                   1.2.0       2018-05-01 [1]</span></a>
<a class="sourceLine" id="cb18-48" title="48"><span class="co">#&gt;    DESeq                * 1.34.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-49" title="49"><span class="co">#&gt;    devtools               2.0.1.9000  2018-11-25 [1]</span></a>
<a class="sourceLine" id="cb18-50" title="50"><span class="co">#&gt;    digest                 0.6.18      2018-10-10 [1]</span></a>
<a class="sourceLine" id="cb18-51" title="51"><span class="co">#&gt;    diptest                0.75-7      2016-12-05 [1]</span></a>
<a class="sourceLine" id="cb18-52" title="52"><span class="co">#&gt;    DO.db                  2.9         2018-12-07 [1]</span></a>
<a class="sourceLine" id="cb18-53" title="53"><span class="co">#&gt;    DOSE                 * 3.8.0       2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-54" title="54"><span class="co">#&gt;    dplyr                * 0.7.8       2018-11-10 [1]</span></a>
<a class="sourceLine" id="cb18-55" title="55"><span class="co">#&gt;    dynamicTreeCut       * 1.63-1      2016-03-11 [1]</span></a>
<a class="sourceLine" id="cb18-56" title="56"><span class="co">#&gt;    e1071                  1.7-0       2018-07-28 [1]</span></a>
<a class="sourceLine" id="cb18-57" title="57"><span class="co">#&gt;    enrichplot             1.2.0       2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-58" title="58"><span class="co">#&gt;    europepmc              0.3         2018-04-20 [1]</span></a>
<a class="sourceLine" id="cb18-59" title="59"><span class="co">#&gt;    evaluate               0.12        2018-10-09 [1]</span></a>
<a class="sourceLine" id="cb18-60" title="60"><span class="co">#&gt;    farver                 1.1.0       2018-11-20 [1]</span></a>
<a class="sourceLine" id="cb18-61" title="61"><span class="co">#&gt;    fastcluster            1.1.25      2018-06-07 [1]</span></a>
<a class="sourceLine" id="cb18-62" title="62"><span class="co">#&gt;    fastmatch              1.1-0       2017-01-28 [1]</span></a>
<a class="sourceLine" id="cb18-63" title="63"><span class="co">#&gt;    fgsea                  1.8.0       2018-10-31 [1]</span></a>
<a class="sourceLine" id="cb18-64" title="64"><span class="co">#&gt;    flexmix                2.3-14      2017-04-28 [1]</span></a>
<a class="sourceLine" id="cb18-65" title="65"><span class="co">#&gt;    foreach                1.4.4       2017-12-12 [1]</span></a>
<a class="sourceLine" id="cb18-66" title="66"><span class="co">#&gt;    fpc                    2.1-11.1    2018-07-20 [1]</span></a>
<a class="sourceLine" id="cb18-67" title="67"><span class="co">#&gt;    fs                     1.2.6       2018-08-23 [1]</span></a>
<a class="sourceLine" id="cb18-68" title="68"><span class="co">#&gt;    genefilter             1.64.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-69" title="69"><span class="co">#&gt;    geneplotter            1.60.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-70" title="70"><span class="co">#&gt;    generics               0.0.1       2018-10-26 [1]</span></a>
<a class="sourceLine" id="cb18-71" title="71"><span class="co">#&gt;    GenomeInfoDb         * 1.18.1      2018-11-12 [1]</span></a>
<a class="sourceLine" id="cb18-72" title="72"><span class="co">#&gt;    GenomeInfoDbData       1.2.0       2018-11-25 [1]</span></a>
<a class="sourceLine" id="cb18-73" title="73"><span class="co">#&gt;    GenomicRanges        * 1.34.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-74" title="74"><span class="co">#&gt;    ggforce                0.1.3       2018-07-07 [1]</span></a>
<a class="sourceLine" id="cb18-75" title="75"><span class="co">#&gt;    ggplot2              * 3.1.0       2018-10-25 [1]</span></a>
<a class="sourceLine" id="cb18-76" title="76"><span class="co">#&gt;    ggplotify              0.0.3       2018-08-03 [1]</span></a>
<a class="sourceLine" id="cb18-77" title="77"><span class="co">#&gt;    ggraph                 1.0.2       2018-07-07 [1]</span></a>
<a class="sourceLine" id="cb18-78" title="78"><span class="co">#&gt;    ggrepel                0.8.0       2018-05-09 [1]</span></a>
<a class="sourceLine" id="cb18-79" title="79"><span class="co">#&gt;    ggridges               0.5.1       2018-09-27 [1]</span></a>
<a class="sourceLine" id="cb18-80" title="80"><span class="co">#&gt;    glmnet                 2.0-16      2018-04-02 [1]</span></a>
<a class="sourceLine" id="cb18-81" title="81"><span class="co">#&gt;    glue                   1.3.0       2018-07-17 [1]</span></a>
<a class="sourceLine" id="cb18-82" title="82"><span class="co">#&gt;    GO.db                  3.7.0       2018-12-07 [1]</span></a>
<a class="sourceLine" id="cb18-83" title="83"><span class="co">#&gt;    GOSemSim               2.8.0       2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-84" title="84"><span class="co">#&gt;    gower                  0.1.2       2017-02-23 [1]</span></a>
<a class="sourceLine" id="cb18-85" title="85"><span class="co">#&gt;    graph                  1.60.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-86" title="86"><span class="co">#&gt;    graphite               1.28.1      2018-11-24 [1]</span></a>
<a class="sourceLine" id="cb18-87" title="87"><span class="co">#&gt;    gridExtra              2.3         2017-09-09 [1]</span></a>
<a class="sourceLine" id="cb18-88" title="88"><span class="co">#&gt;    gridGraphics           0.3-0       2018-04-03 [1]</span></a>
<a class="sourceLine" id="cb18-89" title="89"><span class="co">#&gt;    gtable                 0.2.0       2016-02-26 [1]</span></a>
<a class="sourceLine" id="cb18-90" title="90"><span class="co">#&gt;    hms                    0.4.2       2018-03-10 [1]</span></a>
<a class="sourceLine" id="cb18-91" title="91"><span class="co">#&gt;    htmltools              0.3.6       2017-04-28 [1]</span></a>
<a class="sourceLine" id="cb18-92" title="92"><span class="co">#&gt;    htmlwidgets            1.3         2018-09-30 [1]</span></a>
<a class="sourceLine" id="cb18-93" title="93"><span class="co">#&gt;    httr                   1.3.1       2017-08-20 [1]</span></a>
<a class="sourceLine" id="cb18-94" title="94"><span class="co">#&gt;    igraph                 1.2.2       2018-07-27 [1]</span></a>
<a class="sourceLine" id="cb18-95" title="95"><span class="co">#&gt;    ipred                  0.9-8       2018-11-05 [1]</span></a>
<a class="sourceLine" id="cb18-96" title="96"><span class="co">#&gt;    IRanges              * 2.16.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-97" title="97"><span class="co">#&gt;    iterators              1.0.10      2018-07-13 [1]</span></a>
<a class="sourceLine" id="cb18-98" title="98"><span class="co">#&gt;    jsonlite               1.5         2017-06-01 [1]</span></a>
<a class="sourceLine" id="cb18-99" title="99"><span class="co">#&gt;    kernlab                0.9-27      2018-08-10 [1]</span></a>
<a class="sourceLine" id="cb18-100" title="100"><span class="co">#&gt;    knitr                  1.21        2018-12-10 [1]</span></a>
<a class="sourceLine" id="cb18-101" title="101"><span class="co">#&gt;    labeling               0.3         2014-08-23 [1]</span></a>
<a class="sourceLine" id="cb18-102" title="102"><span class="co">#&gt;    lattice              * 0.20-35     2017-03-25 [2]</span></a>
<a class="sourceLine" id="cb18-103" title="103"><span class="co">#&gt;    lava                   1.6.4       2018-11-25 [1]</span></a>
<a class="sourceLine" id="cb18-104" title="104"><span class="co">#&gt;    lazyeval               0.2.1       2017-10-29 [1]</span></a>
<a class="sourceLine" id="cb18-105" title="105"><span class="co">#&gt;    locfit               * 1.5-9.1     2013-04-20 [1]</span></a>
<a class="sourceLine" id="cb18-106" title="106"><span class="co">#&gt;    lubridate              1.7.4       2018-04-11 [1]</span></a>
<a class="sourceLine" id="cb18-107" title="107"><span class="co">#&gt;    magrittr               1.5         2014-11-22 [1]</span></a>
<a class="sourceLine" id="cb18-108" title="108"><span class="co">#&gt;    MASS                   7.3-50      2018-04-30 [2]</span></a>
<a class="sourceLine" id="cb18-109" title="109"><span class="co">#&gt;    Matrix                 1.2-14      2018-04-13 [2]</span></a>
<a class="sourceLine" id="cb18-110" title="110"><span class="co">#&gt;    matrixStats          * 0.54.0      2018-07-23 [1]</span></a>
<a class="sourceLine" id="cb18-111" title="111"><span class="co">#&gt;    mclust                 5.4.2       2018-11-17 [1]</span></a>
<a class="sourceLine" id="cb18-112" title="112"><span class="co">#&gt;    memoise                1.1.0       2017-04-21 [1]</span></a>
<a class="sourceLine" id="cb18-113" title="113"><span class="co">#&gt;    ModelMetrics           1.2.2       2018-11-03 [1]</span></a>
<a class="sourceLine" id="cb18-114" title="114"><span class="co">#&gt;    modeltools             0.2-22      2018-07-16 [1]</span></a>
<a class="sourceLine" id="cb18-115" title="115"><span class="co">#&gt;    munsell                0.5.0       2018-06-12 [1]</span></a>
<a class="sourceLine" id="cb18-116" title="116"><span class="co">#&gt;    mvtnorm                1.0-8       2018-05-31 [1]</span></a>
<a class="sourceLine" id="cb18-117" title="117"><span class="co">#&gt;    networkD3            * 0.4         2017-03-18 [1]</span></a>
<a class="sourceLine" id="cb18-118" title="118"><span class="co">#&gt;    nlme                   3.1-137     2018-04-07 [2]</span></a>
<a class="sourceLine" id="cb18-119" title="119"><span class="co">#&gt;    nnet                   7.3-12      2016-02-02 [2]</span></a>
<a class="sourceLine" id="cb18-120" title="120"><span class="co">#&gt;    org.Hs.eg.db         * 3.7.0       2018-12-07 [1]</span></a>
<a class="sourceLine" id="cb18-121" title="121"><span class="co">#&gt;    pillar                 1.3.0       2018-07-14 [1]</span></a>
<a class="sourceLine" id="cb18-122" title="122"><span class="co">#&gt;    pkgbuild               1.0.2       2018-10-16 [1]</span></a>
<a class="sourceLine" id="cb18-123" title="123"><span class="co">#&gt;    pkgconfig              2.0.2       2018-08-16 [1]</span></a>
<a class="sourceLine" id="cb18-124" title="124"><span class="co">#&gt;    pkgdown                1.3.0       2018-12-07 [1]</span></a>
<a class="sourceLine" id="cb18-125" title="125"><span class="co">#&gt;    pkgload                1.0.2       2018-10-29 [1]</span></a>
<a class="sourceLine" id="cb18-126" title="126"><span class="co">#&gt;    plyr                   1.8.4       2016-06-08 [1]</span></a>
<a class="sourceLine" id="cb18-127" title="127"><span class="co">#&gt;    prabclus               2.2-6       2015-01-14 [1]</span></a>
<a class="sourceLine" id="cb18-128" title="128"><span class="co">#&gt;    prettyunits            1.0.2       2015-07-13 [1]</span></a>
<a class="sourceLine" id="cb18-129" title="129"><span class="co">#&gt;    processx               3.2.0       2018-08-16 [1]</span></a>
<a class="sourceLine" id="cb18-130" title="130"><span class="co">#&gt;    prodlim                2018.04.18  2018-04-18 [1]</span></a>
<a class="sourceLine" id="cb18-131" title="131"><span class="co">#&gt;    progress               1.2.0       2018-06-14 [1]</span></a>
<a class="sourceLine" id="cb18-132" title="132"><span class="co">#&gt;    ps                     1.2.1       2018-11-06 [1]</span></a>
<a class="sourceLine" id="cb18-133" title="133"><span class="co">#&gt;    purrr                  0.2.5       2018-05-29 [1]</span></a>
<a class="sourceLine" id="cb18-134" title="134"><span class="co">#&gt;    qvalue                 2.14.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-135" title="135"><span class="co">#&gt;    R6                     2.3.0       2018-10-04 [1]</span></a>
<a class="sourceLine" id="cb18-136" title="136"><span class="co">#&gt;    rappdirs               0.3.1       2016-03-28 [1]</span></a>
<a class="sourceLine" id="cb18-137" title="137"><span class="co">#&gt;    RColorBrewer           1.1-2       2014-12-07 [1]</span></a>
<a class="sourceLine" id="cb18-138" title="138"><span class="co">#&gt;    Rcpp                   1.0.0       2018-11-07 [1]</span></a>
<a class="sourceLine" id="cb18-139" title="139"><span class="co">#&gt;    RcppArmadillo          0.9.200.4.0 2018-11-10 [1]</span></a>
<a class="sourceLine" id="cb18-140" title="140"><span class="co">#&gt;  D RcppParallel           4.4.1       2018-07-19 [1]</span></a>
<a class="sourceLine" id="cb18-141" title="141"><span class="co">#&gt;    RCurl                  1.95-4.11   2018-07-15 [1]</span></a>
<a class="sourceLine" id="cb18-142" title="142"><span class="co">#&gt;    reactome.db            1.66.0      2018-12-07 [1]</span></a>
<a class="sourceLine" id="cb18-143" title="143"><span class="co">#&gt;    ReactomePA           * 1.26.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-144" title="144"><span class="co">#&gt;    recipes                0.1.4       2018-11-19 [1]</span></a>
<a class="sourceLine" id="cb18-145" title="145"><span class="co">#&gt;    remotes                2.0.2       2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-146" title="146"><span class="co">#&gt;    reshape2               1.4.3       2017-12-11 [1]</span></a>
<a class="sourceLine" id="cb18-147" title="147"><span class="co">#&gt;    rlang                  0.3.0.1     2018-10-25 [1]</span></a>
<a class="sourceLine" id="cb18-148" title="148"><span class="co">#&gt;    rmarkdown              1.10        2018-06-11 [1]</span></a>
<a class="sourceLine" id="cb18-149" title="149"><span class="co">#&gt;    robustbase             0.93-3      2018-09-21 [1]</span></a>
<a class="sourceLine" id="cb18-150" title="150"><span class="co">#&gt;    roxygen2               6.1.1       2018-11-07 [1]</span></a>
<a class="sourceLine" id="cb18-151" title="151"><span class="co">#&gt;    rpart                  4.1-13      2018-02-23 [2]</span></a>
<a class="sourceLine" id="cb18-152" title="152"><span class="co">#&gt;    rprojroot              1.3-2       2018-01-03 [1]</span></a>
<a class="sourceLine" id="cb18-153" title="153"><span class="co">#&gt;    RSQLite                2.1.1       2018-05-06 [1]</span></a>
<a class="sourceLine" id="cb18-154" title="154"><span class="co">#&gt;    rstudioapi             0.8         2018-10-02 [1]</span></a>
<a class="sourceLine" id="cb18-155" title="155"><span class="co">#&gt;    Rtsne                  0.15        2018-11-10 [1]</span></a>
<a class="sourceLine" id="cb18-156" title="156"><span class="co">#&gt;    rvcheck                0.1.3       2018-12-06 [1]</span></a>
<a class="sourceLine" id="cb18-157" title="157"><span class="co">#&gt;    S4Vectors            * 0.20.1      2018-11-09 [1]</span></a>
<a class="sourceLine" id="cb18-158" title="158"><span class="co">#&gt;    scales                 1.0.0       2018-08-09 [1]</span></a>
<a class="sourceLine" id="cb18-159" title="159"><span class="co">#&gt;    scGPS                * 0.99.20     2019-06-25 [1]</span></a>
<a class="sourceLine" id="cb18-160" title="160"><span class="co">#&gt;    sessioninfo            1.1.1       2018-11-05 [1]</span></a>
<a class="sourceLine" id="cb18-161" title="161"><span class="co">#&gt;    SingleCellExperiment * 1.4.0       2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-162" title="162"><span class="co">#&gt;    stringi                1.2.4       2018-07-20 [1]</span></a>
<a class="sourceLine" id="cb18-163" title="163"><span class="co">#&gt;    stringr                1.3.1       2018-05-10 [1]</span></a>
<a class="sourceLine" id="cb18-164" title="164"><span class="co">#&gt;    SummarizedExperiment * 1.12.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-165" title="165"><span class="co">#&gt;    survival               2.42-3      2018-04-16 [2]</span></a>
<a class="sourceLine" id="cb18-166" title="166"><span class="co">#&gt;    testthat               2.0.1       2018-10-13 [1]</span></a>
<a class="sourceLine" id="cb18-167" title="167"><span class="co">#&gt;    tibble                 1.4.2       2018-01-22 [1]</span></a>
<a class="sourceLine" id="cb18-168" title="168"><span class="co">#&gt;    tidyr                  0.8.2       2018-10-28 [1]</span></a>
<a class="sourceLine" id="cb18-169" title="169"><span class="co">#&gt;    tidyselect             0.2.5       2018-10-11 [1]</span></a>
<a class="sourceLine" id="cb18-170" title="170"><span class="co">#&gt;    timeDate               3043.102    2018-02-21 [1]</span></a>
<a class="sourceLine" id="cb18-171" title="171"><span class="co">#&gt;    triebeard              0.3.0       2016-08-04 [1]</span></a>
<a class="sourceLine" id="cb18-172" title="172"><span class="co">#&gt;    trimcluster            0.1-2.1     2018-07-20 [1]</span></a>
<a class="sourceLine" id="cb18-173" title="173"><span class="co">#&gt;    tweenr                 1.0.0       2018-09-27 [1]</span></a>
<a class="sourceLine" id="cb18-174" title="174"><span class="co">#&gt;    units                  0.6-2       2018-12-05 [1]</span></a>
<a class="sourceLine" id="cb18-175" title="175"><span class="co">#&gt;    UpSetR                 1.3.3       2017-03-21 [1]</span></a>
<a class="sourceLine" id="cb18-176" title="176"><span class="co">#&gt;    urltools               1.7.1       2018-08-03 [1]</span></a>
<a class="sourceLine" id="cb18-177" title="177"><span class="co">#&gt;    usethis                1.4.0       2018-08-14 [1]</span></a>
<a class="sourceLine" id="cb18-178" title="178"><span class="co">#&gt;    viridis                0.5.1       2018-03-29 [1]</span></a>
<a class="sourceLine" id="cb18-179" title="179"><span class="co">#&gt;    viridisLite            0.3.0       2018-02-01 [1]</span></a>
<a class="sourceLine" id="cb18-180" title="180"><span class="co">#&gt;    whisker                0.3-2       2013-04-28 [1]</span></a>
<a class="sourceLine" id="cb18-181" title="181"><span class="co">#&gt;    withr                  2.1.2       2018-03-15 [1]</span></a>
<a class="sourceLine" id="cb18-182" title="182"><span class="co">#&gt;    xfun                   0.4         2018-10-23 [1]</span></a>
<a class="sourceLine" id="cb18-183" title="183"><span class="co">#&gt;    XML                    3.98-1.16   2018-08-19 [1]</span></a>
<a class="sourceLine" id="cb18-184" title="184"><span class="co">#&gt;    xml2                   1.2.0       2018-01-24 [1]</span></a>
<a class="sourceLine" id="cb18-185" title="185"><span class="co">#&gt;    xtable                 1.8-3       2018-08-29 [1]</span></a>
<a class="sourceLine" id="cb18-186" title="186"><span class="co">#&gt;    XVector                0.22.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-187" title="187"><span class="co">#&gt;    yaml                   2.2.0       2018-07-25 [1]</span></a>
<a class="sourceLine" id="cb18-188" title="188"><span class="co">#&gt;    zlibbioc               1.28.0      2018-10-30 [1]</span></a>
<a class="sourceLine" id="cb18-189" title="189"><span class="co">#&gt;  source                         </span></a>
<a class="sourceLine" id="cb18-190" title="190"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-191" title="191"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-192" title="192"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-193" title="193"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-194" title="194"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-195" title="195"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-196" title="196"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-197" title="197"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-198" title="198"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-199" title="199"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-200" title="200"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-201" title="201"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-202" title="202"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-203" title="203"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-204" title="204"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-205" title="205"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-206" title="206"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-207" title="207"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-208" title="208"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-209" title="209"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-210" title="210"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-211" title="211"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-212" title="212"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-213" title="213"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-214" title="214"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-215" title="215"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-216" title="216"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-217" title="217"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-218" title="218"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-219" title="219"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-220" title="220"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-221" title="221"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-222" title="222"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-223" title="223"><span class="co">#&gt;  Github (r-lib/devtools@8a3d2c2)</span></a>
<a class="sourceLine" id="cb18-224" title="224"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-225" title="225"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-226" title="226"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-227" title="227"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-228" title="228"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-229" title="229"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-230" title="230"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-231" title="231"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-232" title="232"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-233" title="233"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-234" title="234"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-235" title="235"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-236" title="236"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-237" title="237"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-238" title="238"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-239" title="239"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-240" title="240"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-241" title="241"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-242" title="242"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-243" title="243"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-244" title="244"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-245" title="245"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-246" title="246"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-247" title="247"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-248" title="248"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-249" title="249"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-250" title="250"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-251" title="251"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-252" title="252"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-253" title="253"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-254" title="254"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-255" title="255"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-256" title="256"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-257" title="257"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-258" title="258"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-259" title="259"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-260" title="260"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-261" title="261"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-262" title="262"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-263" title="263"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-264" title="264"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-265" title="265"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-266" title="266"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-267" title="267"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-268" title="268"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-269" title="269"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-270" title="270"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-271" title="271"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-272" title="272"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-273" title="273"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-274" title="274"><span class="co">#&gt;  CRAN (R 3.5.2)                 </span></a>
<a class="sourceLine" id="cb18-275" title="275"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-276" title="276"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-277" title="277"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-278" title="278"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-279" title="279"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-280" title="280"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-281" title="281"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-282" title="282"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-283" title="283"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-284" title="284"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-285" title="285"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-286" title="286"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-287" title="287"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-288" title="288"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-289" title="289"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-290" title="290"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-291" title="291"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-292" title="292"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-293" title="293"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-294" title="294"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-295" title="295"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-296" title="296"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-297" title="297"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-298" title="298"><span class="co">#&gt;  CRAN (R 3.5.2)                 </span></a>
<a class="sourceLine" id="cb18-299" title="299"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-300" title="300"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-301" title="301"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-302" title="302"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-303" title="303"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-304" title="304"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-305" title="305"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-306" title="306"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-307" title="307"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-308" title="308"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-309" title="309"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-310" title="310"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-311" title="311"><span class="co">#&gt;  CRAN (R 3.5.0)                 </span></a>
<a class="sourceLine" id="cb18-312" title="312"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-313" title="313"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-314" title="314"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-315" title="315"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-316" title="316"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-317" title="317"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-318" title="318"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-319" title="319"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-320" title="320"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-321" title="321"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-322" title="322"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-323" title="323"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-324" title="324"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-325" title="325"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-326" title="326"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-327" title="327"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-328" title="328"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-329" title="329"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-330" title="330"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-331" title="331"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-332" title="332"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-333" title="333"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-334" title="334"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-335" title="335"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-336" title="336"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-337" title="337"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-338" title="338"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-339" title="339"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-340" title="340"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-341" title="341"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-342" title="342"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-343" title="343"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-344" title="344"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-345" title="345"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-346" title="346"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-347" title="347"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-348" title="348"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-349" title="349"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-350" title="350"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-351" title="351"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-352" title="352"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-353" title="353"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-354" title="354"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-355" title="355"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-356" title="356"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-357" title="357"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-358" title="358"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-359" title="359"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-360" title="360"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-361" title="361"><span class="co">#&gt;  CRAN (R 3.5.1)                 </span></a>
<a class="sourceLine" id="cb18-362" title="362"><span class="co">#&gt;  Bioconductor                   </span></a>
<a class="sourceLine" id="cb18-363" title="363"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-364" title="364"><span class="co">#&gt; [1] C:/Users/s5125601/Documents/R/win-library/3.5</span></a>
<a class="sourceLine" id="cb18-365" title="365"><span class="co">#&gt; [2] C:/Program Files/R/R-3.5.1/library</span></a>
<a class="sourceLine" id="cb18-366" title="366"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-367" title="367"><span class="co">#&gt;  D -- DLL MD5 mismatch, broken installation.</span></a></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation-instruction">1. Installation instruction</a></li>
      <li>
<a href="#a-simple-workflow-of-the-scgps">2. A simple workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#create-scgps-objects">2.1 Create scGPS objects</a></li>
      <li><a href="#run-prediction">2.2 Run prediction</a></li>
      <li><a href="#summarise-results">2.3 Summarise results</a></li>
      </ul>
</li>
      <li>
<a href="#a-complete-workflow-of-the-scgps">3. A complete workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#identify-clusters-in-a-dataset-using-core">3.1 Identify clusters in a dataset using CORE</a></li>
      <li><a href="#identify-clusters-in-a-dataset-using-score-stable-clustering-at-optimal-resolution">3.2 Identify clusters in a dataset using SCORE (Stable Clustering at Optimal REsolution)</a></li>
      <li><a href="#visualise-all-cluster-results-in-all-iterations">3.3 Visualise all cluster results in all iterations</a></li>
      <li><a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--tsne">3.4 Compare clustering results with other dimensional reduction methods (e.g., tSNE)</a></li>
      <li><a href="#find-gene-markers-and-annotate-clusters">3.5 Find gene markers and annotate clusters</a></li>
      </ul>
</li>
      <li>
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples">4. Relationship between clusters within one sample or between two samples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#start-the-scgps-prediction-to-find-relationship-between-clusters">4.1 Start the scGPS prediction to find relationship between clusters</a></li>
      <li><a href="#display-summary-results-for-the-prediction">4.2 Display summary results for the prediction</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-one-sample">4.3 Plot the relationship between clusters in one sample</a></li>
      <li><a href="#plot-the-relationship-between-clusters-in-two-samples">4.3 Plot the relationship between clusters in two samples</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen, Michael Thompson, Anne Senabouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
